---
title: "CHILD cord blood linear regression using limma with covariate missing values filled in with means"
author: "Samantha LEE"
date: "January 13, 2021"
output: html_document
---

*Purpose*: The purpose of this script is to examine DNAm changes in cord blood associated with prenatal air pollution exposure (proxied by no2).


Set global chunk options. 
```{r setup, include=FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


Load required libraries.
```{r libraries}
library(tidyverse) # be tidy
library(limma) # for DNAm differences
library(here) # file calling
library(pbapply) # for running lm for each cpg
library(piecewiseSEM) # for getting partial residuals
library(broom) # tidy function for summarizing lm
library(sva) # for calculating SVs
library(ggcorrplot) # correlation matrix
```


Load betas
```{r betas}
#load the preprocessed betas
load(file=here("output_data", "preprocessed_data",
               "2021-01-14_CHILD_preprocessed_betas_pdata_annotation.Rdata"))
```


Subset out cord blood betas
```{r nocombat_betas}
# check ranges to confirm DNAm data is in beta value format
dim(betas)
max(betas)
min(betas)

# subset pData into CBMCs (cord blood)
# these children have data at both time points
cb_pdata <- subset(pdata_nodups, Tissue == "C")
# check size of cord blood pdata
dim(cb_pdata)

# subset cordb lood DNAm samples for those included in cord blood pdata
cb_betas <- betas[ ,colnames(betas) %in% rownames(cb_pdata)]

# check size for correct sample number
dim(cb_betas) #144

# check that CBMC betas and pdata are in same order
identical(colnames(cb_betas), rownames(cb_pdata)) # TRUE

# relabel CBMC beta columns with sample label (5 digit identifier)
colnames(cb_betas) <- cb_pdata$Sample_Label
```


Load covariate data. Missing values in covariates of interest have been filled with the mean of that variable. 
```{r covariates}
load(file=here("output_data", "covariates", "2020-11-12_CHILD_covariates_na_with_means.Rdata"))
```


Subset covariates for individuals with CBMC DNAm.
```{r sub_covariates}
# subset covariates based on samples with DNAm data
covariates_sub <- covariates_nameans[as.character(covariates_nameans$sampleid) 
                                     %in% colnames(cb_betas), ]

# check size
dim(covariates_sub) # 144 113  

# check if subset of covariates and betas are in the same order 
identical(as.character(covariates_sub$sampleid), colnames(cb_betas)) # TRUE
```


Load CBMC PCs.
```{r load_PC_correctedbetas}
# load the cell type PCs
load(here("output_data", "deconvolution", "2021-02-04_CBMC_PCs.RData"))

# check size
dim(cb_pcs) #144 6

#check if rownames of cell PCs are same as pdata
identical(rownames(cb_pcs), rownames(cb_pdata)) # TRUE

# rename rows using sample label (5 digit number)
rownames(cb_pcs) <- cb_pdata$Sample_Label

# rename columns for clarity
colnames(cb_pcs) <- paste("cbmc.", colnames(cb_pcs), sep="")
```


Load genotyping PCs and subset out REEGLE PCs. 
Genotyping data contains PCs for all CHILD participants that were genotyped.
```{r genotyping_PCS}
# read in genotyping data for CHILD cohort
genotyping <- read.csv(here("input_data", "genotyping", "CHILD_subjects_Qced_first10PCS_Duan.csv"))

# make rownames sample names (5 digit number)
rownames(genotyping) <- genotyping$FID
# removing sample naming columns - no longer needed now they are rownames 
genotyping <- subset(genotyping, select=-c(FID, IID))

# susbet genotyping data for REEGLE subset 
genosub <- genotyping[rownames(genotyping) %in% cb_pdata$Sample_Label,]

# check size
dim(genosub) # 131 of 144 children have genotyping

# rename columns of genotyping data for clarity
colnames(genosub) <- paste("genotyping.", colnames(genosub), sep="")

#put rows in numerical order according to sample id
genosub_order <- genosub[order(rownames(genosub)), ]
```


Examine genetic ancestry groups of participants based on genotyping PCs. Want to confirm that three PCs are required for genetic ancestry.
```{r genotyping_pc_groups, eval=FALSE}
# load genotyping info from Amirtha/Qingling 
# this info is for all CHILD participants that were genotyped
pca_info <- readxl::read_excel(here("input_data", "genotyping", "PCA_ethnicity_child_comparison_duan_lab07102020.xlsx"))

# sub for individuals in this study (REEGLE subset)
pca_sub <- pca_info[as.character(pca_info$FID) %in% rownames(genosub),]

# plot and colour according to race specified on questionnaires
ggplot(pca_sub, aes(x=PC1, y=PC2, 
                    col=as.factor(race_mom), 
                    shape=as.factor(`PCA ethnicity`))) +
  geom_point(alpha=0.7, size=2) +
  scale_color_manual(labels = c("First Nations", "Mixed","South East Asian",
                                 "East Asian","South Asian","Middle Eastern",
                                 "Hispanic","Caucasian White"),
                      values=c("grey55","#4E2A7A" , "#B20D30","#C17817",
                               "#3F784C", "black", "#6DB1BF","#254A90"),
                      name ="Maternal race")  +
  scale_shape_manual(values=c(21,22,23,24),
                     labels=c("Admixed","Central European" , 
                              "East Asian","South Asian"),
                     name ="PCA Genetic Ancestry") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size=6),
        axis.text.y = element_text(size=6),
        axis.title.x = element_text(size=8),
        axis.title.y = element_text(size=8),
        legend.text=element_text(size=4),
        legend.title=element_text(size=5),
        legend.key.size = unit(0.5, 'cm'), #change legend key size
        legend.key.height = unit(0.3, 'cm'), #change legend key height
        legend.key.width = unit(0.5, 'cm')) 

#  ggsave( filename="2021-01-21_CHILD_genopcs_race_ancestry.tiff",
#          device = "tiff",
#          path =  here("figures", "genotyping"),
#          width = 3.5,
#          height = 2.5,
#          units = "in",
#          limitsize = TRUE)
```


Subset betas, pdata, covariates, and cell type PCS for participants with genotyping data (the limiting factor).
```{r subset}
# subset betas
cb_betas_geno <- cb_betas[, rownames(genosub_order)]
# check size of subbed betas
dim(cb_betas_geno) # 424644    131

# subset pdata
cb_pdata_geno <- cb_pdata[cb_pdata$Sample_Label %in% rownames(genosub_order), ]
# check size of subbed pdata
dim(cb_pdata_geno) # 131 17

# subset covariates
covariates_sub_geno <- covariates_sub[covariates_sub$sampleid %in% 
                                        rownames(genosub_order), ]
# check size of subbed covariates 
dim(covariates_sub_geno) # 131 114

# subset cell type PCs
cb_pcs_geno <- cb_pcs[rownames(genosub_order), ]
# check size of subbed cell type PCs
dim(cb_pcs_geno) # 131 10
```


Check that betas, pdata, covariates, cell type PCs are in same order as genotyping data for REEGLE participants.
```{r cb_batchcorr_order}
# pdata and genosub_order
identical(as.character(cb_pdata_geno$Sample_Label),rownames(genosub_order)) # TRUE

# betas and genosub_order
identical(colnames(cb_betas_geno), rownames(genosub_order)) # TRUE

# covariates and genosub_order
identical(as.character(covariates_sub_geno$sampleid), rownames(genosub_order)) # TRUE

# cell type PCs and genosub_order
identical(rownames(genosub_order), rownames(cb_pcs_geno)) # TRUE
```


Combined covariates, genotyping, cell type PCs, and pData together.
```{r combine_cell_pcs_covariates}
###################################
# clean up pdata before combining #
###################################

# create new column for batch row variable in pdata
cb_pdata_geno$row <- as.numeric(as.factor(substr(cb_pdata_geno$Sentrix_Position,
                                                 1, 3)))
# examine
head(cb_pdata_geno$row)

# rename pdata sentrix Id column to chip
colnames(cb_pdata_geno)[8] <- "chip"

# rename pdata sentrix Id column to chip
colnames(cb_pdata_geno)[11] <- "run"

################################
# combine data frames together #
################################

# for pdata, only include batch variables
covariates_geno_ctpc_tech <- cbind(covariates_sub_geno, 
                                  cb_pcs_geno,
                                  genosub_order,
                                  cb_pdata_geno[,c("run", "row", "chip")])

# check classes to make sure they are what we expect before running linear model
lapply(covariates_geno_ctpc_tech, class)

# reclass run 
covariates_geno_ctpc_tech$run <- factor(covariates_geno_ctpc_tech$run, 
                                        ordered=F)
# reclass chip
covariates_geno_ctpc_tech$chip <- factor(covariates_geno_ctpc_tech$chip,
                                         ordered=F)
# reclass sex
covariates_geno_ctpc_tech$sex <- factor(covariates_geno_ctpc_tech$sex, 
                                        ordered=F)
```


Remove rows where air pollution is 0.
```{r no_zeroes_in_no2}
covariates_geno_ctpc_tech_not0 <- covariates_geno_ctpc_tech[!covariates_geno_ctpc_tech$no2_preg_entire==0, ]

dim(covariates_geno_ctpc_tech) # 131 132
dim(covariates_geno_ctpc_tech_not0) # 125 132

```


Subset complete cases in covariate data and betas in preparation for DNAm analysis.
```{r complete_cases}
# subset complete cases
covariates_geno_ctpc_tech_cc <- 
  covariates_geno_ctpc_tech_not0[complete.cases(
    covariates_geno_ctpc_tech_not0 %>%
      dplyr::select(no2_preg_entire, 
                    sex, 
                    gestational_days,
                    prenatal_maternal_smoking, 
                    maternal_education_length,
                    pss_18wk, 
                    cbmc.PC1, cbmc.PC2, cbmc.PC3, 
                    cbmc.PC4, cbmc.PC5,
                    genotyping.PC1, 
                    genotyping.PC2,
                    genotyping.PC3)), ] 

# check number of complete cases
dim(covariates_geno_ctpc_tech_cc) # 95 132

# subset beta values for complete cases
cb_betas_geno_cc <- cb_betas_geno[,colnames(cb_betas_geno) %in% 
                                    covariates_geno_ctpc_tech_cc$sampleid]

# check size of complete case betas
dim(cb_betas_geno_cc) # 424644 95

# remove probes from betas that were used to predict cell type
cb_betas_geno_cc_clean <- 
  cb_betas_geno_cc[!rownames(cb_betas_geno_cc) %in%
                     rownames(cb_deconvolutionprobes$coefEsts), ]

# check size
dim(cb_betas_geno_cc_clean) # 424079 95; 565 probes removed
```


Create model matrix for DNAm analysis. This model matrix does not contain batch variables as we are going to let surrogate variables capture their variance within the DNAm data. 
```{r model_matrix}
# create model matrix
mod <- model.matrix(~ no2_preg_entire + sex + gestational_days + pss_18wk + 
                      prenatal_maternal_smoking + maternal_education_length + 
                      cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 + 
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3,
                    data=covariates_geno_ctpc_tech_cc)

# check size
dim(mod) # 95 15
```


Surrogate variable analysis using the sva packages. This will allow us to capture variance due to batch effects, as well as variance due to unkown biological variables. 
```{r sva}
###############
# null matrix #
###############

# create null matrix
# this matrix does not include any of our covariates 
# this means that our covariates are "protected" and SVs will not capture them
mod0 <- model.matrix(~ 1, data=covariates_geno_ctpc_tech_cc)

################
# sva analysis #
################

# to get SVs in a way that allows calculation of their variance
# we need to pull the code out of sva and run it line-by-line
# we are using the "Be" method for calculating SVs

######################################
# irwsva required internal functions #
######################################

# edge.lfdr function is internal to irwsva function below
# mono function is internal to edge.lfdr
# they have to be initialized before we can run the rest of the code

mono <- function(lfdr){
  .Call("monotone", as.numeric(lfdr), PACKAGE="sva")
}

edge.lfdr <- function(p, trunc=TRUE, monotone=TRUE, transf=c("probit", "logit"), adj=1.5, eps=10^-8, lambda=0.8, ...) {
  pi0 <- mean(p >= lambda)/(1 - lambda)
  pi0 <- min(pi0, 1)
  n <- length(p)
  transf <- match.arg(transf)
  if(transf=="probit") {
    p <- pmax(p, eps)
    p <- pmin(p, 1-eps)
    x <- qnorm(p)
    myd <- density(x, adjust=adj)
    mys <- smooth.spline(x=myd$x, y=myd$y)
    y <- predict(mys, x)$y
    lfdr <- pi0*dnorm(x)/y
  }
  if(transf=="logit") {
    x <- log((p+eps)/(1-p+eps))
    myd <- density(x, adjust=adj)
    mys <- smooth.spline(x=myd$x, y=myd$y)
    y <- predict(mys, x)$y
    dx <- exp(x) / (1+exp(x))^2
    lfdr <- pi0 * dx/y
  }
  if(trunc) {
    lfdr[lfdr > 1] <- 1
  }
  if(monotone) {	
    lfdr <- lfdr[order(p)]
    lfdr <- mono(lfdr)
    lfdr <- lfdr[rank(p)]
  }
  return(lfdr)
}



##################################
# numsv function internal to sva #
##################################

#determine the number of required SVs
#this is numsv function code pulled out

#this is what is fed into this function from the sva function
dat = cb_betas_geno_cc_clean
mod = mod
#method="be" # this is the default and what is called by numsv
#vfilter=NULL # we dont apply a filter and its not used for be method
B=20
set.seed(1234)

# initialize some stuff
# get the residual matrix (the remaining variance)
n <- ncol(dat)
m <- nrow(dat)
H <- mod %*% solve(t(mod) %*% mod) %*% t(mod)
res <- dat - t(H %*% t(dat))

# run svd and get some eigenvalues
uu <- svd(res)
# get the num of "sig" eigenvalues
ndf <- min(m, n) - ceiling(sum(diag(H)))
# get percent variance explained by each eigenvalue
dstat <- uu$d[1:ndf]^2/sum(uu$d[1:ndf]^2)
# initialize null distribution matrix
dstat0 <- matrix(0, nrow = B, ncol = ndf)

# set seed here since we are calling sample
# create the null distribution
for (i in 1:B) {
  # this part of the code removes all relationships in the 
  res0 <- t(apply(res, 1, sample, replace = FALSE))
  res0 <- res0 - t(H %*% t(res0))
  # calculate svs on this
  uu0 <- svd(res0)
  # bind their explained variance in null matrix
  dstat0[i, ] <- uu0$d[1:ndf]^2/sum(uu0$d[1:ndf]^2)
  print(i) # print out the iteration we are on
}

# determine the number of svs required here based on pvalues
psv <- rep(1, n)
for (i in 1:ndf) {
  psv[i] <- mean(dstat0[, i] >= dstat[i])
}
for (i in 2:ndf) {
  psv[i] <- max(psv[(i - 1)], psv[i])
}

# this is the variable holding the number of SVs required 
# number is each to all the svs with a pval less than/equal to 0.1
nsv <- sum(psv <= 0.1)


####################################
# code from the main SVA function  #
####################################

n.sv = nsv # from code above (run internally when sva called)
B = 5 # number of iterations
  
# dat =  betas # already initialized
# mod = mod  # already initialized
# mod0 = mod0  # already initialized 
# controls = NULL # we dont include controls - only for supervised method
# method = "irw" # this isnt actually used but would be called by sva
# vfilter = NULL # we dont set this when running sva
# numSVmethod = "be" # this is the default called by sva (code above)
  

###################################
# irwsva function internal to sva #
###################################

# this code is really want does all the work in sva
# we use the irw method so this is the code for that method
# there are also supervised and two step methods (not used by us)

# irw function takes input from sva, so we have already defined below vars
# dat = dat 
# mod = mod
# mod0 = mod0
# n.sv = n.sv
# B = B

# intialize variables again
n <- ncol(dat)
m <- nrow(dat)
Id <- diag(n)
resid <- dat %*% (Id - mod %*% solve(t(mod) %*% mod) %*% t(mod))
uu <- eigen(t(resid) %*% resid)
vv <- uu$vectors
ndf <- n - dim(mod)[2]
pprob <- rep(1, m)
one <- rep(1, n)
Id <- diag(n)
df1 <- dim(mod)[2] + n.sv
df0 <- dim(mod0)[2] + n.sv
rm(resid)

cat(paste("Iteration (out of", B, "):"))
for (i in 1:B) {
  mod.b <- cbind(mod, uu$vectors[, 1:n.sv])
  mod0.b <- cbind(mod0, uu$vectors[, 1:n.sv])
  ptmp <- f.pvalue(dat, mod.b, mod0.b)
  pprob.b <- (1 - edge.lfdr(ptmp))
  mod.gam <- cbind(mod0, uu$vectors[, 1:n.sv])
  mod0.gam <- cbind(mod0)
  ptmp <- f.pvalue(dat, mod.gam, mod0.gam)
  pprob.gam <- (1 - edge.lfdr(ptmp))
  pprob <- pprob.gam * (1 - pprob.b)
  dats <- dat * pprob
  dats <- dats - rowMeans(dats)
  uu <- eigen(t(dats) %*% dats)
  cat(paste(i, " "))
}

# this is what is normally returned by sva
# problem is that it doesnt include the diagonal matrix we want
sv = svd(dats)$v[, 1:n.sv, drop = FALSE] 
svobj <- list(sv = sv, pprob.gam = pprob.gam, pprob.b = pprob.b, 
               n.sv = n.sv)

# to get all info (including diagonal)
svobj_all <- svd(dats)

svs <- as.data.frame(svobj$sv)

# populate rownames with sampleids
rownames(svs) <- covariates_geno_ctpc_tech_cc$sampleid

# save 
save(svs, file = here("output_data", "linear_regression", "2021-06-06_prenatal_svs.RData"))

###############
#scree of SVs #
###############

# from elodie portales-casamar

# heat scree prepartion
vars <- svobj_all[['d']]^2
vars <- vars/sum(vars)
names(vars) <- paste0('SV', 1:length(vars))
xp <- 1:svobj$n.sv
varsdf <- as.data.frame(vars)
varsdf$svnum <- rep(1:length(vars))

# heat scree ggplot!! 
ggplot(varsdf[1:nsv,], aes(x=svnum, y=vars)) +
  geom_col() +
  theme_bw() +
  labs(x="Surrogate Variables", y = "Percentage of Variance")

# it looks liek the first two svs are important
# examine a correlation matrix between all svs and covariates to see
# what variance they are capturing

######################
# correlation matrix #
######################

# add svs to covariates
covariates_geno_ctpc_tech_cc <- cbind(covariates_geno_ctpc_tech_cc, svobj$sv)

# rename columns with svs
colnames(covariates_geno_ctpc_tech_cc)[133:146] <-
  paste("sv",colnames(covariates_geno_ctpc_tech_cc)[133:146], sep="")

# early exp and pn_18 are the same thing. 
# exclude early_exp from correlation matrices
corr <- round(cor2(covariates_geno_ctpc_tech_cc %>% 
                select(-no_vaccines_6mos, 
                       -clinician_wheeze_dx_y3, 
                       -reported_asthma_dx_y1, 
                       -move_6m,
                       -reported_asthma_dx_5y,
                       -move_3m)))

corrsub <- corr[,127:140]

ggcorrplot(corr=corrsub,
           #p.mat=pval,
           #insig = "blank",
           #type = "lower", 
           #lab=TRUE,
           #lab_size = 2,
           tl.cex =6, 
           colors = c("#6D9EC1", "white", "#E46726")) +
  theme(panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        legend.position="left") +
  scale_y_discrete(position = "right")


######################################
# heat scree plot of svs against pcs #
######################################

#convert raw betas to mvals
mvals <- lumi::beta2m(cb_betas_geno_cc_clean)
max(mvals)
min(mvals)

#pca 
uncor.dat <- t(scale(t(mvals)))
PCA_full<-princomp(na.omit(uncor.dat))
Loadings<-as.data.frame(unclass(PCA_full$loadings))
vars <- PCA_full$sdev^2
Importance <-vars/sum(vars)

#Specify which covariates are categorical and/or continuous
meta_categorical <- 
  covariates_geno_ctpc_tech_cc[,c("sex", "prenatal_maternal_smoking",
                                  "run", "chip")] 
meta_continuous <- 
  covariates_geno_ctpc_tech_cc[,c("no2_preg_entire",
                                  "gestational_days", 
                                  "maternal_education_length", 
                                  "pss_18wk", "csed_18wk","row",
                                  "cbmc.PC1", "cbmc.PC2", "cbmc.PC3",
                                  "cbmc.PC4", "cbmc.PC5",
                                  "genotyping.PC1", "genotyping.PC2",
                                  "genotyping.PC3", "sv1", "sv2", "sv3",
                                  "sv4", "sv5", "sv6", "sv7", "sv8",
                                  "sv9", "sv10", "sv11", "sv12", "sv13",
                                  "sv14")]

# Specify the number of PCs you want shown 
# pick enough to view most of the variance
Num<-10

# Designate what order you want the variables to appear (continuous variables rbinded to categorical variables in function)
Order<-c(1:40)

#Apply function on PCA results of uncorrected data, pulls in the meta data and beta values from above
heat_scree_plot(Loadings, Importance, Num, Order)


############################################################
# point plots to see how well SVs separate batch variables #
############################################################

# cord blood row
ggpairs(covariates_geno_ctpc_tech_cc, columns=133:146,
          mapping = ggplot2::aes(color = as.factor(row)),
        upper = "blank",
        lower = list(continuous = wrap("points", alpha = 0.5, size=1)), 
        diag = NULL) +
  theme_bw()  +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()) 

# cord blood run
ggpairs(covariates_geno_ctpc_tech_cc, columns=133:146,
          mapping = ggplot2::aes(color = as.factor(run)),
        upper = "blank",
        lower = list(continuous = wrap("points", alpha = 0.5, size=1)), 
        diag = NULL) +
  theme_bw()  +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())


#cord blood chip
ggpairs(covariates_geno_ctpc_tech_cc, columns=133:146,
          mapping = ggplot2::aes(color = chip),
        upper = "blank",
        lower = list(continuous = wrap("points", alpha = 0.5, size=1)), 
        diag = NULL) +
  theme_bw()  +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()) +
  scale_colour_manual(values=c("deeppink1", "green1", "blue1", "purple1",
                               "cyan1", "firebrick1", "seagreen3", "tomato1",
                               "black", "dimgrey", "yellow1", "hotpink1",
                               "aquamarine1", "goldenrod", "darkorange1", "mediumorchid3",
                               "coral2", "deepskyblue", "dodgerblue", "springgreen",
                               "royalblue", "palevioletred", "red", "orange", "maroon"))


###########################
# add SVs to model matrix #
###########################

# use 6 SVs 
modSv = cbind(mod,svobj$sv[,1:6])
```


Run Limma for DNAm analysis.
```{r limma}
# fit methylation to no2
cb_fit <-  lmFit(cb_betas_geno_cc_clean, modSv)
cb_ebayes <- eBayes(cb_fit)

# subset annotation file for cpgs used in limma
annotation_cpgs <- annotation[rownames(annotation) %in% rownames(cb_ebayes),]

# check that annotation file and limma cpgs are in the same order
identical(rownames(annotation_cpgs), rownames(cb_ebayes)) # false

# reorder annotation file to match limma output
annotation_cpgs_order <- annotation_cpgs[rownames(cb_ebayes), ]

# recheck order of annotation file against limma cpgs
identical(rownames(annotation_cpgs_order), rownames(cb_ebayes)) #true

# pull out pval, coefficients from limma and combine with annotation
cb_pvals_tech <- cbind(as.data.frame(cb_ebayes$p.value), 
                       as.data.frame(cb_ebayes$coefficients),
                       as.data.frame(annotation_cpgs_order))

# rename columns to indicate pvalues vs coeffcients 
colnames(cb_pvals_tech) <- c(paste("pval", 
                                   colnames(cb_pvals_tech)[1:21], 
                                   sep = "_"), 
                             paste("coeff", 
                                   colnames(cb_pvals_tech)[22:42], 
                                   sep = "_"),
                             colnames(cb_pvals_tech)[43:75])

cb_pvals_tech$pval_no2_preg_entire_adj <-
  p.adjust(cb_pvals_tech$pval_no2_preg_entire, method="BH")
```


Examine qqplot and pvalue histogram.
```{r qqplot_pval_histo}
########
#qqplot#
########

# use code from qqman::qq function 
# plot with ggplot to make it more aesthetically appealling 
pvector <- cb_pvals_tech$pval_no2_preg_entire

pvector2 <- pvector[!is.na(pvector) & !is.nan(pvector) & !is.null(pvector) & 
                      is.finite(pvector) & pvector < 1 & pvector > 0]

pval_qq <- as.data.frame(cbind(o = -log10(sort(pvector2, decreasing = FALSE)),
                               e = -log10(ppoints(length(pvector2)))))


#calculate lambda
chisq <- qchisq(1-pvector,1)
lambda <-  median(chisq)/qchisq(0.5,1)
 
ggplot(pval_qq, aes(x=e, y=o)) + 
  geom_point(col="#254A90", size=1) + 
  geom_abline(slope = 1, intercept = 0, 
              col = "red", size=0.25, linetype="dashed") +
  labs(x = "Expected -log10(p-value)",
       y = "Observed -log10(p-value)") +
  annotate("text", x=4, y=1, label = paste("Î» = ", round(lambda, 2)), size=4) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        axis.title.x = element_text(size=11),
        axis.title.y = element_text(size=11)) +
  #ylim(0,7) + 

  ggsave( filename="2021-05-15_CHILD_cb_qqplot.png",
          path =  here("figures", "linear_regression", "prenatal"),
          width = 7.9, height = 6, units = "cm")
```


Delta beta
```{r db}
range = max(covariates_geno_ctpc_tech_cc$no2_preg_entire) -
    min(covariates_geno_ctpc_tech_cc$no2_preg_entire)

cb_pvals_tech$db <- cb_pvals_tech$coeff_no2_preg_entire * range
```


Volcano plot
```{r volcano_db_tech}

#make new column to label sig points with cpgid
cb_pvals_tech$cpgid <- rownames(cb_pvals_tech)

# plot that volcano
ggplot(cb_pvals_tech, aes(x=as.numeric(db)*100 , y=-log10(as.numeric(pval_no2_preg_entire)))) +
  geom_point(alpha=0.4, col = "#254A90", size=1) +
  
  #nominal p value cut off
  geom_hline(yintercept = -log10(0.000001), linetype="dashed", size=0.25) + 
  
  #10% effect size cut-off
  #geom_point(data=subset(cb_pvals_tech, 
  #                       (pval_no2_preg_entire < 0.000001 & db > 0.1) | 
  #                       (pval_no2_preg_entire < 0.000001 & db < -0.1)), 
  #           alpha=0.4, col = "red", size=1) + 
  #geom_vline(xintercept = c(10, -10), linetype = "dashed", size=0.25) +
  
  # 5% effect size cut off
  #geom_point(data=subset(cb_pvals_tech, 
  #                       (pval_no2_preg_entire < 0.000001 & db > 0.05) | 
  #                         (pval_no2_preg_entire < 0.000001 & db < -0.05)), 
  #           alpha=0.4, col = "red", size=1) + 
  #geom_vline(xintercept = c(5, -5), linetype = "dashed", size=0.25) +
  
  
  # 2.5% effect size cut off
  geom_point(data=subset(cb_pvals_tech, 
                         (pval_no2_preg_entire < 0.000001 & db > 0.025) | 
                           (pval_no2_preg_entire < 0.000001 & db < -0.025)), 
             alpha=0.4, col = "red", size=1) + 
  geom_vline(xintercept = c(2.5, -2.5), linetype = "dashed", size=0.25) +

  scale_x_continuous(breaks = seq(-50, 50, by = 10)) +
  labs(y="-log10(unadjusted p-value)", x = "Effect size (%)") +
  
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        axis.title.x = element_text(size=11),
        axis.title.y = element_text(size=11)) +
  
  ggsave( filename="2021-05-24_CHILD_cb_volcanoplot.png",
          path =  here("figures", "linear_regression", "prenatal"),
          width = 7.9, height = 6, units = "cm")


# poster

# plot that volcano
ggplot(cb_pvals_tech, aes(x=as.numeric(db)*100 , y=-log10(as.numeric(pval_no2_preg_entire)))) +
  geom_point(alpha=0.4, col = "#254A90", size=5) +
  
  #nominal p value cut off
  geom_hline(yintercept = -log10(0.000001), linetype="dashed", size=1) + 
  
  # 2.5% effect size cut off
  geom_point(data=subset(cb_pvals_tech, 
                         (pval_no2_preg_entire < 0.000001 & db > 0.025) | 
                           (pval_no2_preg_entire < 0.000001 & db < -0.025)), 
             alpha=0.4, col = "red", size=5) + 
  geom_vline(xintercept = c(2.5, -2.5), linetype = "dashed", size=1) +

  scale_x_continuous(breaks = seq(-50, 50, by = 10)) +
  labs(y="-log10(unadjusted p-value)", x = "Effect size (%)") +
  
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(size=22),
        axis.text.y = element_text(size=22),
        axis.title.x = element_text(size=25),
        axis.title.y = element_text(size=25)) +
  
  ggsave( filename="2021-06-19_CHILD_cb_volcanoplot_poster.png",
          path =  here("figures", "linear_regression", "prenatal"),
          width = 7, height = 5, units = "in")
```

Subset out important cpgs
```{r sig_cpgs_tables}
# add effect size to ebayes
cb_ebayes$db <- cb_pvals_tech$db

# subset based on effect size
cb_ebayes_sig <- cb_ebayes[cb_ebayes$db > 0.025 | cb_ebayes$db < -0.025, ]

# get top ten significant genes with coefficients and write csv
write.csv(topTable(cb_ebayes_sig, number=10, 
              coef ="no2_preg_entire", sort.by = "P", confint = T),
     here("tables", "prenatal", "2021-05-15_toptable_prenatal_10cpgs.csv"))

# get top 10 cpgs with manifest annotation
write.csv(subset(cb_pvals_tech, 
            (pval_no2_preg_entire < 0.000011 & db > 0.025) | 
                 (pval_no2_preg_entire < 0.000011 & db < -0.025)), 
          here("tables", "prenatal", 
               "2021-05-15_top10_prenatal_cpgs.csv"))

# write out all prenatal findings
write.csv(cb_pvals_tech, here("tables", "prenatal", 
                              "2021-05-15_ALL_prenatal_cpgs.csv"))
```


Plot nominally significant CpGs from volcano plot.
```{r sig_cpgs_graphs}

#######################
# C21orf70 cg03392571 #
#######################

# add dnam to covariates for linear regression
covariates_geno_ctpc_tech_cc$cb_cg03392571 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg03392571", ]

# linear regression
cb_cg03392571_lm <- lm(cb_cg03392571 ~ 
                         no2_preg_entire + sex + gestational_days + pss_18wk +
                         prenatal_maternal_smoking + maternal_education_length + 
                         cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 + 
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                         sv1 + sv2 + sv3 + sv4 + sv5 + sv6,
                       data=covariates_geno_ctpc_tech_cc)

# get summary for slope and intercept
summary(cb_cg03392571_lm)

# get partial residuals for no2 and dnam 
cb_cg03392571_res <- partialResid(cb_cg03392571 ~ no2_preg_entire, 
                                  cb_cg03392571_lm,
                                  data=covariates_geno_ctpc_tech_cc)

# fit again to get r2
cb_cg03392571_res_fit <- lm(cb_cg03392571_res$yresid ~ cb_cg03392571_res$xresid)

# summary of partial resid fit
summary(cb_cg03392571_res_fit)

# partial regression plot
ggplot(cb_cg03392571_res, aes(x=xresid, y=yresid*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
  scale_y_continuous(limits = c(-25, 15), breaks = seq(-20, 10, by = 10)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank()) +
  ggsave(filename="2021-05-24_CHILD_cb_g03392571_C21orf70_PR.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.0, height = 3.2, units = "cm")

# normal plot
ggplot(covariates_geno_ctpc_tech_cc, 
       aes(x=no2_preg_entire, y=cb_cg03392571*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
  scale_y_continuous(limits = c(50, 100), breaks = seq(55, 100, by = 10)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank()) +
  ggsave(filename="2021-05-24_CHILD_cb_cg03392571_C21orf70.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 3.7, height = 3.2, units = "cm")


# normal plot for poster
ggplot(covariates_geno_ctpc_tech_cc, 
       aes(x=no2_preg_entire, y=cb_cg03392571*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=3) + 
  stat_smooth(method="lm", se=F, col="red", size=1, linetype="dashed") + 
  scale_y_continuous(limits = c(50, 100), breaks = seq(55, 100, by = 10)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=20),
        axis.title = element_blank()) +
  ggsave(filename="2021-06-19_CHILD_cb_cg03392571_C21orf70_poster.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4, height = 3.2, units = "in")


# correlation 
cor(covariates_geno_ctpc_tech_cc$cb_cg03392571, 
    covariates_geno_ctpc_tech_cc$no2_preg_entire,
    method="pearson") #0.4964235


####################
# PACS2 cg22908922 #
####################

#add dnam to covariates for linear regression
covariates_geno_ctpc_tech_cc$cb_cg22908922 <- 
        cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg22908922", ]

#linear regression
cb_cg22908922_lm <- lm(cb_cg22908922 ~ 
                         no2_preg_entire + sex + gestational_days + pss_18wk +
                         prenatal_maternal_smoking + maternal_education_length + 
                         cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 + 
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                         sv1 + sv2 + sv3 + sv4 + sv5 + sv6,
                       data=covariates_geno_ctpc_tech_cc)

# get summary of  linear regression
summary(cb_cg22908922_lm)

# get partial residuals for no2 and dnam 
cb_cg22908922_res <- partialResid(cb_cg22908922 ~ no2_preg_entire, 
                                  cb_cg22908922_lm,
                                  data=covariates_geno_ctpc_tech_cc)

# fit again to get r2
cb_cg22908922_res_fit <- lm(cb_cg22908922_res$yresid ~ cb_cg22908922_res$xresid)

# summary of partial residual fit
summary(cb_cg22908922_res_fit)

# plot
ggplot(cb_cg22908922_res, aes(x=xresid, y=yresid*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
  scale_y_continuous(limits = c(-25, 15), breaks = seq(-20, 10, by = 10)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank())  +
  ggsave(filename="2021-05-24_CHILD_cb_cg22908922_PACS2_PR.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.0, height = 3.2, units = "cm")

# normal plot
ggplot(covariates_geno_ctpc_tech_cc,
       aes(x=no2_preg_entire, y=cb_cg22908922*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
  scale_y_continuous(limits = c(50, 100), breaks = seq(55, 100, by = 10)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank()) +
  ggsave(filename="2021-05-24_CHILD_cb_cg22908922_PACS2.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 3.7, height = 3.2, units = "cm")


# poster normal plot
ggplot(covariates_geno_ctpc_tech_cc,
       aes(x=no2_preg_entire, y=cb_cg22908922*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=3) + 
  stat_smooth(method="lm", se=F, col="red", size=1, linetype="dashed") + 
  scale_y_continuous(limits = c(50, 100), breaks = seq(55, 100, by = 10)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=20),
        axis.title = element_blank()) +
  ggsave(filename="2021-06-19_CHILD_cb_cg22908922_PACS2_poster.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4, height = 3.2, units = "in")

# correlation 
cor(covariates_geno_ctpc_tech_cc$cb_cg22908922, 
    covariates_geno_ctpc_tech_cc$no2_preg_entire,
    method="pearson") #-0.1659826

#############################
# AFAP1/LOC84740 cg12228123 #
#############################

# add dnam to covariates for linear regression
covariates_geno_ctpc_tech_cc$cb_cg12228123<- 
        cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg12228123", ]

# linear regression
cb_cg12228123_lm <- lm(cb_cg12228123 ~ no2_preg_entire + 
                         no2_preg_entire + sex + gestational_days + pss_18wk +
                         prenatal_maternal_smoking + maternal_education_length + 
                         cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 + 
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                         sv1 + sv2 + sv3 + sv4 + sv5 + sv6,
                       data=covariates_geno_ctpc_tech_cc)

# get summary of fit
summary(cb_cg12228123_lm)

# get partial residuals for no2 and dnam 
cb_cg12228123_res <- partialResid(cb_cg12228123 ~ no2_preg_entire, 
                                  cb_cg12228123_lm, 
                                  data=covariates_geno_ctpc_tech_cc)

# fit again to get r2
cb_cg12228123_res_fit <- lm(cb_cg12228123_res$yresid ~ cb_cg12228123_res$xresid)

# get summary of partial residual fit
summary(cb_cg12228123_res_fit)

# partial residual plot
ggplot(cb_cg12228123_res, aes(x=xresid, y=yresid*100)) +
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
  scale_y_continuous(limits = c(-25, 15), breaks = seq(-20, 10, by = 10)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank())  +
  ggsave(filename="2021-05-24_CHILD_cb_cg12228123_AFAP1_PR.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.0, height = 3.2, units = "cm")

# normal plot
ggplot(covariates_geno_ctpc_tech_cc,
       aes(x=no2_preg_entire, 
           y=cb_cg12228123*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
  scale_y_continuous(limits = c(50, 100), breaks = seq(55, 100, by = 10)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank()) +
  ggsave(filename="2021-05-24_CHILD_cb_cg12228123_AFAP1.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 3.7, height = 3.2, units = "cm")

# normal plot nposter
ggplot(covariates_geno_ctpc_tech_cc,
       aes(x=no2_preg_entire, 
           y=cb_cg12228123*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=3) + 
  stat_smooth(method="lm", se=F, col="red", size=1, linetype="dashed") + 
  scale_y_continuous(limits = c(50, 100), breaks = seq(55, 100, by = 10)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=20),
        axis.title = element_blank()) +
  ggsave(filename="2021-06-19_CHILD_cb_cg12228123_AFAP1_poster.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4, height = 3.2, units = "in")

# correlation 
cor(covariates_geno_ctpc_tech_cc$cb_cg12228123, 
    covariates_geno_ctpc_tech_cc$no2_preg_entire,
    method="pearson") # -0.4303284

```


#### Examine persistence of prenatal changes at age one

First we need to prepare the age one data for the linear model. Then we need to calculate the number of SVs needed for this model. 

Get age one pdata
```{r y1_pdata}
#subset out age one pdata
y1_pdata <- subset(pdata_nodups, Tissue == "P")

#check size
dim(y1_pdata) # 145 17
```


Get age one betas
```{r y1_betas}
#subset out PBMC betas 
y1_betas <- betas[ ,colnames(betas) %in% rownames(y1_pdata)]

#check if age one betas and pdata are in same order
identical(colnames(y1_betas), rownames(y1_pdata)) # TRUE

#rename colums with sample label (5 dig num)
colnames(y1_betas) <- y1_pdata$Sample_Label
```


Load age one cell type PCs
```{r y1_celltypePCs}
#load age one PBMC estimates
load(here("output_data", "deconvolution", "2021-02-04_PBMC_PCs.RData"))

#rename columns of PBMCs for clarity
colnames(y1_pcs) <- paste("pbmc", colnames(y1_pcs), sep=".")

#rename row of cell type PCs to sample label (5 digit num)
identical(rownames(y1_pcs), rownames(y1_pdata)) #true
rownames(y1_pcs) <- y1_pdata$Sample_Label

#subset cell type PCs for complete cases
y1_pcs_sub <- y1_pcs[rownames(y1_pcs) %in% covariates_geno_ctpc_tech_cc$sampleid,]
dim(y1_pcs_sub) #101

# add age one PBMCs to covariate data frame
covariates_geno_ctpc_tech_cc_y1 <- cbind(covariates_geno_ctpc_tech_cc, y1_pcs_sub)
```


Add age one batch variables to covariate data frame
```{r y1_data_prep}
#add age one batch variables to covariate matrix
y1_pdata$y1_run <- as.factor(y1_pdata$Run) 
y1_pdata$y1_row  <- as.factor(substr(y1_pdata$Sentrix_Position,1,3)) 
y1_pdata$y1_row <- as.numeric(y1_pdata$y1_row)
y1_pdata$y1_chip <- as.factor(y1_pdata$Sentrix_ID)

covariates_geno_ctpc_tech_cc_y1 <- cbind(covariates_geno_ctpc_tech_cc_y1, 
                                       y1_pdata[y1_pdata$Sample_Label %in% covariates_geno_ctpc_tech_cc_y1$sampleid, c("y1_run", "y1_row", "y1_chip")])
```


Subset for complete cases in age one datga
```{r y1_complete_cases}
#subset covariates with age one cell type PCs for linear regression
covariates_geno_ctpc_tech_cc_y1_sub <- 
  covariates_geno_ctpc_tech_cc_y1[complete.cases(covariates_geno_ctpc_tech_cc_y1 %>% 
                                                 dplyr::select(no2_preg_entire, 
                                                               sex,  
                                                               maternal_smoking_y1,
                                                            maternal_education_length,
                                                               pss_6mos,
                                                               pbmc.PC1, pbmc.PC2,
                                                               pbmc.PC3, pbmc.PC4,
                                                               genotyping.PC1,
                                                               genotyping.PC2,
                                                               genotyping.PC3)), ] 

#check size
dim(covariates_geno_ctpc_tech_cc_y1_sub) # 92

#subset age one betas for samples in the complete case cord blood betas 
y1_betas_cc <- y1_betas[,colnames(y1_betas) %in%
                          as.character(covariates_geno_ctpc_tech_cc_y1_sub$sampleid)]

#check size
dim(y1_betas_cc) #424644    92

#remove probes that were used for PBMC cell type predict from age one betas
y1_betas_cc_sub <- y1_betas_cc[!rownames(y1_betas_cc) %in% 
                                 rownames(y1_deconvolutionprobes$coefEsts), ]

#check size
dim(y1_betas_cc_sub) # 424079    92
# 565 probes removed

#check that covariates and betas in same order
identical(colnames(y1_betas_cc_sub), 
          as.character(covariates_geno_ctpc_tech_cc_y1_sub$sampleid)) # TRUE
```


Create model matrix for age one analysis
```{r model_matrix}
# create model matrix
y1_mod <- model.matrix(~ no2_preg_entire + sex + pss_6mos + 
                      maternal_smoking_y1 + maternal_education_length +
                      pbmc.PC1 + pbmc.PC2 + pbmc.PC3 + pbmc.PC4  +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3,
                    data=covariates_geno_ctpc_tech_cc_y1_sub)

# check size
dim(y1_mod) # 92 13
```


Calculate the number of SVs needed for age one model.
```{r y1_sv_analysis}

# create null matrix
# this matrix does not include any of our covariates 
# this means that our covariates are "protected" and SVs will not capture them
y1_mod0 <- model.matrix(~ 1, data=covariates_geno_ctpc_tech_cc_y1_sub)

################
# sva analysis #
################

##################################
# numsv function internal to sva #
##################################

#determine the number of required SVs
#this is numsv function code pulled out

#this is what is fed into this function from the sva function
dat = y1_betas_cc_sub
mod = y1_mod
mod0 = y1_mod0
#method="be" # this is the default and what is called by numsv
#vfilter=NULL # we dont apply a filter and its not used for be method
B=20
set.seed(1234)

# initialize some stuff
# get the residual matrix (the remaining variance)
n <- ncol(dat)
m <- nrow(dat)
H <- mod %*% solve(t(mod) %*% mod) %*% t(mod)
res <- dat - t(H %*% t(dat))

# run svd and get some eigenvalues
uu <- svd(res)
# get the num of "sig" eigenvalues
ndf <- min(m, n) - ceiling(sum(diag(H)))
# get percent variance explained by each eigenvalue
dstat <- uu$d[1:ndf]^2/sum(uu$d[1:ndf]^2)
# initialize null distribution matrix
dstat0 <- matrix(0, nrow = B, ncol = ndf)

# set seed here since we are calling sample
# create the null distribution
for (i in 1:B) {
  # this part of the code removes all relationships in the 
  res0 <- t(apply(res, 1, sample, replace = FALSE))
  res0 <- res0 - t(H %*% t(res0))
  # calculate svs on this
  uu0 <- svd(res0)
  # bind their explained variance in null matrix
  dstat0[i, ] <- uu0$d[1:ndf]^2/sum(uu0$d[1:ndf]^2)
  print(i) # print out the iteration we are on
}

# determine the number of svs required here based on pvalues
psv <- rep(1, n)
for (i in 1:ndf) {
  psv[i] <- mean(dstat0[, i] >= dstat[i])
}
for (i in 2:ndf) {
  psv[i] <- max(psv[(i - 1)], psv[i])
}

# this is the variable holding the number of SVs required 
# number is each to all the svs with a pval less than/equal to 0.1
nsv <- sum(psv <= 0.1)


####################################
# code from the main SVA function  #
####################################

n.sv = nsv # from code above (run internally when sva called)
B = 5 # number of iterations
  
# dat =  betas # already initialized
# mod = mod  # already initialized
# mod0 = mod0  # already initialized 
# controls = NULL # we dont include controls - only for supervised method
# method = "irw" # this isnt actually used but would be called by sva
# vfilter = NULL # we dont set this when running sva
# numSVmethod = "be" # this is the default called by sva (code above)
  

###################################
# irwsva function internal to sva #
###################################

# this code is really want does all the work in sva
# we use the irw method so this is the code for that method
# there are also supervised and two step methods (not used by us)

# irw function takes input from sva, so we have already defined below vars
# dat = dat 
# mod = mod
# mod0 = mod0
# n.sv = n.sv
# B = B

# intialize variables again
n <- ncol(dat)
m <- nrow(dat)
Id <- diag(n)
resid <- dat %*% (Id - mod %*% solve(t(mod) %*% mod) %*% t(mod))
uu <- eigen(t(resid) %*% resid)
vv <- uu$vectors
ndf <- n - dim(mod)[2]
pprob <- rep(1, m)
one <- rep(1, n)
Id <- diag(n)
df1 <- dim(mod)[2] + n.sv
df0 <- dim(mod0)[2] + n.sv
rm(resid)

cat(paste("Iteration (out of", B, "):"))
for (i in 1:B) {
  mod.b <- cbind(mod, uu$vectors[, 1:n.sv])
  mod0.b <- cbind(mod0, uu$vectors[, 1:n.sv])
  ptmp <- f.pvalue(dat, mod.b, mod0.b)
  pprob.b <- (1 - edge.lfdr(ptmp))
  mod.gam <- cbind(mod0, uu$vectors[, 1:n.sv])
  mod0.gam <- cbind(mod0)
  ptmp <- f.pvalue(dat, mod.gam, mod0.gam)
  pprob.gam <- (1 - edge.lfdr(ptmp))
  pprob <- pprob.gam * (1 - pprob.b)
  dats <- dat * pprob
  dats <- dats - rowMeans(dats)
  uu <- eigen(t(dats) %*% dats)
  cat(paste(i, " "))
}

# this is what is normally returned by sva
# problem is that it doesnt include the diagonal matrix we want
sv = svd(dats)$v[, 1:n.sv, drop = FALSE] 
svobj_y1 <- list(sv = sv, pprob.gam = pprob.gam, pprob.b = pprob.b, 
               n.sv = n.sv)

# to get all info (including diagonal)
svobj_all_y1 <- svd(dats)

# populate rownames with sample ids
colnames(svobj_all_y1$v) <- covariates_geno_ctpc_tech_cc_y1_sub$sampleid


# save 
save(svobj_all_y1, file = here("output_data", "linear_regression", "2021-06-06_ageone_svs.RData"))

###############
#scree of SVs #
###############

# from elodie portales-casamar

# heat scree prepartion
vars <- svobj_all_y1[['d']]^2
vars <- vars/sum(vars)
names(vars) <- paste0('SV', 1:length(vars))
xp <- 1:svobj$n.sv
varsdf <- as.data.frame(vars)
varsdf$svnum <- rep(1:length(vars))

# heat scree ggplot!! 
ggplot(varsdf[1:nsv,], aes(x=svnum, y=vars)) +
  geom_col() +
  theme_bw() +
  labs(x="Surrogate Variables", y = "Percentage of Variance")

# it looks liek the first two svs are important
# examine a correlation matrix between all svs and covariates to see
# what variance they are capturing

######################
# correlation matrix #
######################

# add svs to covariates
covariates_geno_ctpc_tech_cc_y1_sub <- cbind(covariates_geno_ctpc_tech_cc_y1_sub, svobj_y1$sv)

# rename columns with svs
colnames(covariates_geno_ctpc_tech_cc_y1_sub)[158:173] <-
  paste("y1sv",colnames(covariates_geno_ctpc_tech_cc_y1_sub)[158:173], sep="")

# early exp and pn_18 are the same thing. 
# exclude early_exp from correlation matrices
corr <- round(cor2(covariates_geno_ctpc_tech_cc_y1_sub %>% 
                dplyr::select(-no_vaccines_6mos, 
                       -clinician_wheeze_dx_y3, 
                       -reported_asthma_dx_y1, 
                       -move_6m,
                       -reported_asthma_dx_5y,
                       -move_3m)))

corrsub <- corr[,152:166]

ggcorrplot(corr=corrsub,
           #p.mat=pval,
           #insig = "blank",
           type = "lower", 
           #lab=TRUE,
           #lab_size = 2,
           tl.cex =6, 
           colors = c("#6D9EC1", "white", "#E46726")) +
  theme(panel.grid.minor = element_blank(),
        panel.grid.major = element_blank(),
        legend.position="left") +
  scale_y_discrete(position = "right")


######################################
# heat scree plot of svs against pcs #
######################################

#convert raw betas to mvals
mvals <- lumi::beta2m(y1_betas_cc_sub)
max(mvals)
min(mvals)

#pca 
uncor.dat <- t(scale(t(mvals)))
PCA_full<-princomp(na.omit(uncor.dat))
Loadings<-as.data.frame(unclass(PCA_full$loadings))
vars <- PCA_full$sdev^2
Importance <-vars/sum(vars)

#Specify which covariates are categorical and/or continuous
meta_categorical <- 
  covariates_geno_ctpc_tech_cc_y1_sub[,c("sex", "maternal_smoking_y1",
                                  "y1_run", "y1_chip")] 
meta_continuous <- 
  covariates_geno_ctpc_tech_cc_y1_sub[,c("no2_preg_entire",
                                  "maternal_education_length", 
                                  "pss_6mos", "csed_6mos","y1_row",
                                  "pbmc.PC1", "pbmc.PC2", "pbmc.PC3",
                                  "pbmc.PC4",
                                  "genotyping.PC1", "genotyping.PC2",
                                  "genotyping.PC3", "y1sv1", "y1sv2", "y1sv3",
                                  "y1sv4", "y1sv5", "y1sv6", "y1sv7", "y1sv8",
                                  "y1sv9", "y1sv10", "y1sv11", "y1sv12", "y1sv13",
                                  "y1sv14", "y1sv15")]

# Specify the number of PCs you want shown 
# pick enough to view most of the variance
Num<-10

# Designate what order you want the variables to appear (continuous variables rbinded to categorical variables in function)
Order<-c(1:40)

#Apply function on PCA results of uncorrected data, pulls in the meta data and beta values from above
heat_scree_plot(Loadings, Importance, Num, Order)



############################################################
# point plots to see how well SVs separate batch variables #
############################################################

# cord blood row
ggpairs(covariates_geno_ctpc_tech_cc, columns=133:146,
          mapping = ggplot2::aes(color = as.factor(row)),
        upper = "blank",
        lower = list(continuous = wrap("points", alpha = 0.5, size=1)), 
        diag = NULL) +
  theme_bw()  +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()) 

# cord blood run
ggpairs(covariates_geno_ctpc_tech_cc, columns=133:146,
          mapping = ggplot2::aes(color = as.factor(run)),
        upper = "blank",
        lower = list(continuous = wrap("points", alpha = 0.5, size=1)), 
        diag = NULL) +
  theme_bw()  +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank())


#cord blood chip
ggpairs(covariates_geno_ctpc_tech_cc, columns=133:146,
          mapping = ggplot2::aes(color = chip),
        upper = "blank",
        lower = list(continuous = wrap("points", alpha = 0.5, size=1)), 
        diag = NULL) +
  theme_bw()  +
  theme(panel.grid.major = element_blank(),
        panel.grid.minor = element_blank()) +
  scale_colour_manual(values=c("deeppink1", "green1", "blue1", "purple1",
                               "cyan1", "firebrick1", "seagreen3", "tomato1",
                               "black", "dimgrey", "yellow1", "hotpink1",
                               "aquamarine1", "goldenrod", "darkorange1", "mediumorchid3",
                               "coral2", "deepskyblue", "dodgerblue", "springgreen",
                               "royalblue", "palevioletred", "red", "orange", "maroon"))


###########################
# add SVs to model matrix #
###########################

# use 9 SVs 
y1_modSv = cbind(y1_mod,svobj_all_y1$v[,1:9])

```


Year one limma test. Except pvalues to follow null distribution in qqplot
```{r y1_limma, eval=FALSE}
# fit methylation to no2
y1_fit <-  lmFit(y1_betas_cc_sub, y1_modSv)
y1_ebayes <- eBayes(y1_fit)

# subset annotation file for cpgs used in limma
annotation_cpgs <- annotation[rownames(annotation) %in% rownames(y1_ebayes),]

# check that annotation file and limma cpgs are in the same order
identical(rownames(annotation_cpgs), rownames(y1_ebayes)) # false

# reorder annotation file to match limma output
annotation_cpgs_order <- annotation_cpgs[rownames(y1_ebayes), ]

# recheck order of annotation file against limma cpgs
identical(rownames(annotation_cpgs_order), rownames(y1_ebayes)) #true

# pull out pval, coefficients from limma and combine with annotation
y1_pvals_tech <- cbind(as.data.frame(y1_ebayes$p.value), 
                       as.data.frame(y1_ebayes$coefficients),
                       as.data.frame(annotation_cpgs_order))

# rename columns to indicate pvalues vs coeffcients 
colnames(y1_pvals_tech) <- c(paste("pval", 
                                   colnames(y1_pvals_tech)[1:22], 
                                   sep = "_"), 
                             paste("coeff", 
                                   colnames(y1_pvals_tech)[23:44], 
                                   sep = "_"),
                             colnames(y1_pvals_tech)[45:77])

```


```{r y1_qqplot_pval_histo, eval=FALSE}
########
#qqplot#
########

# use code from qqman::qq function 
# plot with ggplot to make it more aesthetically appealling 
pvector <- y1_pvals_tech$pval_no2_preg_entire

pvector2 <- pvector[!is.na(pvector) & !is.nan(pvector) & !is.null(pvector) & 
                      is.finite(pvector) & pvector < 1 & pvector > 0]

pval_qq <- as.data.frame(cbind(o = -log10(sort(pvector2, decreasing = FALSE)),
                               e = -log10(ppoints(length(pvector2)))))

#calculate lambda
chisq <- qchisq(1-pvector,1)
lambda <-  median(chisq)/qchisq(0.5,1)
 
ggplot(pval_qq, aes(x=e, y=o)) + 
  geom_point(col="#254A90", size=1) + 
  geom_abline(slope = 1, intercept = 0, col = "red", size=0.25) +
  labs(x = "Expected -log10(p-value)",
       y = "Observed -log10(p-value)") +
  annotate("text", x=4, y=1, label = paste("Î» = ", round(lambda, 2)), size=3) +
  scale_x_continuous(breaks=c(0,2,4,6),
                     labels=c("-0.00", "-2.00", "-4.00", "-6.00")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size=5,angle=90, vjust = 0.5, hjust=1),
        axis.text.y = element_text(size=5),
        axis.title.x = element_text(size=7),
        axis.title.y = element_text(size=7)) +
  ylim(0,12)

```


Examine persistence of prenatal changes at age one.
```{r persistence}

#######################
# C21orf70 cg03392571 #
#######################

# add dnam to covariates for linear regression
covariates_geno_ctpc_tech_cc_y1_sub$y1_cg03392571 <- 
        y1_betas_cc_sub[rownames(y1_betas_cc_sub)=="cg03392571", ]

# linear regression
y1_cg03392571_lm <- lm(y1_cg03392571 ~  
                         no2_preg_entire + sex + pss_6mos +
                         maternal_smoking_y1 + maternal_education_length +
                         pbmc.PC1 + pbmc.PC2 + pbmc.PC3 + pbmc.PC4 + 
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
                         y1sv1 + y1sv2 + y1sv3 + y1sv4 + y1sv5 + 
                         y1sv6 + y1sv7 + y1sv9, 
                       data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_cg03392571_lm)

# get partial residuals for no2 and dnam 
y1_cg03392571_res <- partialResid(y1_cg03392571 ~ no2_preg_entire, 
                                  y1_cg03392571_lm, 
                                  data=covariates_geno_ctpc_tech_cc_y1_sub)

# fit again to get r2
y1_cg03392571_fit <- lm(y1_cg03392571_res$yresid ~ y1_cg03392571_res$xresid)

# summary of partial residual
summary(y1_cg03392571_fit)


# plots
ggplot(y1_cg03392571_res, aes(x=xresid, y=yresid*100)) +
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
  scale_y_continuous(limits = c(-25, 15), breaks = seq(-20, 10, by = 10)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank())  +
  ggsave(filename="2021-05-25_CHILD_y1_cg03392571_C21orf70_PR.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.0, height = 3.2, units = "cm")

# normal plot
ggplot(covariates_geno_ctpc_tech_cc_y1_sub,
       aes(x=no2_preg_entire, y=y1_cg03392571*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
  scale_y_continuous(limits = c(50, 100), breaks = seq(55, 100, by = 10)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank()) +
  ggsave(filename="2021-05-24_CHILD_y1_cg03392571_C21orf70.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 3.7, height = 3.2, units = "cm")


# normal plot poster 
ggplot(covariates_geno_ctpc_tech_cc_y1_sub,
       aes(x=no2_preg_entire, y=y1_cg03392571*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=3) + 
  stat_smooth(method="lm", se=F, col="red", size=1, linetype="dashed") + 
  scale_y_continuous(limits = c(50, 100), breaks = seq(55, 100, by = 10)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=20),
        axis.title = element_blank()) +
  ggsave(filename="2021-06-19_CHILD_y1_cg03392571_C21orf70_poster.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4, height = 3.2, units = "in")

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_cg03392571, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_preg_entire,
    method="pearson") #0.02433326


####################
# PACS2 cg22908922 #
####################

# add dnam to covariates for linear regression
covariates_geno_ctpc_tech_cc_y1_sub$y1_cg22908922 <- 
        y1_betas_cc_sub[rownames(y1_betas_cc_sub)=="cg22908922", ]

# linear regression
y1_cg22908922_lm <- lm(y1_cg22908922 ~  
                         no2_preg_entire + sex + pss_6mos +
                         maternal_smoking_y1 + maternal_education_length +
                         pbmc.PC1 + pbmc.PC2 + pbmc.PC3 + pbmc.PC4 + 
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
                         y1sv1 + y1sv2 + y1sv3 + y1sv4 + y1sv5 + 
                         y1sv6 + y1sv7 + y1sv9, 
                       data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_cg22908922_lm)

# get partial residuals for no2 and dnam 
y1_cg22908922_res <- partialResid(y1_cg22908922 ~ no2_preg_entire, 
                                  y1_cg22908922_lm, 
                                  data=covariates_geno_ctpc_tech_cc_y1_sub)

# fit again to get r2
y1_cg22908922_fit <- lm(y1_cg22908922_res$yresid ~ y1_cg22908922_res$xresid)

# summary of partial residual 
summary(y1_cg22908922_fit)


# plots
ggplot(y1_cg22908922_res, aes(x=xresid, y=yresid*100)) +
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
  scale_y_continuous(limits = c(-25, 15), breaks = seq(-20, 10, by = 10)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank()) +
  ggsave(filename="2021-05-24_CHILD_y1_cg22908922_PACS2_PR.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.0, height = 3.2, units = "cm")

# normal plot
ggplot(covariates_geno_ctpc_tech_cc_y1_sub,
       aes(x=no2_preg_entire, y=y1_cg22908922*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
  theme_bw() +
  scale_y_continuous(limits = c(50, 100), breaks = seq(55, 100, by = 10)) +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank()) +
  ggsave(filename="2021-05-24_CHILD_y1_cg22908922_PACS2.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 3.7, height = 3.2, units = "cm")


# normal plot poster
ggplot(covariates_geno_ctpc_tech_cc_y1_sub,
       aes(x=no2_preg_entire, y=y1_cg22908922*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=3) + 
  stat_smooth(method="lm", se=F, col="red", size=1, linetype="dashed") + 
  theme_bw() +
  scale_y_continuous(limits = c(50, 100), breaks = seq(55, 100, by = 10)) +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=20),
        axis.title = element_blank()) +
  ggsave(filename="2021-06-19_CHILD_y1_cg22908922_PACS2.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4, height = 3.2, units = "in")

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_cg22908922, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_preg_entire,
    method="pearson") # -0.3001429


#############################
# AFAP1/LOC84740 cg12228123 #
#############################

#add dnam to covariates for linear regression
covariates_geno_ctpc_tech_cc_y1_sub$y1_cg12228123<- 
        y1_betas_cc_sub[rownames(y1_betas_cc_sub)=="cg12228123", ]

#linear regression
y1_cg12228123_lm <- lm(y1_cg12228123 ~ 
                         no2_preg_entire + sex + pss_6mos + 
                         maternal_smoking_y1 + maternal_education_length +
                         pbmc.PC1 + pbmc.PC2 + pbmc.PC3 + pbmc.PC4 + 
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
                         y1sv1 + y1sv2 + y1sv3 + y1sv4 + y1sv5 + 
                         y1sv6 + y1sv7 + y1sv9, 
                    data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_cg12228123_lm)

# get partial residuals for no2 and dnam 
y1_cg12228123_res <- partialResid(y1_cg12228123 ~ no2_preg_entire, 
                                  y1_cg12228123_lm, 
                                  data=covariates_geno_ctpc_tech_cc_y1_sub)

# fit again to get r2
y1_cg12228123_fit <- lm(y1_cg12228123_res$yresid ~ y1_cg12228123_res$xresid)

#summary of partial resid fit
summary(y1_cg12228123_fit)

# partial residual plots
ggplot(y1_cg12228123_res, aes(x=xresid, y=yresid*100)) +
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
  scale_y_continuous(limits = c(-25, 15), breaks = seq(-20, 10, by = 10)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank()) +
  ggsave(filename="2021-05-24_CHILD_y1_cg12228123_AFAP1_PR.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.0, height = 3.2, units = "cm")

# normal plot
ggplot(covariates_geno_ctpc_tech_cc_y1_sub,
       aes(x=no2_preg_entire, y=y1_cg12228123*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
  theme_bw() +
  scale_y_continuous(limits = c(50, 100), breaks = seq(55, 100, by = 10)) +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank()) +
  ggsave(filename="2021-05-24_CHILD_y1_cg12228123_AFAP1.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 3.7, height = 3.2, units = "cm")


# normal plot poster
ggplot(covariates_geno_ctpc_tech_cc_y1_sub,
       aes(x=no2_preg_entire, y=y1_cg12228123*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=3) + 
  stat_smooth(method="lm", se=F, col="red", size=1, linetype="dashed") + 
  theme_bw() +
  scale_y_continuous(limits = c(50, 100), breaks = seq(55, 100, by = 10)) +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=20),
        axis.title = element_blank()) +
  ggsave(filename="2021-06-19_CHILD_y1_cg12228123_AFAP1_poster.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4, height = 3.2, units = "in")


# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_cg12228123, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_preg_entire,
    method="pearson") # -0.2256992
```


### Examine genome wide methylation changes vs no2 exposure

Examine mean methylation vs no2
```{r mean_meth}
#convert betas to df
cb_betas_geno_cc_clean_df <- as.data.frame(cb_betas_geno_cc_clean)

methmeans <- colMeans(cb_betas_geno_cc_clean_df)

covariates_geno_ctpc_tech_cc <- cbind(covariates_geno_ctpc_tech_cc, methmeans)

#colmeans
meanmeth_lm <- 
  lm(methmeans ~ no2_preg_entire + sex + gestational_days + pss_18wk + 
       prenatal_maternal_smoking + maternal_education_length + 
       cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 +
       genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
       sv1 + sv2 + sv3 + sv4 + sv5 + sv6, 
     data=covariates_geno_ctpc_tech_cc)

#get summary
summary(meanmeth_lm) #no2 not sig

#get partial residuals for no2 and dnam 
methmean_res <- partialResid( methmeans ~ no2_preg_entire, meanmeth_lm, 
              data=covariates_geno_ctpc_tech_cc)

#fit again to get r2
methmean_fit <- lm(methmean_res$yresid ~ methmean_res$xresid)

summary(methmean_fit)

# partial residual plot
ggplot(methmean_res, aes(x=xresid, y=yresid*100)) +
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") +
  #scale_y_continuous(limits = c(65, 85), breaks = seq(65, 85, by = 5)) +
  scale_y_continuous(limits = c(-1, 1)) + 
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank()) +
  ggsave(filename="2021-05-24_CHILD_cb_global_dnam_PR.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.2, height = 3.2, units = "cm")


# normal plot
ggplot(covariates_geno_ctpc_tech_cc,
       aes(x=no2_preg_entire, y=methmeans*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
  scale_y_continuous(limits = c(40, 60), breaks = seq(40, 60, by = 5)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank()) +
  ggsave(filename="2021-05-24_CHILD_cb_global_dnam.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.0, height = 3.2, units = "cm")

# correlation 
cor(covariates_geno_ctpc_tech_cc$methmeans, 
    covariates_geno_ctpc_tech_cc$no2_preg_entire,
    method="pearson") # 0.1057361
```


### Examine LINE methylation vs NO2

Examine methylation of LINE1 elements
Based off of analysis by Andrea baccareli
https://link-springer-com.uml.idm.oclc.org/article/10.1186/s12989-014-0071-3
```{r line_annotation, eval=FALSE}

#load repeat element UCSC track
rmsk <- read.delim(file=here("raw_data", "repeat_element_anno", "rmsk.txt"), 
                   sep="\t",
                   header = F)

#populate colnames
colnames(rmsk) <- c("bin", "swscore", "milli_div", "milli_del",
                    "milli_ins", "genoname", "genostart", "genoend",
                    "genoleft", "strand", "repname", "repclass", "repfamily",
                    "repstart", "repend", "repleft", "id")

#get just LINE elements
rmsk_lines <- rmsk %>% filter(repclass=="LINE")

#add column to annotation indicating whether probe has at least 15 basepairs in LINE element

anno_lines <- pbapply(annotation, 1, function(x){
  
  probe_line <- rmsk_lines %>% 
  filter(genoname==x["chr"]) %>% 
  filter(genostart >= x["pos"] & genoend <= x["pos"])

  if(nrow(probe_line)!=0){
    return(TRUE)
  } else {return(FALSE)}
  
  
})

annotation$LINEs <- anno_lines

write.csv(annotation, file=here("output_data", "2020-11-29_annotation_LINEs.csv"))
```


```{r get_lineannotation}
#read in line annotation
annotationlines <- read.csv(here("output_data", "linear_regression", "2020-11-29_annotation_LINEs.csv"))

#subset 
annotationlines <- annotationlines[annotationlines$LINEs==T, ]

#subset betas
cb_betas_geno_cc_clean_df_lines <- 
  cb_betas_geno_cc_clean_df[rownames(cb_betas_geno_cc_clean_df) %in%
                     annotationlines$X, ]

#get size of data frame with line info
dim(cb_betas_geno_cc_clean_df_lines) # 182299 101

# get means of line methylation
linemeth <- colMeans(cb_betas_geno_cc_clean_df_lines)

# cbind line methylation to covariate data frame
covariates_geno_ctpc_tech_cc <- cbind(covariates_geno_ctpc_tech_cc, linemeth)

#colmeans
linemeth_lm <- lm(linemeth ~ no2_preg_entire + 
                    sex + gestational_days + pss_18wk +   
                    prenatal_maternal_smoking + maternal_education_length + 
                    cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 +
                    genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
                    sv1 + sv2 + sv3 + sv4 + sv5 + sv6, 
              data=covariates_geno_ctpc_tech_cc)

# summary of line regression
summary(linemeth_lm) #no2 not sig

# get partial residuals for no2 and dnam 
linemeth_res <- partialResid(linemeth ~ no2_preg_entire, linemeth_lm, 
              data=covariates_geno_ctpc_tech_cc)

# fit again to get r2
linemeth_fit <- lm(linemeth_res$yresid ~ linemeth_res$xresid)

# summary of partial residual fit
summary(linemeth_fit)

# partial residual plot
ggplot(linemeth_res, aes(x=xresid, y=yresid*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
  theme_bw() +
    scale_y_continuous(limits=c(-1,1)) + 
   theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank()) +
  ggsave(filename="2021-05-24_CHILD_cb_line_dnam_PR.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.2, height = 3.2, units = "cm")

# normal plot
ggplot(covariates_geno_ctpc_tech_cc,
       aes(x=no2_preg_entire, y=linemeth*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
  scale_y_continuous(limits = c(40, 60), breaks = seq(40, 60, by = 5)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank()) +
  ggsave(filename="2021-05-24_CHILD_cb_line_dnam.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.0, height = 3.2, units = "cm")

# correlation 
cor(covariates_geno_ctpc_tech_cc$linemeth, 
    covariates_geno_ctpc_tech_cc$no2_preg_entire,
    method="pearson") # 0.09918525
```


### Gene specific changes

These genes are related to the proposed pathway of air pollution action or have previously been shown to exhibit DNA methylation changes associated with prenatal air pollution exposure. 

```{r gene_specific}
#FHC is FTH1 (ferritin heavy chain 1)
#nothing maps to FOXP3
#TNFA is TNF
goi <- annotation[grep("RNF39|CYP2E1|LONP1|HIBADH|SLC25A28|PLVAP|^GPR55$|^CAT$|TPO|NFKB1|
                       ^NOX2|SOD1|TRX1|TRX2|GSTP1|^MT3|NQO1|HMOX1|GPX1|DDH1|
                       ^NOS2|CYP2C11|CYP7B|^ALOX12$|^ALOX5|^NRF2$|KEAP1|AHR$|^ARNT;|GCLC|
                       GCLM|GSR1|SLC7A11|GPC2|GSTA1|GSTA2|GSTA3|GSTA5|GSTM1|GSTM2|
                       GSTM3|TXN$|TXNRD1$|SRXN1|G6PD|CYP1A1|IL1A|^IL2$|^IL3$|^IL4.*IL4$|
                       ^IL5$|^IL6$|IL9$|IL10$|IL12A|IL13|IL15$|IL17A|IL18$|IL19|
                       IL21$|IL22$|IL23A$|IL25$|IL27$|IL31$|IL33$|IFNG$|^TNF$|TGFB1$|
                       FOXP3|FTH1", annotation$UCSC_RefGene_Name), ]

goi <- as.data.frame(goi)
unique(goi$UCSC_RefGene_Name)

# remove pvlap
goi$row <- c(1:nrow(goi))
goi_sub <- goi[-c(902:915), ]
unique(goi_sub$UCSC_RefGene_Name)
# only first annotate gene
 unique(gsub(";.*$", "", unique(goi_sub$UCSC_RefGene_Name))) # 67


# transpose for lm function
cb_betas_geno_cc_clean_t <- t(cb_betas_geno_cc_clean)

#apply function for calculating significance of associations using mice and imputed covariate data
prenatal_goi_list <- 
  pbapply(cb_betas_geno_cc_clean_t[,colnames(cb_betas_geno_cc_clean_t ) %in% rownames(goi_sub)], 2, function(x) {
  
  combined <- cbind(covariates_geno_ctpc_tech_cc, x)
  
  #run the linear regression on this cpg using all 5 imputed covariate data sets
  fit <- lm(x ~ no2_preg_entire + 
              sex + gestational_days + pss_18wk +  
              prenatal_maternal_smoking + maternal_education_length + 
              cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 + 
              genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
              sv1 + sv2 + sv3 + sv4 + sv5 + sv6, 
            data=combined)
  
  #get summary of lm fit
  #this includes coefficients and pvalues
  #coerce into df for manipulation
  #add in terms (rownames) as column for annotation
  fit_sum <- summary(fit)
  fit_sum_coeff <- as.data.frame(fit_sum$coefficients)
  fit_sum_coeff$term <- rownames(fit_sum_coeff)
  
  #this data frame has rows for each linear regression variable, and columns for statistics
  #want a dataframe of 1 row and many columns (one for each variable with each statistic type)
  #do this by using the spread function for each statistic
  #add cpg id onto dataframe for pvalues (to keep track of information)
  #then cbind dataframes together
  
  #coefficient
  estimate <- spread(fit_sum_coeff[,c("term", "Estimate")], term, Estimate)
  colnames(estimate) <- paste("estimate", colnames(estimate), sep=".")
  
  #std error
  std.error <- spread(fit_sum_coeff[,c("term", "Std. Error")], term, 'Std. Error')
  colnames(std.error) <- paste("std.error", colnames(std.error), sep=".")
  
  #t statistics
  statistic <- spread(fit_sum_coeff[,c("term", "t value")], term, 't value')
  colnames(statistic) <- paste("t.val", colnames(statistic), sep=".")

  #pvalue
  p.value <- spread(fit_sum_coeff[,c("term", "Pr(>|t|)")], term, 'Pr(>|t|)')
  colnames(p.value) <- paste("p.value", colnames(p.value), sep=".")
  
  #add in cpgid for annotation 
  p.value$cpg <- colnames(x)
  
  #cbind into wide dataframe
  widepool <- cbind(estimate, std.error, statistic, p.value)
  
  #return the wide dataframe
  #these will be do.call(rbind()) together after
  return(widepool)
  
})


prenatal_goi <- do.call(rbind, prenatal_goi_list)

prenatal_goi$p.value.no2_preg_entire_adjusted <- 
  p.adjust(prenatal_goi$p.value.no2_preg_entire, method="BH")

#reorder according ot unadjusted no2 pvalue
prenatal_goi_order <- prenatal_goi[order(prenatal_goi$p.value.no2_preg_entire),]

#reorder goi annotation to be in same order as goi stats
goi_order <- goi[rownames(prenatal_goi_order),]

#check identical
identical(rownames(goi_order), rownames(prenatal_goi_order)) #true

#bind
goi_df <- cbind(prenatal_goi_order, goi_order)

#sort goi df by pvalue
goi_df_pval <- goi_df %>% arrange(p.value.no2_preg_entire)

# add delta beta
goi_df_pval_db <- merge(goi_df_pval, cb_pvals_tech, by="row.names", all.x=T) 

# write out table for all GOIs
write.csv(goi_df_pval_db, file=here("tables", "prenatal",
                                 "2021-05-18_goi_prenatal_cpgs.csv"))

# write table of top ten 
write.csv(goi_df_pval_db %>% 
            subset(db > 0.025 | db < -0.025) %>% 
            subset(-log10(p.value.no2_preg_entire) > 1.6),
          here("tables", "prenatal", "2021-05-18_top10goi_prenatal_cpgs.csv"))

# use top table to get 95% CI
goi_df_pval_db_sig <- goi_df_pval_db %>% 
            subset(db > 0.025 | db < -0.025) %>% 
            subset(-log10(p.value.no2_preg_entire) > 1.6)

write.csv(goi_df_pval_db_sig, here("tables", "prenatal", "2021-05-18_toptable_prenatal_goi.csv"))

#check cpgs from gruvieva 2017 paper

##################
#LONP1 cg12283362#
##################

covariates_geno_ctpc_tech_cc$cg12283362 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg12283362", ] 

cg12283362_lm <- lm(cg12283362 ~ 
                      no2_preg_entire + sex + gestational_days + pss_18wk + 
                      prenatal_maternal_smoking + maternal_education_length + 
                      cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 + 
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6, 
                    data=covariates_geno_ctpc_tech_cc)

# get lm info
summary(cg12283362_lm)

# get partial residuals for no2 and dnam 
cg12283362_res <- partialResid(cg12283362 ~ no2_preg_entire, cg12283362_lm, 
              data=covariates_geno_ctpc_tech_cc)

# fit again to get r2
cg12283362_res_fit <- lm(cg12283362_res$yresid ~ cg12283362_res$xresid)

# partial residual summary 
summary(cg12283362_res_fit)

# partial residual plot
ggplot(cg12283362_res, aes(x=xresid, y=yresid*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
  scale_y_continuous(limits=c(-10, 10)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank()) +
  ggsave(filename="2021-05-24_CHILD_cg12283362_LONP1_PR.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.0, height = 3.2, units = "cm")


# normal plot
ggplot(covariates_geno_ctpc_tech_cc,
       aes(x=no2_preg_entire, y=cg12283362*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
  scale_y_continuous(limits = c(65, 85), breaks = seq(65, 85, by = 5)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank()) +
  ggsave(filename="2021-05-24_CHILD_cg12283362_LONP1.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.0, height = 3.2, units = "cm")

# correlation 
cor(covariates_geno_ctpc_tech_cc$cg12283362, 
    covariates_geno_ctpc_tech_cc$no2_preg_entire,
    method="pearson") # -0.02974077


#####################
#SLC25A28 cg08973675#
#####################

covariates_geno_ctpc_tech_cc$cg08973675 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg08973675", ] 

cg08973675_lm <- lm(cg08973675~ 
                      no2_preg_entire + sex + gestational_days + pss_18wk + 
                      prenatal_maternal_smoking + maternal_education_length + 
                      cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 + 
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6, 
                    data=covariates_geno_ctpc_tech_cc)

# get partial residuals for no2 and dnam 
cg08973675_res <- partialResid(cg08973675 ~ no2_preg_entire, cg08973675_lm, 
              data=covariates_geno_ctpc_tech_cc)

# fit again to get r2
cg08973675_res_fit <- lm(cg08973675_res$yresid ~ cg08973675_res$xresid)

# partial residual summary
summary(cg08973675_res_fit)

# partial residual plot
ggplot(cg08973675_res, aes(x=xresid, y=yresid*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") +
    scale_y_continuous(limits=c(-10, 10)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank()) +
  ggsave(filename="2021-05-24_CHILD_cg08973675_SLC25A28_PR.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.0, height = 3.2, units = "cm")


# normal plot
ggplot(covariates_geno_ctpc_tech_cc,
       aes(x=no2_preg_entire, y=cg08973675*100)) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  stat_smooth(method="lm", se=F, col="red", size=0.5, linetype="dashed") + 
    scale_y_continuous(limits = c(0, 20), breaks = seq(0, 20, by = 5)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank()) +
  ggsave(filename="2021-05-24_CHILD_cg08973675_SLC25A28.png",
         path =  here("figures", "linear_regression", "prenatal"),
          width = 4.0, height = 3.2, units = "cm")

# correlation 
cor(covariates_geno_ctpc_tech_cc$cg08973675, 
    covariates_geno_ctpc_tech_cc$no2_preg_entire,
    method="pearson") # 0.1007488
```


Try re-running limma with birth month/ interaction between birth month and air pollution.
```{r interaction_terms_to_assess_birthmonth}
#want to use anova to compare models usign anova
#anova only accepts models from lm
#use lm to compare how birth month affects cpgs above 10^-3 sig 

#You should be able to ask for confidence intervals for the slope coefficients for each month.  #even in a tabular display, you may see that âsmoothly varyingâ pattern we were discussing.
#Then, you could use those to plot a sort of âhereâs how the slope varies by monthâ picture.Sort #of like forest plots for meta analysis results, if that makes sense.

#Also, before we decide it doesnât matter, the individual slopes might look fairly similar, but the appropriate overall test would be to check the anova(fit_with_interaction) and check the p-value on the interaction term.

#start by investigating cpgs identified as nominally significant in volcano plot


###########################
#cg12228123 AFAP1/LOC84740#             
###########################

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg12228123 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg12228123", ] 

# fit wiht lm
cg12228123_fit_bm <- lm(cb_cg12228123 ~ 
                          no2_preg_entire + sex + gestational_days + pss_18wk +
                          prenatal_maternal_smoking + maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 + 
                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                          sv1 + sv2 + sv3 + sv4 + sv5 + sv6 +
                          birth_month + no2_preg_entire*birth_month,
                        data=covariates_geno_ctpc_tech_cc)

# anova
anova(cg12228123_fit_bm)

# get plot of birth month effect with 95% CI
tidy(cg12228123_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) +
    
    
  ggsave(filename="2021-05-18_CHILD_cb_cg12228123_AFAP1_birthmonth_aug.png",
         path =  here("figures", "linear_regression", "prenatal"),
        width = 4.8, height = 4, units = "cm")



##################
#cg22908922 PACS2#             
##################

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg22908922 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg22908922", ] 

# fit wiht lm
cg22908922_fit_bm <- lm(cb_cg22908922 ~ 
                          no2_preg_entire + sex + gestational_days + pss_18wk +
                          prenatal_maternal_smoking + maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 + 
                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                          sv1 + sv2 + sv3 + sv4 + sv5 + sv6 +
                          birth_month + no2_preg_entire*birth_month,
                        data=covariates_geno_ctpc_tech_cc)

# anova
anova(cg22908922_fit_bm)

# get plot of birth month effect with 95% CI
tidy(cg22908922_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) +
    
  ggsave(filename="2021-05-18_CHILD_cb_cg22908922_PACS2_birthmonth_aug.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.8, height = 4, units = "cm")



#####################
#cg03392571 C21orf70#             
#####################

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg03392571 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg03392571", ] 

# fit wiht lm
cg03392571_fit_bm <- lm(cb_cg03392571 ~ 
                          no2_preg_entire + sex + gestational_days + pss_18wk +
                          prenatal_maternal_smoking + maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 + 
                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                          sv1 + sv2 + sv3 + sv4 + sv5 + sv6 +
                          birth_month + no2_preg_entire*birth_month,
                        data=covariates_geno_ctpc_tech_cc)

# anova
anova(cg03392571_fit_bm)

# get plot of birth month effect with 95% CI
tidy(cg03392571_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) +
    
    
  ggsave(filename="2021-05-18_CHILD_cb_cg03392571_c21orf70_birthmonth_aug.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.8, height = 4, units = "cm")



#################
#cg08440690 ETFA#             
#################

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg08440690 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg08440690", ] 

# fit wiht lm
cg08440690_fit_bm <- lm(cb_cg08440690 ~ 
                          no2_preg_entire + sex + gestational_days + pss_18wk +
                          prenatal_maternal_smoking + maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 + 
                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                          sv1 + sv2 + sv3 + sv4 + sv5 + sv6 +
                          birth_month + no2_preg_entire*birth_month,
                        data=covariates_geno_ctpc_tech_cc)

# anova
anova(cg08440690_fit_bm)

# get plot of birth month effect with 95% CI
tidy(cg08440690_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) +
    
  ggsave(filename="2021-05-18_CHILD_cb_cg08440690_ETFA_birthmonth_aug.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.8, height = 4, units = "cm")



####################
#cg02243522 MIR1207#             
####################

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg02243522 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg02243522", ] 

# fit wiht lm
cg02243522_fit_bm <- lm(cb_cg02243522 ~ 
                          no2_preg_entire + sex + gestational_days + pss_18wk +
                          prenatal_maternal_smoking + maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 + 
                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                          sv1 + sv2 + sv3 + sv4 + sv5 + sv6 +
                          birth_month + no2_preg_entire*birth_month,
                        data=covariates_geno_ctpc_tech_cc)

# anova
anova(cg02243522_fit_bm)

# get plot of birth month effect with 95% CI
tidy(cg02243522_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) +
    
    
  ggsave(filename="2021-05-18_CHILD_cb_cg02243522_mir1207_birthmonth_aug.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.8, height = 4, units = "cm")



#############
#cg26881997 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg26881997 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg26881997", ] 

# fit wiht lm
cg26881997_fit_bm <- lm(cb_cg26881997 ~ 
                          no2_preg_entire + sex + gestational_days + pss_18wk +
                          prenatal_maternal_smoking + maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 + 
                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                          sv1 + sv2 + sv3 + sv4 + sv5 + sv6 +
                          birth_month + no2_preg_entire*birth_month,
                        data=covariates_geno_ctpc_tech_cc)

# anova
anova(cg26881997_fit_bm)

# get plot of birth month effect with 95% CI
tidy(cg26881997_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) +
    
    
  ggsave(filename="2021-05-18_CHILD_cb_cg26881997_birthmonth_aug.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.8, height = 4, units = "cm")


#############
#cg00679662 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg00679662 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg00679662", ] 

# fit wiht lm
cg00679662_fit_bm <- lm(cb_cg00679662 ~ 
                          no2_preg_entire + sex + gestational_days + pss_18wk +
                          prenatal_maternal_smoking + maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 + 
                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                          sv1 + sv2 + sv3 + sv4 + sv5 + sv6 +
                          birth_month + no2_preg_entire*birth_month,
                        data=covariates_geno_ctpc_tech_cc)

# anova
anova(cg00679662_fit_bm)

# get plot of birth month effect with 95% CI
tidy(cg00679662_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) +
    
    
  ggsave(filename="2021-05-18_CHILD_cb_cg00679662_birthmonth_aug.png",
         path =  here("figures", "linear_regression", "prenatal"),
         width = 4.8, height = 4, units = "cm")


######################
#cg23386895 CDC42EP2 #             
######################

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg23386895<- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg23386895", ] 

# fit wiht lm
cg23386895_fit_bm <- lm(cb_cg23386895~ 
                          no2_preg_entire + sex + gestational_days + pss_18wk +
                          prenatal_maternal_smoking + maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 + 
                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                          sv1 + sv2 + sv3 + sv4 + sv5 + sv6 +
                          birth_month + no2_preg_entire*birth_month,
                        data=covariates_geno_ctpc_tech_cc)

# anova
anova(cg23386895_fit_bm)

# get plot of birth month effect with 95% CI
tidy(cg23386895_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) +
    
  ggsave(filename="2021-05-18_CHILD_cb_cg23386895_CDC42EP2_birthmonth_aug.png",
         path =  here("figures", "linear_regression", "prenatal"),
        width = 4.8, height = 4, units = "cm")



###################
#cg08374494 ZFPM1 #             
###################

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg08374494<- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg08374494", ] 

# fit wiht lm
cg08374494_fit_bm <- lm(cb_cg08374494~ 
                          no2_preg_entire + sex + gestational_days + pss_18wk +
                          prenatal_maternal_smoking + maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 + 
                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                          sv1 + sv2 + sv3 + sv4 + sv5 + sv6 +
                          birth_month + no2_preg_entire*birth_month,
                        data=covariates_geno_ctpc_tech_cc)

# anova
anova(cg08374494_fit_bm)

# get plot of birth month effect with 95% CI
tidy(cg08374494_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) +
    
    
  ggsave(filename="2021-05-18_CHILD_cb_cg08374494_ZFPM1_birthmonth_aug.png",
         path =  here("figures", "linear_regression", "prenatal"),
        width = 4.8, height = 4, units = "cm")


####################
#cg08447934 ACVRL1 #             
####################

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg08447934<- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg08447934", ] 

# fit wiht lm
cg08447934_fit_bm <- lm(cb_cg08447934~ 
                          no2_preg_entire + sex + gestational_days + pss_18wk +
                          prenatal_maternal_smoking + maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + cbmc.PC4 + cbmc.PC5 + 
                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                          sv1 + sv2 + sv3 + sv4 + sv5 + sv6 +
                          birth_month + no2_preg_entire*birth_month,
                        data=covariates_geno_ctpc_tech_cc)

# anova
anova(cg08447934_fit_bm)

# get plot of birth month effect with 95% CI
tidy(cg08447934_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) +
    
    
  ggsave(filename="2021-05-18_CHILD_cb_cg08447934_ACVRL1_birthmonth_aug.png",
         path =  here("figures", "linear_regression", "prenatal"),
        width = 4.8, height = 4, units = "cm")


```

Boxplots of NO2, PM2.5
```{r airpollution_boxplots}

# prenatal no2
ggplot(covariates_geno_ctpc_tech_cc, aes(y=no2_preg_entire, x=studycentre, fill=studycentre)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(height=0, width=0.25, alpha=0.8) +
  theme_bw() +
  scale_fill_manual(values=c("#254A90", "#508FC2", "#92C5EB", "white")) +
  labs(x="Study Centre", y=bquote("Prenatal" ~NO[2]~ "(ppb)")) +
  theme(panel.grid = element_blank(),
        legend.position = "none",
        axis.text = element_text(size=14),
        axis.title = element_text(size=16))

# year one no2
ggplot(covariates_geno_ctpc_tech_cc_y1_sub, aes(y=no2_y1_entire, x=studycentre,fill=studycentre)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(height=0, width=0.25, alpha=0.8) +
  scale_fill_manual(values=c("#254A90", "#508FC2", "#92C5EB", "white")) +
  labs(x="Study Centre", y=bquote("Year one" ~NO[2]~ "(ppb)")) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        legend.position = "none",
        axis.text = element_text(size=14),
        axis.title = element_text(size=16))

# Pm2.5 B
ggplot(covariates_geno_ctpc_tech, aes(y=PM25DALbYY_01, x=studycentre, fill=studycentre)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(height=0, width=0.25, alpha=0.8) +
  scale_fill_manual(values=c("#254A90", "#508FC2", "#92C5EB", "white")) +
  labs(x="Study Centre", y=bquote(~PM[2.5]~ "(ppb)")) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        legend.position = "none",
        axis.text = element_text(size=14),
        axis.title = element_text(size=16))

```


Correlation functionm
```{r corr_functions}

cor2 = function(df){
  
  print("checking dataframe clases")
  stopifnot(inherits(df, "data.frame"))
  stopifnot(sapply(df, class) %in% c("integer"
                                     , "numeric"
                                     , "factor"
                                     , "character"))
  
  print("calculating correlation matrix")
  cor_fun <- function(pos_1, pos_2){
    
    # both are numeric
    if(class(df[[pos_1]]) %in% c("integer", "numeric") &&
       class(df[[pos_2]]) %in% c("integer", "numeric")){
      print("both are numeric")
      print(paste(colnames(df)[[pos_1]], colnames(df)[[pos_2]]))
      r <- stats::cor(df[[pos_1]], df[[pos_2]], use = "pairwise.complete.obs")
    }
    
    #one is numberic the other is factor/character
    #must use sum of squares type 3 because groups are not of the same size
    #calculates ANOVA
    #then calcualtes eta-squared: measures the proportion of the total variance in a      dependent variable that
    #is associated with the membership of different groups defined by an independent      variable.
    #takes square root to get correlation
    if(class(df[[pos_1]]) %in% c("integer", "numeric") &&
       class(df[[pos_2]]) %in% c("factor", "character")){
      
      #print statements for debugging
      print("first is numeric, second is factor")
      print(paste(colnames(df)[[pos_1]], colnames(df)[[pos_2]]))
      
      #check factor levels of complete cases
      print("checking levels of pos_2 factor")
      tmp <- data.frame(df[[pos_1]], df[[pos_2]], row.names = NULL)
      complete <- complete.cases(tmp)
      tmp2 <- tmp[complete,]
      tmp2[,2] <- as.factor(tmp2[,2])
      
      #if statement to catch cases where dropped values result in a factor with
      #one or fewer levels
      if(is.factor(tmp2[,2]) & (nlevels(droplevels(tmp2[,2]))<2)){
        r <- 999
      } else {
        #anova for correlation
        r <- stats::aov(df[[pos_1]] ~ as.factor(df[[pos_2]])) %>%
          lsr::etaSquared(type=3) %>% `[`(1) %>% sqrt()
      }
      rm(tmp,tmp2)
    }
    
    
    if(class(df[[pos_2]]) %in% c("integer", "numeric") &&
       class(df[[pos_1]]) %in% c("factor", "character")){
      
      #print statements for debugging
      print("first is factor, second is numeric")
      print(paste(colnames(df)[[pos_1]], colnames(df)[[pos_2]]))
      
      #check factor levels of complete cases
      print("checking levels of pos_1 factor")
      tmp <- data.frame(df[[pos_1]], df[[pos_2]], row.names = NULL)
      complete <- complete.cases(tmp)
      tmp2 <- tmp[complete,]
      tmp2[,1] <- as.factor(tmp2[,1])
      
      #if statement to catch cases where dropped values result in a factor with
      #one or fewer levels
      if(is.factor(tmp2[,1]) && (nlevels(droplevels(tmp2[,1]))<2)){
        r <- 999
      } else {
        r <- stats::aov(df[[pos_2]] ~ as.factor(df[[pos_1]])) %>%
          lsr::etaSquared(type=3) %>% `[`(1) %>% sqrt()
      }
      rm(tmp,tmp2)
    }
    
    
    #both are factor/character
    if(class(df[[pos_1]]) %in% c("factor", "character") &&
       class(df[[pos_2]]) %in% c("factor", "character")){
      
      #print statement for debugging
      print("both are characters")
      print(paste(colnames(df)[[pos_1]], colnames(df)[[pos_2]]))
      
       #check factor levels of complete cases
      print("checking levels of of both positions")
      tmp <- data.frame(df[[pos_1]], df[[pos_2]], row.names = NULL)
      complete <- complete.cases(tmp)
      tmp2 <- tmp[complete,]
      tmp2[,1] <- as.factor(tmp2[,1])
      tmp2[,2] <- as.factor(tmp2[,2])
      
      #if statement to catch cases where dropped values result in a factor with
      #one or fewer levels
      if(is.factor(tmp2[,1]) && (nlevels(droplevels(tmp2[,1]))<2)){
        r <- 999
      } else {
        if(is.factor(tmp2[,2]) && (nlevels(droplevels(tmp2[,2]))<2)){
          r <-999
        } else { 
          r <- lsr::cramersV(df[[pos_1]], df[[pos_2]], simulate.p.value = TRUE)
        }
      }
      rm(tmp,tmp2)
    }
    return(r)
  } 
  
  cor_fun <- Vectorize(cor_fun)
  
  # now compute corr matrix
  corrmat <- outer(1:ncol(df)
                   , 1:ncol(df)
                   , function(x, y) cor_fun(x, y)
  )
  
  rownames(corrmat) <- colnames(df)
  colnames(corrmat) <- colnames(df)
  return(corrmat)
}


pval2 = function(df){
  
  #check that classes are appropriate
  print("checking dataframe clases")
  stopifnot(inherits(df, "data.frame"))
  stopifnot(sapply(df, class) %in% c("integer"
                                     , "numeric"
                                     , "factor"
                                     , "character"))
  
  print("calculating correlation matrix")
  
  ##################################
  #function for calculating pvalues#
  ##################################
  
  pval_fun <- function(pos_1, pos_2){
    

    ##################
    #both are numeric#
    ##################
    
    if(class(df[[pos_1]]) %in% c("integer", "numeric") &&
       class(df[[pos_2]]) %in% c("integer", "numeric")){
        
      print("both are numeric")
      
      if(colnames(df)[[pos_1]] == colnames(df)[[pos_2]]){
        #dont want to star comparisons of something against itsself
        p <- 1
      } else {
        print(paste(colnames(df)[[pos_1]], colnames(df)[[pos_2]]))
        p <- unname((summary(lm(df[[pos_1]] ~ df[[pos_2]]))$coefficients[,4])[2])
      }
    }
  
    ########################################
    #first is integer and second is numeric#
    ########################################
    
    if(class(df[[pos_1]]) %in% c("integer", "numeric") &&
       class(df[[pos_2]]) %in% c("factor", "character")){
      
      #print statements for debugging
      print("first is numeric, second is factor")
      print(paste(colnames(df)[[pos_1]], colnames(df)[[pos_2]]))
      
      #check factor levels of complete cases
      print("checking levels of pos_2 factor")
      tmp <- data.frame(df[[pos_1]], df[[pos_2]], row.names = NULL)
      complete <- complete.cases(tmp)
      tmp2 <- tmp[complete,]
      tmp2[,2] <- as.factor(tmp2[,2])
      
      #if statement to catch cases where dropped values result in a factor with
      #one or fewer levels
      if(is.factor(tmp2[,2]) & (nlevels(droplevels(tmp2[,2]))<2)){
        p <- 999
      } else {
       if(nlevels(droplevels(tmp2[,2]))>2){
          #anova for p-pval with 3 or  more factors
          lm_p <- lm(df[[pos_1]] ~ as.factor(df[[pos_2]]))
          p <- (anova(lm_p))$`Pr(>F)`[1]
        } else{
          #ttest for 2 factors
          p <- t.test(df[[pos_1]] ~ as.factor(df[[pos_2]]),
                  alternative = "two.sided")$p.value
        }
      }
      rm(tmp,tmp2)
    }
    
    
    ########################################
    #second is integer and first is numeric#
    ########################################
    
    if(class(df[[pos_2]]) %in% c("integer", "numeric") &&
       class(df[[pos_1]]) %in% c("factor", "character")){
      
      #print statements for debugging
      print("first is factor, second is numeric")
      print(paste(colnames(df)[[pos_1]], colnames(df)[[pos_2]]))
      
      #check factor levels of complete cases
      print("checking levels of pos_1 factor")
      tmp <- data.frame(df[[pos_1]], df[[pos_2]], row.names = NULL)
      complete <- complete.cases(tmp)
      tmp2 <- tmp[complete,]
      tmp2[,1] <- as.factor(tmp2[,1])
      
      #if statement to catch cases where dropped values result in a factor with
      #one or fewer levels
      if(is.factor(tmp2[,1]) && (nlevels(droplevels(tmp2[,1]))<2)){
        p <- 999
      } else {
        if(nlevels(droplevels(tmp2[,1]))>2){
          #anova for p-pval with 3 or  more factors
          lm_p <- lm(df[[pos_2]] ~ as.factor(df[[pos_1]]))
          p <- (anova(lm_p))$`Pr(>F)`[1]
        } else{
          #ttest for 2 factors
          p <- t.test(df[[pos_2]] ~ as.factor(df[[pos_1]]),
                  alternative = "two.sided")$p.value
        }
      }
      rm(tmp,tmp2)
    }
    
    
    ###########################
    #both are factor/character#
    ###########################
    
    if(class(df[[pos_1]]) %in% c("factor", "character") &&
       class(df[[pos_2]]) %in% c("factor", "character")){
      
      #print statement for debugging
      print("both are characters")
      print(paste(colnames(df)[[pos_1]], colnames(df)[[pos_2]]))
      
       #check factor levels of complete cases
      print("checking levels of of both positions")
      tmp <- data.frame(df[[pos_1]], df[[pos_2]], row.names = NULL)
      complete <- complete.cases(tmp)
      tmp2 <- tmp[complete,]
      tmp2[,1] <- as.factor(tmp2[,1])
      tmp2[,2] <- as.factor(tmp2[,2])
      
      
      #if statement to catch cases where dropped values result in a factor with
      #one or fewer levels
      if(colnames(df)[[pos_1]] == colnames(df)[[pos_2]]){
        p <- 1
      } else {
        if(is.factor(tmp2[,1]) && (nlevels(droplevels(tmp2[,1]))<2)){
          p <- 999
        } else {
          if(is.factor(tmp2[,2]) && (nlevels(droplevels(tmp2[,2]))<2)){
            p <- 999
          } else {
            #get p value
            fish_table <- table(df[[pos_1]], df[[pos_2]])
            p <- fisher.test(fish_table, simulate.p.value = T)$p.value
          }
        }
      }
      rm(tmp,tmp2)
    }
    return(p)
  } 
  
  pval_fun <- Vectorize(pval_fun)
  
  # now compute corr matrix
  pvalmat <- outer(1:ncol(df)
                   , 1:ncol(df)
                   , function(x, y) pval_fun(x, y)
  )
  
  rownames(pvalmat) <- colnames(df)
  colnames(pvalmat) <- colnames(df)
  return(pvalmat)
}
```



Set up heat scree plot function
```{r heat_plot}

#initialize the heatplot function 
#outputs heatmap of PCs
heat_scree_plot <- function(Loadings, Importance, Num, Order){

  ######### adjust according to importance of first PC #########
  adjust <- 1-Importance[1]
  pca_adjusted <- Importance[2:length(Importance)]/adjust
  pca_df <- data.frame(adjusted_variance = pca_adjusted, 
                      PC = seq(1:length(pca_adjusted)))
  
  ######### Scree plot #########
  
  #plot variance that each adjusted PC accounts for 
  scree <- ggplot(pca_df[which(pca_df$PC<=Num),],aes(PC,adjusted_variance)) + 
    geom_bar(stat = "identity",color="black",fill="grey") +
    theme_bw()+
    theme(axis.text = element_text(size =12),
          axis.title = element_text(size =15),
          plot.margin=unit(c(1,1.5,0.2,2.25),"cm"))+ylab("Variance") +
    scale_x_continuous(breaks = seq(1,Num,1))
  
  ######### Heat map of variance in each variable explained by PCs ######### 
  
  #correlate metadata (variables) with PCS
  #Run anova of each PC on each meta data variable
  
  ### Categorical variables ###
  
  #Run ANOVA on each PC
  aov_PC_meta <- lapply(1:ncol(meta_categorical), 
                        function(covar) sapply(1:ncol(Loadings), 
                        function(PC) summary(aov(Loadings[,PC]~
                                                 meta_categorical[,covar]))[[1]]$"Pr(>F)"[1]))
  
  #set names according to names of categorical variables
  names(aov_PC_meta) <- colnames(meta_categorical)
  #create matrix from list
  aov_PC_meta <- do.call(rbind, aov_PC_meta)
  
  
  ### Continuous variables ### 
  
  #Conduct spearman correlation 
  cor_PC_meta <- lapply(1:ncol(meta_continuous), 
                        function(covar) sapply(1:ncol(Loadings), 
                        function(PC) (cor.test(Loadings[,PC],
                                               as.numeric(meta_continuous[,covar]),
                                               alternative = "two.sided", method="spearman",
                                               na.action=na.omit, exact=FALSE)$p.value)))
  
  #rename according to names of continuous variables
  names(cor_PC_meta)<-colnames(meta_continuous)
  #create matrix from list
  cor_PC_meta<-do.call(rbind, cor_PC_meta)
  
  
  ### Prepare continous and categorical data for heat map ###
  
  #combine as df
  allvar_PC <- as.data.frame(rbind(aov_PC_meta, cor_PC_meta))
  
  #omit first pc to adjust
  allvar_PC_adjust <- allvar_PC[,2:ncol(allvar_PC)]
  
  ### Clean all variable PCs for plotting ###
  
  #plot number of PCs specified by user 
  #reduces number of columns to be equal to "num"
  plotting_PCs <- allvar_PC_adjust[,1:Num]
  
  #convert plotting PCs to numeric
  plotting_PCs_num <- apply(plotting_PCs,2, as.numeric)
  
  #convert plotting PCs to dataframe
  plotting_PCs_num <- as.data.frame(plotting_PCs_num)
  
  #Rename columna to PC 1, 2, 3, etc...
  colnames(plotting_PCs_num) <- sapply(1:Num, function(x) paste("PC",x, sep=""))
  
  #Rename rownames accordiong to variable names (metadata)
  plotting_PCs_num$meta <- rownames(plotting_PCs[1:nrow(plotting_PCs),])
  
  #Melt dataframe to long format for plotting
  plotting_PCs_melt <- reshape2::melt(plotting_PCs_num, id.vars="meta")
  
  #Cluster metadata according to order specified by user
  ord <- Order
  plotting_PCs_order <- unique(plotting_PCs_melt$meta)[rev(ord)]
  plotting_PCs_melt$meta <- factor(plotting_PCs_melt$meta, levels = plotting_PCs_order)
  
  #hard code colours for heat map into dataframe according to PC significance
  plotting_PCs_melt$Pvalue<-sapply(1:nrow(plotting_PCs_melt), function(x)
    if(plotting_PCs_melt$value[x]<=0.001){"<=0.001"}else{
      if(plotting_PCs_melt$value[x]<=0.01){"<=0.01"}else{
        if(plotting_PCs_melt$value[x]<=0.05){"<=0.05"}else{">0.05"}}})
  
  heat <- ggplot(plotting_PCs_melt, aes(variable, meta, fill = Pvalue)) +
    geom_tile(color = "black",size=0.5) +
    theme_gray(8) + 
    scale_fill_manual(breaks = c("<=0.001", "<=0.01", "<=0.05", ">0.05"),
                        values=c("#084594","#4292c6","#9ecae1","#ffffff")) + 
    theme(axis.text = element_text(size =10, color="black"),
          axis.text.x = element_text(),
          axis.title = element_text(size =15),
          legend.text = element_text(size =14),
          legend.title = element_text(size =12),
          legend.position = "bottom",
          plot.margin=unit(c(0,2.25,1,1),"cm"))+
    xlab("Principal Component")+ylab(NULL)
  
  cowplot::plot_grid(scree, heat, ncol=1)
}

```
