---
title: "CHILD cord blood linear regression using limma with covariate missing values filled in with means"
author: "Samantha LEE"
date: "January 13, 2021"
output: html_document
---

*Purpose*: The purpose of this script is to examine DNAm changes in cord blood associated with prenatal air pollution exposure (proxied by no2).


Set global chunk options. 
```{r setup, include=FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Load required libraries.
```{r libraries}
library(tidyverse)
library(limma)
library(here)
library(pbapply) #for running lm for each cpg
library(piecewiseSEM) #for getting partial residuals
library(broom) #for tidy function for lm
```


Load betas
```{r betas}
#load the preprocessed betas
load(file=here("output_data", "preprocessed_data",
               "2021-01-14_CHILD_preprocessed_betas_pdata_annotation.Rdata"))
```


Subset out cord blood betas
```{r nocombat_betas}
# check ranges to confirm DNAm data is in beta value format
dim(betas)
max(betas)
min(betas)

# subset pData into CBMCs (cord blood)
# these children have data at both time points
cb_pdata <- subset(pdata_nodups, Tissue == "C")
dim(cb_pdata)

# subset cordblood DNAM samples
cb_betas <- betas[ ,colnames(betas) %in% rownames(cb_pdata)]

# check size for correct sample num
dim(cb_betas) #144

# check that CBMC betas and pdata are in same order
identical(colnames(cb_betas), rownames(cb_pdata)) # true

# relabel CBMC beta columns with sample label (5 digit identifier)
colnames(cb_betas) <- cb_pdata$Sample_Label
```


Load covariate data. Missing values in covariates of interest have been filled in mean means. 
```{r covariates}
load(file=here("output_data", "covariates", "2020-11-12_CHILD_covariates_na_with_means.Rdata"))
```


Subset covariates for individuals with CBMC DNAm.
```{r sub_covariates}
#subset covariates based on samples with DNAm data
covariates_sub <- 
  covariates_nameans[as.character(covariates_nameans$sampleid) %in% colnames(cb_betas), ]

#check size
dim(covariates_sub) #144

#check if subset of covariates and CBMC betas are in the same order 
identical(as.character(covariates_sub$sampleid), colnames(cb_betas)) #true
```


Load CBMC PCs.
```{r load_PC_correctedbetas}
load(here("output_data", "deconvolution", "2021-02-04_CBMC_PCs.RData"))

dim(cb_pcs) #144 6

#check if rownames of cell pcs are same as pdata
identical(rownames(cb_pcs), rownames(cb_pdata)) #true

#rename rows using sample label (5 digit number)
rownames(cb_pcs) <- cb_pdata$Sample_Label

#rename columns for clarity
colnames(cb_pcs) <- paste("cbmc.", colnames(cb_pcs), sep="")
```



Load genotyping PCs and subset out REEGLE PCs (genotyping data contains PCs for a few thousand participants).
```{r genotyping_PCS}
genotyping <- read.csv(here("input_data", "genotyping", "CHILD_subjects_Qced_first10PCS_Duan.csv"))

#make rownames sample names then rmove this row
rownames(genotyping) <- genotyping$FID
genotyping <- subset(genotyping, select=-c(FID, IID))

genosub <- genotyping[rownames(genotyping) %in% cb_pdata$Sample_Label,]
#missing genotyping data for 14 REEGLE children?

#rename columns of genotyping data for clarity
colnames(genosub) <- paste("genotyping.", colnames(genosub), sep="")

#check size
dim(genosub) #131

#put rows in numerical order according to sample id
genosub_order <- genosub[order(rownames(genosub)), ]


#load genotyping info from Amirtha/Qingling 
pca_info <- readxl::read_excel(here("input_data", "genotyping", "PCA_ethnicity_child_comparison_duan_lab07102020.xlsx"))

#sub for individuals in this study
pca_sub <- pca_info[as.character(pca_info$FID) %in% rownames(genosub),]

#plot and colour according to race
ggplot(pca_sub, aes(x=PC1, y=PC2, 
                    col=as.factor(race_mom), 
                    shape=as.factor(`PCA ethnicity`))) +
  geom_point(alpha=0.7, size=2) +
  scale_color_manual(labels = c("First Nations", "Mixed","South East Asian",
                                 "East Asian","South Asian","Middle Eastern",
                                 "Hispanic","Caucasian White"),
                      values=c("grey55","#4E2A7A" , "#B20D30","#C17817",
                               "#3F784C", "black", "#6DB1BF","#254A90"),
                      name ="Maternal race")  +
  scale_shape_manual(values=c(21,22,23,24),
                     labels=c("Admixed","Central European" , 
                              "East Asian","South Asian"),
                     name ="PCA Genetic Ancestry") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size=6),
        axis.text.y = element_text(size=6),
        axis.title.x = element_text(size=8),
        axis.title.y = element_text(size=8),
        legend.text=element_text(size=4),
        legend.title=element_text(size=5),
        legend.key.size = unit(0.5, 'cm'), #change legend key size
        legend.key.height = unit(0.3, 'cm'), #change legend key height
        legend.key.width = unit(0.5, 'cm')) 

#  ggsave( filename="2021-01-21_CHILD_genopcs_race_ancestry.tiff",
#          device = "tiff",
#          path =  here("figures", "genotyping"),
#          width = 3.5,
#          height = 2.5,
#          units = "in",
#          limitsize = TRUE)
```

Subset betas, pdata, covariates, and cell type PCS for genotyping data
```{r subset}
#betas
cb_betas_geno <- cb_betas[, rownames(genosub_order)]
dim(cb_betas_geno) # 424644    131

#pdata
cb_pdata_geno <- cb_pdata[cb_pdata$Sample_Label %in% rownames(genosub_order), ]
dim(cb_pdata_geno) # 131 17

#covariates
covariates_sub_geno <- covariates_sub[covariates_sub$sampleid %in% rownames(genosub_order), ]
dim(covariates_sub_geno) # 131 114

#genotyping
cb_pcs_geno <- cb_pcs[rownames(genosub_order), ]
dim(cb_pcs_geno) # 131 10
```


Check that betas, pdata, covariates, cell type PCs are in same order.
```{r cb_batchcorr_order}
#pdata and genosub_order
identical(as.character(cb_pdata_geno$Sample_Label),rownames(genosub_order)) #true

#betas and genosub_order
identical(colnames(cb_betas_geno), rownames(genosub_order))

#covariates and genosub_order
identical(as.character(covariates_sub_geno$sampleid), rownames(genosub_order)) #true

#geno and genosub_order
identical(rownames(genosub_order), rownames(cb_pcs_geno)) #true
```


Combined all data together
```{r combine_cell_pcs_covariates}
#create new column for row in pdata
cb_pdata_geno$row <- as.numeric(as.factor(substr(cb_pdata_geno$Sentrix_Position, 1, 3)))
#examine
head(cb_pdata_geno$row)

#rename pdata sentrix Id column to chip
colnames(cb_pdata_geno)[8] <- "chip"

#rename pdata sentrix Id column to chip
colnames(cb_pdata_geno)[11] <- "run"

#add technical columns of pdata to covariates data
covariates_geno_ctpc_tech <- cbind(covariates_sub_geno, 
                                  cb_pcs_geno,
                                  genosub_order,
                                  cb_pdata_geno[,c("run", "row")])

#reclass run 
covariates_geno_ctpc_tech$run <- factor(covariates_geno_ctpc_tech$run, ordered=F)
```




Linear regression with limma.
```{r limma}
#####################
#Data prep for limma#
#####################

# check how many complete cases available for analysis
covariates_geno_ctpc_tech_cc <- 
  covariates_geno_ctpc_tech[complete.cases(covariates_geno_ctpc_tech %>% 
                                             dplyr::select(no2_preg_entire, sex, gestational_days,
                                                           pss_18wk, csed_18wk,
                                                           prenatal_maternal_smoking,
                                                           maternal_education_length, 
                                                           cbmc.PC1, cbmc.PC2, cbmc.PC3, 
                                                           cbmc.PC4, cbmc.PC5, cbmc.PC6,
                                                           genotyping.PC1, genotyping.PC2,
                                                           genotyping.PC3,
                                                           run,row)), ] #101

dim(covariates_geno_ctpc_tech_cc) #101 131

# create model matrix
cb_design <- 
  model.matrix(~no2_preg_entire + as.factor(sex) + gestational_days +
                 pss_18wk + csed_18wk + 
                 as.factor(prenatal_maternal_smoking) + 
                 maternal_education_length + 
                 cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                 cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                 genotyping.PC1 + genotyping.PC2 + 
                 genotyping.PC3 + as.factor(run) + as.numeric(row),
               data=covariates_geno_ctpc_tech_cc)

# check size
dim(cb_design) # 101 18


# subset beta values for complete cases
cb_betas_geno_cc <- cb_betas_geno[,colnames(cb_betas_geno) %in% 
                                    covariates_geno_ctpc_tech_cc$sampleid]

# check size of complete case betas
dim(cb_betas_geno_cc) # 424644 101

# remove probes from beta that were used to predict cell type
cb_betas_geno_cc_clean <- cb_betas_geno_cc[!rownames(cb_betas_geno_cc) %in% 
                                           rownames(cb_deconvolutionprobes$coefEsts), ]
# check size
dim(cb_betas_geno_cc_clean) # 424079    101
# 565 probes removed


#######
#LIMMA#
#######

# fit methylation to no2
cb_fit <-  lmFit(cb_betas_geno_cc_clean, cb_design)
cb_ebayes <- eBayes(cb_fit)

# subset annotation file for cpgs used in limma
annotation_cpgs <- annotation[rownames(annotation) %in% rownames(cb_ebayes),]

# check that annotation file and limma cpgs are in the same order
identical(rownames(annotation_cpgs), rownames(cb_ebayes)) # false

# reorder annotation file to match limma output
annotation_cpgs_order <- annotation_cpgs[rownames(cb_ebayes), ]

# recheck order of annotation file against limma cpgs
identical(rownames(annotation_cpgs_order), rownames(cb_ebayes)) #true

#pull out pval, coefficients from limma and combine with annotation
cb_pvals_tech <- cbind(as.data.frame(cb_ebayes$p.value), 
                  as.data.frame(cb_ebayes$coefficients),
                  as.data.frame(annotation_cpgs_order))

#rename columns to indicate pvalues vs coeffcients 
colnames(cb_pvals_tech) <- c(paste("pval", colnames(cb_pvals_tech)[1:21], sep = "_"), 
                         paste("coeff", colnames(cb_pvals_tech)[22:42], sep = "_"),
                         colnames(cb_pvals_tech)[43:75])

cb_pvals_tech$pval_no2_preg_entire_adj <- p.adjust(cb_pvals_tech$pval_no2_preg_entire, method="BH")
```


Examine qqplot and pvalue histogram.
```{r qqplot_pval_histo}
########
#qqplot#
########

# use code from qqman::qq function 
# plot with ggplot to make it more aesthetically appealling 
pvector <- cb_pvals_tech$pval_no2_preg_entire

pvector2 <- pvector[!is.na(pvector) & !is.nan(pvector) & !is.null(pvector) & 
                      is.finite(pvector) & pvector < 1 & pvector > 0]

pval_qq <- as.data.frame(cbind(o = -log10(sort(pvector2, decreasing = FALSE)),
                               e = -log10(ppoints(length(pvector2)))))

#calculate lambda
chisq <- qchisq(1-pvector,1)
lambda <-  median(chisq)/qchisq(0.5,1)
 
ggplot(pval_qq, aes(x=e, y=o)) + 
  geom_point(col="#254A90", size=1) + 
  geom_abline(slope = 1, intercept = 0, col = "red", size=0.25) +
  labs(x = "Expected -log10(p-value)",
       y = "Observed -log10(p-value)") +
  annotate("text", x=4, y=1, label = paste("λ = ", round(lambda, 2)), size=3) +
  scale_x_continuous(breaks=c(0,2,4,6),
                     labels=c("-0.00", "-2.00", "-4.00", "-6.00")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size=5,angle=90, vjust = 0.5, hjust=1),
        axis.text.y = element_text(size=5),
        axis.title.x = element_text(size=7),
        axis.title.y = element_text(size=7)) +
    ggsave( filename="2021-01-21_CHILD_cb_qqplot.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression"),
          width = 2,
          height = 2,
          units = "in",
          limitsize = TRUE)
 

###########
#histogram#
###########

ggplot( data = cb_pvals_tech, aes(x=pval_no2_preg_entire)) +
  geom_histogram(bins=100, fill = "#254A90") +
  labs(y="Count", x="Unadjusted p-value") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size=6),
        axis.text.y = element_text(size=6),
        axis.title.x = element_text(size=8),
        axis.title.y = element_text(size=8)) +
    ggsave( filename="2021-01-21_CHILD_cb_pval_histogram.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression"),
          width = 3,
          height = 2,
          units = "in",
          limitsize = TRUE)
 
```


Delta beta
```{r db}
range = max(covariates_geno_ctpc_tech_cc$no2_preg_entire) -
    min(covariates_geno_ctpc_tech_cc$no2_preg_entire)

cb_pvals_tech$db <- cb_pvals_tech$coeff_no2_preg_entire * range
```


Volcano plot
```{r volcano_db_tech}

#make new column to label sig points with cpgid
cb_pvals_tech$cpgid <- rownames(cb_pvals_tech)


ggplot(cb_pvals_tech, aes(x=as.numeric(db)*100 , y=-log10(as.numeric(pval_no2_preg_entire)))) +
  geom_point(alpha=0.4, col = "#508FC2", size=3) +
  
  #nominal p value cut off
  geom_hline(yintercept = -log10(0.00001), linetype="dashed", size=0.25) + 
  
  #10% effect size cut-off
  geom_point(data=subset(cb_pvals_tech, 
                         (pval_no2_preg_entire < 0.00001 & db > 0.1) | 
                           (pval_no2_preg_entire < 0.00001 & db < -0.1)), 
             alpha=0.4, col = "red", size=3) + 
  geom_vline(xintercept = c(10, -10), linetype = "dashed", size=0.25) +
  
  # 5% effect size cut off
  geom_point(data=subset(cb_pvals_tech, 
                         (pval_no2_preg_entire < 0.00001 & db > 0.05) | 
                           (pval_no2_preg_entire < 0.00001 & db < -0.05)), 
             alpha=0.4, col = "red", size=3) + 
  geom_vline(xintercept = c(5, -5), linetype = "dashed", size=0.25) +
  
  
  # 2.5% effect size cut off
  geom_point(data=subset(cb_pvals_tech, 
                         (pval_no2_preg_entire < 0.00001 & db > 0.025) | 
                           (pval_no2_preg_entire < 0.00001 & db < -0.025)), 
             alpha=0.4, col = "red", size=3) + 
  geom_vline(xintercept = c(2.5, -2.5), linetype = "dashed", size=0.25) +
  
  #label significant points
 # ggrepel::geom_text_repel(data=subset(cb_pvals_tech, 
#                         (pval_no2_preg_entire < 0.00001 & db > 0.025) | 
#                           (pval_no2_preg_entire < 0.00001 & db < -0.025)),
#                         mapping=aes(x=as.numeric(db)*100 , y=-log10(as.numeric(pval_no2_preg_entire)),
#                                    label=cpgid),
#                         size=6) + 
  
  scale_x_continuous(breaks=c(-25, -10, -5, 0, 5, 10, 25)) +
  labs(y="-log10(unadjusted p-value)", x = "Effect size (methylation %)") +
  
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size=17, angle=90, vjust = 0.5, hjust=1),
        axis.text.y = element_text(size=17),
        axis.title.x = element_text(size=19),
        axis.title.y = element_text(size=19)) +
  
  ggsave( filename="2021-02-04_CHILD_cb_volcanoplot.png",
          device = "png",
          path =  here("figures", "linear_regression"),
          width = 15,
          height = 11,
          units = "cm",
          limitsize = TRUE)
 
```


Plot nominally significant CpGs from volcano plot.
```{r sig_cpgs}


##################
#ROBO1 cg15325658#
##################

#add dnam to covariates for linear regression
covariates_geno_ctpc_tech_cc$cb_cg15325658<- 
        cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg15325658", ]

#linear regression
cb_cg15325658_lm <- lm(cb_cg15325658 ~ no2_preg_entire + sex + gestational_days +
                         pss_18wk + csed_18wk + 
                         prenatal_maternal_smoking + 
                         maternal_education_length + 
                         cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                         cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                         genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + run + row, 
                       data=covariates_geno_ctpc_tech_cc)


#get partial residuals for no2 and dnam 
cb_cg15325658_res <- partialResid(cb_cg15325658 ~ no2_preg_entire, cb_cg15325658_lm, 
              data=covariates_geno_ctpc_tech_cc)

#fit again to get r2
cb_cg15325658_res_fit <- lm(cb_cg15325658_res$yresid ~ cb_cg15325658_res$xresid)

summary(cb_cg15325658_res_fit)

ggplot(cb_cg15325658_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, fill = "#508FC2", shape=21, size=3) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Prenatal" ~NO[2]~ "(ppb) partial residual"), 
       y="Methylation (%) partial residual") +
  annotate("text",x=7,y=0.06, label="slope = -0.09817", size=6) +
  ggtitle("ROBO1 cg15325658") +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=19),
        axis.title = element_text(size=17),
        title = element_text(size=20)) +
  ylim(-0.1, 0.1) +
  xlim(NA, 16) +
  
  ggsave( filename="2021-02-04_CHILD_cb_cg15325658_ROBO1.png",
          device = "png",
          path =  here("figures", "linear_regression"),
          width = 12,
          height = 10,
          units = "cm",
          limitsize = TRUE)
 


###################
#FAM53B cg16404170#
###################

#add dnam to covariates for linear regression
covariates_geno_ctpc_tech_cc$cb_cg16404170<- 
        cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg16404170", ]

#linear regression
cb_cg16404170_lm <- lm(cb_cg16404170 ~ no2_preg_entire + 
                         sex + gestational_days +
                         pss_18wk + csed_18wk + 
                         prenatal_maternal_smoking + 
                         maternal_education_length + 
                         cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                         cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                         genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + run + row, 
                    data=covariates_geno_ctpc_tech_cc)


#get partial residuals for no2 and dnam 
cb_cg16404170_res <- partialResid(cb_cg16404170 ~ no2_preg_entire, cb_cg16404170_lm, 
              data=covariates_geno_ctpc_tech_cc)

#fit again to get r2
cb_cg16404170_res_fit <- lm(cb_cg16404170_res$yresid ~ cb_cg16404170_res$xresid)

summary(cb_cg16404170_res_fit)

ggplot(cb_cg16404170_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.6, fill = "#508FC2", shape=21, size=3) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Prenatal" ~NO[2]~ "(ppb) partial residual"), 
       y="Methylation (%) partial residual") +
  annotate("text",x=7,y=0.06, label="slope = -0.2138", size=6) +
  ggtitle("FAM53B cg16404170") +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=19),
        axis.title = element_text(size=17),
        title = element_text(size=20)) +
  ylim(-0.1, 0.1) +
  xlim(NA, 16) +
  
  ggsave( filename="2021-02-04_CHILD_cb_cg16404170_FAM53B.png",
          device = "png",
          path =  here("figures", "linear_regression"),
          width = 12,
          height = 10,
          units = "cm",
          limitsize = TRUE)




############
#cg20430496#
############

#add dnam to covariates for linear regression
covariates_geno_ctpc_tech_cc$cb_cg20430496<- 
        cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg20430496", ]

#linear regression
cb_cg20430496_lm <- lm(cb_cg20430496 ~ no2_preg_entire + 
                         sex + gestational_days +
                         pss_18wk + csed_18wk + 
                         prenatal_maternal_smoking + 
                         maternal_education_length + 
                         cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                         cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                         genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + run + row, 
                    data=covariates_geno_ctpc_tech_cc)


#get partial residuals for no2 and dnam 
cb_cg20430496_res <- partialResid(cb_cg20430496 ~ no2_preg_entire, cb_cg20430496_lm, 
              data=covariates_geno_ctpc_tech_cc)

#fit again to get r2
cb_cg20430496_res_fit <- lm(cb_cg20430496_res$yresid ~ cb_cg20430496_res$xresid)

summary(cb_cg20430496_res_fit)

ggplot(cb_cg20430496_res, aes(x=xresid, y=yresid)) +
  geom_point(alpha=0.6, fill = "#508FC2", shape=21, size=3) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Prenatal" ~NO[2]~ "(ppb) partial residual"), 
       y="Methylation (%) partial residual") +
  annotate("text",x=7,y=0.06, label="slope = -0.1856", size=6) +
  ggtitle("cg20430496") +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=19),
        axis.title = element_text(size=17),
        title = element_text(size=20)) +
  ylim(-0.1, 0.1) +
  xlim(NA, 16) +
  
  ggsave( filename="2021-02-04_CHILD_cb_cg20430496.png",
          device = "png",
          path =  here("figures", "linear_regression"),
          width = 12,
          height = 10,
          units = "cm",
          limitsize = TRUE)




#overplot residual and raw?
df <- data.frame(no2 = covariates_geno_ctpc_tech_cc$no2_preg_entire,
                 dnam = covariates_geno_ctpc_tech_cc$cb_cg20430496,
                 xresid = cb_cg20430496_res$xresid,
                 yresid = cb_cg20430496_res$yresid)

p1 <- ggplot(df, aes(x=no2, y=dnam)) +
  geom_point(col="black", alpha=0.5) +
  #ylim(-0.1, 0.1) +
  #xlim(-10, 30) +
  theme_bw() + 
    theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=8),
        axis.title = element_text(size=10),
        strip.text.x = element_text(size = 7.5))

p2 <- ggplot(df, aes(x=xresid, y=yresid)) +
  geom_point(col="red", alpha=0.5) +
  #ylim(-0.1, 0.1) +
  #xlim(-10, 30) +
  theme_bw() + 
    theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=8),
        axis.title = element_text(size=10),
        strip.text.x = element_text(size = 7.5))


cowplot::plot_grid(p1, p2, align="h")
```


Examine persistence of these changes at age one.
```{r persistent_changes}

#################
#Age 1 data prep#
#################

#load age one PBMC estimates
load(here("output_data", "deconvolution", "2021-02-04_PBMC_PCs.RData"))

#subset out age one pdata
y1_pdata <- subset(pdata_nodups, Tissue == "P")
#check size
dim(y1_pdata) # 145 17

#subset out PBMC betas 
y1_betas <- betas[ ,colnames(betas) %in% rownames(y1_pdata)]

#check if age one betas and pdata are in same order
identical(colnames(y1_betas), rownames(y1_pdata)) # TRUE

#rename colums with sample label (5 dig num)
colnames(y1_betas) <- y1_pdata$Sample_Label

#rename columns of PBMCs
colnames(y1_pcs) <- paste("pbmc", colnames(y1_pcs), sep=".")

#rename row of cell type PCs to sample label (5 digit num)
identical(rownames(y1_pcs), rownames(y1_pdata)) #true
rownames(y1_pcs) <- y1_pdata$Sample_Label

#subset for complete cases
y1_pcs_sub <- y1_pcs[rownames(y1_pcs) %in% covariates_geno_ctpc_tech_cc$sampleid,]
dim(y1_pcs_sub) #101

# add age one PBMCs to covariate data frame
covariates_geno_ctpc_tech_cc_y1 <- cbind(covariates_geno_ctpc_tech_cc, y1_pcs_sub)

#add age one batch variables to covariate matrix
y1_pdata$Run_y1  <- as.factor(y1_pdata$Run) 
y1_pdata$Row_y1  <- as.factor(substr(y1_pdata$Sentrix_Position,1,3)) 
y1_pdata$Row_y1 <- as.numeric(y1_pdata$Row_y1)

covariates_geno_ctpc_tech_cc_y1 <- cbind(covariates_geno_ctpc_tech_cc_y1, 
                                       y1_pdata[y1_pdata$Sample_Label %in% covariates_geno_ctpc_tech_cc_y1$sampleid, c("Run_y1", "Row_y1")])


#subset covariates with age one cell type PCs for linear regression
covariates_geno_ctpc_tech_cc_y1_sub <- 
  covariates_geno_ctpc_tech_cc_y1[complete.cases(covariates_geno_ctpc_tech_cc_y1 %>% 
                                                 dplyr::select(no2_preg_entire, 
                                                               sex,  
                                                               maternal_smoking_y1,
                                                               maternal_education_length, 
                                                               pss_6mos,
                                                               csed_6mos, 
                                                               pbmc.PC1, pbmc.PC2,
                                                               pbmc.PC3, pbmc.PC4,
                                                               pbmc.PC5,
                                                               genotyping.PC1,
                                                               genotyping.PC2,
                                                               genotyping.PC3,
                                                               Run_y1, Row_y1)), ] 

#check size
dim(covariates_geno_ctpc_tech_cc_y1_sub) # 97

#subset age one betas for samples in the complete case cord blood betas 
y1_betas_cc <- y1_betas[,colnames(y1_betas) %in%
                          as.character(covariates_geno_ctpc_tech_cc_y1_sub$sampleid)]

#check size
dim(y1_betas_cc) #424644    97

#remove probes that were used for PBMC cell type predict from age one betas
y1_betas_cc_sub <- y1_betas_cc[!rownames(y1_betas_cc) %in% 
                                           rownames(y1_deconvolutionprobes$coefEsts), ]

#check size
dim(y1_betas_cc_sub) # 424079    101
# 565 probes removed

#check that covariates and betas in same order
identical(colnames(y1_betas_cc_sub), 
          as.character(covariates_geno_ctpc_tech_cc_y1_sub$sampleid)) # TRUE

####################
#Linear regressions#
####################


##################
#ROBO1 cg15325658#
##################

#add dnam to covariates for linear regression
covariates_geno_ctpc_tech_cc_y1_sub$y1_cg15325658<- 
        y1_betas_cc_sub[rownames(y1_betas_cc_sub)=="cg15325658", ]

#linear regression
y1_cg15325658_lm <- lm(y1_cg15325658 ~ no2_preg_entire + 
                      sex + 
                      maternal_smoking_y1 + 
                      maternal_education_length + 
                      pss_6mos + 
                      csed_6mos + 
                      pbmc.PC1 + pbmc.PC2 +
                      pbmc.PC3 + pbmc.PC4 + pbmc.PC5 +
                      genotyping.PC1 +
                      genotyping.PC2 +
                      genotyping.PC3 +
                      Run_y1 + Row_y1, 
                    data=covariates_geno_ctpc_tech_cc_y1_sub)


#get partial residuals for no2 and dnam 
y1_cg15325658_res <- partialResid(y1_cg15325658 ~ no2_preg_entire, y1_cg15325658_lm, 
              data=covariates_geno_ctpc_tech_cc_y1_sub)

#fit again to get r2
y1_cg15325658_fit <- lm(y1_cg15325658_res$yresid ~ y1_cg15325658_res$xresid)

summary(y1_cg15325658_lm)

ggplot(y1_cg15325658_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.6, fill = "#508FC2", shape=21, size=3) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Prenatal" ~NO[2]~ "(ppb) partial residual"), 
       y="Methylation (%) partial residual") +
  annotate("text",x=7,y=0.06, label="slope = -0.03209", size=6) +
  ggtitle("cg15325658") +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=19),
        axis.title = element_text(size=17),
        title = element_text(size=20)) +
  ylim(-0.1, 0.1) +
  xlim(NA, 16) +
  
  ggsave( filename="2021-02-04_CHILD_y1_cg15325658_ROBO1.png",
          device = "png",
          path =  here("figures", "linear_regression"),
          width = 12,
          height = 10,
          units = "cm",
          limitsize = TRUE)
  
###################
#FAM53B cg16404170#
###################

#add dnam to covariates for linear regression
covariates_geno_ctpc_tech_cc_y1_sub$y1_cg16404170<- 
        y1_betas_cc_sub[rownames(y1_betas_cc_sub)=="cg16404170", ]

#linear regression
y1_cg16404170_lm <- lm(y1_cg16404170 ~ no2_preg_entire + 
                      sex + 
                      maternal_smoking_y1 + 
                      maternal_education_length + 
                      pss_6mos + 
                      csed_6mos + 
                      pbmc.PC1 + pbmc.PC2 +
                      pbmc.PC3 + pbmc.PC4 + pbmc.PC5 +
                      genotyping.PC1 +
                      genotyping.PC2 +
                      genotyping.PC3 +
                      Run_y1 + Row_y1, 
                    data=covariates_geno_ctpc_tech_cc_y1_sub)


#get partial residuals for no2 and dnam 
y1_cg16404170_res <- partialResid(y1_cg16404170 ~ no2_preg_entire, y1_cg16404170_lm, 
              data=covariates_geno_ctpc_tech_cc_y1_sub)

#fit again to get r2
y1_cg16404170_fit <- lm(y1_cg16404170_res$yresid ~ y1_cg16404170_res$xresid)

summary(y1_cg16404170_fit)

ggplot(y1_cg16404170_res, aes(x=xresid, y=yresid)) + 
    geom_point(alpha=0.6, fill = "#508FC2", shape=21, size=3) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Prenatal" ~NO[2]~ "(ppb) partial residual"), 
       y="Methylation (%) partial residual") +
  annotate("text",x=8,y=0.08, label="slope = -0.01538", size=6) +
  ggtitle("cg16404170") +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=19),
        axis.title = element_text(size=17),
        title = element_text(size=20)) +
  ylim(-0.1, 0.1) +
  xlim(NA, 16) +
  
  ggsave( filename="2021-02-04_CHILD_y1_cg16404170_FAM35B.png",
          device = "png",
          path =  here("figures", "linear_regression"),
          width = 12,
          height = 10,
          units = "cm",
          limitsize = TRUE)



############
#cg20430496#
############

#add dnam to covariates for linear regression
covariates_geno_ctpc_tech_cc_y1_sub$y1_cg20430496<- 
        y1_betas_cc_sub[rownames(y1_betas_cc_sub)=="cg20430496", ]

#linear regression
y1_cg20430496_lm <- lm(y1_cg20430496 ~ no2_preg_entire + 
                      sex + 
                      maternal_smoking_y1 + 
                      maternal_education_length + 
                      pss_6mos + 
                      csed_6mos + 
                      pbmc.PC1 + pbmc.PC2 +
                      pbmc.PC3 + pbmc.PC4 + pbmc.PC5 +
                      genotyping.PC1 +
                      genotyping.PC2 +
                      genotyping.PC3 +
                      Run_y1 + Row_y1, 
                    data=covariates_geno_ctpc_tech_cc_y1_sub)


#get partial residuals for no2 and dnam 
y1_cg20430496_res <- partialResid(y1_cg20430496 ~ no2_preg_entire, y1_cg20430496_lm, 
              data=covariates_geno_ctpc_tech_cc_y1_sub)

#fit again to get r2
y1_cg20430496_fit <- lm(y1_cg20430496_res$yresid ~ y1_cg20430496_res$xresid)

summary(y1_cg20430496_fit)

ggplot(y1_cg20430496_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.6, fill = "#508FC2", shape=21, size=3) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Prenatal" ~NO[2]~ "(ppb) partial residual"), 
       y="Methylation (%) partial residual") +
  annotate("text",x=8,y=0.06, label="slope = -0.02204", size=6) +
  ggtitle("cg20430496") +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=19),
        axis.title = element_text(size=17),
        title = element_text(size=20)) +
  ylim(-0.1, 0.1) +
  xlim(NA, 16) + 
  
  ggsave( filename="2021-02-04_CHILD_y1_cg20430496.png",
          device = "png",
          path =  here("figures", "linear_regression"),
          width = 12,
          height = 10,
          units = "cm",
          limitsize = TRUE)
```


### Examine genome wide methylation changes vs no2 exposure

Examine mean methylation vs no2
```{r mean_meth}
#convert betas to df
cb_betas_geno_cc_clean_df <- as.data.frame(cb_betas_geno_cc_clean)

methmeans <- colMeans(cb_betas_geno_cc_clean_df)

covariates_geno_ctpc_tech_cc <- cbind(covariates_geno_ctpc_tech_cc, methmeans)

#colmeans
meanmeth_lm <- 
  lm(methmeans ~ no2_preg_entire + sex + gestational_days +
       pss_18wk + csed_18wk + 
       prenatal_maternal_smoking + 
       maternal_education_length + 
       cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
       cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
       genotyping.PC1 + genotyping.PC2 + 
       genotyping.PC3 + run + row, 
     data=covariates_geno_ctpc_tech_cc)

#get summary
summary(meanmeth_lm) #no2 not sig

#get partial residuals for no2 and dnam 
methmean_res <- partialResid( methmeans ~ no2_preg_entire, meanmeth_lm, 
              data=covariates_geno_ctpc_tech_cc)

#fit again to get r2
methmean_fit <- lm(methmean_res$yresid ~ methmean_res$xresid)

summary(methmean_fit)

ggplot(methmean_res, aes(x=xresid, y=yresid)) +
  geom_point(alpha=0.6, fill = "#254A90", shape=21, size=1.5) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Prenatal" ~NO[2]~ "(ppb) partial residual"), 
       y= str_wrap("Global DNA methylation (%) partial residual", 30)) +
  annotate("text",x=8,y=0.008, label="slope = -0.002011", size=2.25) +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7.5),
        title = element_text(size=7.5)) +
  ylim(-0.01, 0.01) +
  
  ggsave( filename="2021-01-21_CHILD_cb_global_dnam.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression"),
          width = 2.5,
          height = 2,
          units = "in",
          limitsize = TRUE)
```


### Examine LINE methylation vs NO2

Examine methylation of LINE1 elements
Based off of analysis by Andrea baccareli
https://link-springer-com.uml.idm.oclc.org/article/10.1186/s12989-014-0071-3
```{r line_annotation}

#load repeat element UCSC track
rmsk <- read.delim(file=here("raw_data", "repeat_element_anno", "rmsk.txt"), 
                   sep="\t",
                   header = F)

#populate colnames
colnames(rmsk) <- c("bin", "swscore", "milli_div", "milli_del",
                    "milli_ins", "genoname", "genostart", "genoend",
                    "genoleft", "strand", "repname", "repclass", "repfamily",
                    "repstart", "repend", "repleft", "id")

#get just LINE elements
rmsk_lines <- rmsk %>% filter(repclass=="LINE")

#add column to annotation indicating whether probe has at least 15 basepairs in LINE element

anno_lines <- pbapply(annotation, 1, function(x){
  
  probe_line <- rmsk_lines %>% 
  filter(genoname==x["chr"]) %>% 
  filter(genostart >= x["pos"] & genoend <= x["pos"])

  if(nrow(probe_line)!=0){
    return(TRUE)
  } else {return(FALSE)}
  
  
})

annotation$LINEs <- anno_lines

write.csv(annotation, file=here("output_data", "2020-11-29_annotation_LINEs.csv"))
```


```{r get_lineannotation}
#read in line annotation
annotationlines <- read.csv(here("output_data", "linear_regression", "2020-11-29_annotation_LINEs.csv"))

#subset 
annotationlines <- annotationlines[annotationlines$LINEs==T, ]

#subset betas
cb_betas_geno_cc_clean_df_lines <- 
  cb_betas_geno_cc_clean_df[rownames(cb_betas_geno_cc_clean_df) %in%
                     annotationlines$X, ]

dim(cb_betas_geno_cc_clean_df_lines) # 182299 101


linemeth <- colMeans(cb_betas_geno_cc_clean_df_lines)

covariates_geno_ctpc_tech_cc <- cbind(covariates_geno_ctpc_tech_cc, linemeth)

#colmeans
linemeth_lm <- lm(linemeth ~ no2_preg_entire + sex + gestational_days +
                    pss_18wk + csed_18wk + 
                    prenatal_maternal_smoking + 
                    maternal_education_length + 
                    cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                    cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                    genotyping.PC1 + genotyping.PC2 + 
                    genotyping.PC3 + run + row, 
              data=covariates_geno_ctpc_tech_cc)

summary(linemeth_lm) #no2 not sig

#get partial residuals for no2 and dnam 
linemeth_res <- partialResid(linemeth ~ no2_preg_entire, linemeth_lm, 
              data=covariates_geno_ctpc_tech_cc)

#fit again to get r2
linemeth_fit <- lm(linemeth_res$yresid ~ linemeth_res$xresid)

summary(linemeth_fit)

ggplot(linemeth_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.6, fill = "#254A90", shape=21, size=1.5) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Prenatal" ~NO[2]~ "(ppb) partial residual"), 
       y= str_wrap("LINE methylation (%) partial residual", 25)) +
  annotate("text",x=8,y=0.008, label="slope = -0.001954", size=2.25) +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7.5),
        title = element_text(size=7.5)) +
  ylim(-0.01, 0.01) +
  
  ggsave( filename="2021-01-21_CHILD_cb_line_dnam.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression"),
          width = 2.5,
          height = 2,
          units = "in",
          limitsize = TRUE)
```


### Gene specific changes

These genes are related to the proposed pathway of air pollution action or have previously been shown to exhibit DNA methylation changes associated with prenatal air pollution exposure. 

```{r gene_specific}
#FHC is FTH1 (ferritin heavy chain 1)
#nothing maps to FOXP3
#TNFA is TNF
goi <- annotation[grep("RNF39|CYP2E1|LONP1|HIBADH|SLC25A28|PLVAP|GPR55|^CAT$|TPO|NFKB1|^NOX2|SOD1|TRX1|TRX2|GSTP1|^MT3|NQO1|HMOX1|GPX1|DDH1|^NOS2|CYP2C11|CYP7B|ALOX12$|ALOX5|^NRF2$|KEAP1|AHR$|^ARNT;|GCLC|GCLM|GSR1|SLC7A11|GPC2|GSTA1|GSTA2|GSTA3|GSTA5|GSTM1|GSTM2|GSTM3|TXN$|TXNRD1$|SRXN1|G6PD|CYP1A1|IL1A|^IL2$|^IL3$|^IL4.*IL4$|^IL5$|^IL6$|IL9$|IL10$|IL12A|IL13|IL15$|IL17A|IL18$|IL19|IL21$|IL22$|IL23A$|IL25$|IL27$|IL31$|IL33$|IFNG$|^TNF$|TGFB1$|FOXP3|FTH1", annotation$UCSC_RefGene_Name), ]


cb_betas_geno_cc_clean_t <- t(cb_betas_geno_cc_clean)


#apply function for calculating significance of associations using mice and imputed covariate data
prenatal_goi_list <- 
  pbapply(cb_betas_geno_cc_clean_t[,colnames(cb_betas_geno_cc_clean_t ) %in% rownames(goi)], 2, function(x) {
  
  combined <- cbind(covariates_geno_ctpc_tech_cc, x)
  
  #run the linear regression on this cpg using all 5 imputed covariate data sets
  fit <- lm(x ~ no2_preg_entire + sex + gestational_days +
                    pss_18wk + csed_18wk + 
                    prenatal_maternal_smoking + 
                    maternal_education_length + 
                    cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                    cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                    genotyping.PC1 + genotyping.PC2 + 
                    genotyping.PC3 + run + row, data=combined)
  
  #get summary of lm fit
  #this includes coefficients and pvalues
  #coerce into df for manipulation
  #add in terms (rownames) as column for annotation
  fit_sum <- summary(fit)
  fit_sum_coeff <- as.data.frame(fit_sum$coefficients)
  fit_sum_coeff$term <- rownames(fit_sum_coeff)
  
  #this data frame has rows for each linear regression variable, and columns for statistics
  #want a dataframe of 1 row and many columns (one for each variable with each statistic type)
  #do this by using the spread function for each statistic
  #add cpg id onto dataframe for pvalues (to keep track of information)
  #then cbind dataframes together
  
  #coefficient
  estimate <- spread(fit_sum_coeff[,c("term", "Estimate")], term, Estimate)
  colnames(estimate) <- paste("estimate", colnames(estimate), sep=".")
  
  #std error
  std.error <- spread(fit_sum_coeff[,c("term", "Std. Error")], term, 'Std. Error')
  colnames(std.error) <- paste("std.error", colnames(std.error), sep=".")
  
  #t statistics
  statistic <- spread(fit_sum_coeff[,c("term", "t value")], term, 't value')
  colnames(statistic) <- paste("t.val", colnames(statistic), sep=".")

  #pvalue
  p.value <- spread(fit_sum_coeff[,c("term", "Pr(>|t|)")], term, 'Pr(>|t|)')
  colnames(p.value) <- paste("p.value", colnames(p.value), sep=".")
  
  #add in cpgid for annotation 
  p.value$cpg <- colnames(x)
  
  #cbind into wide dataframe
  widepool <- cbind(estimate, std.error, statistic, p.value)
  
  #return the wide dataframe
  #these will be do.call(rbind()) together after
  return(widepool)
  
})


prenatal_goi <- do.call(rbind, prenatal_goi_list)

prenatal_goi$p.value.no2_preg_entire_adjusted <- p.adjust(prenatal_goi$p.value.no2_preg_entire, method="bonferroni")

#reorder according ot unadjusted no2 pvalue
prenatal_goi_order <- prenatal_goi[order(prenatal_goi$p.value.no2_preg_entire),]

#reorder goi annotation to be in same order as goi stats
goi_order <- goi[rownames(prenatal_goi_order),]

#check identical
identical(rownames(goi_order), rownames(prenatal_goi_order)) #true

#bind
goi_df <- cbind(prenatal_goi_order, goi_order)


#check cpgs from gruvieva 2017 paper

##################
#LONP1 cg12283362#
##################

covariates_ctpcs_geno_tech_cc$cg12283362 <- 
  cb_betas_geno_cc_sub[rownames(cb_betas_geno_cc_sub)=="cg12283362", ] 

cg12283362_lm <- lm(cg12283362 ~ no2_preg_entire + sex + gestational_days +
                      pss_18wk + csed_18wk + 
                      prenatal_maternal_smoking + 
                      maternal_education_length + 
                      cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                      cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                      genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + run + row, 
                    data=covariates_ctpcs_geno_tech_cc)

#get partial residuals for no2 and dnam 
cg12283362_res <- partialResid(cg12283362 ~ no2_preg_entire, cg12283362_lm, 
              data=covariates_ctpcs_geno_tech_cc)

#fit again to get r2
cg12283362_res_fit <- lm(cg12283362_res$yresid ~ cg12283362_res$xresid)

summary(cg12283362_res_fit)

ggplot(cg12283362_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.6, fill = "#254A90", shape=21, size=1.5) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Prenatal" ~NO[2]~ "(ppb) partial residual"), 
       y= "Methylation (%) partial residual") +
  ggtitle("LONP1 cg12283362") + 
  annotate("text",x=8,y=0.09, label="slope = 0.04769", size=2.25) +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7.5),
        title = element_text(size=7.5)) +
  ylim(-0.1, 0.1) +
  
  ggsave( filename="2021-01-21_CHILD_cb_cg12283362_LONP1.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression"),
          width = 2.5,
          height = 2,
          units = "in",
          limitsize = TRUE)


###############################
#cg24172570 upstream of HIBADH#
###############################

# this cpg is not in goi_df since it's not mapped to a gene

covariates_ctpcs_geno_tech_cc$cg24172570 <- 
  cb_betas_geno_cc_sub[rownames(cb_betas_geno_cc_sub)=="cg24172570", ] 

cg24172570_lm <- lm(cg24172570~ no2_preg_entire + sex + gestational_days +
                      pss_18wk + csed_18wk + 
                      prenatal_maternal_smoking + 
                      maternal_education_length + 
                      cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                      cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                      genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + run + row, 
              data=covariates_ctpcs_geno_tech_cc)

#get partial residuals for no2 and dnam 
cg24172570_res <- partialResid(cg24172570 ~ no2_preg_entire, cg24172570_lm, 
              data=covariates_ctpcs_geno_tech_cc)

#fit again to get r2
cg24172570_res_fit <- lm(cg24172570_res$yresid ~ cg24172570_res$xresid)

summary(cg24172570_res_fit)

ggplot(cg24172570_res, aes(x=xresid, y=yresid)) +
  geom_point(alpha=0.6, fill = "#254A90", shape=21, size=1.5) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Prenatal" ~NO[2]~ "(ppb) partial residual"), 
       y= "Methylation (%) partial residual") +
  ggtitle("cg24172570") + 
  annotate("text",x=8,y=0.09, label="slope = 0.04615", size=2.25) +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7.5),
        title = element_text(size=7.5)) +
  ylim(-0.1, 0.1) +
  
  ggsave( filename="2021-01-21_CHILD_cb_cg24172570_HIBADH.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression"),
          width = 2.5,
          height = 2,
          units = "in",
          limitsize = TRUE)


#####################
#SLC25A28 cg08973675#
#####################

covariates_ctpcs_geno_tech_cc$cg08973675 <- 
  cb_betas_geno_cc_sub[rownames(cb_betas_geno_cc_sub)=="cg08973675", ] 

cg08973675_lm <- lm(cg08973675~ no2_preg_entire + sex + gestational_days +
                      pss_18wk + csed_18wk + 
                      prenatal_maternal_smoking + 
                      maternal_education_length + 
                      cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                      cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                      genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + run + row, 
              data=covariates_ctpcs_geno_tech_cc)

#get partial residuals for no2 and dnam 
cg08973675_res <- partialResid(cg08973675 ~ no2_preg_entire, cg08973675_lm, 
              data=covariates_ctpcs_geno_tech_cc)

#fit again to get r2
cg08973675_res_fit <- lm(cg08973675_res$yresid ~ cg08973675_res$xresid)

summary(cg08973675_res_fit)

ggplot(cg08973675_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.6, fill = "#254A90", shape=21, size=1.5) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Prenatal" ~NO[2]~ "(ppb) partial residual"), 
       y= "Methylation (%) partial residual") +
  ggtitle("SLC25A28 cg08973675") + 
  annotate("text",x=8,y=0.09, label="slope = 0.04615", size=2.25) +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7.5),
        title = element_text(size=7.5)) +
  ylim(-0.1, 0.1) +
  
  ggsave( filename="2021-01-21_CHILD_cb_cg08973675_SLC25A28.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression"),
          width = 2.5,
          height = 2,
          units = "in",
          limitsize = TRUE)
```





Try re-running limma with birth month/ interaction between birth month and air pollution.
```{r interaction_terms_to_assess_birthmonth}
#want to use anova to compare models usign anova
#anova only accepts models from lm
#use lm to compare how birth month affects cpgs above 10^-3 sig 

#You should be able to ask for confidence intervals for the slope coefficients for each month.  #even in a tabular display, you may see that “smoothly varying” pattern we were discussing.
#Then, you could use those to plot a sort of “here’s how the slope varies by month” picture.Sort #of like forest plots for meta analysis results, if that makes sense.

#Also, before we decide it doesn’t matter, the individual slopes might look fairly similar, but the appropriate overall test would be to check the anova(fit_with_interaction) and check the p-value on the interaction term.

#start by investigating cpgs identified as nominally significant in volcano plot

##################
#ROBO1 cg15325658#
##################

covariates_ctpcs_geno_tech_cc$cg15325658 <- 
  cb_betas_geno_cc_sub[rownames(cb_betas_geno_cc_sub)=="cg15325658", ] 

cg15325658_fit_bm <- lm(cg15325658 ~ no2_preg_entire +
                          sex + gestational_days +
                          pss_18wk + csed_18wk + 
                          prenatal_maternal_smoking + 
                          maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                          cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                          genotyping.PC1 + genotyping.PC2 + 
                          genotyping.PC3 + run + row +
                          birth_month + no2_preg_entire*birth_month, data=covariates_ctpcs_geno_tech_cc)


tidy(cg15325658_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>%
  ggplot(aes(y = estimate, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low , ymax = conf.high)) +
  theme_bw() +
  ggtitle("cg15325658") +
  scale_x_discrete(labels=c("NO2:February","NO2:March", "NO2:April", "NO2:May",
                            "NO2:June","NO2:July", "NO2:August", "NO2:September",
                            "NO2:October", "NO2:November","NO2:December")) +
  labs(x="Interaction term", y="Estimate") +
  theme(panel.border=element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=16)) + 
  ylim(-0.05, 0.05)

anova(cg15325658_fit_bm)
  

###################
#cg16404170 FAM53B#
###################

covariates_ctpcs_geno_tech_cc$cg16404170 <- 
  cb_betas_geno_cc_sub[rownames(cb_betas_geno_cc_sub)=="cg16404170", ] 

  
cg16404170_fit_bm <- lm(cg16404170 ~ no2_preg_entire +
                          sex + gestational_days +
                          pss_18wk + csed_18wk + 
                          prenatal_maternal_smoking + 
                          maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                          cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                          genotyping.PC1 + genotyping.PC2 + 
                          genotyping.PC3 + run + row +
                          birth_month + no2_preg_entire*birth_month, data=covariates_ctpcs_geno_tech_cc)


tidy(cg16404170_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low , ymax = conf.high)) +
  scale_x_discrete(labels=c("NO2:February","NO2:March", "NO2:April", "NO2:May",
                            "NO2:June","NO2:July", "NO2:August", "NO2:September",
                            "NO2:October", "NO2:November","NO2:December")) +
  theme_bw() +
  theme(panel.border=element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=16)) +
  ggtitle("cg16404170") +
    labs(x="Interaction term", y="Estimate") + 
  ylim(-0.05, 0.05)

anova(cg16404170_fit_bm)


############
#cg20430496#
############

covariates_ctpcs_geno_tech_cc$cg20430496 <- 
  cb_betas_geno_cc_sub[rownames(cb_betas_geno_cc_sub)=="cg20430496", ] 

  
cg20430496_fit_bm <- lm(cg20430496 ~ no2_preg_entire +
                          sex + gestational_days +
                          pss_18wk + csed_18wk + 
                          prenatal_maternal_smoking + 
                          maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                          cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                          genotyping.PC1 + genotyping.PC2 + 
                          genotyping.PC3 + run + row +
                          birth_month + no2_preg_entire*birth_month, data=covariates_ctpcs_geno_tech_cc)


tidy(cg20430496_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low , ymax = conf.high)) +
  scale_x_discrete(labels=c("NO2:February","NO2:March", "NO2:April", "NO2:May",
                            "NO2:June","NO2:July", "NO2:August", "NO2:September",
                            "NO2:October", "NO2:November","NO2:December")) +
  theme_bw() +
  theme(panel.border=element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=16)) +
  ggtitle("cg20430496") +
    labs(x="Interaction term", y="Estimate") + 
  ylim(-0.05, 0.05)

anova(cg20430496_fit_bm)



### Now examine some otherinteresting CpGs

##################
#cg07376282 GPR37#
##################

# cg07376282 is CpG with largest postive DB and smallest p-val

covariates_ctpcs_geno_tech_cc$cg07376282 <- 
  cb_betas_geno_cc_sub[rownames(cb_betas_geno_cc_sub)=="cg07376282", ] 

  
cg07376282_fit_bm <- lm(cg07376282 ~ no2_preg_entire +
                          sex + gestational_days +
                          pss_18wk + csed_18wk + 
                          prenatal_maternal_smoking + 
                          maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                          cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                          genotyping.PC1 + genotyping.PC2 + 
                          genotyping.PC3 + run + row +
                          birth_month + no2_preg_entire*birth_month, data=covariates_ctpcs_geno_tech_cc)


tidy(cg07376282_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low , ymax = conf.high)) +
  scale_x_discrete(labels=c("NO2:February","NO2:March", "NO2:April", "NO2:May",
                            "NO2:June","NO2:July", "NO2:August", "NO2:September",
                            "NO2:October", "NO2:November","NO2:December")) +
  theme_bw() +
  theme(panel.border=element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=16)) +
  ggtitle("cg07376282") +
    labs(x="Interaction term", y="Estimate") 

anova(cg07376282_fit_bm)


### CpGs with p value < 0.0001 and db > 0.1 or < -0.1

###################
#cg23755933 GALGA3#             
###################

covariates_ctpcs_geno_tech_cc$cg23755933 <- 
  cb_betas_geno_cc_sub[rownames(cb_betas_geno_cc_sub)=="cg23755933", ] 

  
cg23755933_fit_bm <- lm(cg23755933 ~ no2_preg_entire +
                          sex + gestational_days +
                          pss_18wk + csed_18wk + 
                          prenatal_maternal_smoking + 
                          maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                          cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                          genotyping.PC1 + genotyping.PC2 + 
                          genotyping.PC3 + run + row +
                          birth_month + no2_preg_entire*birth_month, data=covariates_ctpcs_geno_tech_cc)


tidy(cg23755933_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low , ymax = conf.high)) +
  scale_x_discrete(labels=c("NO2:February","NO2:March", "NO2:April", "NO2:May",
                            "NO2:June","NO2:July", "NO2:August", "NO2:September",
                            "NO2:October", "NO2:November","NO2:December")) +
  theme_bw() +
  theme(panel.border=element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=16)) +
  ggtitle("cg23755933") +
    labs(x="Interaction term", y="Estimate") 

anova(cg23755933_fit_bm)

############
#cg27183818#             
############

covariates_ctpcs_geno_tech_cc$cg27183818 <- 
  cb_betas_geno_cc_sub[rownames(cb_betas_geno_cc_sub)=="cg27183818", ] 

  
cg27183818_fit_bm <- lm(cg27183818 ~ no2_preg_entire +
                          sex + gestational_days +
                          pss_18wk + csed_18wk + 
                          prenatal_maternal_smoking + 
                          maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                          cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                          genotyping.PC1 + genotyping.PC2 + 
                          genotyping.PC3 + run + row +
                          birth_month + no2_preg_entire*birth_month, data=covariates_ctpcs_geno_tech_cc)


tidy(cg27183818_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low , ymax = conf.high)) +
  scale_x_discrete(labels=c("NO2:February","NO2:March", "NO2:April", "NO2:May",
                            "NO2:June","NO2:July", "NO2:August", "NO2:September",
                            "NO2:October", "NO2:November","NO2:December")) +
  theme_bw() +
  theme(panel.border=element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=16)) +
  ggtitle("cg27183818") +
    labs(x="Interaction term", y="Estimate") 

anova(cg27183818_fit_bm)


###################
#cg02734358 GPRIN3#             
###################

covariates_ctpcs_geno_tech_cc$cg02734358 <- 
  cb_betas_geno_cc_sub[rownames(cb_betas_geno_cc_sub)=="cg02734358", ] 

  
cg02734358_fit_bm <- lm(cg02734358 ~ no2_preg_entire +
                          sex + gestational_days +
                          pss_18wk + csed_18wk + 
                          prenatal_maternal_smoking + 
                          maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                          cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                          genotyping.PC1 + genotyping.PC2 + 
                          genotyping.PC3 + run + row +
                          birth_month + no2_preg_entire*birth_month, data=covariates_ctpcs_geno_tech_cc)


tidy(cg02734358_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low , ymax = conf.high)) +
  scale_x_discrete(labels=c("NO2:February","NO2:March", "NO2:April", "NO2:May",
                            "NO2:June","NO2:July", "NO2:August", "NO2:September",
                            "NO2:October", "NO2:November","NO2:December")) +
  theme_bw() +
  theme(panel.border=element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=16)) +
  ggtitle("cg02734358") +
    labs(x="Interaction term", y="Estimate") 

anova(cg02734358_fit_bm)

###########################
#cg12228123 AFAP1/LOC84740#             
###########################

covariates_ctpcs_geno_tech_cc$cg12228123 <- 
  cb_betas_geno_cc_sub[rownames(cb_betas_geno_cc_sub)=="cg12228123", ] 

  
cg12228123_fit_bm <- lm(cg12228123 ~ no2_preg_entire +
                          sex + gestational_days +
                          pss_18wk + csed_18wk + 
                          prenatal_maternal_smoking + 
                          maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                          cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                          genotyping.PC1 + genotyping.PC2 + 
                          genotyping.PC3 + run + row +
                          birth_month + no2_preg_entire*birth_month, data=covariates_ctpcs_geno_tech_cc)


tidy(cg12228123_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low , ymax = conf.high)) +
  scale_x_discrete(labels=c("NO2:February","NO2:March", "NO2:April", "NO2:May",
                            "NO2:June","NO2:July", "NO2:August", "NO2:September",
                            "NO2:October", "NO2:November","NO2:December")) +
  theme_bw() +
  theme(panel.border=element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=16)) +
  ggtitle("cg12228123") +
    labs(x="Interaction term", y="Estimate") 

anova(cg12228123_fit_bm)


# CpGs with p-val < 0.000029 and delta beta > 0.05 and < 0.1 or < -0.1 and > -0.05

###################
#cg05732646 KBTBD6#             
###################

covariates_ctpcs_geno_tech_cc$cg05732646 <- 
  cb_betas_geno_cc_sub[rownames(cb_betas_geno_cc_sub)=="cg05732646", ] 

  
cg05732646_fit_bm <- lm(cg05732646 ~ no2_preg_entire +
                          sex + gestational_days +
                          pss_18wk + csed_18wk + 
                          prenatal_maternal_smoking + 
                          maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                          cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                          genotyping.PC1 + genotyping.PC2 + 
                          genotyping.PC3 + run + row +
                          birth_month + no2_preg_entire*birth_month, data=covariates_ctpcs_geno_tech_cc)


tidy(cg05732646_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low , ymax = conf.high)) +
  scale_x_discrete(labels=c("NO2:February","NO2:March", "NO2:April", "NO2:May",
                            "NO2:June","NO2:July", "NO2:August", "NO2:September",
                            "NO2:October", "NO2:November","NO2:December")) +
  theme_bw() +
  theme(panel.border=element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=16)) +
  ggtitle("cg05732646") +
    labs(x="Interaction term", y="Estimate") 

anova(cg05732646_fit_bm)


#################
#cg20701183 LMO4#             
#################

covariates_ctpcs_geno_tech_cc$cg20701183 <- 
  cb_betas_geno_cc_sub[rownames(cb_betas_geno_cc_sub)=="cg20701183", ] 

  
cg20701183_fit_bm <- lm(cg20701183 ~ no2_preg_entire +
                          sex + gestational_days +
                          pss_18wk + csed_18wk + 
                          prenatal_maternal_smoking + 
                          maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                          cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                          genotyping.PC1 + genotyping.PC2 + 
                          genotyping.PC3 + run + row +
                          birth_month + no2_preg_entire*birth_month, data=covariates_ctpcs_geno_tech_cc)


tidy(cg20701183_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low , ymax = conf.high)) +
  scale_x_discrete(labels=c("NO2:February","NO2:March", "NO2:April", "NO2:May",
                            "NO2:June","NO2:July", "NO2:August", "NO2:September",
                            "NO2:October", "NO2:November","NO2:December")) +
  theme_bw() +
  theme(panel.border=element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=16)) +
  ggtitle("cg20701183") +
    labs(x="Interaction term", y="Estimate") 

anova(cg20701183_fit_bm)



###################
#cg03783907 BTN3A3#             
###################

covariates_ctpcs_geno_tech_cc$cg03783907 <- 
  cb_betas_geno_cc_sub[rownames(cb_betas_geno_cc_sub)=="cg03783907", ] 

  
cg03783907_fit_bm <- lm(cg03783907 ~ no2_preg_entire +
                          sex + gestational_days +
                          pss_18wk + csed_18wk + 
                          prenatal_maternal_smoking + 
                          maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                          cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                          genotyping.PC1 + genotyping.PC2 + 
                          genotyping.PC3 + run + row +
                          birth_month + no2_preg_entire*birth_month, data=covariates_ctpcs_geno_tech_cc)


tidy(cg03783907_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low , ymax = conf.high)) +
  scale_x_discrete(labels=c("NO2:February","NO2:March", "NO2:April", "NO2:May",
                            "NO2:June","NO2:July", "NO2:August", "NO2:September",
                            "NO2:October", "NO2:November","NO2:December")) +
  theme_bw() +
  theme(panel.border=element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=16)) +
  ggtitle("cg03783907") +
    labs(x="Interaction term", y="Estimate") 

anova(cg03783907_fit_bm)


##################
#cg07645296 ITPR2#             
##################

covariates_ctpcs_geno_tech_cc$cg07645296 <- 
  cb_betas_geno_cc_sub[rownames(cb_betas_geno_cc_sub)=="cg07645296", ] 

  
cg07645296_fit_bm <- lm(cg07645296 ~ no2_preg_entire +
                          sex + gestational_days +
                          pss_18wk + csed_18wk + 
                          prenatal_maternal_smoking + 
                          maternal_education_length + 
                          cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
                          cbmc.PC4 + cbmc.PC5 + cbmc.PC6 + 
                          genotyping.PC1 + genotyping.PC2 + 
                          genotyping.PC3 + run + row +
                          birth_month + no2_preg_entire*birth_month, data=covariates_ctpcs_geno_tech_cc)


tidy(cg07645296_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low , ymax = conf.high)) +
  scale_x_discrete(labels=c("NO2:February","NO2:March", "NO2:April", "NO2:May",
                            "NO2:June","NO2:July", "NO2:August", "NO2:September",
                            "NO2:October", "NO2:November","NO2:December")) +
  theme_bw() +
  theme(panel.border=element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=16)) +
  ggtitle("cg07645296") +
    labs(x="Interaction term", y="Estimate") 

anova(cg07645296_fit_bm)
```
