---
title: "CHILD cord blood linear regression using limma with covariate missing values filled in with means"
author: "Samantha LEE"
date: "2021-09-24"
output: html_document
---

*Purpose*: The purpose of this script is to examine DNAm changes in cord blood associated with prenatal air pollution exposure (proxied by no2).


Set global chunk options. 
```{r setup, include=FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

### Libraries used in analysis

Load required libraries.
```{r libraries}
library(tidyverse) # be tidy
library(limma) # for DNAm differences
library(here) # file calling
library(pbapply) # for running lm for each cpg
library(ggeffects) # for estimating marginal effects
library(broom) # tidy function for summarizing lm
library(sva) # for calculating SVs
library(ggcorrplot) # correlation matrix
library(ggsignif) # for significance comparison bars
library(GenomicRanges) # for data formatting prior to DMR analysis
library(IRanges) # for data formatting prior to DMR analysis
library(ENmix) # for DMR identification using comb-p
library(missMethyl) # gene set enrichment of DMRs
library(lme4) # linear mixed models
library(lmerTest) # get pvalues for linear mixed models
library(Gviz) # plot gene annotations and ChromHMM data
library(org.Hs.eg.db) # get gene annotation from GVIZ figs
library(mediation) # causal mediation 
```


## Analysis of prenatal air pollution exposure

### Load data

Load betas
```{r betas}
load(file=here("output_data", 
               "preprocessed_data",
               "2023-06-28_CHILD_preprocessed_betas_pdata_annotation.Rdata"))
```

Subset out cord blood betas
```{r nocombat_betas}
# check ranges to confirm DNAm data is in beta value format
dim(betas)
max(betas)
min(betas)

# subset pData into CBMCs (cord blood)
# these children have data at both time points
cb_pdata <- subset(pdata_nodups, Tissue == "C")
# check size of cord blood pdata
dim(cb_pdata)

# subset cordb lood DNAm samples for those included in cord blood pdata
cb_betas <- betas[ ,colnames(betas) %in% rownames(cb_pdata)]

# check size for correct sample number
dim(cb_betas) #144

# check that CBMC betas and pdata are in same order
identical(colnames(cb_betas), rownames(cb_pdata)) # TRUE

# relabel CBMC beta columns with sample label (5 digit identifier)
colnames(cb_betas) <- cb_pdata$Sample_Label
```

Load covariate data. Missing values in covariates of interest have been filled with the mean of that variable. 
```{r covariates}
load(here("output_data", 
          "covariates", 
          "2023-06-28_CHILD_covariates_na_with_means.Rdata"))

dim(covariates_nameans)
```

Subset covariates for individuals with CBMC DNAm.
```{r sub_covariates}
# subset covariates based on samples with DNAm data
covariates_sub <- covariates_nameans[as.character(covariates_nameans$sample_id) 
                                     %in% colnames(cb_betas), ]

# check size
dim(covariates_sub) # 144 120  

# check if subset of covariates and betas are in the same order 
identical(as.character(covariates_sub$sample_id), 
          colnames(cb_betas)) # TRUE
```

Add sample name to cord blood covariate data
```{r add_cb_sample_name}
identical(as.numeric(covariates_sub$sample_id),
          as.numeric(cb_pdata$Sample_Label)) # TRUE

covariates_sub$cb_sample_name <- rownames(cb_pdata)
```


Load CBMC PCs.
```{r load_PC_correctedbetas}
# load the cell type PCs
load(here("output_data", 
          "deconvolution", 
          "2023-08-14_CBMC_PCs.RData"))

# load deconvolution for later use - remove probes 
load(here("output_data",
          "deconvolution", 
          "2023-06-28_cbmc_pbmc_deconvolution.Rdata"))

# check size
dim(cb_ilr_pca$scores) #144 6

# pull out pcs
cb_pcs <- cb_ilr_pca$scores

# check if rownames of cell PCs are same as pdata
identical(rownames(cb_pcs), 
          rownames(cb_pdata)) # TRUE
# cb pcs do not have rownames

# rename rows using sample label (5 digit number)
rownames(cb_pcs) <- cb_pdata$Sample_Label
```

Load genotyping PCs and subset out REEGLE PCs. Genotyping data contains PCs for all CHILD participants that were genotyped.
```{r genotyping_PCS}
# read in genotyping data for CHILD cohort
genotyping <- read.csv(here("input_data", 
                            "genotyping", 
                            "CHILD_subjects_Qced_first10PCS_Duan.csv"))

# make rownames sample names (5 digit number)
rownames(genotyping) <- genotyping$FID
# removing sample naming columns - no longer needed now they are rownames 
genotyping <- subset(genotyping, select=-c(FID, IID))

# susbet genotyping data for REEGLE subset 
genosub <- genotyping[rownames(genotyping) %in% cb_pdata$Sample_Label,]

# check size
dim(genosub) # 131 of 144 children have genotyping

# rename columns of genotyping data for clarity
colnames(genosub) <- paste("genotyping.", colnames(genosub), sep="")

write.csv(genosub, 
          here("tables",
               "2023-08-14_genotyping_pcs.csv"))

#put rows in numerical order according to sample id
genosub_order <- genosub[order(rownames(genosub)), ]
```

Examine genetic ancestry groups of participants based on genotyping PCs. Want to confirm that three PCs are required for genetic ancestry.
```{r genotyping_pc_groups, eval=FALSE}
# load genotyping info from Amirtha/Qingling 
# this info is for all CHILD participants that were genotyped
pca_info <- readxl::read_excel(here("input_data",
                                    "genotyping", 
                                    "PCA_ethnicity_child_comparison_duan_lab07102020.xlsx"))

# sub for individuals in this study (REEGLE subset)
pca_sub <- pca_info[as.character(pca_info$FID) %in% rownames(genosub),]

# plot and colour according to race specified on questionnaires
geno_pc1_pc2 <- ggplot(pca_sub, aes(x=PC1, y=PC2, 
                                    col=as.factor(race_mom), 
                                    shape=as.factor(`PCA ethnicity`))) +
  geom_point(alpha=0.7, size=2) +
  scale_color_manual(labels = c("First Nations", "Mixed","South East Asian",
                                "East Asian","South Asian","Middle Eastern",
                                "Hispanic","Caucasian White"),
                     values=c("grey55","#4E2A7A" , "#B20D30","#C17817",
                              "#3F784C", "black", "#6DB1BF","#254A90"),
                     name ="Maternal race")  +
  scale_shape_manual(values=c(21,22,23,24),
                     labels=c("Admixed","Central European" , 
                              "East Asian","South Asian"),
                     name ="PCA Genetic Ancestry") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        axis.title.x = element_text(size=12),
        axis.title.y = element_text(size=12),
        legend.text=element_text(size=9),
        legend.title=element_text(size=9),
        legend.key.size = unit(1, 'cm'), #change legend key size
        legend.key.height = unit(0.5, 'cm'), #change legend key height
        legend.key.width = unit(1, 'cm'),
        legend.position = "none") +
  labs(x="Genotyping PC1", y = "Genotyping PC2")

ggsave(plot = geno_pc1_pc2, 
       filename="2023-06-28_CHILD_genopcs_race_ancestry_PC1_PC2.png",
       device = "png",
       path =  here("figures", "genotyping"),
       width = 6,
       height = 6,
       units = "cm")


# plot and colour according to race specified on questionnaires
geno_pc1_pc3 <- ggplot(pca_sub, aes(x=PC1, y=PC3, 
                                    col=as.factor(race_mom), 
                                    shape=as.factor(`PCA ethnicity`))) +
  geom_point(alpha=0.7, size=2) +
  scale_color_manual(labels = c("First Nations", "Mixed","South East Asian",
                                "East Asian","South Asian","Middle Eastern",
                                "Hispanic","Caucasian White"),
                     values=c("grey55","#4E2A7A" , "#B20D30","#C17817",
                              "#3F784C", "black", "#6DB1BF","#254A90"),
                     name ="Maternal race")  +
  scale_shape_manual(values=c(21,22,23,24),
                     labels=c("Admixed","Central European" , 
                              "East Asian","South Asian"),
                     name ="PCA Genetic Ancestry") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        axis.title.x = element_text(size=12),
        axis.title.y = element_text(size=12),
        legend.text=element_text(size=9),
        legend.title=element_text(size=9),
        legend.key.size = unit(1, 'cm'), #change legend key size
        legend.key.height = unit(0.5, 'cm'), #change legend key height
        legend.key.width = unit(1, 'cm'), 
        legend.position = "right") +
  labs(x="Genotyping PC1", y = "Genotyping PC3")

ggsave(plot = geno_pc1_pc3, 
       filename="2023-06-28_CHILD_genopcs_race_ancestry_PC1_PC3_legend.png",
       device = "png",
       path =  here("figures", "genotyping"),
       width = 6,
       height = 12,
       units = "cm")
```

Subset betas, pdata, covariates, and cell type PCS for participants with genotyping data (the limiting factor).
```{r subset}
# subset betas
cb_betas_geno <- cb_betas[, rownames(genosub_order)]
# check size of subbed betas
dim(cb_betas_geno) # 424644    131

# subset pdata
cb_pdata_geno <- cb_pdata[cb_pdata$Sample_Label %in% rownames(genosub_order), ]
# check size of subbed pdata
dim(cb_pdata_geno) # 131 17

# subset covariates
covariates_sub_geno <- covariates_sub[covariates_sub$sample_id %in% 
                                        rownames(genosub_order), ]
# check size of subbed covariates 
dim(covariates_sub_geno) # 131 120

# subset cell type PCs
cb_pcs_geno <- cb_pcs[rownames(genosub_order), ]
# check size of subbed cell type PCs
dim(cb_pcs_geno) # 131 6
```

Check that betas, pdata, covariates, cell type PCs are in same order as genotyping data for REEGLE participants.
```{r cb_batchcorr_order}
# pdata and genosub_order
identical(as.character(cb_pdata_geno$Sample_Label),
          rownames(genosub_order)) # TRUE

# betas and genosub_order
identical(colnames(cb_betas_geno), 
          rownames(genosub_order)) # TRUE

# covariates and genosub_order
identical(as.character(covariates_sub_geno$sample_id), 
          rownames(genosub_order)) # TRUE

# cell type PCs and genosub_order
identical(rownames(genosub_order), 
          rownames(cb_pcs_geno)) # TRUE
```

Combined covariates, genotyping, cell type PCs, and pData together.
```{r combine_cell_pcs_covariates}

###################################
# clean up pdata before combining #
###################################

# create new column for batch row variable in pdata
cb_pdata_geno$row <- as.numeric(as.factor(substr(cb_pdata_geno$Sentrix_Position,
                                                 1, 3)))
# examine
head(cb_pdata_geno$row)

# rename pdata sentrix Id column to chip
colnames(cb_pdata_geno)[8] <- "chip"

# rename pdata sentrix Id column to chip
colnames(cb_pdata_geno)[11] <- "run"

################################
# combine data frames together #
################################

# for pdata, only include batch variables
covariates_geno_ctpc_tech <- cbind(covariates_sub_geno, 
                                   cb_pcs_geno,
                                   genosub_order,
                                   cb_pdata_geno[,c("run", "row", "chip")])

dim(covariates_geno_ctpc_tech) # 131 138

# check classes to make sure they are what we expect before running linear model
lapply(covariates_geno_ctpc_tech, class)

# reclass run 
covariates_geno_ctpc_tech$run <- factor(covariates_geno_ctpc_tech$run, 
                                        ordered=F)
# reclass chip
covariates_geno_ctpc_tech$chip <- factor(covariates_geno_ctpc_tech$chip,
                                         ordered=F)
# reclass sex
covariates_geno_ctpc_tech$sex <- factor(covariates_geno_ctpc_tech$sex, 
                                        ordered=F)
```

There should be no zeroes in NO2 data - double check
```{r no_zeroes_in_no2}
# newest no2 variable
sum(na.omit(covariates_geno_ctpc_tech$no2_prenatal_avg_2019v)==0)
# old no2 variable
sum(na.omit(covariates_geno_ctpc_tech$no2_prenatal_avg_2015v)==0) # 6
```


### Subset for complete cases

Subset complete cases in covariate data and betas in preparation for DNAm analysis.
```{r complete_cases}
# subset complete cases
covariates_geno_ctpc_tech_cc <- covariates_geno_ctpc_tech %>%
  subset(complete.cases(no2_prenatal_avg_2019v, sex, gestational_age,
                        prenatal_maternal_smoke, maternal_education_length,
                        Comp.1, Comp.2, Comp.3, Comp.4, 
                        genotyping.PC1, genotyping.PC2, genotyping.PC3))

# check number of complete cases
dim(covariates_geno_ctpc_tech_cc) # 128 139

# subset beta values for complete cases
cb_betas_geno_cc <- cb_betas_geno[,colnames(cb_betas_geno) %in% 
                                    covariates_geno_ctpc_tech_cc$sample_id]

# check size of complete case betas
dim(cb_betas_geno_cc) # 424644 128

# remove probes from betas that were used to predict cell type
cb_betas_geno_cc_clean <- 
  cb_betas_geno_cc[!rownames(cb_betas_geno_cc) %in%
                     (child_fscbc_ecc2_cb$probelist_all %>% unlist() %>% unname() %>% unique()), ]

# check size
dim(cb_betas_geno_cc_clean) # 424079 128; 565 probes removed
```

### Model matrix

Create model matrix for DNAm analysis. This model matrix does not contain batch variables as we are going to let surrogate variables capture their variance within the DNAm data. 
```{r model_matrix}
# create model matrix
mod <- model.matrix(~ no2_prenatal_avg_2019v + sex + gestational_age +
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3,
                    data=covariates_geno_ctpc_tech_cc)

# check size
dim(mod) # 128 15
```

### Surrogate variable analysis (SVA) to account for technical variation

Surrogate variable analysis using the sva packages. This will allow us to capture variance due to batch effects, as well as variance due to unkown biological variables. 
```{r sva}

###############
# null matrix #
###############

# create null matrix
# this matrix does not include any of our covariates 
# this means that our covariates are "protected" and SVs will not capture them
mod0 <- model.matrix(~ sex + gestational_age + prenatal_maternal_smoke + 
                       maternal_education_length + 
                       Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                       genotyping.PC1 + genotyping.PC2 + genotyping.PC3,  
                     data=covariates_geno_ctpc_tech_cc)

################
# sva analysis #
################

# to get SVs in a way that allows calculation of their variance
# we need to pull the code out of sva and run it line-by-line
# we are using the "Be" method for calculating SVs

######################################
# irwsva required internal functions #
######################################

# edge.lfdr function is internal to irwsva function below
# mono function is internal to edge.lfdr
# they have to be initialized before we can run the rest of the code

set.seed(1234)

mono <- function(lfdr){
  .Call("monotone", as.numeric(lfdr), PACKAGE="sva")
}

edge.lfdr <- function(p, trunc=TRUE, monotone=TRUE, transf=c("probit", "logit"), adj=1.5, eps=10^-8, lambda=0.8, ...) {
  pi0 <- mean(p >= lambda)/(1 - lambda)
  pi0 <- min(pi0, 1)
  n <- length(p)
  transf <- match.arg(transf)
  if(transf=="probit") {
    p <- pmax(p, eps)
    p <- pmin(p, 1-eps)
    x <- qnorm(p)
    myd <- density(x, adjust=adj)
    mys <- smooth.spline(x=myd$x, y=myd$y)
    y <- predict(mys, x)$y
    lfdr <- pi0*dnorm(x)/y
  }
  if(transf=="logit") {
    x <- log((p+eps)/(1-p+eps))
    myd <- density(x, adjust=adj)
    mys <- smooth.spline(x=myd$x, y=myd$y)
    y <- predict(mys, x)$y
    dx <- exp(x) / (1+exp(x))^2
    lfdr <- pi0 * dx/y
  }
  if(trunc) {
    lfdr[lfdr > 1] <- 1
  }
  if(monotone) {	
    lfdr <- lfdr[order(p)]
    lfdr <- mono(lfdr)
    lfdr <- lfdr[rank(p)]
  }
  return(lfdr)
}



##################################
# numsv function internal to sva #
##################################

#determine the number of required SVs
#this is numsv function code pulled out

#this is what is fed into this function from the sva function
dat = as.matrix(cb_betas_geno_cc_clean)
mod = mod
#method="be" # this is the default and what is called by numsv
#vfilter=NULL # we dont apply a filter and its not used for be method
B=20
set.seed(1234)

# initialize some stuff
# get the residual matrix (the remaining variance)
n <- ncol(dat)
m <- nrow(dat)
H <- mod %*% solve(t(mod) %*% mod) %*% t(mod)
res <- dat - t(H %*% t(dat))

# run svd and get some eigenvalues
uu <- svd(res)
# get the num of "sig" eigenvalues
ndf <- min(m, n) - ceiling(sum(diag(H)))
# get percent variance explained by each eigenvalue
dstat <- uu$d[1:ndf]^2/sum(uu$d[1:ndf]^2)
# initialize null distribution matrix
dstat0 <- matrix(0, nrow = B, ncol = ndf)

# set seed here since we are calling sample
# create the null distribution
for (i in 1:B) {
  # this part of the code removes all relationships in the 
  res0 <- t(apply(res, 1, sample, replace = FALSE))
  res0 <- res0 - t(H %*% t(res0))
  # calculate svs on this
  uu0 <- svd(res0)
  # bind their explained variance in null matrix
  dstat0[i, ] <- uu0$d[1:ndf]^2/sum(uu0$d[1:ndf]^2)
  print(i) # print out the iteration we are on
}

# determine the number of svs required here based on pvalues
psv <- rep(1, n)
for (i in 1:ndf) {
  psv[i] <- mean(dstat0[, i] >= dstat[i])
}
for (i in 2:ndf) {
  psv[i] <- max(psv[(i - 1)], psv[i])
}

# this is the variable holding the number of SVs required 
# number is each to all the svs with a pval less than/equal to 0.1
nsv <- sum(psv <= 0.1)


####################################
# code from the main SVA function  #
####################################

n.sv = nsv # from code above (run internally when sva called)
B = 5 # number of iterations

# dat =  betas # already initialized
# mod = mod  # already initialized
# mod0 = mod0  # already initialized 
# controls = NULL # we dont include controls - only for supervised method
# method = "irw" # this isnt actually used but would be called by sva
# vfilter = NULL # we dont set this when running sva
# numSVmethod = "be" # this is the default called by sva (code above)


###################################
# irwsva function internal to sva #
###################################

# this code is really want does all the work in sva
# we use the irw method so this is the code for that method
# there are also supervised and two step methods (not used by us)

# irw function takes input from sva, so we have already defined below vars
# dat = dat 
# mod = mod
# mod0 = mod0
# n.sv = n.sv
# B = B

# intialize variables again
n <- ncol(dat)
m <- nrow(dat)
Id <- diag(n)
resid <- dat %*% (Id - mod %*% solve(t(mod) %*% mod) %*% t(mod))
uu <- eigen(t(resid) %*% resid)
vv <- uu$vectors
ndf <- n - dim(mod)[2]
pprob <- rep(1, m)
one <- rep(1, n)
Id <- diag(n)
df1 <- dim(mod)[2] + n.sv
df0 <- dim(mod0)[2] + n.sv
rm(resid)

cat(paste("Iteration (out of", B, "):"))
for (i in 1:B) {
  mod.b <- cbind(mod, uu$vectors[, 1:n.sv])
  mod0.b <- cbind(mod0, uu$vectors[, 1:n.sv])
  ptmp <- f.pvalue(dat, mod.b, mod0.b)
  pprob.b <- (1 - edge.lfdr(ptmp))
  mod.gam <- cbind(mod0, uu$vectors[, 1:n.sv])
  mod0.gam <- cbind(mod0)
  ptmp <- f.pvalue(dat, mod.gam, mod0.gam)
  pprob.gam <- (1 - edge.lfdr(ptmp))
  pprob <- pprob.gam * (1 - pprob.b)
  dats <- dat * pprob
  dats <- dats - rowMeans(dats)
  uu <- eigen(t(dats) %*% dats)
  cat(paste(i, " "))
}

# this is what is normally returned by sva
# problem is that it doesnt include the diagonal matrix we want
sv_cb = svd(dats)$v[, 1:n.sv, drop = FALSE] 
svobj <- list(sv = sv_cb, pprob.gam = pprob.gam, pprob.b = pprob.b, 
              n.sv = n.sv)

# to get all info (including diagonal)
svobj_all <- svd(dats)

svs_cb <- as.data.frame(svobj$sv)

# save 
save(sv_cb, file = here("output_data", 
                        "linear_regression", 
                        "2023-08-14_prenatal_svs.RData"))

###############
#scree of SVs #
###############

# from elodie portales-casamar

# heat scree prepartion
vars <- svobj_all[['d']]^2
vars <- vars/sum(vars)
names(vars) <- paste0('SV', 1:length(vars))
xp <- 1:svobj$n.sv
varsdf <- as.data.frame(vars)
varsdf$svnum <- rep(1:length(vars))

# heat scree ggplot!! 
cb_sv_scree <- ggplot(varsdf[1:20,], aes(x=svnum, y=vars*100)) +
  geom_col() +
  theme_bw() +
  labs(x="Surrogate variable", y = "Variance (%)")

cb_sv_scree

ggsave(plot = cb_sv_scree, 
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal", 
                       "2023-08-14_cb_sv_heat_scree.png"),
       height = 5, width = 15, units = "cm", dpi = 600)

######################
# correlation matrix #
######################

# add svs to covariates
covariates_geno_ctpc_tech_cc <- cbind(covariates_geno_ctpc_tech_cc, sv_cb)

# rename columns with svs
colnames(covariates_geno_ctpc_tech_cc)[141:157]  <-
  paste("sv",colnames(covariates_geno_ctpc_tech_cc)[141:157], sep="")

source(here("scripts", "2023-07-06_corr_mat_funs.R"))

# early exp and pn_18 are the same thing. 
# exclude early_exp from correlation matrices
corr <- corr.mat(covariates_geno_ctpc_tech_cc %>% 
                   dplyr::select(no2_prenatal_avg_2019v, sex, gestational_age,
                                 prenatal_maternal_smoke, maternal_education_length,
                                 Comp.1, Comp.2, Comp.3, Comp.4, 
                                 genotyping.PC1, genotyping.PC2, genotyping.PC3, 
                                 chip, run, row, 
                                 sv1, sv2, sv3, sv4, sv5, sv6, sv7, sv8, sv9,
                                 sv10, sv11, sv12, sv13, sv14, sv15, sv16, sv17) %>%
                   dplyr::rename("Prenatal NO2" = no2_prenatal_avg_2019v,
                          "Biological sex" = sex,
                          "Gestational age (days)" = gestational_age,
                          "Prenatal maternal cigarette smoking" = prenatal_maternal_smoke,
                          "Maternal education (years)" = maternal_education_length,
                          "Cell type PC1" = Comp.1, "Cell type PC2" = Comp.2,
                          "Cell type PC3" = Comp.3, "Cell type PC4" = Comp.4,
                          "Genetic ancestry PC1" = genotyping.PC1, 
                          "Genetic ancestry PC2" = genotyping.PC2, 
                          "Genetic ancestry PC3" = genotyping.PC3, 
                          "Chip" = chip, "Run" = run, "Row" = row,
                          "SV1" = sv1, "SV2" = sv2, "SV3" = sv3, "SV4" = sv4,
                          "SV5" = sv5, "SV6" = sv6, "SV7" = sv7, "SV8" = sv8,
                          "SV9" = sv9, "SV10" = sv10, "SV11" = sv11, 
                          "SV12" = sv12,"SV13" = sv13, "SV14" = sv14, 
                          "SV15" = sv15, "SV16" = sv16, "SV17" = sv17))

write.csv(corr, file = here("tables",
                            "prenatal",
                            "2023-08-14_prenatal_no2_svcorr_correlations.csv"))

pval <- pval.mat(covariates_geno_ctpc_tech_cc %>% 
                   dplyr::select(no2_prenatal_avg_2019v, sex, gestational_age,
                                 prenatal_maternal_smoke, maternal_education_length,
                                 Comp.1, Comp.2, Comp.3, Comp.4, 
                                 genotyping.PC1, genotyping.PC2, genotyping.PC3,
                                 chip, run, row, 
                                 sv1, sv2, sv3, sv4, sv5, sv6, sv7, sv8, sv9,
                                 sv10, sv11, sv12, sv13, sv14, sv15, sv16,sv17) %>%
                   dplyr::rename("Prenatal NO2" = no2_prenatal_avg_2019v,
                          "Biological sex" = sex,
                          "Gestational age (days)" = gestational_age,
                          "Prenatal maternal cigarette smoking" = prenatal_maternal_smoke,
                          "Maternal education (years)" = maternal_education_length,
                          "Cell type PC1" = Comp.1, "Cell type PC2" = Comp.2,
                          "Cell type PC3" = Comp.3, "Cell type PC4" = Comp.4,
                          "Genetic ancestry PC1" = genotyping.PC1, 
                          "Genetic ancestry PC2" = genotyping.PC2, 
                          "Genetic ancestry PC3" = genotyping.PC3, 
                          "Chip" = chip, "Run" = run, "Row" = row,
                          "SV1" = sv1, "SV2" = sv2, "SV3" = sv3, "SV4" = sv4,
                          "SV5" = sv5, "SV6" = sv6, "SV7" = sv7, "SV8" = sv8,
                          "SV9" = sv9, "SV10" = sv10, "SV11" = sv11, 
                          "SV12" = sv12,"SV13" = sv13, "SV14" = sv14, 
                          "SV15" = sv15, "SV16" = sv16,"SV17" = sv17))

write.csv(pval, file = here("tables",
                            "prenatal",
                            "2023-08-14_prenatal_no2_svcorr_pvals.csv"))

svcorr <- corr.plot(pval = pval, 
                    corr = corr, 
                    label.insig.pval = T, 
                    label.sig.pval = T,
                    axis.text.size = 8, x.axis.text.angle = 90) +
  scale_y_discrete(position = "right") +
  theme(legend.position = "none",
        panel.grid = element_blank())

ggsave(plot = svcorr,
       filename = here("figures",
                       "linear_regression",
                       "prenatal", 
                       "2023-08-14_sv_correlations.png"),
       height = 15, width = 15, units = "cm", dpi = 600)

###########################
# add SVs to model matrix #
###########################

colnames(sv_cb) <- c(1:17)

# use 6 SVs 
modSv = cbind(mod,sv_cb)
```

### EWAS examining the effects of prenatal NO2 exposure 

Run Limma for DNAm analysis.
```{r limma}
set.seed(1234)

# fit methylation to no2
cb_fit <-  lmFit(cb_betas_geno_cc_clean, modSv)
cb_ebayes <- eBayes(cb_fit)

# subset annotation file for cpgs used in limma
annotation_cpgs <- annotation[rownames(annotation) %in% rownames(cb_ebayes),]

# check that annotation file and limma cpgs are in the same order
identical(rownames(annotation_cpgs), rownames(cb_ebayes)) # false

# reorder annotation file to match limma output
annotation_cpgs_order <- annotation_cpgs[rownames(cb_ebayes), ]

# recheck order of annotation file against limma cpgs
identical(rownames(annotation_cpgs_order), rownames(cb_ebayes)) #true

# pull out pval, coefficients from limma and combine with annotation
cb_pvals_tech <- cbind(as.data.frame(cb_ebayes$p.value), 
                       as.data.frame(cb_ebayes$coefficients),
                       as.data.frame(annotation_cpgs_order))

# rename columns to indicate pvalues vs coeffcients 
colnames(cb_pvals_tech) <- c(paste("pval", 
                                   colnames(cb_pvals_tech)[1:30], 
                                   sep = "_"), 
                             paste("coeff", 
                                   colnames(cb_pvals_tech)[31:60], 
                                   sep = "_"),
                             colnames(cb_pvals_tech)[61:93])

cb_pvals_tech$pval_no2_prenatal_avg_2019v_adj  <-
  p.adjust(cb_pvals_tech$pval_no2_prenatal_avg_2019v , method="BH")

# write.csv(cb_pvals_tech,
#           file=here("tables",
#                     "prenatal",
#                     "2023-08-14_prenatal_no2_epigenome_wide_analysis.csv"))
```

Examine the qqplot.
```{r qqplot_pval_histo}
pvector <- cb_pvals_tech$pval_no2_prenatal_avg_2019v

pvector2 <- pvector[!is.na(pvector) & !is.nan(pvector) & !is.null(pvector) & 
                      is.finite(pvector) & pvector < 1 & pvector > 0]

pval_qq <- as.data.frame(cbind(o = -log10(sort(pvector2, decreasing = FALSE)),
                               e = -log10(ppoints(length(pvector2)))))

# calculate lambda
z = qnorm(pvector2/2)
lambda = round(median(z^2) / 0.454, 3)

# qqplot of bacon corrected pvalues for main paper figure
qqplot <-  ggplot(pval_qq, aes(x=e, y=o)) + 
  geom_point(col="#254A90", size=1) + 
  geom_abline(slope = 1, intercept = 0, 
              col = "red", size=0.25, linetype="dashed") +
  labs(x = "Expected -log10(p-value)",
       y = "Observed -log10(p-value)") +
  annotate("text", x=3, y=1.5, 
           label = paste("λ = ", signif(lambda, 2)), 
           size=3, hjust=0) +
  theme_bw() +
  theme(axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        axis.title = element_blank(),
        panel.grid = element_blank(),
        panel.border = element_blank(),
        axis.line = element_line(color ="black")) 
qqplot

ggsave(plot = qqplot, 
       filename= here("figures", 
                      "linear_regression", 
                      "prenatal", 
                      "2023-08-14_prenatal_no2_qqplot.png"),
       width = 5, height = 4, units = "cm", dpi = 600)
```

Effect size
```{r effect_size}
range = max(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v)

cb_pvals_tech$effect_size <- cb_pvals_tech$coeff_no2_prenatal_avg_2019v * range
```

Volcano plot
```{r volcano}
# subset for points used in persistence analysis
persist_points <- cb_pvals_tech %>% 
  subset(effect_size > 0.025 | effect_size < -0.025) %>%
  arrange(pval_no2_prenatal_avg_2019v) %>%
  slice_head(n=100) %>% 
  rownames()

cb_volcano <- ggplot(cb_pvals_tech %>%
                       subset(!rownames(.) %in% persist_points), 
                     aes(x=as.numeric(effect_size) , 
                         y=-log10(as.numeric(pval_no2_prenatal_avg_2019v)))) +
  geom_point(alpha=0.75, col = "#254A90", size=1) +
  # colour significant cpgs; 2.5% effect size cut off and FDR < 0.25
  geom_point(data = cb_pvals_tech %>% 
               subset(rownames(.) %in% persist_points), 
             alpha=0.9, col = "#906B25", size=1, shape = 4, fill = "#254A90") +  #"#254A90", "#508FC2", "#92C5EB"

  labs(y="-log10(unadjusted p-value)", x = "Effect size (% DNAm)") +
  theme_bw() +
  theme(panel.grid = element_blank(),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        axis.title = element_blank()) 
cb_volcano

ggsave(plot = cb_volcano,
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal", 
                       "2023-08-14_cb_volcanoplot.png"),
       width = 5, height = 4, units = "cm")
```

#### Male specific EWAS of prenatal NO2 exposure

Subset data for males
```{r sub_males}
covariates_geno_ctpc_tech_cc_males <- covariates_geno_ctpc_tech_cc %>%
  subset(sex == "M")

cb_betas_geno_cc_clean_males <- cb_betas_geno_cc_clean[, as.character(covariates_geno_ctpc_tech_cc_males$sample_id)]

identical(as.character(covariates_geno_ctpc_tech_cc_males$sample_id),
          colnames(cb_betas_geno_cc_clean_males)) # TRUE
```

Model matrix
```{r male-model_matrix}
mod_males <- model.matrix(~ no2_prenatal_avg_2019v + gestational_age +
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                        sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                        sv10 + sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc_males)
```

Run Limma for DNAm analysis.
```{r limma_males}
set.seed(1234)
# fit methylation to no2
cb_fit_males <-  lmFit(cb_betas_geno_cc_clean_males, mod_males)
cb_ebayes_males <- eBayes(cb_fit_males)

# subset annotation file for cpgs used in limma
annotation_cpgs_males <- annotation[rownames(annotation) %in% rownames(cb_ebayes_males),]

# check that annotation file and limma cpgs are in the same order
identical(rownames(annotation_cpgs_males), rownames(cb_ebayes_males)) # false

# reorder annotation file to match limma output
annotation_cpgs_males_order <- annotation_cpgs_males[rownames(cb_ebayes_males), ]

# recheck order of annotation file against limma cpgs
identical(rownames(annotation_cpgs_males_order), rownames(cb_ebayes_males)) #true

# pull out pval, coefficients from limma and combine with annotation
cb_pvals_tech_males <- cbind(as.data.frame(cb_ebayes_males$p.value), 
                       as.data.frame(cb_ebayes_males$coefficients),
                       as.data.frame(annotation_cpgs_males_order))

# rename columns to indicate pvalues vs coeffcients 
colnames(cb_pvals_tech_males) <- c(paste("pval", 
                                   colnames(cb_pvals_tech_males)[1:29], 
                                   sep = "_"), 
                             paste("coeff", 
                                   colnames(cb_pvals_tech_males)[30:58], 
                                   sep = "_"),
                             colnames(cb_pvals_tech_males)[59:91])

cb_pvals_tech_males$pval_no2_prenatal_avg_2019v_adj  <-
  p.adjust(cb_pvals_tech_males$pval_no2_prenatal_avg_2019v , method="BH")

# write.csv(cb_pvals_tech_males,
#           file=here("tables",
#                     "prenatal",
#                     "2023-08-14_prenatal_no2_epigenome_wide_analysis_MALES.csv"))
```

Examine the qqplot.
```{r qqplot_males}
pvector_males <- cb_pvals_tech_males$pval_no2_prenatal_avg_2019v

pvector2_males <- pvector_males[!is.na(pvector_males) & !is.nan(pvector_males) & !is.null(pvector_males) & 
                      is.finite(pvector_males) & pvector_males < 1 & pvector_males > 0]

pval_qq_males <- as.data.frame(cbind(o = -log10(sort(pvector2_males, decreasing = FALSE)),
                               e = -log10(ppoints(length(pvector2_males)))))

# calculate lambda
z_males = qnorm(pvector2_males/2)
lambda_males = round(median(z_males^2) / 0.454, 3)


# qqplot of bacon corrected pvalues for main paper figure
qqplot_males <-  ggplot(pval_qq_males, aes(x=e, y=o)) + 
  geom_point(col="#254A90", size=1) + 
  geom_abline(slope = 1, intercept = 0, 
              col = "red", size=0.25, linetype="dashed") +
  labs(x = "Expected -log10(p-value)",
       y = "Observed -log10(p-value)") +
  annotate("text", x=3, y=1.5, 
           label = paste("λ = ", signif(lambda_males, 2)), 
           size=3, hjust=0) +
  theme_bw() +
  theme(axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        axis.title = element_blank(),
        panel.grid = element_blank(),
        panel.border = element_blank(),
        axis.line = element_line(color ="black")) 
qqplot_males

ggsave(plot = qqplot_males, 
       filename= here("figures", 
                      "linear_regression", 
                      "prenatal", 
                      "2023-08-14_prenatal_no2_qqplot_MALES.png"),
       width = 5, height = 4, units = "cm", dpi = 600)
```

Effect size
```{r effect_size}
range_males = max(covariates_geno_ctpc_tech_cc_males$no2_prenatal_avg_2019v) -
    min(covariates_geno_ctpc_tech_cc_males$no2_prenatal_avg_2019v)

cb_pvals_tech_males$effect_size <- cb_pvals_tech_males$coeff_no2_prenatal_avg_2019v * range_males
```

Volcano plot
```{r volcano_males}
# subset for points used in persistence analysis
persist_points_males <- cb_pvals_tech_males %>% 
  subset(effect_size > 0.025 | effect_size < -0.025) %>%
  arrange(pval_no2_prenatal_avg_2019v) %>%
  slice_head(n=100) %>% 
  rownames()

cb_volcano_males <- ggplot(cb_pvals_tech_males %>%
                       subset(!rownames(.) %in% persist_points_males), 
                     aes(x=as.numeric(effect_size) , 
                         y=-log10(as.numeric(pval_no2_prenatal_avg_2019v)))) +
  geom_point(alpha=0.75, col = "#254A90", size=1) +
  geom_point(data = cb_pvals_tech_males %>% 
               subset(rownames(.) %in% persist_points_males), 
             alpha=0.9, col = "#906B25", size=1, shape = 4, fill = "#254A90") +  #"#254A90", "#508FC2", "#92C5EB"
  labs(y="-log10(unadjusted p-value)", x = "Effect size (% DNAm)") +
  theme_bw() +
  theme(panel.grid = element_blank(),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        axis.title = element_blank()) 
cb_volcano_males

ggsave(plot = cb_volcano_males,
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal", 
                       "2023-08-14_cb_volcanoplot_MALES.png"),
       width = 5, height = 4, units = "cm")
```

#### Female specific EWAS of prenatal NO2 exposure

Subset data for females
```{r sub_females}
covariates_geno_ctpc_tech_cc_females <- covariates_geno_ctpc_tech_cc %>%
  subset(sex == "F")

cb_betas_geno_cc_clean_females <- cb_betas_geno_cc_clean[, as.character(covariates_geno_ctpc_tech_cc_females$sample_id)]

identical(as.character(covariates_geno_ctpc_tech_cc_females$sample_id),
          colnames(cb_betas_geno_cc_clean_females)) # TRUE
```

Model matrix
```{r female-model_matrix}
mod_females <- model.matrix(~ no2_prenatal_avg_2019v + gestational_age +
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                        sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                        sv10 + sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc_females)
```


Run Limma for DNAm analysis.
```{r limma_females}
set.seed(1234)
# fit methylation to no2
cb_fit_females <-  lmFit(cb_betas_geno_cc_clean_females, mod_females)
cb_ebayes_females <- eBayes(cb_fit_females)

# subset annotation file for cpgs used in limma
annotation_cpgs_females <- annotation[rownames(annotation) %in% rownames(cb_ebayes_females),]

# check that annotation file and limma cpgs are in the same order
identical(rownames(annotation_cpgs_females), rownames(cb_ebayes_females)) # false

# reorder annotation file to match limma output
annotation_cpgs_females_order <- annotation_cpgs_females[rownames(cb_ebayes_females), ]

# recheck order of annotation file against limma cpgs
identical(rownames(annotation_cpgs_females_order), rownames(cb_ebayes_females)) #true

# pull out pval, coefficients from limma and combine with annotation
cb_pvals_tech_females <- cbind(as.data.frame(cb_ebayes_females$p.value), 
                       as.data.frame(cb_ebayes_females$coefficients),
                       as.data.frame(annotation_cpgs_females_order))

# rename columns to indicate pvalues vs coeffcients 
colnames(cb_pvals_tech_females) <- c(paste("pval", 
                                   colnames(cb_pvals_tech_females)[1:29], 
                                   sep = "_"), 
                             paste("coeff", 
                                   colnames(cb_pvals_tech_females)[30:58], 
                                   sep = "_"),
                             colnames(cb_pvals_tech_females)[59:91])

cb_pvals_tech_females$pval_no2_prenatal_avg_2019v_adj  <-
  p.adjust(cb_pvals_tech_females$pval_no2_prenatal_avg_2019v , method="BH")

# write.csv(cb_pvals_tech_females,
#           file=here("tables",
#                     "prenatal",
#                     "2023-06-28_prenatal_no2_epigenome_wide_analysis_females.csv"))
```

Examine the qqplot.
```{r qqplot_females}
pvector_females <- cb_pvals_tech_females$pval_no2_prenatal_avg_2019v

pvector2_females <- pvector_females[!is.na(pvector_females) & !is.nan(pvector_females) & !is.null(pvector_females) & 
                      is.finite(pvector_females) & pvector_females < 1 & pvector_females > 0]

pval_qq_females <- as.data.frame(cbind(o = -log10(sort(pvector2_females, decreasing = FALSE)),
                               e = -log10(ppoints(length(pvector2_females)))))

# calculate lambda
z_females = qnorm(pvector2_females/2)
lambda_females = round(median(z_females^2) / 0.454, 3)


# qqplot of bacon corrected pvalues for main paper figure
qqplot_females <-  ggplot(pval_qq_females, aes(x=e, y=o)) + 
  geom_point(col="#254A90", size=1) + 
  geom_abline(slope = 1, intercept = 0, 
              col = "red", size=0.25, linetype="dashed") +
  labs(x = "Expected -log10(p-value)",
       y = "Observed -log10(p-value)") +
  annotate("text", x=3, y=1.5, 
           label = paste("λ = ", signif(lambda_females, 2)), 
           size=3, hjust=0) +
  theme_bw() +
  theme(axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        axis.title = element_blank(),
        panel.grid = element_blank(),
        panel.border = element_blank(),
        axis.line = element_line(color ="black")) 
qqplot_females

ggsave(plot = qqplot_females, 
       filename= here("figures", 
                      "linear_regression", 
                      "prenatal", 
                      "2023-08-14_prenatal_no2_qqplot_femaleS.png"),
       width = 5, height = 4, units = "cm", dpi = 600)
```

Effect size
```{r effect_size}
range_females = max(covariates_geno_ctpc_tech_cc_females$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_females$no2_prenatal_avg_2019v)

cb_pvals_tech_females$effect_size <- cb_pvals_tech_females$coeff_no2_prenatal_avg_2019v * range_females
```

Volcano plot
```{r volcano_females}
# subset for points used in the persistence analysis
persist_points_females <- cb_pvals_tech_females %>% 
  subset(effect_size > 0.025 | effect_size < -0.025) %>%
  arrange(pval_no2_prenatal_avg_2019v) %>%
  slice_head(n=100) %>% 
  rownames()

cb_volcano_females <- ggplot(cb_pvals_tech_females %>%
                       subset(!rownames(.) %in% persist_points_females), 
                     aes(x=as.numeric(effect_size) , 
                         y=-log10(as.numeric(pval_no2_prenatal_avg_2019v)))) +
  geom_point(alpha=0.75, col = "#254A90", size=1) +
  # colour significant cpgs; 2.5% effect size cut off and FDR < 0.25
  geom_point(data = cb_pvals_tech_females %>% 
               subset(rownames(.) %in% persist_points_females), 
             alpha=0.9, col = "#906B25", size=1, shape = 4, fill = "#254A90") +
  labs(y="-log10(unadjusted p-value)", x = "Effect size (% DNAm)") +
  theme_bw() +
  theme(panel.grid = element_blank(),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        axis.title = element_blank()) 
cb_volcano_females

ggsave(plot = cb_volcano_females,
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal", 
                       "2023-08-14_cb_volcanoplot_FEMALES.png"),
       width = 5, height = 4, units = "cm")
```

### Gene specific changes

These genes are related to the proposed pathway of air pollution action or have previously been shown to exhibit DNA methylation changes associated with prenatal air pollution exposure. 

```{r gene_specific}
#FHC is FTH1 (ferritin heavy chain 1)
#nothing maps to FOXP3
#TNFA is TNF
goi <- annotation[grep("^RNF39|^CYP2E1|^LONP1|^SLC25A28|^PLVAP|^CAT$|^CAT;|^TPO|^NFKB1|^NOX2|^SOD1|^SOD2|^TRX1|^TRX2|^GSTP1|^MT3|^NQO1|^HMOX1|^TGPX1|^DDH1|^NOS2|^CYP2C11|^CYP7B|^ALOX12$|^ALOX5;|^ALOX5$|^NRF2$|^NRF2;|^KEAP1|^AHR$|^AHR;|^ARNT;|^ARNT$|^GCLC|^GCLM|^GSR1|^SLC7A11|^GSTA1|^GSTA2|^GSTA3|^GSTA5|^GSTM1|^GSTM2|^GSTM3|^TXN$|^TXN;|^TXNRD1$|^TXNRD1;|^SRXN1|^G6PD|^CYP1A1|^IL1A|^IL2$|^IL2;|^IL3$|^IL3;|^IL4;|^IL4$|^IL5$|^IL5;|^IL6$|^IL6;|^IL9$|^IL9;|^IL10$|^IL10;|^IL12A|^IL13|^IL15$|^IL15;|^IL17A|^IL18$|^IL18;|^IL19|^IL21$|^IL21;|^IL22$|^IL22;|^IL23A$|^IL23A;|^IL25$|^IL25;|^IL27$|^IL27;|^IL31$|^IL31;|^IL33$|^IL33;|^IFNG$|^IFNG;|^TNF$|^TNF;|^TGFB1$|^TGFB1;|^FOXP3|^FTH1", annotation$UCSC_RefGene_Name), ] %>% as.data.frame()

unique(goi$UCSC_RefGene_Name)
# only first annotate gene
unique(gsub(";.*$", "", unique(goi$UCSC_RefGene_Name))) # 59

# transpose for lm function
cb_betas_geno_cc_clean_t <- t(cb_betas_geno_cc_clean)

#apply function for calculating significance of associations using mice and imputed covariate data
prenatal_goi_list <- 
  pbapply(cb_betas_geno_cc_clean_t[,colnames(cb_betas_geno_cc_clean_t ) %in% rownames(goi)], 2, function(x) {
    
    combined <- cbind(covariates_geno_ctpc_tech_cc, x)
    
    #run the linear regression on this cpg using all 5 imputed covariate data sets
    fit <- lm(x ~ no2_prenatal_avg_2019v + sex + gestational_age  + 
                prenatal_maternal_smoke + maternal_education_length + 
                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
              data=combined)
    
    #get summary of lm fit
    #this includes coefficients and pvalues
    #coerce into df for manipulation
    #add in terms (rownames) as column for annotation
    fit_sum <- summary(fit)
    fit_sum_coeff <- as.data.frame(fit_sum$coefficients)
    fit_sum_coeff$term <- rownames(fit_sum_coeff)
    
    #this data frame has rows for each linear regression variable, and columns for statistics
    #want a dataframe of 1 row and many columns (one for each variable with each statistic type)
    #do this by using the spread function for each statistic
    #add cpg id onto dataframe for pvalues (to keep track of information)
    #then cbind dataframes together
    
    #coefficient
    estimate <- spread(fit_sum_coeff[,c("term", "Estimate")], term, Estimate)
    colnames(estimate) <- paste("estimate", colnames(estimate), sep="_")
    
    #std error
    std.error <- spread(fit_sum_coeff[,c("term", "Std. Error")], term, 'Std. Error')
    colnames(std.error) <- paste("std_error", colnames(std.error), sep="_")
    
    #t statistics
    statistic <- spread(fit_sum_coeff[,c("term", "t value")], term, 't value')
    colnames(statistic) <- paste("t_val", colnames(statistic), sep="_")
    
    #pvalue
    p.value <- spread(fit_sum_coeff[,c("term", "Pr(>|t|)")], term, 'Pr(>|t|)')
    colnames(p.value) <- paste("p_value", colnames(p.value), sep="_")
    
    #add in cpgid for annotation 
    p.value$cpg <- colnames(x)
    
    #cbind into wide dataframe
    widepool <- cbind(estimate, std.error, statistic, p.value)
    
    #return the wide dataframe
    #these will be do.call(rbind()) together after
    return(widepool)
    
  })

# convert to data frame
prenatal_goi <- do.call(rbind, prenatal_goi_list)

# adjust pvals
prenatal_goi$p_value_no2_prenatal_avg_2019v_adj <- 
  p.adjust(prenatal_goi$p_value_no2_prenatal_avg_2019v, method="BH")

#reorder according ot unadjusted no2 pvalue
prenatal_goi_order <- prenatal_goi[order(prenatal_goi$p_value_no2_prenatal_avg_2019v),]

# caclculate effect size
range = max(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v)

prenatal_goi_order$prenatal_no2_effect_size <- 
  prenatal_goi_order$estimate_no2_prenatal_avg_2019v * range

# write out table for all GOIs
write.csv(prenatal_goi_order, 
          file=here("tables", 
                    "prenatal",
                    "2023-08-14_prenatal_no2_candidate_gene_analysis.csv"))

#check cpgs from gruvieva 2017 paper

##################
#LONP1 cg12283362#
##################

covariates_geno_ctpc_tech_cc$cg12283362 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg12283362", ] 

cg12283362_lm <- lm(cg12283362 ~ no2_prenatal_avg_2019v + sex + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                    data=covariates_geno_ctpc_tech_cc)

# get lm info
summary(cg12283362_lm)

# estimate marginal effects of prenatal NO2 exposure on DNAm
cg12283362_no2_me <- ggpredict(cg12283362_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  dplyr::rename("cg12283362" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc$cg12283362, 
    covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v,
    method="pearson") # 0.00301464

# effect size
range_cb = max(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v)

summary(cg12283362_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_cb # 0.001169593

#####################
#SLC25A28 cg08973675#
#####################

covariates_geno_ctpc_tech_cc$cg08973675 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg08973675", ] 

cg08973675_lm <- lm(cg08973675~ no2_prenatal_avg_2019v + sex + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                    data=covariates_geno_ctpc_tech_cc)

summary(cg08973675_lm)

# estimate marginal effects of prenatal NO2 exposure on DNAm
cg08973675_no2_me <- ggpredict(cg08973675_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  dplyr::rename("cg08973675" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc$cg08973675, 
    covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v,
    method="pearson") # 0.04369708

# effect size
range_cb = max(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v)

summary(cg08973675_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_cb # 0.004890668
```

#### Gene specific changes in MALES

These genes are related to the proposed pathway of air pollution action or have previously been shown to exhibit DNA methylation changes associated with prenatal air pollution exposure. 

```{r gene_specific_MALES}
#FHC is FTH1 (ferritin heavy chain 1)
#nothing maps to FOXP3
#TNFA is TNF
goi <- annotation[grep("^RNF39|^CYP2E1|^LONP1|^SLC25A28|^PLVAP|^CAT$|^CAT;|^TPO|^NFKB1|^NOX2|^SOD1|^SOD2|^TRX1|^TRX2|^GSTP1|^MT3|^NQO1|^HMOX1|^TGPX1|^DDH1|^NOS2|^CYP2C11|^CYP7B|^ALOX12$|^ALOX5;|^ALOX5$|^NRF2$|^NRF2;|^KEAP1|^AHR$|^AHR;|^ARNT;|^ARNT$|^GCLC|^GCLM|^GSR1|^SLC7A11|^GSTA1|^GSTA2|^GSTA3|^GSTA5|^GSTM1|^GSTM2|^GSTM3|^TXN$|^TXN;|^TXNRD1$|^TXNRD1;|^SRXN1|^G6PD|^CYP1A1|^IL1A|^IL2$|^IL2;|^IL3$|^IL3;|^IL4;|^IL4$|^IL5$|^IL5;|^IL6$|^IL6;|^IL9$|^IL9;|^IL10$|^IL10;|^IL12A|^IL13|^IL15$|^IL15;|^IL17A|^IL18$|^IL18;|^IL19|^IL21$|^IL21;|^IL22$|^IL22;|^IL23A$|^IL23A;|^IL25$|^IL25;|^IL27$|^IL27;|^IL31$|^IL31;|^IL33$|^IL33;|^IFNG$|^IFNG;|^TNF$|^TNF;|^TGFB1$|^TGFB1;|^FOXP3|^FTH1", annotation$UCSC_RefGene_Name), ] %>% as.data.frame()

unique(goi$UCSC_RefGene_Name)
# only first annotate gene
unique(gsub(";.*$", "", unique(goi$UCSC_RefGene_Name))) # 59

# transpose for lm function
cb_betas_geno_cc_clean_males_t <- t(cb_betas_geno_cc_clean_males)

#apply function for calculating significance of associations using mice and imputed covariate data
prenatal_goi_list_males <- 
  pbapply(cb_betas_geno_cc_clean_males_t[,colnames(cb_betas_geno_cc_clean_males_t ) %in% rownames(goi)], 2, function(x) {
    
    combined <- cbind(covariates_geno_ctpc_tech_cc_males, x)
    
    #run the linear regression on this cpg using all 5 imputed covariate data sets
    fit <- lm(x ~ no2_prenatal_avg_2019v + gestational_age  + 
                prenatal_maternal_smoke + maternal_education_length + 
                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
              data=combined)
    
    #get summary of lm fit
    #this includes coefficients and pvalues
    #coerce into df for manipulation
    #add in terms (rownames) as column for annotation
    fit_sum <- summary(fit)
    fit_sum_coeff <- as.data.frame(fit_sum$coefficients)
    fit_sum_coeff$term <- rownames(fit_sum_coeff)
    
    #coefficient
    estimate <- spread(fit_sum_coeff[,c("term", "Estimate")], term, Estimate)
    colnames(estimate) <- paste("estimate", colnames(estimate), sep="_")
    
    #std error
    std.error <- spread(fit_sum_coeff[,c("term", "Std. Error")], term, 'Std. Error')
    colnames(std.error) <- paste("std_error", colnames(std.error), sep="_")
    
    #t statistics
    statistic <- spread(fit_sum_coeff[,c("term", "t value")], term, 't value')
    colnames(statistic) <- paste("t_val", colnames(statistic), sep="_")
    
    #pvalue
    p.value <- spread(fit_sum_coeff[,c("term", "Pr(>|t|)")], term, 'Pr(>|t|)')
    colnames(p.value) <- paste("p_value", colnames(p.value), sep="_")
    
    #add in cpgid for annotation 
    p.value$cpg <- colnames(x)
    
    #cbind into wide dataframe
    widepool <- cbind(estimate, std.error, statistic, p.value)
    
    #return the wide dataframe
    #these will be do.call(rbind()) together after
    return(widepool)
    
  })

# convert to dataframe
prenatal_goi_males <- do.call(rbind, prenatal_goi_list_males)

# adjust pvalues
prenatal_goi_males$p_value_no2_prenatal_avg_2019v_adj <- 
  p.adjust(prenatal_goi_males$p_value_no2_prenatal_avg_2019v, method="BH")

#reorder according ot unadjusted no2 pvalue
prenatal_goi_males_order <- prenatal_goi_males[order(prenatal_goi_males$p_value_no2_prenatal_avg_2019v),]

# caclculate effect size
range_males = max(covariates_geno_ctpc_tech_cc_males$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_males$no2_prenatal_avg_2019v)

prenatal_goi_males_order$prenatal_no2_effect_size <- prenatal_goi_males_order$estimate_no2_prenatal_avg_2019v * range_males

# write out table for all GOIs
write.csv(prenatal_goi_males_order, 
          file=here("tables", 
                    "prenatal",
                    "2023-08-14_prenatal_no2_candidate_gene_analysis_MALES.csv"))

##################
#LONP1 cg12283362#
##################

covariates_geno_ctpc_tech_cc_males$cg12283362 <- 
  cb_betas_geno_cc_clean_males[rownames(cb_betas_geno_cc_clean_males)=="cg12283362", ] 

cg12283362_lm_males <- lm(cg12283362 ~ no2_prenatal_avg_2019v + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                    data=covariates_geno_ctpc_tech_cc_males)

# get lm info
summary(cg12283362_lm)

# estimate marginal effects of prenatal NO2 exposure on DNAm
cg12283362_lm_me_males <- ggpredict(cg12283362_lm_males, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  dplyr::rename("cg12283362" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_males$cg12283362, 
    covariates_geno_ctpc_tech_cc_males$no2_prenatal_avg_2019v,
    method="pearson") # 0.03643646

# effect size
range_cb_males = max(covariates_geno_ctpc_tech_cc_males$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_males$no2_prenatal_avg_2019v)

summary(cg12283362_lm_males)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_cb_males # -0.001931516

#####################
#SLC25A28 cg08973675#
#####################

covariates_geno_ctpc_tech_cc_males$cg08973675 <- 
  cb_betas_geno_cc_clean_males[rownames(cb_betas_geno_cc_clean_males)=="cg08973675", ] 

cg08973675_lm_males <- lm(cg08973675~ no2_prenatal_avg_2019v + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                    data=covariates_geno_ctpc_tech_cc_males)

summary(cg08973675_lm_males)

# estimate marginal effects of prenatal NO2 exposure on DNAm
cg08973675_no2_me_males <- ggpredict(cg08973675_lm_males, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  dplyr::rename("cg08973675" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_males$cg08973675, 
    covariates_geno_ctpc_tech_cc_males$no2_prenatal_avg_2019v,
    method="pearson") # 0.07099577

# effect size
range_cb_males = max(covariates_geno_ctpc_tech_cc_males$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_males$no2_prenatal_avg_2019v)

summary(cg08973675_lm_males)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_cb_males # 0.003957345
```

#### Gene specific changes in females

These genes are related to the proposed pathway of air pollution action or have previously been shown to exhibit DNA methylation changes associated with prenatal air pollution exposure. 

```{r gene_specific_females}
#FHC is FTH1 (ferritin heavy chain 1)
#nothing maps to FOXP3
#TNFA is TNF
goi <- annotation[grep("^RNF39|^CYP2E1|^LONP1|^SLC25A28|^PLVAP|^CAT$|^CAT;|^TPO|^NFKB1|^NOX2|^SOD1|^SOD2|^TRX1|^TRX2|^GSTP1|^MT3|^NQO1|^HMOX1|^TGPX1|^DDH1|^NOS2|^CYP2C11|^CYP7B|^ALOX12$|^ALOX5;|^ALOX5$|^NRF2$|^NRF2;|^KEAP1|^AHR$|^AHR;|^ARNT;|^ARNT$|^GCLC|^GCLM|^GSR1|^SLC7A11|^GSTA1|^GSTA2|^GSTA3|^GSTA5|^GSTM1|^GSTM2|^GSTM3|^TXN$|^TXN;|^TXNRD1$|^TXNRD1;|^SRXN1|^G6PD|^CYP1A1|^IL1A|^IL2$|^IL2;|^IL3$|^IL3;|^IL4;|^IL4$|^IL5$|^IL5;|^IL6$|^IL6;|^IL9$|^IL9;|^IL10$|^IL10;|^IL12A|^IL13|^IL15$|^IL15;|^IL17A|^IL18$|^IL18;|^IL19|^IL21$|^IL21;|^IL22$|^IL22;|^IL23A$|^IL23A;|^IL25$|^IL25;|^IL27$|^IL27;|^IL31$|^IL31;|^IL33$|^IL33;|^IFNG$|^IFNG;|^TNF$|^TNF;|^TGFB1$|^TGFB1;|^FOXP3|^FTH1", annotation$UCSC_RefGene_Name), ] %>% as.data.frame()

unique(goi$UCSC_RefGene_Name)
# only first annotate gene
unique(gsub(";.*$", "", unique(goi$UCSC_RefGene_Name))) # 59

# transpose for lm function
cb_betas_geno_cc_clean_females_t <- t(cb_betas_geno_cc_clean_females)

#apply function for calculating significance of associations using mice and imputed covariate data
prenatal_goi_list_females <- 
  pbapply(cb_betas_geno_cc_clean_females_t[,colnames(cb_betas_geno_cc_clean_females_t ) %in% rownames(goi)], 2, function(x) {
    
    combined <- cbind(covariates_geno_ctpc_tech_cc_females, x)
    
    #run the linear regression on this cpg using all 5 imputed covariate data sets
    fit <- lm(x ~ no2_prenatal_avg_2019v + gestational_age  + 
                prenatal_maternal_smoke + maternal_education_length + 
                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
              data=combined)
    
    #get summary of lm fit
    #this includes coefficients and pvalues
    #coerce into df for manipulation
    #add in terms (rownames) as column for annotation
    fit_sum <- summary(fit)
    fit_sum_coeff <- as.data.frame(fit_sum$coefficients)
    fit_sum_coeff$term <- rownames(fit_sum_coeff)
    
    #coefficient
    estimate <- spread(fit_sum_coeff[,c("term", "Estimate")], term, Estimate)
    colnames(estimate) <- paste("estimate", colnames(estimate), sep="_")
    
    #std error
    std.error <- spread(fit_sum_coeff[,c("term", "Std. Error")], term, 'Std. Error')
    colnames(std.error) <- paste("std_error", colnames(std.error), sep="_")
    
    #t statistics
    statistic <- spread(fit_sum_coeff[,c("term", "t value")], term, 't value')
    colnames(statistic) <- paste("t_val", colnames(statistic), sep="_")
    
    #pvalue
    p.value <- spread(fit_sum_coeff[,c("term", "Pr(>|t|)")], term, 'Pr(>|t|)')
    colnames(p.value) <- paste("p_value", colnames(p.value), sep="_")
    
    #add in cpgid for annotation 
    p.value$cpg <- colnames(x)
    
    #cbind into wide dataframe
    widepool <- cbind(estimate, std.error, statistic, p.value)
    
    #return the wide dataframe
    #these will be do.call(rbind()) together after
    return(widepool)
    
  })

# convert ot data frame
prenatal_goi_females <- do.call(rbind, prenatal_goi_list_females)

# adjust pvalues
prenatal_goi_females$p_value_no2_prenatal_avg_2019v_adj <- 
  p.adjust(prenatal_goi_females$p_value_no2_prenatal_avg_2019v, method="BH")

#reorder according ot unadjusted no2 pvalue
prenatal_goi_females_order <- prenatal_goi_females[order(prenatal_goi_females$p_value_no2_prenatal_avg_2019v),]

# caclculate effect size
range_females = max(covariates_geno_ctpc_tech_cc_females$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_females$no2_prenatal_avg_2019v)

prenatal_goi_females_order$prenatal_no2_effect_size <- 
  prenatal_goi_females_order$estimate_no2_prenatal_avg_2019v * range_females

# write out table for all GOIs
write.csv(prenatal_goi_females_order, 
          file=here("tables", 
                    "prenatal",
                    "2023-08-14_prenatal_no2_candidate_gene_analysis_females.csv"))

##################
#LONP1 cg12283362#
##################

covariates_geno_ctpc_tech_cc_females$cg12283362 <- 
  cb_betas_geno_cc_clean_females[rownames(cb_betas_geno_cc_clean_females)=="cg12283362", ] 

cg12283362_lm_females <- lm(cg12283362 ~ no2_prenatal_avg_2019v + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                    data=covariates_geno_ctpc_tech_cc_females)

# get lm info
summary(cg12283362_lm)

# estimate marginal effects of prenatal NO2 exposure on DNAm
cg12283362_lm_me_females <- ggpredict(cg12283362_lm_females, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  dplyr::rename("cg12283362" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_females$cg12283362, 
    covariates_geno_ctpc_tech_cc_females$no2_prenatal_avg_2019v,
    method="pearson") # 0.03643646

# effect size
range_cb_females = max(covariates_geno_ctpc_tech_cc_females$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_females$no2_prenatal_avg_2019v)

summary(cg12283362_lm_females)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_cb_females # 0.04042044

#####################
#SLC25A28 cg08973675#
#####################

covariates_geno_ctpc_tech_cc_females$cg08973675 <- 
  cb_betas_geno_cc_clean_females[rownames(cb_betas_geno_cc_clean_females)=="cg08973675", ] 

cg08973675_lm_females <- lm(cg08973675~ no2_prenatal_avg_2019v + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                    data=covariates_geno_ctpc_tech_cc_females)

summary(cg08973675_lm_females)

# estimate marginal effects of prenatal NO2 exposure on DNAm
cg08973675_no2_me_females <- ggpredict(cg08973675_lm_females, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  dplyr::rename("cg08973675" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_females$cg08973675, 
    covariates_geno_ctpc_tech_cc_females$no2_prenatal_avg_2019v,
    method="pearson") # 0.07099577

# effect size
range_cb_females = max(covariates_geno_ctpc_tech_cc_females$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_females$no2_prenatal_avg_2019v)

summary(cg08973675_lm_females)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_cb_females # 0.003047114

```

#### Plot of gene specific changes 
```{r goi_plot}
##################
#LONP1 cg12283362#
##################

cb_cg12283362_dnam_plot <-  ggplot(covariates_geno_ctpc_tech_cc,
                                  aes(x = no2_prenatal_avg_2019v, 
                                      y= cg12283362,
                                      shape = sex, colour = sex)) +
  scale_shape_manual(values = c(1,4)) + 
  scale_colour_manual(values = c("#D81B60", "#1E88E5")) +
  geom_line(data = cg12283362_no2_me, 
            aes(x=no2_prenatal_avg_2019v, y = cg12283362),
            col="#FFC107", size=0.5, linetype="solid", inherit.aes = F) +
  #males
    geom_line(data = cg12283362_lm_me_males, 
            aes(x=no2_prenatal_avg_2019v, y = cg12283362),
            col="#1E88E5", size=0.5, linetype="dashed", inherit.aes = F) +
   # females
    geom_line(data = cg12283362_lm_me_females, 
            aes(x=no2_prenatal_avg_2019v, y = cg12283362),
            col="#D81B60", size=0.5, linetype="twodash", inherit.aes = F) +
      geom_point(alpha=0.7, size=1) +
  #scale_y_continuous(breaks = seq(0,0.16,by=0.04), limits = c(0,0.16)) +
  theme_bw() +
  theme(legend.position = "none",
  panel.grid = element_blank(),
        axis.text = element_text(size=9),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.title = element_blank()) 

ggsave(plot = cb_cg12283362_dnam_plot, 
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal",
                       "2023-08-14_cb_cg12283362_dnam_all_males_females.png"),
         width = 4.0, height = 3.2, units = "cm", dpi = 600)

#####################
#SLC25A28 cg08973675#
#####################

cb_cg08973675_dnam_plot <- ggplot(covariates_geno_ctpc_tech_cc,
                                  aes(x = no2_prenatal_avg_2019v, 
                                      y= cg08973675,
                                      shape = sex, colour = sex)) +
  scale_shape_manual(values = c(1,4)) + 
  scale_colour_manual(values = c("#D81B60", "#1E88E5")) +
  geom_line(data = cg08973675_no2_me, 
            aes(x=no2_prenatal_avg_2019v, y = cg08973675),
            col="#FFC107", size=0.5, linetype="solid", inherit.aes = F) +
  #males
    geom_line(data = cg08973675_no2_me_males, 
            aes(x=no2_prenatal_avg_2019v, y = cg08973675),
            col="#1E88E5", size=0.5, linetype="dashed", inherit.aes = F) +
   # females
    geom_line(data = cg08973675_no2_me_females, 
            aes(x=no2_prenatal_avg_2019v, y = cg08973675),
            col="#D81B60", size=0.5, linetype="twodash", inherit.aes = F) +
  geom_point(alpha=0.7, size=1) +
  scale_y_continuous(breaks = seq(0,0.16,by=0.04), limits = c(0,0.16)) +
  theme_bw() +
  theme(legend.position = "none",
  panel.grid = element_blank(),
        axis.text = element_text(size=9),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.title = element_blank()) 

ggsave(plot = cb_cg08973675_dnam_plot, 
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal",
                       "2023-08-14_cb_cg08973675_dnam_all_male_female.png"),
        width = 4.0, height = 3.2,units = "cm", dpi = 600)
```

### Differentially methylated regions (DMRs) in entire study population

```{r dmrs_study_pop}
comb_dat <- cb_pvals_tech %>%
  dplyr::select(pval_no2_prenatal_avg_2019v, chr, pos, Name) %>%
  mutate(chr = str_sub(chr, 4,5),
         end = pos + 1) %>%
  dplyr::rename("p" = pval_no2_prenatal_avg_2019v,
                "start" = pos,
                "probe" = Name)

# need to run in "console" portion of RStudio
# results stored in home directory (~/CANDLE/)
# rename files, move to correct directory before loading 
combp(comb_dat, 
      seed = 0.01,
      verbose = T)

# read in csv of dmrs
resu_combp <- read_csv(here("tables",
                            "prenatal",
                            "2023-08-14_prenatal_no2_resu_combp.csv"))

resu_combp <- resu_combp %>%
  mutate(dmr = rownames(.)) %>%
  mutate(across(everything(), as.character)) %>%
  mutate(across(where(is.character), str_remove_all, pattern = fixed(" ")))

# get annotation for DMRs in combp results
combp_anno <- apply(resu_combp, 1, function(x){
  annotation_cpgs_order %>%
    as.data.frame() %>%
    subset(chr == paste("chr", x["chr"], sep = "")) %>%
    subset(pos >= as.numeric(x["start"]) & pos <= as.numeric(x["end"])-1) %>%
    mutate(dmr = x["dmr"])
})                  

resu_combp <- resu_combp %>%
  mutate(start = as.numeric(start),
         end = as.numeric(end),
         p = as.numeric(p),
         fdr = as.numeric(fdr),
         nprobe = as.numeric(nprobe))

# conver to data frame
combp_anno_df <- do.call(rbind, combp_anno)

# create new columns to specify DMR and analysis
resu_combp$dmr <- 1:nrow(resu_combp)
resu_combp$analysis <- "prenatal_no2"

# specify which DMR the annotation belongs to
combp_anno_df$dmr <- sapply(1:nrow(resu_combp), function(i) 
  rep(i, resu_combp[i,"nprobe"]), simplify = "vector") %>%
  unlist()

combp_anno_df$analysis <- "prenatal_no2"

# merge with rlm data to get effect size at each cpg
combp_anno_df_es <- merge(combp_anno_df, 
                          cb_pvals_tech %>%
                            subset(rownames(.) %in% rownames(combp_anno_df)) %>%
                            dplyr::select(effect_size, coeff_no2_prenatal_avg_2019v),
                          by = "row.names")

# subset for dmrs with  more than 1 cpg
resu_combp_sub <- resu_combp %>%
  subset(nprobe >1 ) %>%
  mutate(dmr2 = 1:nrow(.))

# subset DMR annotation for those with more than 1 CpG
combp_anno_df_es_sub <- combp_anno_df_es %>%
  subset(dmr %in% resu_combp_sub$dmr)

# check mean effect sizes
small_es <- combp_anno_df_es_sub %>%
  group_by(dmr) %>%
  summarise(mean = mean(effect_size)) %>%
  subset(abs(mean) < 0.025) %>%
  pull(dmr)

# omit dmr 5 and 18 have effect size < 0.025
combp_anno_df_es_sub_escutoff <- combp_anno_df_es_sub %>%
  subset(!dmr %in% small_es)

resu_combp_sub_escutoff <- resu_combp_sub %>%
  subset(!dmr %in% small_es)

# write out annotation for dmrs
write.csv(combp_anno_df_es_sub_escutoff,
          here("tables",
               "prenatal",
               "2023-08-14_prenatal_no2_combp_annotation_sigDMRs_CPGS.csv"))

# write out annotation for dmrs
write.csv(resu_combp_sub_escutoff,
          here("tables",
               "prenatal",
               "2023-08-14_prenatal_no2_resu_combp_sub_escutoff_sigDMRs.csv"))
```

Check whether CpGs that compose DMRs are annotated as potential mQTLs in mQTLdb.com. I copied the list of CpGs contained in DMRs into mQTLdb then saved the outputted data to compare whether the CpGs appear in hits here. 
```{r mqtl_check}
# load mQTL data
mqtls <- read.csv(here("input_data",
                       "mQTLdb_lists", 
                       "2023-08-14_CHILD_cordblood_dmr_cpgs_mqtldb.csv"))

# subset mqtls for pval and effect size according to guant (2016; PMID: 27036880)
mqtls_sub <- mqtls %>%  
  subset(p.value< 1.0e-14 & Effect.Size > 0.02)

# annotate individual cpgs within DMRs as a potential mqtl or not
combp_anno_df_es_sub_escutoff_mqtls <- combp_anno_df_es_sub_escutoff %>%
  mutate(mqtl = ifelse(Name %in% (mqtls_sub %>% 
                                    distinct(CpG) %>% 
                                    pull(CpG)), TRUE, FALSE))

combp_anno_df_es_sub_escutoff_mqtls %>% 
  group_by(dmr) %>%
  summarize(
    cpgs = length(dmr),
    birth_mqtls = sum(mqtl))

# get names of cpgs that are potential mqtls 
child_mqtls <- combp_anno_df_es_sub_escutoff_mqtls %>% subset(mqtl) %>% pull(Name) 

# get effect size min and  max
mqtls_sub %>% 
  subset(CpG %in% child_mqtls) %>% 
  pull(Effect.Size) %>% 
  min() # 0.02009

mqtls_sub %>% 
  subset(CpG %in% child_mqtls) %>% 
  pull(Effect.Size) %>% 
  max() # 0.26827

child_mqtls <- combp_anno_df_es_sub_escutoff_mqtls %>% 
  subset(mqtl) %>% 
  pull(Name) 

# write out annotation for dmrs
write.csv(combp_anno_df_es_sub_escutoff_mqtls,
          here("tables",
               "prenatal",
               "2023-08-14_prenatal_no2_cord_blood_mqtldb_birth_cpgs_mqtls.csv"))

# write out annotation for dmrs
write.csv(mqtls_sub,
          here("tables",
               "prenatal",
               "2023-08-14_prenatal_no2_combp_annotation_sigDMRs_mQTL_annotation.csv"))
```

Gene ontology term analysis of DMRs
```{r gene_ontology_enrichment_dmrs_study_pop}
# create GRanges object from combp output
combp_gr <- GRanges(
  seqnames = Rle(resu_combp_sub_escutoff$chr, c(rep(1, int = nrow(resu_combp_sub_escutoff)))),
  ranges = IRanges(start = resu_combp_sub_escutoff$start, end = resu_combp_sub_escutoff$end),
  strand = Rle(combp_anno_df_es_sub_escutoff %>%
                 distinct(dmr, .keep_all = T) %>%
                 pull(strand), rep(1, int = nrow(resu_combp_sub_escutoff))))

# perform enrichment analysis for GO terms
gst.region <-  goregion(regions = combp_gr, 
                        all.cpg = cb_pvals_tech %>%
                          pull(Name),
                        collection ="GO",
                        array.type = "450k",
                        sig.genes = T)

# get top 10 GO terms
top_go <- gst.region %>%
  arrange(P.DE) %>%
  slice_head(n=10)

# save
write.csv(top_go,
          here("output_data",
               "linear_regression",
               "2023-08-14_prenatal_no2_top_go.csv"))
```

Examine whether mean DMR methylation mediates the association between prenatal NO2 exposure and age one atopy or age one wheeze using causal mediation analysis and the mediation package.
```{r causal_mediation_dmrs_study_pop}
# load definitions of wheeze and atop
extra_meta <- read_csv(here("input_data",
                            "metadata", 
                            "extra_meta.csv"))

colnames(extra_meta)[1] <- "sample_id"

# combine with covariate data
covariates_geno_ctpc_tech_cc_aw <- merge(covariates_geno_ctpc_tech_cc,
                                         extra_meta,
                                         by = "sample_id")

# subset for particpants with only atopy or health controls (no atopy, no wheeze)
covariates_geno_ctpc_tech_cc_atopy_controls <- covariates_geno_ctpc_tech_cc_aw %>%
  subset(Group %in% c("ANW", "NANW"))

# create new variable for numeric atopy variable
covariates_geno_ctpc_tech_cc_atopy_controls <-
covariates_geno_ctpc_tech_cc_atopy_controls %>%
  mutate(group_fact = factor(Group, levels = c("NANW", "ANW"))) %>%
  mutate(group_num = as.numeric(group_fact)-1)

#########
# Atopy #
#########

# create lists to hold variable names
# set length as highest valued DMR - remove those as NULL later
total_effect_cb_no2_dmr_results_atopy <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_escutoff$dmr)))
cb_no2_dmr_results_atopy <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_escutoff$dmr)))
cb_no2_dmr_mediation_results_atopy <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_escutoff$dmr)))

# mediation analysis 
for(i in sort(unique(combp_anno_df_es_sub_escutoff$dmr))){
  
  print(paste0("Regressing atopy on pm10 and mean DNAm across DMR ", i))
  total_effect_cb_no2_dmr_results_atopy[[i]] <-  
         lm(group_num ~ no2_prenatal_avg_2019v + sex + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17 + dmr_mean,
            data = covariates_geno_ctpc_tech_cc_atopy_controls %>%
              mutate(dmr_mean = as.numeric(rowMeans(t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                                                    subset(dmr == i) %>%
                                                                    pull(Name),
                                                                    as.character(covariates_geno_ctpc_tech_cc_atopy_controls$sample_id)]*100)))))
  
  print(paste0("Regressing pm10 on mean DNAm across DMR ", i))
  cb_no2_dmr_results_atopy[[i]] <- 
         lm(dmr_mean ~ no2_prenatal_avg_2019v + sex + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17,
            data = covariates_geno_ctpc_tech_cc_atopy_controls %>%
              mutate(dmr_mean = as.numeric(rowMeans(t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                                                    subset(dmr == i) %>%
                                                                    pull(Name),
                                                                    as.character(covariates_geno_ctpc_tech_cc_atopy_controls$sample_id)]*100)))))
  
  print(paste0("Conducting mediation on mean DNAm across DMR ", i))
  set.seed(1234)
  cb_no2_dmr_mediation_results_atopy[[i]] <- 
         mediate(model.m = cb_no2_dmr_results_atopy[[i]],  
                 model.y = total_effect_cb_no2_dmr_results_atopy[[i]], 
                 treat = "no2_prenatal_avg_2019v", 
                 mediator ="dmr_mean",
                control.value = 7.323140,
                treat.value = 16.245684,
                 sim = 1000) 
}

atopy_med_df <- data.frame()
for(i in 1:length(cb_no2_dmr_mediation_results_atopy)){
  if(is.null(cb_no2_dmr_mediation_results_atopy[[i]])){next}
  print(paste("DMR", i, "results:"))
  x <- summary(cb_no2_dmr_mediation_results_atopy[[i]])
  print(x)
  tmp <- c(x["d.avg"], x["d.avg.ci"][[1]], x["d.avg.p"], 
           x["z.avg"], x["z.avg.ci"][[1]], x["z.avg.p"],
           x["tau.coef"], x["tau.ci"][[1]], x["tau.p"],
           x["n.avg"], x["n.avg.ci"][[1]], x["n.avg.p"])
  tmpdf <- do.call(rbind, tmp) %>% t()
  print(class(tmpdf))
  print(tmpdf)
  atopy_med_df <- rbind(atopy_med_df,tmpdf)
} 

write.csv(atopy_med_df,
          file=here("tables",
               "prenatal",
               "2023-06-28_no2_prenatal_cb_causal_mediation_atopy.csv"))

##########
# Wheeze #
##########

# subset for participants with only wheeze and healthy controls
covariates_geno_ctpc_tech_cc_wheeze_controls <- covariates_geno_ctpc_tech_cc_aw %>%
  subset(Group %in% c("NAW", "NANW"))

# create new variable for numeric wheeze
covariates_geno_ctpc_tech_cc_wheeze_controls <-
covariates_geno_ctpc_tech_cc_wheeze_controls %>%
  mutate(group_fact = factor(Group, levels = c("NANW", "NAW"))) %>%
  mutate(group_num = as.numeric(group_fact)-1)

# create lists to hold variable names
# set length as highest valued DMR - remove those as NULL later
total_effect_cb_no2_dmr_results_wheeze <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_escutoff$dmr)))
cb_no2_dmr_results_wheeze <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_escutoff$dmr)))
cb_no2_dmr_mediation_results_wheeze <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_escutoff$dmr)))

# mediation analysis 
for(i in sort(unique(combp_anno_df_es_sub_escutoff$dmr))){
  
  print(paste0("Regressing wheeze on pm10 and mean DNAm across DMR ", i))
  total_effect_cb_no2_dmr_results_wheeze[[i]] <-  
         lm(group_num ~ no2_prenatal_avg_2019v + sex + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17 + dmr_mean,
             data = covariates_geno_ctpc_tech_cc_wheeze_controls %>%
              mutate(dmr_mean = as.numeric(rowMeans(t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                                                    subset(dmr == i) %>%
                                                                    pull(Name),
                                                                    as.character(covariates_geno_ctpc_tech_cc_wheeze_controls$sample_id)]*100)))))
  
  print(paste0("Regressing pm10 on mean DNAm across DMR ", i))
  cb_no2_dmr_results_wheeze[[i]] <- 
         lm(dmr_mean ~ no2_prenatal_avg_2019v + sex + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17,
              data = covariates_geno_ctpc_tech_cc_wheeze_controls %>%
              mutate(dmr_mean = as.numeric(rowMeans(t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                                                    subset(dmr == i) %>%
                                                                    pull(Name),
                                                                    as.character(covariates_geno_ctpc_tech_cc_wheeze_controls$sample_id)]*100)))))
  
  print(paste0("Conducting mediation on mean DNAm across DMR ", i))
  set.seed(1234)
  cb_no2_dmr_mediation_results_wheeze[[i]] <- 
         mediate(model.m = cb_no2_dmr_results_wheeze[[i]],  
                 model.y = total_effect_cb_no2_dmr_results_wheeze[[i]], 
                 treat = "no2_prenatal_avg_2019v", 
                 mediator ="dmr_mean",
                control.value = 7.323140,
                treat.value = 16.245684,
                 sim = 1000) 
}

wheeze_med_df <- data.frame()
for(i in 1:length(cb_no2_dmr_mediation_results_wheeze)){
  if(is.null(cb_no2_dmr_mediation_results_wheeze[[i]])){next}
  print(paste("DMR", i, "results:"))
  x <- summary(cb_no2_dmr_mediation_results_wheeze[[i]])
  print(x)
  tmp <- c(x["d.avg"], x["d.avg.ci"][[1]], x["d.avg.p"], 
           x["z.avg"], x["z.avg.ci"][[1]], x["z.avg.p"],
           x["tau.coef"], x["tau.ci"][[1]], x["tau.p"],
           x["n.avg"], x["n.avg.ci"][[1]], x["n.avg.p"])
  tmpdf <- do.call(rbind, tmp) %>% t()
  print(class(tmpdf))
  print(tmpdf)
  wheeze_med_df <- rbind(wheeze_med_df,tmpdf)
} 

write.csv(wheeze_med_df,
          file=here("tables",
                    "prenatal",
                    "2023-06-28_no2_prenatal_cb_causal_mediation_wheeze.csv"))
```

#### Identification of DMRs in MALES

Identify DMRs using combp in male participants
```{r dmrs_MALES}
comb_dat_males <- cb_pvals_tech_males %>%
  dplyr::select(pval_no2_prenatal_avg_2019v, chr, pos, Name) %>%
  mutate(chr = str_sub(chr, 4,5),
         end = pos + 1) %>%
  dplyr::rename("p" = pval_no2_prenatal_avg_2019v,
                "start" = pos,
                "probe" = Name)

# need to run in "console" portion of RStudio
# results stored in home directory (~/CANDLE/)
# rename files, move to correct directory before loading 
combp(comb_dat_males, 
      seed = 0.01,
      verbose = T)

# read in csv of dmrs
resu_combp_males <- read_csv(here("tables",
                                  "prenatal",
                                  "2023-08-14_prenatal_no2_resu_combp_MALES.csv"))

resu_combp_males <- resu_combp_males %>%
  mutate(dmr = rownames(.)) %>%
  mutate(across(everything(), as.character)) %>%
  mutate(across(where(is.character), str_remove_all, pattern = fixed(" ")))

# get annotation for DMRs in combp results
combp_anno_males <- apply(resu_combp_males, 1, function(x){
  annotation_cpgs_order %>%
    as.data.frame() %>%
    subset(chr == paste("chr", x["chr"], sep = "")) %>%
    subset(pos >= as.numeric(x["start"]) & pos <= as.numeric(x["end"])-1) %>%
    mutate(dmr = x["dmr"])
})                  

resu_combp_males <- resu_combp_males %>%
  mutate(start = as.numeric(start),
         end = as.numeric(end),
         p = as.numeric(p),
         fdr = as.numeric(fdr),
         nprobe = as.numeric(nprobe))

# conver to data frame
combp_anno_df_males <- do.call(rbind, combp_anno_males)

# create new columns to specify DMR and analysis
resu_combp_males$dmr <- 1:nrow(resu_combp_males)
resu_combp_males$analysis <- "prenatal_no2"

# specify which DMR the annotation belongs to
combp_anno_df_males$dmr <- sapply(1:nrow(resu_combp_males), function(i) 
  rep(i, resu_combp_males[i,"nprobe"]), simplify = "vector") %>%
  unlist()

combp_anno_df_males$analysis <- "prenatal_no2"

# merge with rlm data to get effect size at each cpg
combp_anno_df_es_males <- merge(combp_anno_df_males, 
                          cb_pvals_tech_males %>%
                            subset(rownames(.) %in% rownames(combp_anno_df_males)) %>%
                            dplyr::select(effect_size, coeff_no2_prenatal_avg_2019v),
                          by = "row.names")

# subset for dmrs with  more than 1 cpg
resu_combp_sub_males <- resu_combp_males %>%
  subset(nprobe >1 ) %>%
  mutate(dmr2 = 1:nrow(.))

# subset DMR annotation for those with more than 1 CpG
combp_anno_df_es_sub_males <- combp_anno_df_es_males %>%
  subset(dmr %in% resu_combp_sub_males$dmr)

# check mean effect sizes
small_es_males <- combp_anno_df_es_sub_males %>%
  group_by(dmr) %>%
  summarise(mean = mean(effect_size)) %>%
  subset(abs(mean) < 0.025) %>%
  pull(dmr) # all pass

# write out annotation for dmrs
write.csv(combp_anno_df_es_sub_males,
          here("output_data",
               "linear_regression",
               "2023-08-14_prenatal_no2_combp_annotation_sig_dmr_cpgs_MALES.csv"))

# write out annotation for dmrs
write.csv(resu_combp_sub_males,
          here("output_data",
               "linear_regression",
               "2023-08-14_prenatal_no2_resu_combp_sub_escutoff_sig_dmrs_MALES.csv"))
```

Examine if male DMRs are also mQTLs
```{r mqtls_males}
# load mQTL data
mqtls_males <- read.csv(here("input_data",
                       "mQTLdb_lists", 
                       "2023-08-14_CHILD_cordblood_dmr_cpgs_mqtldb_MALES.csv"))

# subset mqtls for pval and effect size according to guant (2016; PMID: 27036880)
mqtls_sub_males <- mqtls_males %>%  
  subset(p.value< 1.0e-14 & Effect.Size > 0.02)

# annotate individual cpgs within DMRs as a potential mqtl or not
combp_anno_df_es_sub_males_mqtls <- combp_anno_df_es_sub_males %>%
  mutate(mqtl = ifelse(Name %in% (mqtls_sub_males %>% 
                                    distinct(CpG) %>% 
                                    pull(CpG)), TRUE, FALSE))

# get names of cpgs that are potential mqtls 
child_mqtls_males <- combp_anno_df_es_sub_males_mqtls %>% 
  subset(mqtl) %>% 
  pull(Name) 

# get effect size min and  max
mqtls_sub_males %>% 
  subset(CpG %in% child_mqtls_males) %>%
  pull(Effect.Size) %>% 
  min() # 0.02001

mqtls_sub_males %>% 
  subset(CpG %in% child_mqtls_males) %>% 
  pull(Effect.Size) %>% 
  max() # 0.26827

combp_anno_df_es_sub_males_mqtls %>%
    group_by(dmr) %>%
    summarize(
        cpgs = length(dmr),
        birth_mqtls = sum(mqtl))

# write out annotation for dmrs
write.csv(combp_anno_df_es_males_mqtls,
          here("tables",
               "prenatal",
               "2023-08-16_prenatal_no2_cord_blood_mqtldb_birth_cpgs_sub_MALES.csv"))

# write out annotation for dmrs
write.csv(mqtls_sub_males,
          here("tables",
               "prenatal",
               "2023-08-16_prenatal_no2_combp_annotation_sig_dmr_cpgs_mQTL_annotation_MALES.csv"))
```

Gene ontology term analysis of DMRs in male participants
```{r gene_ontology_enrichment_dmrs_males}
# create GRanges object from combp output
combp_gr_males <- GRanges(
  seqnames = Rle(resu_combp_sub_males$chr, 
                 c(rep(1, int = nrow(resu_combp_sub_males)))),
  ranges = IRanges(start = resu_combp_sub_males$start, 
                   end = resu_combp_sub_males$end),
  strand = Rle(combp_anno_df_es_sub_males %>%
                 distinct(dmr, .keep_all = T) %>%
                 pull(strand), rep(1, int = nrow(resu_combp_sub_males))))

# perform enrichment analysis for GO terms
gst.region.males <-  goregion(regions = combp_gr_males, 
                        all.cpg = cb_pvals_tech_males %>%
                          pull(Name),
                        collection ="GO",
                        array.type = "450k",
                        sig.genes = T)

# select top 10 GO terms 
top_go_males <- gst.region.males %>%
  arrange(P.DE) %>%
  slice_head(n=10)

write.csv(top_go_males,
          here("output_data",
               "linear_regression",
               "2023-08-14_prenatal_no2_top_go_MALES.csv"))
```

Causal mediation analysis of DMRs identified in male participants
```{r causal_mediation_dmrs_males}

# subset for male participants with only atopy or or healthy controls
covariates_geno_ctpc_tech_cc_atopy_controls_males <- 
  covariates_geno_ctpc_tech_cc_atopy_controls %>%
  subset(sex == "M")

#########
# Atopy #
#########

# create lists to hold variable names
# set length as highest valued DMR - remove those as NULL later
total_effect_cb_no2_dmr_results_atopy_males <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_males$dmr)))
cb_no2_dmr_results_atopy_males <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_males$dmr)))
cb_no2_dmr_mediation_results_atopy_males <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_males$dmr)))

# mediation analysis 
for(i in sort(unique(combp_anno_df_es_sub_males$dmr))){
  
  print(paste0("Regressing atopy on pm10 and mean DNAm across DMR ", i))
  total_effect_cb_no2_dmr_results_atopy_males[[i]] <-  
         lm(group_num ~ no2_prenatal_avg_2019v + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17 + dmr_mean,
              data = covariates_geno_ctpc_tech_cc_atopy_controls_males %>%
              mutate(dmr_mean = as.numeric(rowMeans(t(cb_betas_geno_cc_clean_males[combp_anno_df_es_sub_males %>%
                                                                    subset(dmr == i) %>%
                                                                    pull(Name),
                                                                    as.character(covariates_geno_ctpc_tech_cc_atopy_controls_males$sample_id)]*100)))))
  
  print(paste0("Regressing pm10 on mean DNAm across DMR ", i))
  cb_no2_dmr_results_atopy_males[[i]] <- 
         lm(dmr_mean ~ no2_prenatal_avg_2019v + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17,
              data = covariates_geno_ctpc_tech_cc_atopy_controls_males %>%
              mutate(dmr_mean = as.numeric(rowMeans(t(cb_betas_geno_cc_clean_males[combp_anno_df_es_sub_males %>%
                                                                    subset(dmr == i) %>%
                                                                    pull(Name),
                                                                    as.character(covariates_geno_ctpc_tech_cc_atopy_controls_males$sample_id)]*100)))))
  
  print(paste0("Conducting mediation on mean DNAm across DMR ", i))
  set.seed(1234)
  cb_no2_dmr_mediation_results_atopy_males[[i]] <- 
         mediate(model.m = cb_no2_dmr_results_atopy_males[[i]],  
                 model.y = total_effect_cb_no2_dmr_results_atopy_males[[i]], 
                 treat = "no2_prenatal_avg_2019v", 
                 mediator ="dmr_mean",
                control.value = 6.621813,
                treat.value = 15.849144,
                 sim = 1000) 
}

# causal mediation results
atopy_med_df_males <- data.frame()
for(i in 1:length(cb_no2_dmr_mediation_results_atopy_males)){
  if(is.null(cb_no2_dmr_mediation_results_atopy_males[[i]])){next}
  print(paste("DMR", i, "results:"))
  x <- summary(cb_no2_dmr_mediation_results_atopy_males[[i]])
  print(x)
  tmp <- c(x["d.avg"], x["d.avg.ci"][[1]], x["d.avg.p"], 
           x["z.avg"], x["z.avg.ci"][[1]], x["z.avg.p"],
           x["tau.coef"], x["tau.ci"][[1]], x["tau.p"],
           x["n.avg"], x["n.avg.ci"][[1]], x["n.avg.p"])
  tmpdf <- do.call(rbind, tmp) %>% t()
  print(class(tmpdf))
  print(tmpdf)
  atopy_med_df_males <- rbind(atopy_med_df_males,tmpdf)
} 

write.csv(atopy_med_df_males,
          file=here("tables",
                    "prenatal",
                    "2023-08-14_no2_prenatal_cb_causal_mediation_atopy_MALES.csv"))

# DMR 1 visualization
covariates_geno_ctpc_tech_cc_atopy_controls_males <- 
cbind(covariates_geno_ctpc_tech_cc_atopy_controls_males, 
      t(cb_betas_geno_cc_clean_males[combp_anno_df_es_males_mqtls %>% 
                                 subset(dmr == 1) %>% 
                                 pull(Name),colnames(cb_betas_geno_cc_clean_males) %in% as.character(covariates_geno_ctpc_tech_cc_atopy_controls_males$sample_id)] ))

# plot atopy
longdf_dmr1 <- pivot_longer(covariates_geno_ctpc_tech_cc_atopy_controls_males %>% 
                            dplyr::select(cg18256205, cg10430077, cg25944717, 
                                      sample_id, group_num, no2_prenatal_avg_2019v),
                             cols = -c(sample_id, group_num, no2_prenatal_avg_2019v), 
                             names_to = "cpg", 
                             values_to = "dnam") %>%
  mutate(cpg = factor(cpg, levels = c("cg18256205", "cg10430077", "cg25944717"))) %>%
  na.omit() %>%
  mutate(no2_colour = ifelse(no2_prenatal_avg_2019v < 7.32, "first",
                             ifelse(no2_prenatal_avg_2019v >=7.32 & no2_prenatal_avg_2019v < 12.01 , "second", ifelse(no2_prenatal_avg_2019v >= 12.01  & no2_prenatal_avg_2019v < 16.25, "third", "fourth")))) %>%
  mutate(no2_colour = factor(no2_colour, levels = c("first", "second",  "third", "fourth"))) %>%
  mutate(no2_colour_atopy = paste(no2_colour, group_num, sep = "_")) %>%
  mutate(no2_colour_atopy = factor(no2_colour_atopy, 
                                    levels = c("first_0", "second_0",  "third_0", "fourth_0",
                                               "first_1", "second_1",  "third_1", "fourth_1")))

# violin plots
dmr1_atopy_vis <-  ggplot(longdf_dmr1, 
                            aes(x=as.factor(group_num), y=dnam, 
                                group = interaction(cpg, group_num), 
                                shape = as.factor(no2_colour),
                                color = no2_colour_atopy)) +
  scale_shape_manual(values = c(0, 1, 2, 4), 
                     labels = c("first_0", "second_0",  "third_0", "fourth_0")) +
  scale_color_manual(values = c("#87a8e8", "#4484fc", "#254A90", "#10203d",
                                 "#c2a87f", "#906b25", "#513d1a", "#2b2112")) +
 # scale_x_discrete(labels = c("No", "Yes")) +
  geom_violin(draw_quantiles = c(0.5), trim = T, alpha =0.6) +
  ggbeeswarm::geom_beeswarm(dodge.width=0.9, cex = 7, size = 1, groupOnX = T, alpha = 0.8, corral = "gutter") +
  theme_bw() +
  scale_y_continuous(name = "DNA methylation (%)") +
  theme(legend.position = "none",
        axis.text = element_text(size =9),
        panel.grid.major.x = element_blank(),
        axis.title = element_blank())  +
  facet_grid(.~cpg)

ggsave(plot = dmr1_atopy_vis,
       filename  = here("figures",
                        "linear_regression",
                        "prenatal",
                        "2023-08-17_no2_cb_males_dmr1_atopy.png"), 
       width = 9, height = 4, units = "cm",dpi=600)


##########
# Wheeze #
##########

# select for male participants with only wheeze or healthy controls
covariates_geno_ctpc_tech_cc_wheeze_controls_males <-
  covariates_geno_ctpc_tech_cc_wheeze_controls %>%
  subset(sex == "M")

# create lists to hold variable names
# set length as highest valued DMR - remove those as NULL later
total_effect_cb_no2_dmr_results_wheeze_males <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_males$dmr)))
cb_no2_dmr_results_wheeze_males <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_males$dmr)))
cb_no2_dmr_mediation_results_wheeze_males <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_males$dmr)))

# mediation analysis 
for(i in sort(unique(combp_anno_df_es_sub_males$dmr))){
  
  print(paste0("Regressing wheeze on pm10 and mean DNAm across DMR ", i))
  total_effect_cb_no2_dmr_results_wheeze_males[[i]] <-  
         lm(group_num ~ no2_prenatal_avg_2019v + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17 + dmr_mean,
              data = covariates_geno_ctpc_tech_cc_wheeze_controls_males %>%
              mutate(dmr_mean = as.numeric(rowMeans(t(cb_betas_geno_cc_clean_males[combp_anno_df_es_sub_males %>%
                                                                    subset(dmr == i) %>%
                                                                    pull(Name),
                                                                    as.character(covariates_geno_ctpc_tech_cc_wheeze_controls_males$sample_id)]*100)))))
  
  print(paste0("Regressing pm10 on mean DNAm across DMR ", i))
  cb_no2_dmr_results_wheeze_males[[i]] <- 
         lm(dmr_mean ~ no2_prenatal_avg_2019v + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17,
              data = covariates_geno_ctpc_tech_cc_wheeze_controls_males %>%
              mutate(dmr_mean = as.numeric(rowMeans(t(cb_betas_geno_cc_clean_males[combp_anno_df_es_sub_males %>%
                                                                    subset(dmr == i) %>%
                                                                    pull(Name),
                                                                    as.character(covariates_geno_ctpc_tech_cc_wheeze_controls_males$sample_id)]*100)))))
  
  print(paste0("Conducting mediation on mean DNAm across DMR ", i))
  set.seed(1234)
  cb_no2_dmr_mediation_results_wheeze_males[[i]] <- 
         mediate(model.m = cb_no2_dmr_results_wheeze_males[[i]],  
                 model.y = total_effect_cb_no2_dmr_results_wheeze_males[[i]], 
                 treat = "no2_prenatal_avg_2019v", 
                 mediator ="dmr_mean",
                control.value = 6.621813,
                treat.value = 15.849144,
                 sim = 1000) 
}

wheeze_med_df_males <- data.frame()
for(i in 1:length(cb_no2_dmr_mediation_results_wheeze_males)){
  if(is.null(cb_no2_dmr_mediation_results_wheeze_males[[i]])){next}
  print(paste("DMR", i, "results:"))
  x <- summary(cb_no2_dmr_mediation_results_wheeze_males[[i]])
  print(x)
  tmp <- c(x["d.avg"], x["d.avg.ci"][[1]], x["d.avg.p"], 
           x["z.avg"], x["z.avg.ci"][[1]], x["z.avg.p"],
           x["tau.coef"], x["tau.ci"][[1]], x["tau.p"],
           x["n.avg"], x["n.avg.ci"][[1]], x["n.avg.p"])
  tmpdf <- do.call(rbind, tmp) %>% t()
  print(class(tmpdf))
  print(tmpdf)
  wheeze_med_df_males <- rbind(wheeze_med_df_males,tmpdf)
} 

write.csv(wheeze_med_df_males,
          file=here("tables",
                    "prenatal",
               "2023-08-14_no2_prenatal_cb_causal_mediation_wheeze_MALES.csv"))
```

#### Identification of DMRs in FEMALES

```{r dmrs_FEMALES}
comb_dat_females <- cb_pvals_tech_females %>%
  dplyr::select(pval_no2_prenatal_avg_2019v, chr, pos, Name) %>%
  mutate(chr = str_sub(chr, 4,5),
         end = pos + 1) %>%
  dplyr::rename("p" = pval_no2_prenatal_avg_2019v,
                "start" = pos,
                "probe" = Name)

# need to run in "console" portion of RStudio
# results stored in home directory (~/CANDLE/)
# rename files, move to correct directory before loading 
combp(comb_dat_females, 
      seed = 0.01,
      verbose = T)

# read in csv of dmrs
resu_combp_females <- read_csv(here("tables",
                                  "prenatal",
                                  "2023-08-14_prenatal_no2_resu_combp_FEMALES.csv"))

resu_combp_females <- resu_combp_females %>%
  mutate(dmr = rownames(.)) %>%
  mutate(across(everything(), as.character)) %>%
  mutate(across(where(is.character), str_remove_all, pattern = fixed(" ")))

# get annotation for DMRs in combp results
combp_anno_females <- apply(resu_combp_females, 1, function(x){
  annotation_cpgs_order %>%
    as.data.frame() %>%
    subset(chr == paste("chr", x["chr"], sep = "")) %>%
    subset(pos >= as.numeric(x["start"]) & pos <= as.numeric(x["end"])-1) %>%
    mutate(dmr = x["dmr"])
})                  

resu_combp_females <- resu_combp_females %>%
  mutate(start = as.numeric(start),
         end = as.numeric(end),
         p = as.numeric(p),
         fdr = as.numeric(fdr),
         nprobe = as.numeric(nprobe))

# conver to data frame
combp_anno_df_females <- do.call(rbind, combp_anno_females)

# create new columns to specify DMR and analysis
resu_combp_females$dmr <- 1:nrow(resu_combp_females)
resu_combp_females$analysis <- "prenatal_no2"

# specify which DMR the annotation belongs to
combp_anno_df_females$dmr <- sapply(1:nrow(resu_combp_females), function(i) 
  rep(i, resu_combp_females[i,"nprobe"]), simplify = "vector") %>%
  unlist()

combp_anno_df_females$analysis <- "prenatal_no2"

# merge with rlm data to get effect size at each cpg
combp_anno_df_es_females <- merge(combp_anno_df_females, 
                          cb_pvals_tech_females %>%
                            subset(rownames(.) %in% rownames(combp_anno_df_females)) %>%
                            dplyr::select(effect_size, coeff_no2_prenatal_avg_2019v),
                          by = "row.names")

# subset for dmrs with  more than 1 cpg
resu_combp_sub_females <- resu_combp_females %>%
  subset(nprobe >1 ) %>%
  mutate(dmr2 = 1:nrow(.))

# subset DMR annotation for those with more than 1 CpG
combp_anno_df_es_sub_females <- combp_anno_df_es_females %>%
  subset(dmr %in% resu_combp_sub_females$dmr)

# check mean effect sizes
small_es_females <- combp_anno_df_es_sub_females %>%
  group_by(dmr) %>%
  summarise(mean = mean(effect_size)) %>%
  subset(abs(mean) < 0.025) %>%
  pull(dmr) # all pass

# write out annotation for dmrs
write.csv(combp_anno_df_es_sub_females,
          here("output_data",
               "linear_regression",
               "2023-08-14_prenatal_no2_combp_annotation_sig_dmr_cpgs_FEMALES.csv"))

# write out annotation for dmrs
write.csv(resu_combp_sub_females,
          here("output_data",
               "linear_regression",
               "2023-08-14_prenatal_no2_resu_combp_sub_escutoff_sig_dmrs_FEMALES.csv"))
```

Examine if female DMRs are also mQTLs
```{r mqtls_females}
# load mQTL data
mqtls_females <- read.csv(here("input_data",
                       "mQTLdb_lists", 
                       "2023-08-14_CHILD_cordblood_dmr_cpgs_mqtldb_FEMALES.csv"))

# subset mqtls for pval and effect size according to guant (2016; PMID: 27036880)
mqtls_sub_females <- mqtls_females %>%  
  subset(p.value< 1.0e-14 & Effect.Size > 0.02)

# annotate individual cpgs within DMRs as a potential mqtl or not
combp_anno_df_es_sub_females_mqtls <- combp_anno_df_es_sub_females %>%
  mutate(mqtl = ifelse(Name %in% (mqtls_sub_females %>% 
                                    distinct(CpG) %>% 
                                    pull(CpG)), TRUE, FALSE))

# get names of cpgs that are potential mqtls 
child_mqtls_females <- combp_anno_df_es_sub_females_mqtls %>% 
  subset(mqtl) %>% 
  pull(Name) 

# get effect size min and  max
mqtls_sub_females %>% 
  subset(CpG %in% child_mqtls_females) %>%
  pull(Effect.Size) %>% 
  min() # 0.02002

mqtls_sub_females %>% 
  subset(CpG %in% child_mqtls_females) %>% 
  pull(Effect.Size) %>% 
  max() # 0.37158

combp_anno_df_es_sub_females_mqtls %>%
    group_by(dmr) %>%
    summarize(
        cpgs = length(dmr),
        birth_mqtls = sum(mqtl))

# write out annotation for dmrs
write.csv(combp_anno_df_es_sub_females_mqtls,
          here("tables",
               "prenatal",
               "2023-08-16_prenatal_no2_cord_blood_mqtldb_birth_cpgs_sub_FEMALES.csv"))

# write out annotation for dmrs
write.csv(mqtls_sub_females,
          here("tables",
               "prenatal",
               "2023-08-16_prenatal_no2_combp_annotation_sig_dmr_cpgs_mQTL_annotation_FEMALES.csv"))
```

Causal mediation analysis of DMRs identified in female participants
```{r causal_mediation_dmrs_females}
# select for female participants with only atopy or healthy controls
covariates_geno_ctpc_tech_cc_atopy_controls_females <- 
  covariates_geno_ctpc_tech_cc_atopy_controls %>%
  subset(sex == "F")

#########
# Atopy #
#########

# create lists to hold variable names
# set length as highest valued DMR - remove those as NULL later
total_effect_cb_no2_dmr_results_atopy_females <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_females$dmr)))
cb_no2_dmr_results_atopy_females <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_females$dmr)))
cb_no2_dmr_mediation_results_atopy_females <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_females$dmr)))

# mediation analysis 
for(i in sort(unique(combp_anno_df_es_sub_females$dmr))){
  
  print(paste0("Regressing atopy on pm10 and mean DNAm across DMR ", i))
  total_effect_cb_no2_dmr_results_atopy_females[[i]] <-  
         lm(group_num ~ no2_prenatal_avg_2019v + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17 + dmr_mean,
              data = covariates_geno_ctpc_tech_cc_atopy_controls_females %>%
              mutate(dmr_mean = as.numeric(rowMeans(t(cb_betas_geno_cc_clean_females[combp_anno_df_es_sub_females %>%
                                                                    subset(dmr == i) %>%
                                                                    pull(Name),
                                                                    as.character(covariates_geno_ctpc_tech_cc_atopy_controls_females$sample_id)]*100)))))
  
  print(paste0("Regressing pm10 on mean DNAm across DMR ", i))
  cb_no2_dmr_results_atopy_females[[i]] <- 
         lm(dmr_mean ~ no2_prenatal_avg_2019v + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17,
              data = covariates_geno_ctpc_tech_cc_atopy_controls_females %>%
              mutate(dmr_mean = as.numeric(rowMeans(t(cb_betas_geno_cc_clean_females[combp_anno_df_es_sub_females %>%
                                                                    subset(dmr == i) %>%
                                                                    pull(Name),
                                                                    as.character(covariates_geno_ctpc_tech_cc_atopy_controls_females$sample_id)]*100)))))
  
  print(paste0("Conducting mediation on mean DNAm across DMR ", i))
  set.seed(1234)
  cb_no2_dmr_mediation_results_atopy_females[[i]] <- 
         mediate(model.m = cb_no2_dmr_results_atopy_females[[i]],  
                 model.y = total_effect_cb_no2_dmr_results_atopy_females[[i]], 
                 treat = "no2_prenatal_avg_2019v", 
                 mediator ="dmr_mean",
                control.value = 8.015644,
                treat.value = 16.609526,
                 sim = 1000) 
}

atopy_med_df_females <- data.frame()
for(i in 1:length(cb_no2_dmr_mediation_results_atopy_females)){
  if(is.null(cb_no2_dmr_mediation_results_atopy_females[[i]])){next}
  print(paste("DMR", i, "results:"))
  x <- summary(cb_no2_dmr_mediation_results_atopy_females[[i]])
  print(x)
  tmp <- c(x["d.avg"], x["d.avg.ci"][[1]], x["d.avg.p"], 
           x["z.avg"], x["z.avg.ci"][[1]], x["z.avg.p"],
           x["tau.coef"], x["tau.ci"][[1]], x["tau.p"],
           x["n.avg"], x["n.avg.ci"][[1]], x["n.avg.p"])
  tmpdf <- do.call(rbind, tmp) %>% t()
  print(class(tmpdf))
  print(tmpdf)
  atopy_med_df_females <- rbind(atopy_med_df_females,tmpdf)
} 

write.csv(atopy_med_df_females,
          file=here("tables",
                    "prenatal",
                    "2023-08-14_no2_prenatal_cb_causal_mediation_atopy_FEMALES.csv"))

##########
# Wheeze #
##########

# select for female participants with only wheeze or healthy controls
covariates_geno_ctpc_tech_cc_wheeze_controls_females <- 
  covariates_geno_ctpc_tech_cc_wheeze_controls %>%
  subset(sex == "F")

# create lists to hold variable names
# set length as highest valued DMR - remove those as NULL later
total_effect_cb_no2_dmr_results_wheeze_females <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_females$dmr)))
cb_no2_dmr_results_wheeze_females <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_females$dmr)))
cb_no2_dmr_mediation_results_wheeze_females <- vector(mode = "list", length = max(unique(combp_anno_df_es_sub_females$dmr)))

# mediation analysis 
for(i in sort(unique(combp_anno_df_es_sub_females$dmr))){
  
  print(paste0("Regressing wheeze on pm10 and mean DNAm across DMR ", i))
  total_effect_cb_no2_dmr_results_wheeze_females[[i]] <-  
         lm(group_num ~ no2_prenatal_avg_2019v + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17 + dmr_mean,
              data = covariates_geno_ctpc_tech_cc_wheeze_controls_females %>%
              mutate(dmr_mean = as.numeric(rowMeans(t(cb_betas_geno_cc_clean_females[combp_anno_df_es_sub_females %>%
                                                                    subset(dmr == i) %>%
                                                                    pull(Name),
                                                                    as.character(covariates_geno_ctpc_tech_cc_wheeze_controls_females$sample_id)]*100)))))
  
  print(paste0("Regressing pm10 on mean DNAm across DMR ", i))
  cb_no2_dmr_results_wheeze_females[[i]] <- 
         lm(dmr_mean ~ no2_prenatal_avg_2019v + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17,
              data = covariates_geno_ctpc_tech_cc_wheeze_controls_females %>%
              mutate(dmr_mean = as.numeric(rowMeans(t(cb_betas_geno_cc_clean_females[combp_anno_df_es_sub_females %>%
                                                                    subset(dmr == i) %>%
                                                                    pull(Name),
                                                                    as.character(covariates_geno_ctpc_tech_cc_wheeze_controls_females$sample_id)]*100)))))
  
  print(paste0("Conducting mediation on mean DNAm across DMR ", i))
  set.seed(1234)
  cb_no2_dmr_mediation_results_wheeze_females[[i]] <- 
         mediate(model.m = cb_no2_dmr_results_wheeze_females[[i]],  
                 model.y = total_effect_cb_no2_dmr_results_wheeze_females[[i]], 
                 treat = "no2_prenatal_avg_2019v", 
                 mediator ="dmr_mean",
                control.value = 8.015644,
                treat.value = 16.609526,
                 sim = 1000) 
}

wheeze_med_df_females <- data.frame()
for(i in 1:length(cb_no2_dmr_mediation_results_wheeze_females)){
  if(is.null(cb_no2_dmr_mediation_results_wheeze_females[[i]])){next}
  print(paste("DMR", i, "results:"))
  x <- summary(cb_no2_dmr_mediation_results_wheeze_females[[i]])
  print(x)
  tmp <- c(x["d.avg"], x["d.avg.ci"][[1]], x["d.avg.p"], 
           x["z.avg"], x["z.avg.ci"][[1]], x["z.avg.p"],
           x["tau.coef"], x["tau.ci"][[1]], x["tau.p"],
           x["n.avg"], x["n.avg.ci"][[1]], x["n.avg.p"])
  tmpdf <- do.call(rbind, tmp) %>% t()
  print(class(tmpdf))
  print(tmpdf)
  wheeze_med_df_females <- rbind(wheeze_med_df_females,tmpdf)
} 

write.csv(wheeze_med_df_females,
          file=here("tables",
                    "prenatal",
               "2023-08-14_no2_prenatal_cb_causal_mediation_wheeze_FEMALES.csv"))
```

#### Compare DMRs between study pop, males, and females

```{r compare_dmrs}
male_dmrs <- as(resu_combp_sub_males, "GRanges")
female_dmrs <- as(resu_combp_sub_females, "GRanges")
pop_dmrs <- as(resu_combp_sub_escutoff, "GRanges")

male_pop <- findOverlaps(male_dmrs, pop_dmrs) # 5 
male_female <- findOverlaps(male_dmrs, female_dmrs) # 0 
pop_female <- findOverlaps(pop_dmrs, female_dmrs) # 0 

# create new cols to hold unique dmr info
resu_combp_sub_males$unique_dmrs_id <- NA
resu_combp_sub_females$unique_dmrs_id <- NA
resu_combp_sub_escutoff$unique_dmrs_id <- NA

# identify unique dmrs 
resu_combp_sub_males[male_pop@from, "unique_dmrs_id"] 
resu_combp_sub_males[-male_pop@from, "unique_dmrs_id"] 

resu_combp_sub_escutoff[male_pop@to, "unique_dmrs_id"] 
resu_combp_sub_escutoff[-male_pop@to, "unique_dmrs_id"] 
```


### Genome-wide DNA methylation changes 

Mean array-wide DNAm chanages
```{r mean_array_wide_dnam}
# convert betas to df
cb_betas_geno_cc_clean_df <- as.data.frame(cb_betas_geno_cc_clean)

# get mean array wide dnam
methmeans <- colMeans(cb_betas_geno_cc_clean_df)

# add mean array wide dnam to data frame
covariates_geno_ctpc_tech_cc <- cbind(covariates_geno_ctpc_tech_cc, methmeans)

# regression for array wide dnam
meanmeth_lm <- 
  lm(methmeans ~ no2_prenatal_avg_2019v + sex + gestational_age  + 
       prenatal_maternal_smoke + maternal_education_length + 
       Comp.1 + Comp.2 + Comp.3 + Comp.4 +
       genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
       sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
       sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
     data=covariates_geno_ctpc_tech_cc)

# get summary
summary(meanmeth_lm) #no2 not sig

# estimate marginal effects of prenatal NO2 exposure on DNAm
globaldna_no2_me <- ggpredict(meanmeth_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  dplyr::rename("methmeans" = predicted,
                "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc$methmeans, 
    covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v,
    method="pearson") 

# effect size
range_cb = max(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v)

summary(meanmeth_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_cb 
```

LINE-1 specific DNAm
```{r mean_line1_dnam}
# read in line annotation from UCSC and repeat masker 
repeats_450k <- read.delim(file = gzfile(here("input_data", 
                                              "ucsc_rmsk_lines", 
                                              "ucsc_repeatmasker_enc_methyl450_gm12878.gz")),
                           header = T, skip = 1, sep = "\t")

# subset repeats for L1 LINE elements that are not in sex chromosomes
lines_450k  <- repeats_450k %>% 
  subset(rmsk.repClass == "LINE") %>%
  subset(rmsk.repFamily == "L1") %>%
  subset(!wgEncodeHaibMethyl450Gm12878SitesRep1.chrom %in% c("chrX", "chrY")) %>%
  subset(!is.na(wgEncodeHaibMethyl450Gm12878SitesRep1.chromStart))

# subset betas for those that overlap with line elements
cb_betas_geno_cc_clean_df_lines <- cb_betas_geno_cc_clean_df %>% 
  subset(rownames(.) %in% lines_450k$wgEncodeHaibMethyl450Gm12878SitesRep1.name)

# num of cpgs annotated to lines 
nrow(cb_betas_geno_cc_clean_df_lines) # 7302 

# get mean of line meth 
line_meth <- colMeans(cb_betas_geno_cc_clean_df_lines)

# cbind line methylation to covariate data frame
covariates_geno_ctpc_tech_cc <- cbind(covariates_geno_ctpc_tech_cc, line_meth)

# line dnam regression
line_meth_lm <- lm(line_meth ~ no2_prenatal_avg_2019v + sex + gestational_age  + 
                     prenatal_maternal_smoke + maternal_education_length + 
                     Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                     genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                     sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                     sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                   data=covariates_geno_ctpc_tech_cc)

# summary of line regression
summary(line_meth_lm) #no2 not sig

# estimate marginal effects of prenatal NO2 exposure on DNAm
line_meth_no2_me <- ggpredict(line_meth_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  dplyr::rename("line_meth" = predicted,
                "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc$line_meth, 
    covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v,
    method="pearson") # 0.03949858

# effect size
summary(line_meth_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_cb # 
```

#### Genome wide DNAm changes in males

Mean array-wide DNAm changes in males
```{r mean_array_wide_dnam_males}
# convert betas to df
cb_betas_geno_cc_clean_df_males <- as.data.frame(cb_betas_geno_cc_clean_males)

methmeans_males <- colMeans(cb_betas_geno_cc_clean_df_males)

covariates_geno_ctpc_tech_cc_males <- cbind(covariates_geno_ctpc_tech_cc_males, 
                                      methmeans_males)

# mean array wide dnam regression
meanmeth_lm_males <- 
  lm(methmeans_males ~ no2_prenatal_avg_2019v + gestational_age  + 
       prenatal_maternal_smoke + maternal_education_length + 
       Comp.1 + Comp.2 + Comp.3 + Comp.4 +
       genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
       sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
       sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
     data=covariates_geno_ctpc_tech_cc_males)

#get summary
summary(meanmeth_lm_males) #no2 not sig  

# estimate marginal effects of prenatal NO2 exposure on DNAm
globaldna_no2_me_males <- ggpredict(meanmeth_lm_males, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  dplyr::rename("methmeans_males" = predicted,
                "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_males$methmeans_males, 
    covariates_geno_ctpc_tech_cc_males$no2_prenatal_avg_2019v,
    method="pearson") # 

# effect size
range_cb_males = max(covariates_geno_ctpc_tech_cc_males$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_males$no2_prenatal_avg_2019v)

summary(meanmeth_lm_males)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_cb_males 
```

LINE-1 specific DNAm in males
```{r mean_line1_dnam_males}
# read in line annotation from UCSC and repeat masker 
repeats_450k <- read.delim(file = gzfile(here("input_data", 
                                              "ucsc_rmsk_lines", 
                                              "ucsc_repeatmasker_enc_methyl450_gm12878.gz")),
                           header = T, skip = 1, sep = "\t")

# subset repeats for L1 LINE elements that are not in sex chromosomes
lines_450k  <- repeats_450k %>% 
  subset(rmsk.repClass == "LINE") %>%
  subset(rmsk.repFamily == "L1") %>%
  subset(!wgEncodeHaibMethyl450Gm12878SitesRep1.chrom %in% c("chrX", "chrY")) %>%
  subset(!is.na(wgEncodeHaibMethyl450Gm12878SitesRep1.chromStart))

# subset betas for those that overlap with line elements
cb_betas_geno_cc_clean_df_lines_males <- cb_betas_geno_cc_clean_df_males %>% 
  subset(rownames(.) %in% lines_450k$wgEncodeHaibMethyl450Gm12878SitesRep1.name)

# num of cpgs annotated to lines 
nrow(cb_betas_geno_cc_clean_df_lines_males) # 7302 

# get mean of line meth 
line_meth_males <- colMeans(cb_betas_geno_cc_clean_df_lines_males)

# cbind line methylation to covariate data frame
covariates_geno_ctpc_tech_cc_males <- cbind(covariates_geno_ctpc_tech_cc_males, 
                                      line_meth_males)

# line dnam regression
line_meth_lm_males <- lm(line_meth_males ~ no2_prenatal_avg_2019v + gestational_age  + 
                     prenatal_maternal_smoke + maternal_education_length + 
                     Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                     genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                     sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                     sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                   data=covariates_geno_ctpc_tech_cc_males)

# summary of line regression
summary(line_meth_lm_males) #no2 not sig

# estimate marginal effects of prenatal NO2 exposure on DNAm
line_meth_no2_me_males <- ggpredict(line_meth_lm_males, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  dplyr::rename("line_meth_males" = predicted,
                "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_males$line_meth_males, 
    covariates_geno_ctpc_tech_cc_males$no2_prenatal_avg_2019v,
    method="pearson") 

# effect size
summary(line_meth_lm_males)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_cb_males 
```

#### Genome wide DNAm changes in females

Mean array-wide DNAm changes in females
```{r mean_array_wide_dnam_females}
# convert betas to df
cb_betas_geno_cc_clean_df_females <- as.data.frame(cb_betas_geno_cc_clean_females)

methmeans_females <- colMeans(cb_betas_geno_cc_clean_df_females)

covariates_geno_ctpc_tech_cc_females <- cbind(covariates_geno_ctpc_tech_cc_females, 
                                      methmeans_females)

# mean array wide dnam regression
meanmeth_lm_females <- 
  lm(methmeans_females ~ no2_prenatal_avg_2019v + gestational_age  + 
       prenatal_maternal_smoke + maternal_education_length + 
       Comp.1 + Comp.2 + Comp.3 + Comp.4 +
       genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
       sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
       sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
     data=covariates_geno_ctpc_tech_cc_females)

#get summary
summary(meanmeth_lm_females) #no2 not sig 

# estimate marginal effects of prenatal NO2 exposure on DNAm
globaldna_no2_me_females <- ggpredict(meanmeth_lm_females, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  dplyr::rename("methmeans_females" = predicted,
                "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_females$methmeans_females, 
    covariates_geno_ctpc_tech_cc_females$no2_prenatal_avg_2019v,
    method="pearson") 

# effect size
range_cb_females = max(covariates_geno_ctpc_tech_cc_females$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_females$no2_prenatal_avg_2019v)

summary(meanmeth_lm_females)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_cb_females 
```

LINE-1 specific DNAm in females
```{r mean_line1_dnam_females}
# read in line annotation from UCSC and repeat masker 
repeats_450k <- read.delim(file = gzfile(here("input_data", 
                                              "ucsc_rmsk_lines", 
                                              "ucsc_repeatmasker_enc_methyl450_gm12878.gz")),
                           header = T, skip = 1, sep = "\t")

# subset repeats for L1 LINE elements that are not in sex chromosomes
lines_450k  <- repeats_450k %>% 
  subset(rmsk.repClass == "LINE") %>%
  subset(rmsk.repFamily == "L1") %>%
  subset(!wgEncodeHaibMethyl450Gm12878SitesRep1.chrom %in% c("chrX", "chrY")) %>%
  subset(!is.na(wgEncodeHaibMethyl450Gm12878SitesRep1.chromStart))

# subset betas for those that overlap with line elements
cb_betas_geno_cc_clean_df_lines_females <- cb_betas_geno_cc_clean_df_females %>% 
  subset(rownames(.) %in% lines_450k$wgEncodeHaibMethyl450Gm12878SitesRep1.name)

# num of cpgs annotated to lines 
nrow(cb_betas_geno_cc_clean_df_lines_females) # 7302 

# get mean of line meth 
line_meth_females <- colMeans(cb_betas_geno_cc_clean_df_lines_females)

# cbind line methylation to covariate data frame
covariates_geno_ctpc_tech_cc_females <- cbind(covariates_geno_ctpc_tech_cc_females, 
                                      line_meth_females)

# line dnam regression
line_meth_lm_females <- lm(line_meth_females ~ no2_prenatal_avg_2019v + gestational_age  + 
                     prenatal_maternal_smoke + maternal_education_length + 
                     Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                     genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                     sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                     sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                   data=covariates_geno_ctpc_tech_cc_females)

# summary of line regression
summary(line_meth_lm_females) #no2 not sig

# estimate marginal effects of prenatal NO2 exposure on DNAm
line_meth_no2_me_females <- ggpredict(line_meth_lm_females, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  dplyr::rename("line_meth_females" = predicted,
                "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_females$line_meth_females, 
    covariates_geno_ctpc_tech_cc_females$no2_prenatal_avg_2019v,
    method="pearson") 

# effect size
summary(line_meth_lm_females)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_cb_females
```


#### Plot genome wide DNAm

```{r plot_genome_wide_dnam}
###################
# array wide dnam #
###################

global_dnam_plot <- ggplot(covariates_geno_ctpc_tech_cc,
                           aes(x = no2_prenatal_avg_2019v, 
                               y= methmeans,
                               shape = sex,
                               colour = sex)) + 
  scale_shape_manual(values = c(1,4)) +
    scale_colour_manual(values = c("#D81B60", "#1E88E5")) +
  geom_line(data = globaldna_no2_me, 
            aes(x=no2_prenatal_avg_2019v, y = methmeans),
            col="#FFC107", size=0.5, linetype="solid", inherit.aes = F) +
  geom_line(data = globaldna_no2_me_males, 
            aes(x=no2_prenatal_avg_2019v, y = methmeans_males),
            col="#1E88E5", size=0.5, linetype="dashed", inherit.aes = F) +
  geom_line(data = globaldna_no2_me_females, 
            aes(x=no2_prenatal_avg_2019v, y = methmeans_females),
            col="#D81B60", size=0.5, linetype="twodash", inherit.aes = F) +
  geom_point(alpha=0.7,  size=1) + 
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.title = element_blank(),
        legend.position = "none") +
  lims(y=c(0.48,0.54))

ggsave(plot = global_dnam_plot, 
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal",
                       "2023-08-14_cb_global_dnam_all_males_females.png"),
        width = 4.0, height = 3.2, units = "cm", dpi = 600)

###########
# L1 LINE #
###########

line_dnam_plot <- ggplot(covariates_geno_ctpc_tech_cc,
                         aes(x = no2_prenatal_avg_2019v, 
                             y= line_meth,
                             shape = sex,
                             colour = sex)) + 
    scale_shape_manual(values = c(1,4)) +
    scale_colour_manual(values = c("#D81B60", "#1E88E5")) +
  geom_line(data = line_meth_no2_me, 
            aes(x=no2_prenatal_avg_2019v, y = line_meth),
            col="#FFC107", size=0.5, linetype="solid", inherit.aes = F) +
  geom_line(data = line_meth_no2_me_males, 
            aes(x=no2_prenatal_avg_2019v, y = line_meth_males),
            col="#1E88E5", size=0.5, linetype="dashed", inherit.aes = F) +
  geom_line(data = line_meth_no2_me_females, 
            aes(x=no2_prenatal_avg_2019v, y = line_meth_females),
            col="#D81B60", size=0.5, linetype="twodash", inherit.aes = F) +
  geom_point(alpha=0.7,  size=1) + 
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.title = element_blank(),
        legend.position = "none") +
  lims(y=c(0.80, 0.86))

ggsave(plot = line_dnam_plot, 
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal",
                       "2023-08-14_cb_line_dnam_all_males_females.png"),
       width = 4.0, height = 3.2, units = "cm", dpi = 600)
```


### Interaction effect with birth month in prenatal model 

Try re-running limma with birth month/ interaction between birth month and air pollution.
```{r interaction_terms_to_assess_birthmonth}
#want to use anova to compare models usign anova
#anova only accepts models from lm
#use lm to compare how birth month affects cpgs above 10^-3 sig 

#You should be able to ask for confidence intervals for the slope coefficients for each month.  #even in a tabular display, you may see that “smoothly varying” pattern we were discussing.
#Then, you could use those to plot a sort of “here’s how the slope varies by month” picture.Sort #of like forest plots for meta analysis results, if that makes sense.

#Also, before we decide it doesn’t matter, the individual slopes might look fairly similar, but the appropriate overall test would be to check the anova(fit_with_interaction) and check the p-value on the interaction term.

#start by investigating cpgs identified as nominally significant in volcano plot


#  get names of top 10 cpgs based on effect size and pvalue from ewas
cb_pvals_tech %>%
  subset(effect_size > 0.025 | effect_size < -0.025) %>%
  arrange(pval_no2_prenatal_avg_2019v) %>%
  slice_head(n=10) %>%
  rownames()

#############
#cg03508112 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg03508112 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg03508112", ] 

# fit wiht lm
cg03508112_lm_bm <- lm(cb_cg03508112 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg03508112_lm <- lm(cb_cg03508112 ~  no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg03508112_lm, cg03508112_lm_bm, test = "LRT") # p = 0.1735
summary(cg03508112_lm_bm) 

# get plot of birth month effect with 95% CI
cg03508112_lm_bm_plot <- tidy(cg03508112_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-30, 30) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(filename="2023-08-14_CHILD_prenatal_no2_cg03508112_bm.png",
       path =  here("figures", "linear_regression", "prenatal"),
       width = 4.8, height = 4, units = "cm", dpi = 600)


#############
#cg14658149 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg14658149 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg14658149", ] 

# fit wiht lm
cg14658149_lm_bm <- lm(cb_cg14658149 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg14658149_lm <- lm(cb_cg14658149 ~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg14658149_lm, cg14658149_lm_bm, test = "LRT") # p = 0.2653
summary(cg14658149_lm_bm) 

# get plot of birth month effect with 95% CI
cg14658149_lm_bm_plot <- tidy(cg14658149_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  #ylim(-35, 35) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg14658149_lm_bm_plot,
       filename="2023-08-14_CHILD_prenatal_no2_cg14658149_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm", dpi = 600)


#############
#cg12228123 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg12228123 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg12228123", ] 

# fit wiht lm
cg12228123_lm_bm <- lm(cb_cg12228123 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg12228123_lm <- lm(cb_cg12228123 ~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg12228123_lm, cg12228123_lm_bm, test = "LRT") # p = 0.0.04706
summary(cg12228123_lm_bm) 

# get plot of birth month effect with 95% CI
cg12228123_lm_bm_plot <- tidy(cg12228123_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  #ylim(-80, 80) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg12228123_lm_bm_plot,
       filename="2023-08-14_CHILD_prenatal_no2_cg12228123_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm", dpi=600)


#############
#cg14491284 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg14491284 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg14491284", ] 

# fit wiht lm
cg14491284_lm_bm <- lm(cb_cg14491284 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg14491284_lm <- lm(cb_cg14491284 ~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg14491284_lm, cg14491284_lm_bm, test = "LRT") # p = 0.7623
summary(cg14491284_lm_bm) 

# get plot of birth month effect with 95% CI
cg14491284_lm_bm_plot <- tidy(cg14491284_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  #ylim(-30, 30) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg14491284_lm_bm_plot,
       filename="2023-08-14_CHILD_prenatal_no2_cg14491284_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm")


#############
#cg02074714 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg02074714 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg02074714", ] 

# fit wiht lm
cg02074714_lm_bm <- lm(cb_cg02074714 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg02074714_lm <- lm(cb_cg02074714 ~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg02074714_lm, cg02074714_lm_bm, test = "LRT") # p = 0.9763
summary(cg02074714_lm_bm) 

# get plot of birth month effect with 95% CI
cg02074714_lm_bm_plot <- tidy(cg02074714_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  #ylim(-50, 50) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg02074714_lm_bm_plot,
       filename="2023-08-14_CHILD_prenatal_no2_cg02074714_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm", dpi=600)


#############
#cg22230559 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg22230559 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg22230559", ] 

# fit wiht lm
cg22230559_lm_bm <- lm(cb_cg22230559 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg22230559_lm <- lm(cb_cg22230559 ~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg22230559_lm, cg22230559_lm_bm) # p = 0.3725
summary(cg22230559_lm_bm) 

# get plot of birth month effect with 95% CI
cg22230559_lm_bm_plot <- tidy(cg22230559_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
 # ylim(-20, 20) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg22230559_lm_bm_plot,
       filename="2023-08-14_CHILD_prenatal_no2_cg22230559_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm", dpi = 600)


#############
#cg15194163 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg15194163 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg15194163", ] 

# fit wiht lm
cg15194163_lm_bm <- lm(cb_cg15194163 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg15194163_lm <- lm(cb_cg15194163 ~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg15194163_lm, cg15194163_lm_bm, test = "LRT") # p = 0.5162
summary(cg15194163_lm_bm) 

# get plot of birth month effect with 95% CI
cg15194163_lm_bm_plot <- tidy(cg15194163_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
 # ylim(-30, 30) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg15194163_lm_bm_plot,
       filename="2023-08-14_CHILD_prenatal_no2_cg15194163_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm", dpi = 600)


#############
#cg06518781 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg06518781 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg06518781", ] 

# fit wiht lm
cg06518781_lm_bm <- lm(cb_cg06518781 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg06518781_lm <- lm(cb_cg06518781 ~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg06518781_lm, cg06518781_lm_bm, test = "LRT") # p = 0.9105
summary(cg06518781_lm_bm) 

# get plot of birth month effect with 95% CI
cg06518781_lm_bm_plot <- tidy(cg06518781_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
#  ylim(-20, 20) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg06518781_lm_bm_plot,
       filename="2023-08-14_CHILD_prenatal_no2_cg06518781_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm")


#############
#cg17646392#             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg17646392<- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg07658809", ] 

# fit wiht lm
cg07658809_lm_bm <- lm(cb_cg17646392~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg07658809_lm <- lm(cb_cg17646392~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg07658809_lm, cg07658809_lm_bm, test = "LRT") # p = 0.7547
summary(cg07658809_lm_bm) 

# get plot of birth month effect with 95% CI
cg07658809_lm_bm_plot <- tidy(cg07658809_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
 # ylim(-30, 30) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg07658809_lm_bm_plot,
       filename="2023-08-14_CHILD_prenatal_no2_cg07658809_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm",dpi =600)


#############
#cg16321846 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg16321846 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg16321846", ] 

# fit wiht lm
cg16321846_lm_bm <- lm(cb_cg16321846 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg16321846_lm <- lm(cb_cg16321846 ~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg16321846_lm, cg16321846_lm_bm, test = "LRT") # p = 0.1736
summary(cg16321846_lm_bm) 

# get plot of birth month effect with 95% CI
cg16321846_lm_bm_plot <- tidy(cg16321846_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
 # ylim(-20, 20) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg16321846_lm_bm_plot,
       filename="2023-08-14_CHILD_prenatal_no2_cg16321846_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm",dpi =600)
```


### Table 1 of prenatal model

"Table 1" of prenatal model
```{r prenatal_table1}
library(table1)

# get cell type estimates (not pcs)
cb_cell_estimates <- child_fscbc_ecc2_cb$prop
dim(cb_cell_estimates)

# subset out duplicates
cb_cell_estimates_sub <- cb_cell_estimates %>% 
  subset(rownames(.) %in% paste(cb_pdata$Sentrix_ID, cb_pdata$Sentrix_Position, sep = "_"))
dim(cb_cell_estimates_sub)

# rename rows
identical(rownames(cb_cell_estimates_sub),
          paste(cb_pdata$Sentrix_ID, cb_pdata$Sentrix_Position, sep = "_")) # TRUE
rownames(cb_cell_estimates_sub) <- cb_pdata$Sample_Label

# subset for participants with complete data
cb_cell_estimates_sub_cc <- cb_cell_estimates_sub %>%
  subset(rownames(.) %in% covariates_geno_ctpc_tech_cc_aw$sample_id)
dim(cb_cell_estimates_sub_cc) # 128 6

# check order

identical(rownames(cb_cell_estimates_sub_cc), 
          as.character(covariates_geno_ctpc_tech_cc_aw$sample_id)) # TRUE

# bind cell type estimates with covariate info
covariates_geno_ctpc_tech_cc_aw_cbmc <- cbind(covariates_geno_ctpc_tech_cc_aw, 
                                           cb_cell_estimates_sub_cc)

# table 1
table1(~sex + gestational_age + no2_prenatal_avg_2019v + 
         maternal_education_length +  maternal_race + 
         prenatal_maternal_smoke + CD4T + CD8T + nRBC + Bcell + Gran +Mono + NK + Group,
       data = covariates_geno_ctpc_tech_cc_aw_cbmc)

# prenatal no2 quartiles
<<<<<<< HEAD
quantile(covariates_geno_ctpc_tech_cc_aw_cbmc$no2_prenatal_avg_2019v)

anova(lm(CD4T~ studycentre, covariates_geno_ctpc_tech_cc_aw_cbmc))
anova(lm(CD8T~ studycentre, covariates_geno_ctpc_tech_cc_aw_cbmc))
anova(lm(Mono~ studycentre, covariates_geno_ctpc_tech_cc_aw_cbmc))
anova(lm(NK~ studycentre, covariates_geno_ctpc_tech_cc_aw_cbmc)) 
anova(lm(Bcell~ studycentre, covariates_geno_ctpc_tech_cc_aw_cbmc)) 
anova(lm(nRBC~ studycentre, covariates_geno_ctpc_tech_cc_aw_cbmc)) 
anova(lm(Gran~ studycentre, covariates_geno_ctpc_tech_cc_aw_cbmc)) 

t.test(CD4T~ sex, covariates_geno_ctpc_tech_cc_aw_cbmc) 
t.test(CD8T~ sex, covariates_geno_ctpc_tech_cc_aw_cbmc) 
t.test(Gran~ sex, covariates_geno_ctpc_tech_cc_aw_cbmc) 
t.test(Mono~ sex, covariates_geno_ctpc_tech_cc_aw_cbmc) 
t.test(NK~ sex, covariates_geno_ctpc_tech_cc_aw_cbmc) 
t.test(Bcell~ sex, covariates_geno_ctpc_tech_cc_aw_cbmc) 
t.test(nRBC~ sex, covariates_geno_ctpc_tech_cc_aw_cbmc) 
=======
quantile(covariates_geno_ctpc_tech_cc_cbmc$no2_prenatal_avg_2019v)

anova(lm(CD4T~ studycentre, covariates_geno_ctpc_tech_cc_cbmc))
anova(lm(CD8T~ studycentre, covariates_geno_ctpc_tech_cc_cbmc))
anova(lm(Mono~ studycentre, covariates_geno_ctpc_tech_cc_cbmc))
anova(lm(NK~ studycentre, covariates_geno_ctpc_tech_cc_cbmc)) 
anova(lm(Bcell~ studycentre, covariates_geno_ctpc_tech_cc_cbmc)) 
anova(lm(nRBC~ studycentre, covariates_geno_ctpc_tech_cc_cbmc)) 

t.test(CD4T~ sex, covariates_geno_ctpc_tech_cc_cbmc) 
t.test(CD8T~ sex, covariates_geno_ctpc_tech_cc_cbmc) 
t.test(Mono~ sex, covariates_geno_ctpc_tech_cc_cbmc) 
t.test(NK~ sex, covariates_geno_ctpc_tech_cc_cbmc) 
t.test(Bcell~ sex, covariates_geno_ctpc_tech_cc_cbmc) 
t.test(nRBC~ sex, covariates_geno_ctpc_tech_cc_cbmc) 
>>>>>>> f62b21d7e4615973aae12bc89a4c51ff42b0ad1e

anova(lm(no2_prenatal_avg_2019v~ studycentre, covariates_geno_ctpc_tech_cc_cbmc)) 
anova(lm(gestational_age~ studycentre, covariates_geno_ctpc_tech_cc_cbmc)) 
anova(lm(maternal_education_length~ studycentre, covariates_geno_ctpc_tech_cc_cbmc)) 

fisher.test(covariates_geno_ctpc_tech_cc_aw_cbmc %>% subset(Group != "AW") %>% pull(Group),
             covariates_geno_ctpc_tech_cc_aw_cbmc %>% subset(Group != "AW") %>% pull(studycentre))

fisher.test(covariates_geno_ctpc_tech_cc_aw_cbmc %>% subset(Group != "AW") %>% pull(Group),
             covariates_geno_ctpc_tech_cc_aw_cbmc %>% subset(Group != "AW") %>% pull(studycentre))

fisher.test(covariates_geno_ctpc_tech_cc_cbmc$maternal_race,
            covariates_geno_ctpc_tech_cc_cbmc$studycentre)

fisher.test(covariates_geno_ctpc_tech_cc_cbmc$maternal_smoking_y1,
            covariates_geno_ctpc_tech_cc_cbmc$studycentre)

fisher.test(covariates_geno_ctpc_tech_cc_cbmc$prenatal_maternal_smoke,
            covariates_geno_ctpc_tech_cc_cbmc$studycentre)

```

## Persistence of prenatal DNAm changes at age one

First we need to prepare the age one data for the linear model. Then we need to calculate the number of SVs needed for this model. 

### Load and clean age one data 

Get age one pdata
```{r y1_pdata}
#subset out age one pdata
y1_pdata <- subset(pdata_nodups, Tissue == "P")

#check size
dim(y1_pdata) # 145 17
```


Get age one betas
```{r y1_betas}
#subset out PBMC betas 
y1_betas <- betas[ ,colnames(betas) %in% rownames(y1_pdata)]

#check if age one betas and pdata are in same order
identical(colnames(y1_betas), rownames(y1_pdata)) # TRUE

#rename colums with sample label (5 dig num)
colnames(y1_betas) <- y1_pdata$Sample_Label

dim(y1_betas) # 145
```


Load age one cell type PCs
```{r y1_celltypePCs}
#load age one PBMC estimates
load(here("output_data", 
          "deconvolution", 
          "2023-08-14_PBMC_PCs.RData")) # y1_ilr_ca

# load deconvolution for later use - remove probes 
load(here("output_data",
          "deconvolution", 
          "2023-06-28_cbmc_pbmc_deconvolution.Rdata"))

y1_pcs <- y1_ilr_pca$scores

#rename columns of PBMCs for clarity
colnames(y1_pcs) <- paste("pbmc", colnames(y1_pcs), sep=".")

#rename row of cell type PCs to sample label (5 digit num)
identical(rownames(y1_pcs), rownames(y1_pdata)) # TRUE
rownames(y1_pcs) <- y1_pdata$Sample_Label

# subset cell type PCs for complete cases
y1_pcs_sub <- y1_pcs[rownames(y1_pcs) %in% covariates_geno_ctpc_tech_cc$sample_id,]
dim(y1_pcs_sub) # 128

# add age one PBMCs to covariate data frame
covariates_geno_ctpc_tech_cc_y1 <- cbind(covariates_geno_ctpc_tech_cc, y1_pcs_sub)
```

Add age one batch variables to covariate data frame
```{r y1_data_prep}
#add age one batch variables to covariate matrix
y1_pdata$y1_run <- as.factor(y1_pdata$Run) 
y1_pdata$y1_row  <- as.factor(substr(y1_pdata$Sentrix_Position,1,3)) 
y1_pdata$y1_row <- as.numeric(y1_pdata$y1_row)
y1_pdata$y1_chip <- as.factor(y1_pdata$Sentrix_ID)

covariates_geno_ctpc_tech_cc_y1 <- cbind(covariates_geno_ctpc_tech_cc_y1, 
                                         y1_pdata[y1_pdata$Sample_Label %in% covariates_geno_ctpc_tech_cc_y1$sample_id, c("y1_run", "y1_row", "y1_chip")])
```

Subset for complete cases in age one data
```{r y1_complete_cases}
#subset covariates with age one cell type PCs for linear regression
covariates_geno_ctpc_tech_cc_y1_sub <- covariates_geno_ctpc_tech_cc_y1 %>%
  subset(complete.cases(no2_prenatal_avg_2019v, sex, maternal_smoking_y1,
                        maternal_education_length,
                        pbmc.Comp.1 , pbmc.Comp.2, pbmc.Comp.3,
                        genotyping.PC1, genotyping.PC2, genotyping.PC3))

#check size
dim(covariates_geno_ctpc_tech_cc_y1_sub) # 124

#subset age one betas for samples in the complete case cord blood betas 
y1_betas_cc <- y1_betas[,colnames(y1_betas) %in%
                          as.character(covariates_geno_ctpc_tech_cc_y1_sub$sample_id)]

#check size
dim(y1_betas_cc) # 424644    124

#remove probes that were used for PBMC cell type predict from age one betas
y1_betas_cc_sub <- 
  y1_betas_cc[!rownames(y1_betas_cc) %in%
                rownames(child_fscbc_ecc2_y1$probelist_deconvo_coeffEsts), ]

#check size
dim(y1_betas_cc_sub) # 424079    124
# 565 probes removed

#check that covariates and betas in same order
identical(colnames(y1_betas_cc_sub), 
          as.character(covariates_geno_ctpc_tech_cc_y1_sub$sample_id)) # TRUE
```

Add sample name (sentrix id and row) to y1 data
```{r add_sentrix_info_y1}
# check if any y1 one samples duplicated
y1_pdata$Sample_Label %>% duplicated() %>% sum()

# subset for covariate data
y1_pdata_sub <- y1_pdata %>%
  subset(Sample_Label %in% covariates_geno_ctpc_tech_cc_y1_sub$sample_id)

# check order
identical(as.numeric(y1_pdata_sub$Sample_Label), 
          as.numeric(covariates_geno_ctpc_tech_cc_y1_sub$sample_id)) # TRUE
# check order
covariates_geno_ctpc_tech_cc_y1_sub$y1_sample_name <- rownames(y1_pdata_sub)
```

### Combined cell type PCs across two timepoint


```{r combined_cellpcs}
# load cord blood and age one blood cell type estimates
load(here("output_data",
          "deconvolution",
          "2023-06-28_cbmc_pbmc_deconvolution.Rdata"))

# pull out cell type proportion estimates'
y1_counts <- as.data.frame(child_fscbc_ecc2_y1$prop)
cb_counts <- as.data.frame(child_fscbc_ecc2_cb$prop)

# create new row in age one blood for nRBCs - set to 0
y1_counts$nRBC <- 0
# relocate nRBC column to match structure of cb data
y1_counts <- y1_counts %>% 
  relocate(nRBC, .after = NK)

# create columns for study ID in cord blood data
# check order of pdata and cell counts
identical(rownames(cb_counts), rownames(cb_pdata)) # FALSE

# remove samples in cell counts not present in pdata
cb_counts_sub <- cb_counts %>%
  subset(rownames(.) %in% rownames(cb_pdata))

# re check order of pdata and cell counts
identical(rownames(cb_counts_sub), rownames(cb_pdata)) # TRUE
cb_counts_sub$Sample_Label <- cb_pdata$Sample_Label

# create columns for study ID in age one blood data
# check order of pdata and cell counts
identical(rownames(y1_counts), rownames(y1_pdata)) # TRUE
y1_counts$Sample_Label <- y1_pdata$Sample_Label

# subset age one cell counts for samples with cord blood cell counts
y1_counts_sub <- y1_counts %>%
  subset(Sample_Label %in% cb_counts_sub$Sample_Label)

dim(cb_counts_sub) # 144 8 
dim(y1_counts_sub) # 144 8
identical(cb_counts_sub$Sample_Label, y1_counts_sub$Sample_Label) # TRUE

# combine cord blood and year one blood cell types into one data frame
comb_counts <- rbind(cb_counts_sub, y1_counts_sub)

comb_counts_ilr <- compositions::ilr(comb_counts[,1:7])

cv_comb <- MASS::cov.rob(comb_counts_ilr %>% as.data.frame() %>% as.matrix(), cor = FALSE)
comb_counts_ilr_pca <- suppressWarnings(princomp(comb_counts_ilr %>% as.data.frame() %>% as.matrix(), 
                                                 covmat = cv_comb, 
                                                 cor = FALSE))

# save for paper
write.csv(comb_counts_ilr_pca$scores,
          here("tables",
               "deconvolution",
               "2023-08-14_cbmc_pmbc_persistence_PCs.csv"))


# scree plot of variance
var_explained = comb_counts_ilr_pca$sdev^2 /sum(comb_counts_ilr_pca$sdev^2)
ggplot(var_explained %>% as.data.frame(), aes(y=., x=(c(1:6)))) + geom_col()
cumsum(var_explained) 
# Comp.1    Comp.2    Comp.3    Comp.4    Comp.5    Comp.6 
# 0.4495292 0.6746827 0.8180867 0.9163590 0.9684108 1.0000000 

comb_counts_pcs <- as.data.frame(comb_counts_ilr_pca$scores)
dim(comb_counts_pcs) # 288
rownames(comb_counts_pcs) <- rownames(comb_counts)
comb_counts_pcs$Sample_Label <- comb_counts$Sample_Label
dim(covariates_geno_ctpc_tech_cc_y1_sub) #124

comb_counts_pcs_cc <- comb_counts_pcs %>%
  subset(Sample_Label %in% covariates_geno_ctpc_tech_cc_y1_sub$sample_id)
dim(comb_counts_pcs_cc) # 248
```

### Surrogate variable analysis to account for technical variables in combined cord blood and age one blood DNAm data

```{r combined_dnam_data}
# subset for samples with data at both time points
cb_betas_geno_cc_sub <- cb_betas_geno_cc[, c(colnames(cb_betas_geno_cc) %in%
                                              colnames(y1_betas_cc_sub))]

# check columns are in the same order 
identical(colnames(cb_betas_geno_cc_sub), 
         colnames(y1_betas_cc_sub)) # TRUE

dim(cb_betas_geno_cc_sub)
dim(y1_betas_cc_sub)

# subset for cpgs shared between data
cb_betas_geno_cc_sub_cpgs <- cb_betas_geno_cc_sub[rownames(y1_betas_cc_sub), ]

dim(cb_betas_geno_cc_sub_cpgs)
dim(y1_betas_cc_sub)

# rename columns
colnames(cb_betas_geno_cc_sub_cpgs) <-
  paste("cb_", colnames(cb_betas_geno_cc_sub_cpgs), sep ="")

colnames(y1_betas_cc_sub) <-
  paste("y1_", colnames(y1_betas_cc_sub), sep ="")

comb_betas <- cbind(cb_betas_geno_cc_sub_cpgs,y1_betas_cc_sub)
dim(comb_betas) # 424079    248
```

Add combined cell type pcs to covariate data
```{r add_comb_pcs_to_y1_covariate}
cg_df <- rbind(cbind(covariates_geno_ctpc_tech_cc_y1_sub %>%
                       dplyr::select(no2_prenatal_avg_2019v,
                              gestational_age,
                              prenatal_maternal_smoke, sex, 
                              maternal_education_length, 
                              genotyping.PC1, genotyping.PC2, genotyping.PC3, 
                              sample_id, cb_sample_name,
                              chip, run, row) %>%
                       dplyr::rename("no2" = no2_prenatal_avg_2019v,
                                     "smoking" = prenatal_maternal_smoke,
                                     "sample_name" = cb_sample_name),
                     timepoint = "birth"),
               cbind(covariates_geno_ctpc_tech_cc_y1_sub %>%
                       dplyr::select(no2_prenatal_avg_2019v, 
                                     gestational_age,
                                     maternal_smoking_y1, sex, 
                                     maternal_education_length,
                                     genotyping.PC1, genotyping.PC2, genotyping.PC3, 
                                     sample_id, y1_sample_name, 
                                     y1_chip, y1_row, y1_run)  %>%
                       dplyr::rename("no2" = no2_prenatal_avg_2019v,
                                     "smoking" = maternal_smoking_y1,
                                     "sample_name" = y1_sample_name,
                                     "chip" = y1_chip, "run" = y1_run,
                                     "row" = y1_row), 
                     timepoint = "ageone")) %>%
  mutate(timepoint = factor(timepoint, levels =c("birth", "ageone"))) %>%
  merge(x = ., y = comb_counts_pcs_cc, by.x = "sample_name", by.y = "row.names")

# reorder data frame to match order of betas  
cg_df_order <- cg_df %>%
  arrange(timepoint, sample_id)

# check that the order is the same
# 124 samples repeated over 2 time points
identical(as.numeric(cg_df_order$sample_id[1:124]), 
          str_sub(colnames(comb_betas)[1:124], start = 4, end = 8) %>% as.numeric())
```

Create model matrix for combined betas
```{r model_matrix}
# create model matrix
comb_mod <- model.matrix(~ timepoint*(no2 + gestational_age) + 
                           sex + maternal_education_length + smoking +
                           genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
                           Comp.1 + Comp.2 + Comp.3 + Comp.4, 
                         data=cg_df_order)

# check size
dim(comb_mod) # 248 16
```

Calculate the number of SVs needed for age one model.
```{r comb_sv_analysis}
# create null matrix
# this matrix does not include any of our covariates 
# this means that our covariates are "protected" and SVs will not capture them
comb_mod0 <- model.matrix(~ timepoint*(gestational_age) + 
                            sex + maternal_education_length + smoking +
                            genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
                            Comp.1 + Comp.2 + Comp.3 + Comp.4, 
                          data=cg_df_order)

################
# sva analysis #
################

##################################
# numsv function internal to sva #
##################################

#determine the number of required SVs
#this is numsv function code pulled out

set.seed(1234)

#this is what is fed into this function from the sva function
dat = as.matrix(comb_betas)
mod = comb_mod
mod0 = comb_mod0
#method="be" # this is the default and what is called by numsv
#vfilter=NULL # we dont apply a filter and its not used for be method
B=20


# initialize some stuff
# get the residual matrix (the remaining variance)
n <- ncol(dat)
m <- nrow(dat)
H <- mod %*% solve(t(mod) %*% mod) %*% t(mod)
res <- dat - t(H %*% t(dat))

# run svd and get some eigenvalues
uu <- svd(res)
# get the num of "sig" eigenvalues
ndf <- min(m, n) - ceiling(sum(diag(H)))
# get percent variance explained by each eigenvalue
dstat <- uu$d[1:ndf]^2/sum(uu$d[1:ndf]^2)
# initialize null distribution matrix
dstat0 <- matrix(0, nrow = B, ncol = ndf)

# set seed here since we are calling sample
# create the null distribution
for (i in 1:B) {
  # this part of the code removes all relationships in the 
  res0 <- t(apply(res, 1, sample, replace = FALSE))
  res0 <- res0 - t(H %*% t(res0))
  # calculate svs on this
  uu0 <- svd(res0)
  # bind their explained variance in null matrix
  dstat0[i, ] <- uu0$d[1:ndf]^2/sum(uu0$d[1:ndf]^2)
  print(i) # print out the iteration we are on
}

# determine the number of svs required here based on pvalues
psv <- rep(1, n)
for (i in 1:ndf) {
  psv[i] <- mean(dstat0[, i] >= dstat[i])
}
for (i in 2:ndf) {
  psv[i] <- max(psv[(i - 1)], psv[i])
}

# this is the variable holding the number of SVs required 
# number is each to all the svs with a pval less than/equal to 0.1
nsv <- sum(psv <= 0.1)

####################################
# code from the main SVA function  #
####################################

n.sv = nsv # from code above (run internally when sva called)
B = 5 # number of iterations

###################################
# irwsva function internal to sva #
###################################

# this code is really want does all the work in sva
# we use the irw method so this is the code for that method
# there are also supervised and two step methods (not used by us)

# intialize variables again
n <- ncol(dat)
m <- nrow(dat)
Id <- diag(n)
resid <- dat %*% (Id - mod %*% solve(t(mod) %*% mod) %*% t(mod))
uu <- eigen(t(resid) %*% resid)
vv <- uu$vectors
ndf <- n - dim(mod)[2]
pprob <- rep(1, m)
one <- rep(1, n)
Id <- diag(n)
df1 <- dim(mod)[2] + n.sv
df0 <- dim(mod0)[2] + n.sv
rm(resid)

cat(paste("Iteration (out of", B, "):"))
for (i in 1:B) {
  mod.b <- cbind(mod, uu$vectors[, 1:n.sv])
  mod0.b <- cbind(mod0, uu$vectors[, 1:n.sv])
  ptmp <- f.pvalue(dat, mod.b, mod0.b)
  pprob.b <- (1 - edge.lfdr(ptmp))
  mod.gam <- cbind(mod0, uu$vectors[, 1:n.sv])
  mod0.gam <- cbind(mod0)
  ptmp <- f.pvalue(dat, mod.gam, mod0.gam)
  pprob.gam <- (1 - edge.lfdr(ptmp))
  pprob <- pprob.gam * (1 - pprob.b)
  dats <- dat * pprob
  dats <- dats - rowMeans(dats)
  uu <- eigen(t(dats) %*% dats)
  cat(paste(i, " "))
}

# this is what is normally returned by sva
# problem is that it doesnt include the diagonal matrix we want
sv_comb = svd(dats)$v[, 1:n.sv, drop = FALSE] 
svobj_comb <- list(sv = sv_comb, pprob.gam = pprob.gam, pprob.b = pprob.b, 
                   n.sv = n.sv)

# to get all info (including diagonal)
svobj_all_comb <- svd(dats)

# populate rownames with sample ids
colnames(svobj_all_comb$v) <- covariates_geno_ctpc_tech_cc_y1_sub$sampleid

# save 
save(sv_comb, file = here("output_data", 
                          "linear_regression", 
                          "2023-08-14_combined_cb_y1_svs_with_interaction.RData"))

###############
#scree of SVs #
###############

# from elodie portales-casamar

# heat scree prepartion
vars <- svobj_all_comb[['d']]^2
vars <- vars/sum(vars)
names(vars) <- paste0('SV', 1:length(vars))
xp <- 1:svobj_comb$n.sv
varsdf <- as.data.frame(vars)
varsdf$svnum <- rep(1:length(vars)) 
# use 16

# heat scree ggplot!! 
comb_sv_scree <- ggplot(varsdf[1:20,], aes(x=svnum, y=vars*100)) +
  geom_col() +
  theme_bw() +
  labs(x="Surrogate Variables", y = "Variance (%)")

comb_sv_scree

ggsave(plot = comb_sv_scree, 
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal", 
                       "2023-08-14_cb_y1_comb_sv_heat_scree.png"),
       height = 5, width = 15, units = "cm", dpi=600)

######################
# correlation matrix #
######################

# add svs to covariates
cg_df_order_svs <- cbind(cg_df_order, sv_comb)

# rename columns with svs
colnames(cg_df_order_svs)[22:94] <-
  paste("comb_sv",colnames(cg_df_order_svs)[22:94], sep="")

source(here("scripts", "2023-07-06_corr_mat_funs.R"))

# early exp and pn_18 are the same thing. 
# exclude early_exp from correlation matrices
corr_comb <- corr.mat(cg_df_order_svs %>% 
                        dplyr::select(no2, gestational_age, sex, 
                                      maternal_education_length, smoking,
                                      genotyping.PC1, genotyping.PC2,
                                      genotyping.PC3, Comp.1, Comp.2, Comp.3, Comp.4,
                                      chip, run, row, 
                                      comb_sv1:comb_sv20) %>% 
                        dplyr::rename("NO2 exposure" = no2,
                                      "Sex" = sex, 
                                      "Gestational age (days)" = gestational_age,
                                      "Maternal education (years)" = maternal_education_length,
                                      "Maternal smoking" =  smoking,
                                      "Genetic ancestry PC1" = genotyping.PC1,
                                      "Genetic ancestry PC2" = genotyping.PC2,
                                      "Genetic ancestry PC3" = genotyping.PC3,
                                      "Cell type PC1" = Comp.1,
                                      "Cell type PC2" = Comp.2,
                                      "Cell type PC3" = Comp.3,
                                       "Cell type PC4" = Comp.4,
                                      "Chip" = chip, 
                                      "Run" = run, 
                                      "Row" = row,
                                      "SV1" = comb_sv1, "SV2" = comb_sv2,
                                      "SV3" = comb_sv3, "SV4" = comb_sv4,
                                      "SV5" = comb_sv5, "SV6" = comb_sv6,
                                      "SV7" = comb_sv7, "SV8" = comb_sv8,
                                      "SV9" = comb_sv9, "SV10" = comb_sv10,
                                      "SV11" = comb_sv11, "SV12" = comb_sv12,
                                      "SV13" = comb_sv13, "SV14" = comb_sv14,
                                      "SV15" = comb_sv15, "SV16" = comb_sv16,
                                      "SV17" = comb_sv17, "SV18" = comb_sv18,
                                      "SV19" = comb_sv19, "SV20" = comb_sv20)) 

write.csv(corr_comb,
          file = here("tables",
                      "prenatal",
                      "2023-08-14_persistence_svcorr_correlation.csv"))

pval_comb <- pval.mat(cg_df_order_svs %>% 
                        dplyr::select(no2, gestational_age, sex, 
                                      maternal_education_length, smoking,
                                      genotyping.PC1, genotyping.PC2,
                                      genotyping.PC3, Comp.1, Comp.2, Comp.3, Comp.4,
                                      chip, run, row, 
                                      comb_sv1:comb_sv20) %>% 
                        dplyr::rename("NO2 exposure" = no2,
                                      "Sex" = sex, 
                                      "Gestational age (days)" = gestational_age,
                                      "Maternal education (years)" = maternal_education_length,
                                      "Maternal smoking" =  smoking,
                                      "Genetic ancestry PC1" = genotyping.PC1,
                                      "Genetic ancestry PC2" = genotyping.PC2,
                                      "Genetic ancestry PC3" = genotyping.PC3,
                                      "Cell type PC1" = Comp.1,
                                      "Cell type PC2" = Comp.2,
                                      "Cell type PC3" = Comp.3,
                                   "Cell type PC4" = Comp.4,
                                      "Chip" = chip, 
                                      "Run" = run, 
                                      "Row" = row,
                                      "SV1" = comb_sv1, "SV2" = comb_sv2,
                                      "SV3" = comb_sv3, "SV4" = comb_sv4,
                                      "SV5" = comb_sv5, "SV6" = comb_sv6,
                                      "SV7" = comb_sv7, "SV8" = comb_sv8,
                                      "SV9" = comb_sv9, "SV10" = comb_sv10,
                                      "SV11" = comb_sv11, "SV12" = comb_sv12,
                                      "SV13" = comb_sv13, "SV14" = comb_sv14,
                                      "SV15" = comb_sv15, "SV16" = comb_sv16,
                                      "SV17" = comb_sv17, "SV18" = comb_sv18,
                                      "SV19" = comb_sv19, "SV20" = comb_sv20)) 

write.csv(pval_comb,
          file = here("tables",
                      "prenatal",
                      "2023-08-14_persistence_svcorr_pvals.csv"))

svcorr_comb <- corr.plot(pval = pval_comb, 
                         corr = corr_comb, 
                         label.sig = T, 
                         shape = 8,shape.size = 1,
                         axis.text.size = 8, x.axis.text.angle = 90) +
  scale_y_discrete(position = "right") +
  theme(legend.position = "none",
        panel.grid = element_blank())
svcorr_comb

ggsave(plot = svcorr_comb,
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal",
                       "2022-11-17_comb_cb_y1_sv_correlations.png"),
       height = 15,
       width = 15, 
       units = "cm",dpi=600)
```

### Linear mixed models regressions examining persistence at significant CpGs/DMRs

Examine persistence of of DMR prenatal changes at age one.
```{r dmr_persistence_study_pop}

# check order of data and dnam
identical(as.numeric(str_sub(colnames(comb_betas), 4, 8)), 
          as.numeric(cg_df_order_svs$sample_id)) # true

cg_df_order_svs_cpgs <- 
  cbind(cg_df_order_svs, 
        t(comb_betas[combp_anno_df_es_sub_escutoff_mqtls %>% pull(Name), ]))

lmer_results <- NULL
lmer_pvals <- NULL
lmer_coeffs <- NULL

for(i in 95:221){
  
  cg_df_order_svs_cpgs$dnam <- cg_df_order_svs_cpgs[,i] 
  cg_lmer <- lmer(dnam ~  timepoint*(no2 + gestational_age)  + sex +  
                    maternal_education_length + smoking +
                    genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                    Comp.1 + Comp.2 + Comp.3 + Comp.4 + comb_sv1 + comb_sv2 +
                    comb_sv3 + comb_sv4 + comb_sv5 + comb_sv6 + comb_sv7 +
                    comb_sv8 + comb_sv9 + comb_sv10 + comb_sv11 + comb_sv12 +
                    comb_sv13 + comb_sv14 + comb_sv15 + comb_sv16 + comb_sv17 + 
                    (1 |sample_id), 
                  data = cg_df_order_svs_cpgs)
  
  lmer_results <- c(lmer_results, cg_lmer)
  lmer_pvals <- cbind(lmer_pvals, summary(cg_lmer)$coefficients[c(1,2,3,32),5])
  lmer_coeffs <- cbind(lmer_coeffs, summary(cg_lmer)$coefficients[c(1,2,3,32),1])
  
}

names(lmer_results) <- combp_anno_df_es_sub_escutoff_mqtls %>% pull(Name)

lmer_pvals <- lmer_pvals %>% 
  t() %>% 
  as.data.frame() %>% 
  rename_with(., ~ paste("pval", .x, sep="_"))

lmer_coeffs <- lmer_coeffs %>% 
  t() %>% 
  as.data.frame() %>% 
  rename_with(., ~ paste("coef", .x, sep="_"))

lmer_pvals_dmrs <- lmer_pvals %>% 
  cbind(., lmer_coeffs) %>%
  as.data.frame() %>%
  #rename("pval" = 1) %>%
  mutate(dmr = combp_anno_df_es_sub_escutoff$dmr,
         cpg = combp_anno_df_es_sub_escutoff$Name,
         singular = lapply(lmer_results, isSingular) %>% do.call(rbind,.),
         pval_no2_adj = p.adjust(pval_no2, method = "bonferroni"),
         pval_timepointageone_adj = p.adjust(pval_timepointageone, method = "bonferroni"),
         `pval_timepointageone:no2_adj` = p.adjust(`pval_timepointageone:no2`, method = "bonferroni"),
         coef_no2_age_one = coef_no2 + `coef_timepointageone:no2`) %>%
  mutate( change = ifelse(abs(coef_no2_age_one) > abs(coef_no2), "larger", "smaller"),
          birth_effect_size = coef_no2 * range,
          y1_effect_size = coef_no2_age_one * range) %>%
  mutate(singular = as.vector(singular))

# combine persistence info with info from ewas
colnames(lmer_pvals_dmrs) <- paste("lme", colnames(lmer_pvals_dmrs), sep="_")
colnames(combp_anno_df_es_sub_escutoff_mqtls)[c(37,38)] <- 
  paste("lm", colnames(combp_anno_df_es_sub_escutoff_mqtls)[c(37,38)], sep="_")
dmrs_lm_lme <- merge(x=combp_anno_df_es_sub_escutoff_mqtls, y=lmer_pvals_dmrs, by.x = 5, by.y = 10)
dmrs_lm_lme_sub <- dmrs_lm_lme %>% dplyr::select(-c(lme_dmr, Row.names, analysis))

write.csv(dmrs_lm_lme_sub, file = here ("tables",
                                        "prenatal",
                                        "2023-08-14_prenatal_no2_dmr_cpg_persistence.csv"))

# examine effect sizes from lme models vs those from ewas
lmer_effect_sizes <- dmrs_lm_lme %>%
  group_by(lme_dmr) %>%
  summarise(mean_birth_es = mean(lme_birth_effect_size),
            mean_y1_es = mean(lme_y1_effect_size)) %>%
  cbind(., combp_anno_df_es_sub_escutoff_mqtls %>% 
          group_by(dmr) %>%
          summarise(ewas_birth_es = mean(lm_effect_size)))

write.csv(lmer_effect_sizes, file = here ("tables",
                                          "prenatal",
                                          "2023-08-14_prenatal_no2_dmr_cpg_persistence_effect_size_by_dmr.csv"))

# check if mean birth effect sizes are significantly different from year one effect sizes
lme_birth_y1_es <- dmrs_lm_lme_sub %>%
  dplyr::select(Name, dmr, lme_birth_effect_size, lme_y1_effect_size) %>% 
  pivot_longer(-c(Name, dmr)) %>%
  mutate(name = factor(name, levels = c("lme_y1_effect_size",
                                        "lme_birth_effect_size"))) %>%
  group_by(dmr) %>%
  group_map(~ t.test(abs(value) ~ name, .x, paired = TRUE)) 

lme_birth_y1_es_df <- do.call(rbind, lme_birth_y1_es)   
lme_birth_y1_es_df[,3] %>% 
  rbind() %>% 
  as.numeric() %>% 
  as.data.frame() %>% 
  mutate(adj = `.`*19) %>% 
  mutate(adj = ifelse(adj >=1, 1, adj)) %>%
  View()

# visualize change in effect size
dmr_persistence <- lmer_pvals_dmrs %>% 
  mutate(sig = ifelse(`lme_pval_timepointageone:no2_adj` < 0.05, T, F),
         start = ifelse(lme_coef_no2 < 0, "Smaller", "Larger")) %>%
  mutate(start = factor(start, levels =c("Smaller", "Larger"))) %>%
  dplyr::select(lme_birth_effect_size, lme_y1_effect_size, lme_cpg, sig, start, lme_change) %>%
  pivot_longer(cols = -c(lme_cpg, sig, start, lme_change)) %>%
  ggplot(aes(x=name, y = value, group=lme_cpg, color = sig, shape = sig, linetype = lme_change)) +
  geom_point(alpha = 0.75, size = 1) + 
  geom_line(alpha = 0.75, size = 0.5) + 
  facet_wrap(~start) + 
  theme_bw() +
  scale_linetype_manual(values = c("solid", "dashed"), 
                        labels = c("Larger", "Smaller"), "Absolute Δ \neffect size") +
   scale_shape_manual(values = c(1,4), labels = c(">=0.05", "<0.05"), name = "p-value", ) +
  scale_x_discrete(labels = c("Birth", "Age one"), name = element_blank()) +
  scale_y_continuous(name = "Prenatal NO2 effect size (% DNAm)") +
  scale_color_manual(values = c("#254A90", "red2"), name = "p-value", labels = c(">=0.05", "<0.05")) +
  theme(panel.grid.major.x = element_blank(),
        axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        strip.background = element_blank(),
        legend.text = element_text(size=9),
        legend.title = element_text(size=9),
        legend.position = "none") 

ggsave(plot = dmr_persistence,
       filename = here("figures",
                       "linear_regression",
                       "prenatal",
                       "2023-08-14_dmr_persistence.png"),
       height = 6, width = 6.5, units="cm", dpi = 600)
```

### Examine changes in effect size at top 100 CpGs from EWAS over the first year of life 

```{r top100_effect_size_persistence}

# get top 100 cord blood cpgs 
cb100 <- cb_pvals_tech %>% 
  subset(effect_size > 0.025 | effect_size < -0.025) %>%
  arrange(pval_no2_prenatal_avg_2019v) %>%
  slice_head(n=100)


# get birth and age one dnam for top 100 cord blood cpgs
cb_top100_betas <- 
  cb_betas_geno_cc_clean[rownames(cb100), 
                            str_sub(colnames(y1_betas_cc_sub[rownames(cb100), ] ),4,8)] %>%
  t() %>%
  as.data.frame() 
# rename_with(., ~ paste("cb", .x, sep="_"))

y1_top100_betas <- 
  y1_betas_cc_sub[rownames(cb100), ] %>%
  t() %>%
  as.data.frame()
# rename_with(., ~ paste("y1", .x, sep="_"))

# check order
identical(rownames(cb_top100_betas), 
          str_sub(rownames(y1_top100_betas), 4, 8)) # TRUE

identical(rownames(cb_top100_betas), 
          cg_df_order_svs_cpgs %>%
            subset(timepoint=="birth") %>% 
            pull(sample_id) %>% 
            as.character())

top100_betas <- rbind(cb_top100_betas, 
                      y1_top100_betas)

colnames(top100_betas) <- paste("top100_",
                                colnames(top100_betas),
                                sep="")

cg_df_order_svs_cpgs_top100cpgs <- 
  cbind(cg_df_order_svs_cpgs, top100_betas)

lmer_results_cpgs <- NULL
lmer_pvals_cpgs <- NULL
lmer_coeffs_cpgs <- NULL

for(i in 223:322){
  
  cg_df_order_svs_cpgs_top100cpgs$dnam <- cg_df_order_svs_cpgs_top100cpgs[,i] 
  
  cg_lmer_cpgs <- lmer(dnam ~  timepoint*(no2 + gestational_age)  + sex +  
                         maternal_education_length + smoking +
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                         Comp.1 + Comp.2 + Comp.3 + Comp.4 + comb_sv1 + comb_sv2 +
                         comb_sv3 + comb_sv4 + comb_sv5 + comb_sv6 + comb_sv7 +
                         comb_sv8 + comb_sv9 + comb_sv10 + comb_sv11 + comb_sv12 +
                         comb_sv13 + comb_sv14 + comb_sv15 + comb_sv16 + 
                         comb_sv17 + (1 |sample_id), 
                       data = cg_df_order_svs_cpgs_top100cpgs)
  
  lmer_results_cpgs <- c(lmer_results_cpgs, cg_lmer_cpgs)
  lmer_pvals_cpgs <- cbind(lmer_pvals_cpgs, summary(cg_lmer_cpgs)$coefficients[c(1,2,3,32),5])
  lmer_coeffs_cpgs <- cbind(lmer_coeffs_cpgs, summary(cg_lmer_cpgs)$coefficients[c(1,2,3,32),1])
}

names(lmer_results_cpgs) <- colnames(top100_betas)

lmer_pvals_cpgs <- lmer_pvals_cpgs %>% 
  t() %>% 
  as.data.frame() %>% 
  rename_with(., ~ paste("pval", .x, sep="_"))

lmer_coeffs_cpgs <- lmer_coeffs_cpgs %>% 
  t() %>% 
  as.data.frame() %>% 
  rename_with(., ~ paste("coef", .x, sep="_"))

lmer_pvals_cpgs_results <- lmer_pvals_cpgs %>% 
  cbind(., lmer_coeffs_cpgs) %>%
  as.data.frame() %>%
  mutate(cpg = cb_pvals_tech %>% 
           subset(effect_size > 0.025 | effect_size < -0.025) %>%
           arrange(pval_no2_prenatal_avg_2019v) %>%
           slice_head(n=100) %>% 
           pull(Name),
         singular = as.vector(lapply(lmer_results_cpgs, isSingular) %>% do.call(rbind,.))) %>%
  #subset(singular != TRUE) %>%
  mutate(pval_no2_adj = p.adjust(pval_no2, "bonferroni"),
         pval_timepointageone_adj = p.adjust(pval_timepointageone, "bonferroni"),
         `pval_timepointageone:no2_adj` = p.adjust(`pval_timepointageone:no2`, "bonferroni"),
         coef_no2_age_one = coef_no2 + `coef_timepointageone:no2`) %>%
  mutate( change = ifelse(abs(coef_no2_age_one) > abs(coef_no2), "larger", "smaller"),
          y1_effect_size = coef_no2_age_one*range,
          birth_effect_size = coef_no2*range) %>%
  mutate(singular = as.vector(singular))

write.csv(lmer_pvals_cpgs_results, file = here ("tables",
                                                "prenatal",
                                                "2023-08-14_prenatal_no2_top100_persistence.csv"))

# combine persistence info with info from ewas
colnames(lmer_pvals_cpgs_results) <- paste("lme", colnames(lmer_pvals_cpgs_results), sep="_")

top100_lm_lme <- merge(x=cb_pvals_tech[,c(61:93,2, 94, 32, 95)], 
                       y=lmer_pvals_cpgs_results, 
                       by.x = 4, by.y = 9)
colnames(top100_lm_lme)[34:37] <- paste("lm", colnames(top100_lm_lme)[34:37], sep="_")

write.csv(top100_lm_lme, file = here ("tables",
                                      "prenatal",
                                      "2023-08-14_prenatal_no2_top100_persistence_lm_lme.csv"))

# visualize
top100_persistence <- lmer_pvals_cpgs_results %>% 
  subset(lme_singular==FALSE) %>%
  mutate(sig = ifelse(`lme_pval_timepointageone:no2_adj` < 0.05, T, F),
         start = ifelse(lme_coef_no2 < 0, "Smaller", "Larger")) %>%
  mutate(start = factor(start, levels =c("Smaller", "Larger"))) %>%
  dplyr::select(lme_birth_effect_size, lme_y1_effect_size, lme_cpg, sig, start, lme_change) %>%
  pivot_longer(cols = -c(lme_cpg, sig, start, lme_change)) %>%
  ggplot(aes(x=name, y = value, group=lme_cpg, color = sig, shape = sig, linetype = lme_change)) +
  geom_point(alpha = 0.75, size = 1) + 
  geom_line(alpha = 0.75, size = 0.5) + 
  facet_wrap(~start) + 
  theme_bw() +
  scale_linetype_manual(values = c("solid", "dashed"), 
                        labels = c("Larger", "Smaller"), "Absolute Δ \neffect size") +
   scale_shape_manual(values = c(1,4), labels = c(">=0.05", "<0.05"), name = "p-value", ) +
  scale_x_discrete(labels = c("Birth", "Age one"), name = element_blank()) +
  scale_y_continuous(name = "Prenatal NO2 effect size (% DNAm)") +
  scale_color_manual(values = c("#254A90", "red2"), name = "p-value", labels = c(">=0.05", "<0.05")) +
  theme(panel.grid.major.x = element_blank(),
        axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        strip.background = element_blank(),
        legend.text = element_text(size=9),
        legend.title = element_text(size=9),
        legend.position = "none") 

ggsave(plot = top100_persistence,
       filename = here("figures",
                       "linear_regression",
                       "prenatal",
                       "2023-08-14_top100_persistence.png"),
       height = 6, width = 6.5, units="cm")
```

### Peristence in males

#### Linear mixed models regressions examining persistence at significant CpGs/DMRs

subset for male data
```{r males_sub_pers}
cg_df_order_svs_males <- cg_df_order_svs %>% 
  subset(sex == "M")

comb_betas_males <- comb_betas[,as.numeric(str_sub(colnames(comb_betas), 4, 8)) %in% as.numeric(cg_df_order_svs_males$sample_id)]
```

Examine persistence of of DMR prenatal changes at age one.
```{r dmr_persistence_study_pop}

# check order of data and dnam
identical(as.numeric(str_sub(colnames(comb_betas_males), 4, 8)), 
          as.numeric(cg_df_order_svs_males$sample_id)) # true

cg_df_order_svs_cpgs_males <- 
  cbind(cg_df_order_svs_males, 
        t(comb_betas_males[combp_anno_df_es_sub_males_mqtls %>% pull(Name), ]))

lmer_results_males <- NULL
lmer_pvals_males <- NULL
lmer_coeffs_males <- NULL

for(i in 95:185){
  
  cg_df_order_svs_cpgs_males$dnam <- cg_df_order_svs_cpgs_males[,i] 
  cg_lmer_males <- lmer(dnam ~  timepoint*(no2 + gestational_age) +  
                    maternal_education_length + smoking +
                    genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                    Comp.1 + Comp.2 + Comp.3 + Comp.4 + comb_sv1 + comb_sv2 +
                    comb_sv3 + comb_sv4 + comb_sv5 + comb_sv6 + comb_sv7 +
                    comb_sv8 + comb_sv9 + comb_sv10 + comb_sv11 + comb_sv12 +
                    comb_sv13 + comb_sv14 + comb_sv15 + comb_sv16 + comb_sv17 + 
                    (1 |sample_id), 
                  data = cg_df_order_svs_cpgs_males)
  
  lmer_results_males <- c(lmer_results_males, 
                    cg_lmer_males)
  lmer_pvals_males <- cbind(lmer_pvals_males,
                      summary(cg_lmer_males)$coefficients[c(1,2,3,31),5])
  lmer_coeffs_males <- cbind(lmer_coeffs_males, 
                       summary(cg_lmer_males)$coefficients[c(1,2,3,31),1])
  
}

names(lmer_results_males) <- combp_anno_df_es_sub_males_mqtls %>% 
  pull(Name)

lmer_pvals_males <- lmer_pvals_males %>% 
  t() %>% 
  as.data.frame() %>% 
  rename_with(., ~ paste("pval", .x, sep="_"))

lmer_coeffs_males <- lmer_coeffs_males %>% 
  t() %>% 
  as.data.frame() %>% 
  rename_with(., ~ paste("coef", .x, sep="_"))

lmer_pvals_dmrs_males <- lmer_pvals_males %>% 
  cbind(., lmer_coeffs_males) %>%
  as.data.frame() %>%
  #rename("pval" = 1) %>%
  mutate(dmr = combp_anno_df_es_sub_males_mqtls$dmr,
         cpg = combp_anno_df_es_sub_males_mqtls$Name,
         singular = lapply(lmer_results_males, isSingular) %>% do.call(rbind,.),
         pval_no2_adj = p.adjust(pval_no2, method = "bonferroni"),
         pval_timepointageone_adj = p.adjust(pval_timepointageone, method = "bonferroni"),
         `pval_timepointageone:no2_adj` = p.adjust(`pval_timepointageone:no2`, method = "bonferroni"),
         coef_no2_age_one = coef_no2 + `coef_timepointageone:no2`) %>%
  mutate( change = ifelse(abs(coef_no2_age_one) > abs(coef_no2), "larger", "smaller"),
          birth_effect_size = coef_no2 * range,
          y1_effect_size = coef_no2_age_one * range) %>%
  mutate(singular = as.vector(singular))

# combine persistence info with info from ewas
colnames(lmer_pvals_dmrs_males) <- paste("lme", 
                                         colnames(lmer_pvals_dmrs_males), 
                                         sep="_")

colnames(combp_anno_df_es_sub_males_mqtls)[c(37,38)] <- 
  paste("lm", 
        colnames(combp_anno_df_es_sub_males)[c(37,38)], 
        sep="_")

dmrs_lm_lme_males <- merge(x=combp_anno_df_es_sub_males, 
                     y=lmer_pvals_dmrs_males, 
                     by.x = 5, by.y = 10)

dmrs_lm_lme_sub_males <- dmrs_lm_lme_males %>% 
  dplyr::select(-c(lme_dmr, Row.names, analysis))

write.csv(dmrs_lm_lme_sub_males, 
          file = here ("tables",
                       "prenatal",
                       "2023-08-14_prenatal_no2_dmr_cpg_persistence_MALES.csv"))


# check if mean birth effect sizes are significantly different from year one effect sizes
lme_birth_y1_es_males <- dmrs_lm_lme_sub_males %>%
  dplyr::select(Name, dmr, lme_birth_effect_size, lme_y1_effect_size) %>% 
  pivot_longer(-c(Name, dmr)) %>%
  mutate(name = factor(name, levels = c("lme_y1_effect_size",
                                        "lme_birth_effect_size"))) %>%
  group_by(dmr) %>%
  group_map(~ t.test(abs(value) ~ name, .x, paired = TRUE)) 

lme_birth_y1_es_males_df <- do.call(rbind, lme_birth_y1_es_males)   
lme_birth_y1_es_males_df[,3] %>% 
  rbind() %>% 
  as.numeric() %>% 
  as.data.frame() %>% 
  mutate(adj = `.`*19) %>% 
  mutate(adj = ifelse(adj >=1, 1, adj)) %>%
  View()

# examine effect sizes from lme models vs those from ewas
lmer_effect_sizes_males <- dmrs_lm_lme_males %>%
  group_by(lme_dmr) %>%
  summarise(mean_birth_es = mean(lme_birth_effect_size),
            mean_y1_es = mean(lme_y1_effect_size)) %>%
  cbind(., combp_anno_df_es_sub_males %>% 
          group_by(dmr) %>%
          summarise(ewas_birth_es = mean(lm_effect_size)))

write.csv(lmer_effect_sizes_males, 
          file = here ("tables",
                       "prenatal",
                       "2023-08-14_prenatal_no2_dmr_cpg_persistence_effect_size_by_dmr_MALES.csv"))

# visualize
dmr_persistence_males <- lmer_pvals_dmrs_males %>%
  subset(lme_singular==FALSE) %>%
   mutate(sig = ifelse(`lme_pval_timepointageone:no2_adj` < 0.05, T, F),
         start = ifelse(lme_coef_no2 < 0, "Smaller", "Larger")) %>%
  mutate(start = factor(start, levels =c("Smaller", "Larger"))) %>%
  dplyr::select(lme_birth_effect_size, lme_y1_effect_size, lme_cpg, sig, start, lme_change) %>%
  pivot_longer(cols = -c(lme_cpg, sig, start, lme_change)) %>%
  ggplot(aes(x=name, y = value, group=lme_cpg, color = sig, shape = sig, linetype = lme_change)) +
  geom_point(alpha = 0.75, size = 1) + 
  geom_line(alpha = 0.75, size = 0.5) + 
  facet_wrap(~start) + 
  theme_bw() +
  scale_linetype_manual(values = c("solid", "dashed"), 
                        labels = c("Larger", "Smaller"), "Absolute Δ \neffect size") +
   scale_shape_manual(values = c(1,4), labels = c(">=0.05", "<0.05"), name = "p-value", ) +
  scale_x_discrete(labels = c("Birth", "Age one"), name = element_blank()) +
  scale_y_continuous(name = "Prenatal NO2 effect size (% DNAm)") +
  scale_color_manual(values = c("#254A90", "red2"), name = "p-value", labels = c(">=0.05", "<0.05")) +
  theme(panel.grid.major.x = element_blank(),
        axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        strip.background = element_blank(),
        legend.text = element_text(size=9),
        legend.title = element_text(size=9),
        legend.position = "none") 

ggsave(plot = dmr_persistence_males,
       filename = here("figures",
                       "linear_regression",
                       "prenatal",
                       "2023-08-14_dmr_persistence_MALES.png"),
       height = 6, width = 6.5, units="cm", dpi = 600)
```

#### Examine changes in effect size at top 100 CpGs from EWAS over the first year of life in males

Subset y1 data for males
```{r sub_y1_males}
covariates_geno_ctpc_tech_cc_y1_sub_males <-
  covariates_geno_ctpc_tech_cc_y1_sub %>%
  subset(sex == "M")

```

```{r top100_effect_size_persistence_males}

# get top 100 cord blood cpgs 
cb100_males <- cb_pvals_tech_males %>% 
  subset(effect_size > 0.025 | effect_size < -0.025) %>%
  arrange(pval_no2_prenatal_avg_2019v) %>%
  slice_head(n=100)


y1_betas_cc_sub_males <- y1_betas_cc_sub[,str_sub(colnames(y1_betas_cc_sub[rownames(cb100_males), ] ),4,8) %in% covariates_geno_ctpc_tech_cc_y1_sub_males$sample_id]

# get birth and age one dnam for top 100 cord blood cpgs
cb_top100_betas_males <- 
  cb_betas_geno_cc_clean_df_males[rownames(cb100_males), 
                            str_sub(colnames(y1_betas_cc_sub_males[rownames(cb100_males), ] ),4,8)] %>%
  t() %>%
  as.data.frame() 
# rename_with(., ~ paste("cb", .x, sep="_"))

y1_top100_betas_males <- 
  y1_betas_cc_sub_males[rownames(cb100_males), ] %>%
  t() %>%
  as.data.frame()
# rename_with(., ~ paste("y1", .x, sep="_"))

# check order
identical(rownames(cb_top100_betas_males), 
          str_sub(rownames(y1_top100_betas_males), 4, 8)) # TRUE

identical(rownames(cb_top100_betas_males), 
          cg_df_order_svs_cpgs_males %>%
            subset(timepoint=="birth") %>% 
            pull(sample_id) %>% 
            as.character())

top100_betas_males <- rbind(cb_top100_betas_males, 
                      y1_top100_betas_males)

colnames(top100_betas_males) <- paste("top100_",
                                colnames(top100_betas_males),
                                sep="")

cg_df_order_svs_cpgs_top100cpgs_males <- 
  cbind(cg_df_order_svs_cpgs_males, top100_betas_males)

lmer_results_cpgs_males <- NULL
lmer_pvals_cpgs_males <- NULL
lmer_coeffs_cpgs_males <- NULL

for(i in 187:286){
  
  cg_df_order_svs_cpgs_top100cpgs_males$dnam <- cg_df_order_svs_cpgs_top100cpgs_males[,i] 
  
  cg_lmer_cpgs_males <- lmer(dnam ~  timepoint*(no2 + gestational_age) +  
                         maternal_education_length + smoking +
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                         Comp.1 + Comp.2 + Comp.3 + Comp.4 + comb_sv1 + comb_sv2 +
                         comb_sv3 + comb_sv4 + comb_sv5 + comb_sv6 + comb_sv7 +
                         comb_sv8 + comb_sv9 + comb_sv10 + comb_sv11 + comb_sv12 +
                         comb_sv13 + comb_sv14 + comb_sv15 + comb_sv16 + 
                        comb_sv17 + (1 |sample_id), 
                       data = cg_df_order_svs_cpgs_top100cpgs_males)
  
  lmer_results_cpgs_males <- c(lmer_results_cpgs_males, 
                         cg_lmer_cpgs_males)
  lmer_pvals_cpgs_males <- cbind(lmer_pvals_cpgs_males, 
                           summary(cg_lmer_cpgs_males)$coefficients[c(1,2,3,31),5])
  lmer_coeffs_cpgs_males <- cbind(lmer_coeffs_cpgs_males, 
                            summary(cg_lmer_cpgs_males)$coefficients[c(1,2,3,31),1])
}

names(lmer_results_cpgs_males) <- colnames(top100_betas_males)

lmer_pvals_cpgs_males <- lmer_pvals_cpgs_males %>% 
  t() %>% 
  as.data.frame() %>% 
  rename_with(., ~ paste("pval", .x, sep="_"))

lmer_coeffs_cpgs_males <- lmer_coeffs_cpgs_males %>% 
  t() %>% 
  as.data.frame() %>% 
  rename_with(., ~ paste("coef", .x, sep="_"))

lmer_pvals_cpgs_results_males <- lmer_pvals_cpgs_males %>% 
  cbind(., lmer_coeffs_cpgs_males) %>%
  as.data.frame() %>%
  mutate(cpg = cb_pvals_tech_males %>% 
           subset(effect_size > 0.025 | effect_size < -0.025) %>%
           arrange(pval_no2_prenatal_avg_2019v) %>%
           slice_head(n=100) %>% 
           pull(Name),
         singular = as.vector(lapply(lmer_results_cpgs_males, isSingular) %>% do.call(rbind,.))) %>%
  #subset(singular != TRUE) %>%
  mutate(pval_no2_adj = p.adjust(pval_no2, "bonferroni"),
         pval_timepointageone_adj = p.adjust(pval_timepointageone, "bonferroni"),
         `pval_timepointageone:no2_adj` = p.adjust(`pval_timepointageone:no2`, "bonferroni"),
         coef_no2_age_one = coef_no2 + `coef_timepointageone:no2`) %>%
  mutate( change = ifelse(abs(coef_no2_age_one) > abs(coef_no2), "larger", "smaller"),
          y1_effect_size = coef_no2_age_one*range,
          birth_effect_size = coef_no2*range) %>%
  mutate(singular = as.vector(singular))

write.csv(lmer_pvals_cpgs_results_males,
          file = here ("tables",
                       "prenatal",
                       "2023-08-14_prenatal_no2_top100_persistence_MALES.csv"))

# combine persistence info with info from ewas
colnames(lmer_pvals_cpgs_results_males) <-
  paste("lme", colnames(lmer_pvals_cpgs_results_males), sep="_")

top100_lm_lme_males <- merge(x=cb_pvals_tech_males[,c(59:93)], 
                       y=lmer_pvals_cpgs_results_males, 
                       by.x = 4, by.y = 9)

colnames(top100_lm_lme_males)[34:35] <- 
  paste("lm", colnames(top100_lm_lme_males)[34:35], sep="_")

write.csv(top100_lm_lme_males, 
          file = here ("tables",
                       "prenatal",
                       "2023-08-14_prenatal_no2_top100_persistence_lm_lme_MALES.csv"))


# visualize
top100_persistence_males <- lmer_pvals_cpgs_results_males %>% 
  subset(lme_singular==FALSE) %>%
  mutate(sig = ifelse(`lme_pval_timepointageone:no2_adj` < 0.05, T, F),
         start = ifelse(lme_coef_no2 < 0, "Smaller", "Larger")) %>%
  mutate(start = factor(start, levels =c("Smaller", "Larger"))) %>%
  dplyr::select(lme_birth_effect_size, lme_y1_effect_size, lme_cpg, sig, start, lme_change) %>%
  pivot_longer(cols = -c(lme_cpg, sig, start, lme_change)) %>%
  ggplot(aes(x=name, y = value, group=lme_cpg, color = sig, shape = sig, linetype = lme_change)) +
  geom_point(alpha = 0.75, size = 1) + 
  geom_line(alpha = 0.75, size = 0.5) + 
  facet_wrap(~start) + 
  theme_bw() +
  scale_linetype_manual(values = c("solid", "dashed"), 
                        labels = c("Larger", "Smaller"), "Absolute Δ effect size") +
   scale_shape_manual(values = c(1,4), labels = c(">=0.05", "<0.05"), name = "p-value", ) +
  scale_x_discrete(labels = c("Birth", "Age one"), name = element_blank()) +
  scale_y_continuous(name = "Prenatal NO2 effect size (% DNAm)") +
  scale_color_manual(values = c("#254A90", "red2"), name = "p-value", labels = c(">=0.05", "<0.05")) +
  theme(panel.grid.major.x = element_blank(),
        axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        strip.background = element_blank(),
        legend.text = element_text(size=9),
        legend.title = element_text(size=9),
        legend.position = "none") 

ggsave(plot = top100_persistence_males,
       filename = here("figures",
                       "linear_regression",
                       "prenatal",
                       "2023-08-14_top100_persistence_MALES.png"),
       height = 6, width = 6.5, units="cm")
```

### Peristence in females

#### Linear mixed models regressions examining persistence at significant CpGs/DMRs

subset for female data
```{r females_sub_pers}
cg_df_order_svs_females <- cg_df_order_svs %>% 
  subset(sex == "F")

comb_betas_females <- comb_betas[,as.numeric(str_sub(colnames(comb_betas), 4, 8)) %in% as.numeric(cg_df_order_svs_females$sample_id)]
```

Examine persistence of of DMR prenatal changes at age one in females
```{r dmr_persistence_females}

# check order of data and dnam
identical(as.numeric(str_sub(colnames(comb_betas_females), 4, 8)), 
          as.numeric(cg_df_order_svs_females$sample_id)) # true

cg_df_order_svs_cpgs_females <- 
  cbind(cg_df_order_svs_females, 
        t(comb_betas_females[combp_anno_df_es_sub_females_mqtls %>% pull(Name), ]))
  
lmer_results_females <- NULL
lmer_pvals_females <- NULL
lmer_coeffs_females <- NULL

for(i in 95:164){
  
  cg_df_order_svs_cpgs_females$dnam <- cg_df_order_svs_cpgs_females[,i] 
  cg_lmer_females <- lmer(dnam ~  timepoint*(no2 + gestational_age) +  
                    maternal_education_length + smoking +
                    genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                    Comp.1 + Comp.2 + Comp.3 + Comp.4 + comb_sv1 + comb_sv2 +
                    comb_sv3 + comb_sv4 + comb_sv5 + comb_sv6 + comb_sv7 +
                    comb_sv8 + comb_sv9 + comb_sv10 + comb_sv11 + comb_sv12 +
                    comb_sv13 + comb_sv14 + comb_sv15 + comb_sv16 +
                    comb_sv17 + (1 |sample_id), 
                  data = cg_df_order_svs_cpgs_females)
  
  lmer_results_females <- c(lmer_results_females, 
                    cg_lmer_females)
  lmer_pvals_females <- cbind(lmer_pvals_females,
                      summary(cg_lmer_females)$coefficients[c(1,2,3,31),5])
  lmer_coeffs_females <- cbind(lmer_coeffs_females, 
                       summary(cg_lmer_females)$coefficients[c(1,2,3,31),1])
  
}

names(lmer_results_females) <- combp_anno_df_es_sub_females %>% 
  pull(Name)

lmer_pvals_females <- lmer_pvals_females %>% 
  t() %>% 
  as.data.frame() %>% 
  rename_with(., ~ paste("pval", .x, sep="_"))

lmer_coeffs_females <- lmer_coeffs_females %>% 
  t() %>% 
  as.data.frame() %>% 
  rename_with(., ~ paste("coef", .x, sep="_"))

lmer_pvals_dmrs_females <- lmer_pvals_females %>% 
  cbind(., lmer_coeffs_females) %>%
  as.data.frame() %>%
  #rename("pval" = 1) %>%
  mutate(dmr = combp_anno_df_es_sub_females_mqtls$dmr,
         cpg = combp_anno_df_es_sub_females_mqtls$Name,
         singular = lapply(lmer_results_females, isSingular) %>% do.call(rbind,.),
         pval_no2_adj = p.adjust(pval_no2, method = "bonferroni"),
         pval_timepointageone_adj = p.adjust(pval_timepointageone, method = "bonferroni"),
         `pval_timepointageone:no2_adj` = p.adjust(`pval_timepointageone:no2`, method = "bonferroni"),
         coef_no2_age_one = coef_no2 + `coef_timepointageone:no2`) %>%
  mutate( change = ifelse(abs(coef_no2_age_one) > abs(coef_no2), "larger", "smaller"),
          birth_effect_size = coef_no2 * range,
          y1_effect_size = coef_no2_age_one * range) %>%
  mutate(singular = as.vector(singular))

# combine persistence info with info from ewas
colnames(lmer_pvals_dmrs_females) <- paste("lme", 
                                         colnames(lmer_pvals_dmrs_females), 
                                         sep="_")

colnames(combp_anno_df_es_sub_females_mqtls)[c(37,38)] <- 
  paste("lm", 
        colnames(combp_anno_df_es_sub_females_mqtls)[c(37,38)], 
        sep="_")

dmrs_lm_lme_females <- merge(x=combp_anno_df_es_sub_females_mqtls, 
                     y=lmer_pvals_dmrs_females, 
                     by.x = 5, by.y = 10)

dmrs_lm_lme_sub_females <- dmrs_lm_lme_females %>% 
  dplyr::select(-c(lme_dmr, Row.names, analysis))

write.csv(dmrs_lm_lme_sub_females, 
          file = here ("tables",
                       "prenatal",
                       "2023-08-14_prenatal_no2_dmr_cpg_persistence_FEMALES.csv"))


# check if mean birth effect sizes are significantly different from year one effect sizes
lmer_effect_sizes_females <- dmrs_lm_lme_sub_females %>%
  dplyr::select(Name, dmr, lme_birth_effect_size, lme_y1_effect_size) %>% 
  pivot_longer(-c(Name, dmr)) %>%
  mutate(name = factor(name, levels = c("lme_y1_effect_size",
                                        "lme_birth_effect_size"))) %>%
  group_by(dmr) %>%
  group_map(~ t.test(abs(value) ~ name, .x, paired = TRUE)) 

lmer_effect_sizes_females_df <- do.call(rbind, lmer_effect_sizes_females)   
lmer_effect_sizes_females_df[,3] %>% 
  rbind() %>% 
  as.numeric() %>% 
  as.data.frame() %>% 
  mutate(adj = `.`*19) %>% 
  mutate(adj = ifelse(adj >=1, 1, adj)) %>%
  View()

# examine effect sizes from lme models vs those from ewas
lmer_effect_sizes_females <- lmer_pvals_dmrs_females %>%
  group_by(lme_dmr) %>%
  summarise(mean_birth_es = mean(lme_birth_effect_size),
            mean_y1_es = mean(lme_y1_effect_size)) %>%
  cbind(., combp_anno_df_es_sub_females %>% 
          group_by(dmr) %>%
          summarise(ewas_birth_es = mean(lm_effect_size)))

write.csv(lmer_effect_sizes_females, 
          file = here ("tables",
                       "prenatal",
                       "2023-08-14_prenatal_no2_dmr_cpg_persistence_effect_size_by_dmr_FEMALES.csv"))

# visualize
dmr_persistence_females <- lmer_pvals_dmrs_females %>% 
  subset(lme_singular==FALSE) %>%
  mutate(sig = ifelse(`lme_pval_timepointageone:no2_adj` < 0.05, T, F),
         start = ifelse(lme_coef_no2 < 0, "Smaller", "Larger")) %>%
  mutate(start = factor(start, levels =c("Smaller", "Larger"))) %>%
  dplyr::select(lme_birth_effect_size, lme_y1_effect_size, lme_cpg, sig, start, lme_change) %>%
  pivot_longer(cols = -c(lme_cpg, sig, start, lme_change)) %>%
  ggplot(aes(x=name, y = value, group=lme_cpg, color = sig, shape = sig, linetype = lme_change)) +
  geom_point(alpha = 0.75, size = 1) + 
  geom_line(alpha = 0.75, size = 0.5) + 
  facet_wrap(~start) + 
  theme_bw() +
  scale_linetype_manual(values = c("solid", "dashed"), 
                        labels = c("Larger", "Smaller"), "Absolute Δ \neffect size") +
   scale_shape_manual(values = c(1,4), labels = c(">=0.05", "<0.05"), name = "p-value", ) +
  scale_x_discrete(labels = c("Birth", "Age one"), name = element_blank()) +
  scale_y_continuous(name = "Prenatal NO2 effect size (% DNAm)") +
  scale_color_manual(values = c("#254A90", "red2"), name = "p-value", labels = c(">=0.05", "<0.05")) +
  theme(panel.grid.major.x = element_blank(),
        axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        strip.background = element_blank(),
        legend.text = element_text(size=9),
        legend.title = element_text(size=9),
        legend.position = "none") 

ggsave(plot = dmr_persistence_females,
       filename = here("figures",
                       "linear_regression",
                       "prenatal",
                       "2023-06-28_dmr_persistence_FEMALES.png"),
       height = 6, width = 6.5, units="cm", dpi = 600)
```

#### Examine changes in effect size at top 100 CpGs from EWAS over the first year of life in females

Subset y1 data for females
```{r sub_y1_females}
covariates_geno_ctpc_tech_cc_y1_sub_females <-
  covariates_geno_ctpc_tech_cc_y1_sub %>%
  subset(sex == "F")

# get top 100 cord blood cpgs 
cb100_females <- cb_pvals_tech_females %>% 
  subset(effect_size > 0.025 | effect_size < -0.025) %>%
  arrange(pval_no2_prenatal_avg_2019v) %>%
  slice_head(n=100)

y1_betas_cc_sub_females <- y1_betas_cc_sub[,str_sub(colnames(y1_betas_cc_sub[rownames(cb100_females), ] ),4,8) %in% covariates_geno_ctpc_tech_cc_y1_sub_females$sample_id]
```

```{r top100_effect_size_persistence_females}
# get birth and age one dnam for top 100 cord blood cpgs
cb_top100_betas_females <- 
  cb_betas_geno_cc_clean_df_females[rownames(cb100_females), 
                            str_sub(colnames(y1_betas_cc_sub_females[rownames(cb100_females), ] ),4,8)] %>%
  t() %>%
  as.data.frame() 
# rename_with(., ~ paste("cb", .x, sep="_"))

y1_top100_betas_females <- 
  y1_betas_cc_sub_females[rownames(cb100_females), ] %>%
  t() %>%
  as.data.frame()
# rename_with(., ~ paste("y1", .x, sep="_"))

# check order
identical(rownames(cb_top100_betas_females), 
          str_sub(rownames(y1_top100_betas_females), 4, 8)) # TRUE

identical(rownames(cb_top100_betas_females), 
          cg_df_order_svs_cpgs_females %>%
            subset(timepoint=="birth") %>% 
            pull(sample_id) %>% 
            as.character())

top100_betas_females <- rbind(cb_top100_betas_females, 
                      y1_top100_betas_females)

colnames(top100_betas_females) <- paste("top100_",
                                colnames(top100_betas_females),
                                sep="")

cg_df_order_svs_cpgs_top100cpgs_females <- 
  cbind(cg_df_order_svs_cpgs_females, top100_betas_females)

lmer_results_cpgs_females <- NULL
lmer_pvals_cpgs_females <- NULL
lmer_coeffs_cpgs_females <- NULL

for(i in 166:265){
  
  cg_df_order_svs_cpgs_top100cpgs_females$dnam <- cg_df_order_svs_cpgs_top100cpgs_females[,i] 
  
  cg_lmer_cpgs_females <- lmer(dnam ~  timepoint*(no2 + gestational_age) +  
                         maternal_education_length + smoking +
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                         Comp.1 + Comp.2 + Comp.3 + Comp.4 + comb_sv1 + comb_sv2 +
                         comb_sv3 + comb_sv4 + comb_sv5 + comb_sv6 + comb_sv7 +
                         comb_sv8 + comb_sv9 + comb_sv10 + comb_sv11 + comb_sv12 +
                         comb_sv13 + comb_sv14 + comb_sv15 + comb_sv16 + 
                           comb_sv17 + (1 |sample_id), 
                       data = cg_df_order_svs_cpgs_top100cpgs_females)
  
  lmer_results_cpgs_females <- c(lmer_results_cpgs_females, 
                         cg_lmer_cpgs_females)
  lmer_pvals_cpgs_females <- cbind(lmer_pvals_cpgs_females, 
                           summary(cg_lmer_cpgs_females)$coefficients[c(1,2,3,31),5])
  lmer_coeffs_cpgs_females <- cbind(lmer_coeffs_cpgs_females, 
                            summary(cg_lmer_cpgs_females)$coefficients[c(1,2,3,31),1])
}

names(lmer_results_cpgs_females) <- colnames(top100_betas_females)

lmer_pvals_cpgs_females <- lmer_pvals_cpgs_females %>% 
  t() %>% 
  as.data.frame() %>% 
  rename_with(., ~ paste("pval", .x, sep="_"))

lmer_coeffs_cpgs_females <- lmer_coeffs_cpgs_females %>% 
  t() %>% 
  as.data.frame() %>% 
  rename_with(., ~ paste("coef", .x, sep="_"))

lmer_pvals_cpgs_results_females <- lmer_pvals_cpgs_females %>% 
  cbind(., lmer_coeffs_cpgs_females) %>%
  as.data.frame() %>%
  mutate(cpg = cb_pvals_tech_females %>% 
           subset(effect_size > 0.025 | effect_size < -0.025) %>%
           arrange(pval_no2_prenatal_avg_2019v) %>%
           slice_head(n=100) %>% 
           pull(Name),
         singular = as.vector(lapply(lmer_results_cpgs_females, isSingular) %>% do.call(rbind,.))) %>%
  #subset(singular != TRUE) %>%
  mutate(pval_no2_adj = p.adjust(pval_no2, "bonferroni"),
         pval_timepointageone_adj = p.adjust(pval_timepointageone, "bonferroni"),
         `pval_timepointageone:no2_adj` = p.adjust(`pval_timepointageone:no2`, "bonferroni"),
         coef_no2_age_one = coef_no2 + `coef_timepointageone:no2`) %>%
  mutate( change = ifelse(abs(coef_no2_age_one) > abs(coef_no2), "larger", "smaller"),
          y1_effect_size = coef_no2_age_one*range,
          birth_effect_size = coef_no2*range) %>%
  mutate(singular = as.vector(singular))

write.csv(lmer_pvals_cpgs_results_females,
          file = here ("tables",
                       "prenatal",
                       "2023-08-14_prenatal_no2_top100_persistence_FEMALES.csv"))

# combine persistence info with info from ewas
colnames(lmer_pvals_cpgs_results_females) <-
  paste("lme", colnames(lmer_pvals_cpgs_results_females), sep="_")

top100_lm_lme_females <- merge(x=cb_pvals_tech_females[,c(59:93)], 
                       y=lmer_pvals_cpgs_results_females, 
                       by.x = 4, by.y = 9)

colnames(top100_lm_lme_females)[34:35] <- 
  paste("lm", colnames(top100_lm_lme_females)[34:35], sep="_")

write.csv(top100_lm_lme_females, 
          file = here ("tables",
                       "prenatal",
                       "2023-08-14_prenatal_no2_top100_persistence_lm_lme_FEMALES.csv"))

# visualize
top100_persistence_females <- lmer_pvals_cpgs_results_females %>% 
  subset(lme_singular==FALSE) %>%
  mutate(sig = ifelse(`lme_pval_timepointageone:no2_adj` < 0.05, T, F),
         start = ifelse(lme_birth_effect_size < 0, "Smaller", "Larger")) %>%
  mutate(start = factor(start, levels =c("Smaller", "Larger"))) %>%
  dplyr::select(lme_birth_effect_size, lme_y1_effect_size, lme_cpg, sig, start, lme_change) %>%
  pivot_longer(cols = -c(lme_cpg, sig, start, lme_change)) %>%
  ggplot(aes(x=name, y = value, group=lme_cpg, color = sig, shape = sig, linetype = lme_change)) +
  geom_point(alpha = 0.75, size = 1) + 
  geom_line(alpha = 0.75, size = 0.5) + 
  facet_wrap(~start) + 
  theme_bw() +
  scale_linetype_manual(values = c("solid", "dashed"),
                        labels = c("Larger", "Smaller"), "Absolute Δ \neffect size") +
   scale_shape_manual(values = c(1,4), labels = c(">=0.05", "<0.05"), name = "p-value", ) +
  scale_x_discrete(labels = c("Birth", "Age one"), name = element_blank()) +
  scale_y_continuous(name = "Prenatal NO2 effect size (% DNAm)") +
  scale_color_manual(values = c("#254A90", "red2"), name = "p-value", labels = c(">=0.05", "<0.05")) +
  theme(panel.grid.major.x = element_blank(),
        axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        strip.background = element_blank(),
        legend.text = element_text(size=9),
        legend.title = element_text(size=9),
        legend.position = "none") 

ggsave(plot = top100_persistence_females,
       filename = here("figures",
                       "linear_regression",
                       "prenatal",
                       "2023-08-14_top100_persistence_FEMALES.png"),
       height = 6, width = 6.5, units="cm")
```

### Table 1 for persistence model

```{r age_one_persistence_table1}
library(table1)
# get cell type estimates (not pcs)
cb_cell_estimates <- child_fscbc_ecc2_cb$prop
dim(cb_cell_estimates)

y1_cell_estimates <- child_fscbc_ecc2_y1$prop
dim(y1_cell_estimates)

# subset out duplicates
cb_cell_estimates_sub <- cb_cell_estimates %>% 
  subset(rownames(.) %in% paste(cb_pdata$Sentrix_ID, cb_pdata$Sentrix_Position, sep = "_"))
dim(cb_cell_estimates_sub) # 144

# rename rows
identical(rownames(cb_cell_estimates_sub),
          paste(cb_pdata$Sentrix_ID, cb_pdata$Sentrix_Position, sep = "_")) # TRUE
rownames(cb_cell_estimates_sub) <- cb_pdata$Sample_Label

identical(rownames(y1_cell_estimates),
          paste(y1_pdata$Sentrix_ID, y1_pdata$Sentrix_Position, sep = "_")) # TRUE
rownames(y1_cell_estimates) <- y1_pdata$Sample_Label

# subset for participants with complete data
cb_cell_estimates_sub_cc <- cb_cell_estimates_sub %>%
  subset(rownames(.) %in% covariates_geno_ctpc_tech_cc_y1_sub$sample_id)
dim(cb_cell_estimates_sub_cc) # 125 6

y1_cell_estimates_cc <- y1_cell_estimates %>%
  subset(rownames(.) %in% covariates_geno_ctpc_tech_cc_y1_sub$sample_id)
dim(y1_cell_estimates_cc) # 125 5

# check order
identical(rownames(cb_cell_estimates_sub_cc), 
          as.character(covariates_geno_ctpc_tech_cc_y1_sub$sample_id)) # TRUE

identical(rownames(y1_cell_estimates_cc), 
          as.character(covariates_geno_ctpc_tech_cc_y1_sub$sample_id)) # TRUE

# rename cols for clarity
colnames(cb_cell_estimates_sub_cc) <- paste("cb", 
                                            colnames(cb_cell_estimates_sub_cc), 
                                            sep = "_")

colnames(y1_cell_estimates_cc) <- paste("y1", 
                                            colnames(y1_cell_estimates_cc), 
                                            sep = "_")


# subset covariates with atopy wheeze group for persistence 
covariates_geno_ctpc_tech_cc_aw_y1 <- covariates_geno_ctpc_tech_cc_aw %>%
  subset(sample_id %in% covariates_geno_ctpc_tech_cc_y1_sub$sample_id)

identical(covariates_geno_ctpc_tech_cc_aw_y1$sample_id,
          covariates_geno_ctpc_tech_cc_y1_sub$sample_id) # TRUE

# bind cell type estimates with covariate info
covariates_geno_ctpc_tech_cc_aw_y1_cbmc <- cbind(covariates_geno_ctpc_tech_cc_aw_y1, 
                                           cb_cell_estimates_sub_cc)

covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc <- cbind(covariates_geno_ctpc_tech_cc_aw_y1_cbmc, 
                                           y1_cell_estimates_cc)
# table 1
# table 1
table1(~sex + no2_prenatal_avg_2019v + sex + gestational_age +
           maternal_education_length +  maternal_race + prenatal_maternal_smoke +
           maternal_smoking_y1 + cb_CD4T + cb_CD8T + cb_Bcell + cb_Gran + cb_Mono + cb_NK + 
           cb_nRBC + y1_CD4T + y1_CD8T + y1_Bcell + y1_Gran + y1_Mono + y1_NK + Group,
       data = covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)

# prenatal no2 quartiles
quantile(covariates_geno_ctpc_tech_cc_y1_sub_cbmc_pbmc %>% subset(sex == "F") %>% pull(no2_prenatal_avg_2019v))
quantile(covariates_geno_ctpc_tech_cc_y1_sub_cbmc_pbmc %>% subset(sex == "M") %>% pull(no2_prenatal_avg_2019v))

<<<<<<< HEAD
anova(lm(cb_CD4T~ studycentre, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc))
anova(lm(cb_CD8T~ studycentre, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(cb_Gran~ studycentre, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(cb_Mono~ studycentre, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(cb_NK~ studycentre, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(cb_Bcell~ studycentre, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc))
anova(lm(cb_nRBC~ studycentre, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 

anova(lm(y1_CD4T~ studycentre, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(y1_CD8T~ studycentre, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(y1_Mono~ studycentre, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(y1_NK~ studycentre, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(y1_Bcell~ studycentre, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(y1_Gran~ studycentre, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 


anova(lm(cb_CD4T~ sex, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc))
anova(lm(cb_CD8T~ sex, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(cb_Gran~ sex, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(cb_Mono~ sex, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(cb_NK~ sex, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(cb_Bcell~ sex, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc))
anova(lm(cb_nRBC~ sex, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 

anova(lm(y1_CD4T~ sex, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(y1_CD8T~ sex, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(y1_Gran~ sex, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(y1_Mono~ sex, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(y1_NK~ sex, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 
anova(lm(y1_Bcell~ sex, covariates_geno_ctpc_tech_cc_aw_y1_cbmc_pbmc)) 

=======
anova(lm(cb_CD4T~ studycentre, covariates_geno_ctpc_tech_cc_y1_sub_cbmc_pbmc))
anova(lm(cb_CD8T~ studycentre, covariates_geno_ctpc_tech_cc_y1_sub_cbmc_pbmc)) 
anova(lm(cb_Mono~ studycentre, covariates_geno_ctpc_tech_cc_y1_sub_cbmc_pbmc)) 
anova(lm(cb_NK~ studycentre, covariates_geno_ctpc_tech_cc_y1_sub_cbmc_pbmc)) 
anova(lm(cb_Bcell~ studycentre, covariates_geno_ctpc_tech_cc_y1_sub_cbmc_pbmc))
anova(lm(cb_nRBC~ studycentre, covariates_geno_ctpc_tech_cc_y1_sub_cbmc_pbmc)) 

anova(lm(y1_CD4T~ studycentre, covariates_geno_ctpc_tech_cc_y1_sub_cbmc_pbmc)) 
anova(lm(y1_CD8T~ studycentre, covariates_geno_ctpc_tech_cc_y1_sub_cbmc_pbmc)) 
anova(lm(y1_Mono~ studycentre, covariates_geno_ctpc_tech_cc_y1_sub_cbmc_pbmc)) 
anova(lm(y1_NK~ studycentre, covariates_geno_ctpc_tech_cc_y1_sub_cbmc_pbmc)) 
anova(lm(y1_Bcell~ studycentre, covariates_geno_ctpc_tech_cc_y1_sub_cbmc_pbmc)) 
>>>>>>> f62b21d7e4615973aae12bc89a4c51ff42b0ad1e
```

save session info
```{r session_info}
sesh <- sessionInfo()

save(sesh,
     file = here("output_data",
                 "linear_regression",
                 "2023-06-28_no2_prenatal_dnam_session_info.Rdata"))

```

## Prenatal and postnatal air pollution exposure boxplots

```{r airpollution_boxplots}

# prenatal no2
studysite_prenatal_no2 <- ggplot(covariates_geno_ctpc_tech_cc, aes(y=no2_prenatal_avg_2019v, 
                                                                   x=studycentre, 
                                                                   fill=studycentre)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(height=0, width=0.2, alpha=0.6, size = 1) +
  theme_bw() +
  scale_fill_manual(values=c("#254A90", "#508FC2", "#92C5EB", "white")) +
  labs(x="Study Centre", y=bquote("Prenatal" ~NO[2]~ "(ppb)")) +
  theme(panel.grid = element_blank(),
        legend.position = "none",
        axis.text = element_text(size=9),
        axis.title.x = element_blank(),
        axis.title = element_text(size=11)) +
  geom_signif(
    comparisons = list(c("Winnipeg", "Vancouver"),
                       c("Vancouver", "Toronto"),
                       c("Edmonton", "Toronto"),
                       c("Winnipeg", "Toronto"),
                       c("Winnipeg", "Edmonton")),
    step_increase = 0.1,
    annotations = c("8.4e-12", 
                    "1.5e-06", 
                    "6.7e-04",
                    "7.6e-15",
                    "3.3e-14"), 
    margin_top = 0, textsize = 2.5, 
  ) + ylim(0, 40)

anova(lm(no2_prenatal_avg_2019v ~ studycentre, data = covariates_geno_ctpc_tech_cc))
TukeyHSD(aov(lm(no2_prenatal_avg_2019v ~ studycentre, data = covariates_geno_ctpc_tech_cc)))

ggsave(plot = studysite_prenatal_no2,
       filename = here("figures",
                       "linear_regression",
                       "prenatal",
                       "2023-06-28_studysite_vs_prenatalno2.png"),
       height = 5, width = 9, units="cm")

covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v %>% mean()
covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v %>% sd()

covariates_geno_ctpc_tech_cc %>%
  group_by(studycentre) %>%
  summarise(mean(no2_prenatal_avg_2019v),
            sd(no2_prenatal_avg_2019v))

# prenatal no2 persistence
studysite_prenatal_no2_persistence <- ggplot(covariates_geno_ctpc_tech_cc_y1_sub, 
                                             aes(y=no2_prenatal_avg_2019v, x=studycentre,fill=studycentre)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(height=0, width=0.2, alpha=0.6, size = 1) +
  theme_bw() +
  scale_fill_manual(values=c("#254A90", "#508FC2", "#92C5EB", "white")) +
  labs(x="Study Centre", y=bquote("Prenatal" ~NO[2]~ "(ppb)")) +
  theme(panel.grid = element_blank(),
        legend.position = "none",
        axis.text = element_text(size=9),
        axis.title.x = element_blank(),
        axis.title = element_text(size=11)) +
  geom_signif(
    comparisons = list(c("Winnipeg", "Vancouver"),
                       c("Vancouver", "Toronto"),
                       c("Edmonton", "Toronto"),
                       c("Winnipeg", "Toronto"),
                       c("Winnipeg", "Edmonton")),
    step_increase = 0.1,
    annotations = c("1.9e-11", 
                    "4.0e-07", 
                    "5.0e-04",
                    "2.4e-14",
                    "4.5e-14"), 
    margin_top = 0, textsize = 2.5,
  ) + ylim(0, 40)

anova(lm(no2_prenatal_avg_2019v ~ studycentre, data = covariates_geno_ctpc_tech_cc_y1_sub))
TukeyHSD(aov(lm(no2_prenatal_avg_2019v ~ studycentre, data = covariates_geno_ctpc_tech_cc_y1_sub)))

ggsave(plot = studysite_prenatal_no2_persistence,
       filename = here("figures",
                       "linear_regression",
                       "prenatal",
                       "2023-06-28_studysite_prenatal_no2_persistence.png"),
       height = 5, width = 9, units="cm")

covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v %>% mean()
covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v %>% sd()

covariates_geno_ctpc_tech_cc_y1_sub %>%
  group_by(studycentre) %>%
  summarise(mean(no2_prenatal_avg_2019v),
            sd(no2_prenatal_avg_2019v))


# compare prenatal vs year one no2 exposure 
t.test(covariates_geno_ctpc_tech_cc %>% 
         dplyr::select(no2_prenatal_avg_2019v, no2_y1_2019v, sample_id) %>% 
         na.omit() %>%
         pull(no2_prenatal_avg_2019v),
       covariates_geno_ctpc_tech_cc %>% 
         dplyr::select(no2_prenatal_avg_2019v, no2_y1_2019v, sample_id) %>% 
         na.omit() %>%
         pull(no2_y1_2019v), paired = T)

prenatal_vs_yearone_no2 <- ggplot(covariates_geno_ctpc_tech_cc %>% 
         dplyr::select(no2_prenatal_avg_2019v, no2_y1_2019v, sample_id, studycentre) %>% 
         na.omit() %>% 
         mutate(higher = ifelse(no2_y1_2019v > no2_prenatal_avg_2019v, 1, 0)) %>%
         pivot_longer(-c(sample_id, higher, studycentre)),
       aes(x=name, y = value, 
           shape = as.factor(higher),
           color = as.factor(higher), 
           linetype = as.factor(higher))) +
  geom_point(alpha = 0.5, size = 2) +
  geom_line(aes(group = sample_id), alpha = 0.8, linewidth = 0.5) +
  scale_linetype_manual(values = c("solid", "dashed"), labels = c("Decreased", "Increased"),
                         name = bquote("Δ" ~NO[2]~ "exposure")) + 
  scale_color_manual(values = c("#254A90", "red"), labels = c("Decreased", "Increased"), 
                     name = bquote("Δ" ~NO[2]~ "exposure")) +
  scale_shape_manual(values = c(1,4),  labels = c("Decreased", "Increased"), 
                     name = bquote("Δ" ~NO[2]~ "exposure")) +
  scale_x_discrete(labels = c("Prenatal", "Year one")) +
  labs(y=bquote( ~NO[2]~ "exposure (ppb)")) +
  theme_bw() +
  theme(panel.grid.major.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text = element_text(size = 9),
        axis.title.y =element_blank(),  #element_text(size = 11),
        legend.position = "none") +
  geom_signif(
    comparisons = list(c("no2_prenatal_avg_2019v", "no2_y1_2019v")),
    #annotations = c("1.8e-05"), 
    textsize = 3, color = "black", margin_top = 0.1, test = t.test, test.args=list(paired = T),
  ) + ylim(0, 35)

ggsave(plot = prenatal_vs_yearone_no2,
       filename = here("figures",
                       "linear_regression",
                       "prenatal",
                       "2023-06-28_prenatal_vs_yearone_no2_exp_nolegend.png"),
       height = 8, width = 5, units="cm")


prenatal_vs_y1_corr <- ggplot(covariates_geno_ctpc_tech_cc %>% 
           dplyr::select(no2_prenatal_avg_2019v, 
                         no2_y1_2019v, sample_id) %>% 
           na.omit() %>%
           mutate(higher = ifelse(no2_y1_2019v > no2_prenatal_avg_2019v, 1, 0)), 
       aes(x=no2_prenatal_avg_2019v, y = no2_y1_2019v, shape = as.factor(higher), colour = as.factor(higher))) + 
    geom_smooth(method = "lm", color = "black", linetype = "dashed", linewidth = 0.5, se = F, inherit.aes = F,
                aes(x=no2_prenatal_avg_2019v, y = no2_y1_2019v)) +
    scale_color_manual(values = c("#254A90", "red"), labels = c("Decreased", "Increased"),
                       name = bquote("Δ" ~NO[2]~ "exposure")) +
    scale_shape_manual(values = c(1,4), labels = c("Decreased", "Increased"),
                       name = bquote("Δ" ~NO[2]~ "exposure")) +
    geom_point(alpha =0.8)  +
    theme_bw() +
    theme(axis.title = element_blank(),
          axis.text = element_text(size=9),
          legend.position = "none") 

ggsave(plot = prenatal_vs_y1_corr,
       filename = here("figures",
                       "linear_regression",
                       "prenatal",
                       "2023-06-28_prenatal_vs_y1_corr.png"),
       height = 8, width = 5, units="cm")


cor(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v, covariates_geno_ctpc_tech_cc)

# prenatal smoking 
smoking <- covariates_geno_ctpc_tech_cc %>% 
    dplyr::select(prenatal_maternal_smoke, prenatal_second_hand_smoke, maternal_smoking_y1, urine_cotinine_3m, sample_id) %>% 
    subset(!is.na(urine_cotinine_3m)) %>%
    mutate(prenatal_shs_y1 = ifelse(prenatal_maternal_smoke == 1 & prenatal_second_hand_smoke ==1 & maternal_smoking_y1 == 1, 1, 0),
           prenatal_shs = ifelse(prenatal_maternal_smoke == 1 & prenatal_second_hand_smoke ==1 & maternal_smoking_y1 == 0, 1, 0),
           prenatal_y1 = ifelse(prenatal_maternal_smoke == 1 & prenatal_second_hand_smoke == 0 & maternal_smoking_y1 == 1, 1, 0),
           shs_y1 = ifelse(prenatal_maternal_smoke == 0 & prenatal_second_hand_smoke == 1 & maternal_smoking_y1 == 1, 1, 0), 
           prenatal_only = ifelse(prenatal_maternal_smoke == 1 & prenatal_second_hand_smoke == 0 & maternal_smoking_y1 == 0, 1, 0),
           shs_only = ifelse(prenatal_maternal_smoke == 0 & prenatal_second_hand_smoke == 1 & maternal_smoking_y1 == 0, 1, 0),
           y1_only = ifelse(prenatal_maternal_smoke == 0 & prenatal_second_hand_smoke == 0 & maternal_smoking_y1 == 1, 1, 0),
           no_smoke = ifelse(prenatal_maternal_smoke == 0 & prenatal_second_hand_smoke == 0 & maternal_smoking_y1 == 0, 1, 0)) %>%
    dplyr::select(urine_cotinine_3m:no_smoke) %>%
    pivot_longer(-c(urine_cotinine_3m,sample_id)) %>% 
    mutate(name = factor(name, 
                         levels = c("no_smoke", "prenatal_only", "shs_only",
                                    "y1_only", "prenatal_shs", "prenatal_y1", 
                                    "shs_y1", "prenatal_shs_y1"))) %>%
    na.omit() %>%
    subset(value == 1) %>%
    subset(urine_cotinine_3m < 50) %>%
    ggplot(aes(x=name, y = urine_cotinine_3m)) +
    geom_violin(draw_quantiles = c(0.5)) +
    ggbeeswarm::geom_beeswarm(alpha = 0.8, cex = 1, size = 0.75,
                              corral = "gutter", corral.width = 1, color = "#254A90") +
    theme_bw() +
    theme(axis.title = element_blank())

ggsave(smoking,
       file = here("figures",
                  "2023-06-28_smoke_exposures_urine_cotinine.png"),
       units = "cm", height = 6, width = 12, dpi=600)
```
