---
title: "CHILD cord blood NO2 DNAm linear regression"
author: "SL"
date: "December 14, 2020"
output: html_document
---

*Notes*: Missing covariate data is filled in with the mean for that variable. Batch correction was not performed during pre-processing. Batch variables will be included in linear regression. DNAm beta values were corrected for cell type before linear regression (before this script). 

*Purpose*: The purpose of this script is to examine DNAm changes in cord blood associated with prenatal air pollution exposure (proxied by no2).


Load required libraries.
```{r libraries}
library(tidyverse)
library(limma)
library(here)
library(pbapply) #for running lm for each cpg
library(piecewiseSEM) #for getting partial residuals
library(broom) #for tidy function for lm
```


Load betas
```{r load_betas}
load(file=here("output_data", "celltype_corrected_betas", "2021-01-08_CHILD_cb_betas_5celltypePC_chiprow_corrected.Rdata"))

#check ranges to confirm DNAm data is in beta value format
dim(adj_cb_6pcs_betas) #424644    144
max(adj_cb_6pcs_betas) #0.9999998
min(adj_cb_6pcs_betas) #1.134957e-06
```


Subset out cord blood pdata
```{r subset_pdata}
#subset
cb_pdata <- subset(pdata_nodups, Tissue == "C")
#check size
dim(cb_pdata) #144 17

#create new column in pdata to match naming convention of betas
cb_pdata$sentrix_name <- paste(cb_pdata$Sentrix_ID, cb_pdata$Sentrix_Position, sep="_")
#check if this is identical with betas
identical(colnames(adj_cb_6pcs_betas), cb_pdata$sentrix_name)#true
#rename colnames of beta values to sample label
colnames(adj_cb_6pcs_betas) <- cb_pdata$Sample_Label
```


Load covariate data
```{r covariates}
load(file=here("output_data", "covariates", "2020-11-12_CHILD_covariates_na_with_means.Rdata"))
```


Subset covariates for individuals with cord blood DNAm.
```{r sub_covariates}
covariates_sub <- covariates_nameans[as.character(covariates_nameans$sampleid) %in% colnames(adj_cb_6pcs_betas), ]

#check if they are in the same order
identical(as.character(covariates_sub$sampleid), colnames(adj_cb_6pcs_betas)) #true
```


Load genotyping PCs and subset out REEGLE PCs (genotyping data contains PCs for a few thousand participants).
```{r genotypingPCS}
genotyping <- read.csv("R:/Jones/People/Samantha Lee/Computational/CHILD/input_data/CHILD_subjects_Qced_first10PCS_Duan.csv")

#make rownames sample names then rmove this row
rownames(genotyping) <- genotyping$FID
genotyping <- subset(genotyping, select=-c(FID, IID))

genosub <- genotyping[rownames(genotyping) %in% cb_pdata$Sample_Label,]
#missing genotyping data for 14 REEGLE children?

#rename columns of genotyping data for clarity
colnames(genosub) <- paste("genotyping.", colnames(genosub), sep="")
```


Check that betas, pdata, and covariates are in same order.
```{r cb_batchcorr_order}
#check that pdata and betas are in correct order
identical(as.character(cb_pdata$Sample_Label), colnames(adj_cb_6pcs_betas)) #true

#check that pdata and covariates are in the same order
identical(as.character(cb_pdata$Sample_Label), as.character(covariates_sub$sampleid)) #true
```


Subset covariates, pdata, and betas for individuals that have genotyping data. Add genotyping pcs to covariates
```{r sub_on_geno}
#subset covariates
covariates_sub_geno <- covariates_sub[covariates_sub$sampleid %in% rownames(genosub), ]
dim(covariates_sub_geno) #131

#check that covariates and geno are in same order
identical(as.character(covariates_sub_geno$sampleid), rownames(genosub)) #false

#put genosub rows in numerical order
genosub_order <- genosub[order(rownames(genosub)), ]

#check that covariates and geno are in same order
identical(as.character(covariates_sub_geno$sampleid), rownames(genosub_order)) #true

#combine genosub with covariates
covariates_sub_geno <- cbind(covariates_sub_geno, genosub_order)

#subet betas for genotyping data
adj_cb_6pcs_betas_geno <- adj_cb_6pcs_betas[,colnames(adj_cb_6pcs_betas) %in% rownames(genosub_order)] 
dim(adj_cb_6pcs_betas_geno) #131

#check that this is in the same order as covariates
identical(as.character(covariates_sub_geno$sampleid), colnames(adj_cb_6pcs_betas_geno)) #true

#subset pdata
cb_pdata_geno <- cb_pdata[as.character(cb_pdata$Sample_Label) %in% rownames(genosub_order), ]
dim(cb_pdata_geno) #131 16

#check that covariates and dpata are in same order
identical(as.character(cb_pdata_geno$Sample_Label), as.character(covariates_sub_geno$sampleid)) #true

#create new column for row in pdata
cb_pdata_geno$row <- substr(cb_pdata_geno$Sentrix_Position, 1, 3)
#examine
head(cb_pdata_geno$row)

#rename pdata sentrix Id column to chip
colnames(cb_pdata_geno)[8] <- "chip"

#rename pdata sentrix Id column to chip
colnames(cb_pdata_geno)[11] <- "run"

#add technical columns of pdata to covariates data
covariates_sub_geno_tech <- cbind(covariates_sub_geno, cb_pdata_geno[,c("run", "row", "chip")])


#check class of technical factors
#should all be character as order doesnt matter
class(covariates_sub_geno_tech$run)
class(covariates_sub_geno_tech$row)
class(covariates_sub_geno_tech$chip)

covariates_sub_geno_tech$run <- factor(covariates_sub_geno_tech$run, ordered=F)
covariates_sub_geno_tech$row <- factor(covariates_sub_geno_tech$row, ordered=T)
covariates_sub_geno_tech$row <- as.numeric(covariates_sub_geno_tech$row)

covariates_sub_geno_tech$chip <- as.factor(covariates_sub_geno_tech$chip)

is.ordered(covariates_sub_geno_tech$birth_month) #false



#looking at genotyping PCS
#ggplot(covariates_sub_geno_tech, 
#       aes(x=genotyping.PC1, y=genotyping.PC2, col=maternal_ethnicity)) + 
#  geom_point(alpha=0.6, size=2) +
#  geom_text_repel(data=subset(covariates_sub_geno_tech, genotyping.PC3 > 0.15), 
#            aes(label=sampleid), 
#            col="black",
#            min.segment.length=0.1) +
#  theme_bw() +
#  theme(panel.border = element_blank())

#ggplot(covariates_sub_geno_tech, 
#       aes(x=genotyping.PC1, y=genotyping.PC3, col=maternal_ethnicity)) + 
#  geom_point(alpha=0.6, size=2) +
#  geom_text_repel(data=subset(covariates_sub_geno_tech, genotyping.PC3 > 0.15), 
#            aes(label=sampleid), 
#            col="black",
#            min.segment.length=0.1) +
#  theme_bw() +
#  theme(panel.border = element_blank())
```


Linear regression with limma.
```{r limma}

#first check how many complete cases available for analysis
#101 complete cases now!
sum(complete.cases(covariates_sub_geno_tech %>% dplyr::select(no2_preg_entire, 
                                                              sex,
                                                              gestational_days,
                                                              pss_18wk,
                                                              maternal_education_length,
                                                              prenatal_maternal_smoking,
                                                              genotyping.PC1, genotyping.PC2,
                                                              genotyping.PC3,
                                                              run,row))) 


#create model matrix
cb_design <- model.matrix(~no2_preg_entire + 
              sex + 
              gestational_days + 
              pss_18wk +
              maternal_education_length + 
              prenatal_maternal_smoking + 
              genotyping.PC1 +
              genotyping.PC2 +
              genotyping.PC3 +
              run + row,
              data=covariates_sub_geno_tech)

#check size
dim(cb_design) #101


#further subset covariates and beta values for complete cases
#subset covariates
covariates_sub_geno_tech_cc <- 
  covariates_sub_geno_tech[complete.cases(covariates_sub_geno_tech %>% 
                                            dplyr::select(no2_preg_entire, 
                                                          sex,
                                                          gestational_days,
                                                          pss_18wk,
                                                          maternal_education_length, 
                                                          prenatal_maternal_smoking,
                                                          genotyping.PC1,
                                                          genotyping.PC2,
                                                          genotyping.PC3,
                                                          run, row)), ] 
#check size
dim(covariates_sub_geno_tech_cc) #101

#subset betas
adj_cb_6pcs_betas_geno_cc <- 
  adj_cb_6pcs_betas_geno[,colnames(adj_cb_6pcs_betas_geno) %in% covariates_sub_geno_tech_cc$sampleid]

#check size
dim(adj_cb_6pcs_betas_geno_cc) #101

#fit methylation to no2
cb_fit <-  lmFit(adj_cb_6pcs_betas_geno_cc, cb_design)
cb_ebayes <- eBayes(cb_fit)

#add annotation to cb_ebayes
#subset annotaiton for cpgs in lm
annotation_cpgs <- annotation[rownames(annotation) %in% rownames(cb_ebayes),]
#check if order of annotationa and lm is the same
identical(rownames(annotation_cpgs), rownames(cb_ebayes)) #false
#reorder annotation to match lm
annotation_cpgs_order <- annotation_cpgs[rownames(cb_ebayes), ]
#recheck order
identical(rownames(annotation_cpgs_order), rownames(cb_ebayes)) #true

cb_ebayes$annotation <- annotation_cpgs_order

#pull out pval, coefficients and annotation
cb_pvals_tech <- cbind(as.data.frame(cb_ebayes$p.value), 
                  as.data.frame(cb_ebayes$coefficients),
                  as.data.frame(cb_ebayes$annotation))

#rename columns to indicate pvalues vs coeffcients 
colnames(cb_pvals_tech) <- c(paste("pval", colnames(cb_pvals_tech)[1:14], sep = "_"), 
                         paste("coeff", colnames(cb_pvals_tech)[15:28], sep = "_"),
                         colnames(cb_pvals_tech)[29:61])

cb_pvals_tech$pval_no2_preg_entire_adj <- p.adjust(cb_pvals_tech$pval_no2_preg_entire, method="BH")
```


Examine qqplot and pvalue histogram.
```{r qqplot_pval_histo}
library(car)
#qqplot with normal distribution

#estimate parameters for normal distribution
params <- as.list(MASS::fitdistr(-log10(cb_pvals_tech$pval_no2_preg_entire), "normal")$estimate)

#normal plot plot
ggplot(aes(sample = -log10(pval_no2_preg_entire)), data = cb_pvals_tech) + 
  stat_qq(col = "#75A2D9", distribution = qnorm, dparams = params) +
  stat_qq_line(distribution = qnorm, dparams = params) +
  theme_bw() +
  labs(x="Expected -log10(p)", y="Observed -log10(p)") + 
  theme(panel.border = element_blank(),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        axis.text=element_text(size=16),
        axis.title=element_text(size=20))




#qqplot with gamma distribution

#estimate gamma params
#use these is stat_qq and stat_qq_line
params <- as.list(MASS::fitdistr(-log10(cb_pvals_tech$pval_no2_preg_entire), "gamma")$estimate)

#gamma plot
ggplot(aes(sample = -log10(pval_no2_preg_entire)), data = cb_pvals_tech) + 
  stat_qq(col = "#75A2D9", distribution = qgamma, dparams = params) +
  stat_qq_line(distribution = qgamma, dparams = params) +
  theme_bw() +
  labs(x="Expected -log10(p)", y="Observed -log10(p)") + 
  theme(panel.border = element_blank(),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        axis.text=element_text(size=16),
        axis.title=element_text(size=20))


#histograms
ggplot(cb_pvals_tech, aes(-log10(pval_no2_preg_entire))) + geom_histogram()
ggplot(cb_pvals_tech, aes((pval_no2_preg_entire))) + geom_histogram()


#alternate way of making qqplots

#gamma qqplot
qqPlot(-log10(cb_pvals_tech$pval_no2_preg_entire), y = NULL, distribution = "gamma", estimate.params = T, add.line=T)

#normal qqplot
qqPlot(-log10(cb_pvals_tech$pval_no2_preg_entire), y = NULL, distribution = "norm", estimate.params = T, add.line=T)
        


#histogram
ggplot( data = cb_pvals_tech, aes(x=pval_no2_preg_entire)) +
  geom_histogram(bins=100, fill = "#75A2D9") +
  labs(y="Count", x="Unadjusted p-value") +
  theme_bw() +
  theme(panel.border = element_blank(),
        axis.text=element_text(size=18),
        axis.title=element_text(size=20)) 
  
```


Delta beta
```{r db}
range = max(covariates_sub_geno_tech_cc$no2_preg_entire) -
    min(covariates_sub_geno_tech_cc$no2_preg_entire)

cb_pvals_tech$db <- cb_pvals_tech$coeff_no2_preg_entire * range
```






Volcano plot
```{r volcano_db_tech}
# 10% effect size cutoff
ggplot(cb_pvals_tech, aes(x=as.numeric(db)*100 , y=-log10(as.numeric(pval_no2_preg_entire)))) +
  geom_point(alpha=0.4, col = "#75A2D9", size=2) +
  geom_point(data=subset(cb_pvals_tech, 
                         (pval_no2_preg_entire < 0.00001 & db > 0.1) | 
                          (pval_no2_preg_entire < 0.00001 & db < -0.1)), 
    alpha=0.4, col = "red", size=2) + 
  geom_vline(xintercept = c(10, -10), linetype = "dashed") +
  geom_hline(yintercept = -log10(0.00001), linetype="dashed") + 
  theme_bw() +
    theme(panel.border = element_blank(),
        axis.text=element_text(size=18),
        axis.title=element_text(size=20)) +
  labs(y="-log10(unadjusted p-value)", x = "Effect size")

# 5% effect size cutoff
ggplot(cb_pvals_tech, aes(x=as.numeric(db)*100 , y=-log10(as.numeric(pval_no2_preg_entire)))) +
  geom_point(alpha=0.4, col = "#75A2D9", size=2) +
  geom_point(data=subset(cb_pvals_tech, 
                         (pval_no2_preg_entire < 0.00001 & db > 0.05) | 
                          (pval_no2_preg_entire < 0.00001 & db < -0.05)), 
    alpha=0.4, col = "red", size=2) + 
  geom_vline(xintercept = c(5, -5), linetype = "dashed") +
  geom_hline(yintercept = -log10(0.00001), linetype="dashed") + 
  theme_bw() +
    theme(panel.border = element_blank(),
        axis.text=element_text(size=18),
        axis.title=element_text(size=20)) +
  labs(y="-log10(unadjusted p-value)", x = "Effect size")


# 2.5% effect size cutoff
ggplot(cb_pvals_tech, aes(x=as.numeric(db)*100 , y=-log10(as.numeric(pval_no2_preg_entire)))) +
  geom_point(alpha=0.4, col = "#75A2D9", size=2) +
  geom_point(data=subset(cb_pvals_tech, 
                         (pval_no2_preg_entire < 0.00001 & db > 0.025) | 
                          (pval_no2_preg_entire < 0.00001 & db < -0.025)), 
    alpha=0.4, col = "red", size=2) + 
  geom_vline(xintercept = c(2.5, -2.5), linetype = "dashed") +
  geom_hline(yintercept = -log10(0.00001), linetype="dashed") + 
  theme_bw() +
    theme(panel.border = element_blank(),
        axis.text=element_text(size=18),
        axis.title=element_text(size=20)) +
  labs(y="-log10(unadjusted p-value)", x = "Effect size")


```

Correlating old significant cpg (with 5 pcs) to cell type pcs
```{r pcscorr}
#load pc data
load("R:/Jones/People/Samantha Lee/Projects/CHILD/CHILD_air_pollution/output_data/deconvolution/2020-11-10_CBMC_PCs.RData")

#load cell counts
load("R:/Jones/People/Samantha Lee/Projects/CHILD/CHILD_air_pollution/output_data/deconvolution/2020-11-10_CBMC_PBMC_deconvolution.Rdata")

#convert cell counts to df
child_fscbc_ecc2_cb_df <- as.data.frame(child_fscbc_ecc2_cb)


#change row names to sample id from sentrix id for cell PCS
identical(rownames(cb_pcs), cb_pdata$sentrix_name) #true
rownames(cb_pcs) <- cb_pdata$Sample_Label

#change row names to sample id from sentrix id for cell counts
identical(rownames(child_fscbc_ecc2_cb_df), cb_pdata$sentrix_name) #false
#reorder cell counts
child_fscbc_ecc2_cb_df <- child_fscbc_ecc2_cb_df[cb_pdata$sentrix_name, ]
identical(rownames(child_fscbc_ecc2_cb_df), cb_pdata$sentrix_name) #ture
rownames(child_fscbc_ecc2_cb_df) <- cb_pdata$Sample_Label



#susbet for that in covariates
cb_pcs_sub <- cb_pcs[rownames(cb_pcs) %in% as.character(covariates_sub_geno_tech_cc$sampleid),]
dim(cb_pcs_sub)

child_fscbc_ecc2_cb_df_sub <- 
  child_fscbc_ecc2_cb_df[rownames(child_fscbc_ecc2_cb_df) %in% as.character(covariates_sub_geno_tech_cc$sampleid),]
dim(child_fscbc_ecc2_cb_df_sub)

#combined
covariates_sub_geno_tech_cc <- cbind(covariates_sub_geno_tech_cc, 
                                     cb_pcs_sub, 
                                     child_fscbc_ecc2_cb_df_sub)


##function for adding regression info to facets
lm_eqn = function(df){
  y=df[,1]
  x=df[,3]
  
  m = summary(lm(y~x))

  slope <- m$coefficients[2,1]
  intercept <- m$coefficients[1,1]
  p.value <- m$coefficients[2,4]
  R2 <- m$r.squared
  R2.Adj <- m$adj.r.squared
  c(slope,intercept,p.value, R2,R2.Adj)
  
}


######
#LMO4#
######

#add dnam to covariates 
covariates_sub_geno_tech_cc$cg20701183 <- 
  adj_cb_6pcs_betas_geno_cc[rownames(adj_cb_6pcs_betas_geno_cc)=="cg20701183", ] 



#melt to long format for graphing
lmo4_melt_PCs <- 
  reshape2::melt(covariates_sub_geno_tech_cc[ ,c("cg20701183", "PC1", "PC2",
                                                 "PC3", "PC4", "PC5", "PC6")],
                 id.vars="cg20701183")

#get linear regression equaions
lmo4_lm_pcs <- plyr::ddply(lmo4_melt_PCs, "variable" , lm_eqn)
#rename columns
colnames(lmo4_lm_pcs) <- c("supp","slope","intercept","P.value", "R2","R2.Adj")

#plot
ggplot(lmo4_melt_PCs, aes(x=value, y=cg20701183)) +
  geom_point() +
  geom_smooth(method='lm', se=F) +
  facet_wrap(vars(variable), scales="free") +
  ggtitle("LMO4 DNAm vs cell type PCs")

#icam2 cell counts
lmo4_melt_cellcounts <- 
  reshape2::melt(covariates_sub_geno_tech_cc[,c("cg20701183", 
                                                "counts.CD4T", 
                                                "counts.CD8T", 
                                                "counts.NK", 
                                                "counts.Bcell", 
                                                "counts.Mono", 
                                                "counts.nRBC")],
       id.vars="cg20701183")

#get linear regression equaions
lmo4_lm_counts <- plyr::ddply(lmo4_melt_cellcounts, "variable" , lm_eqn)
#rename columns
colnames(lmo4_lm_counts) <- c("supp","slope","intercept","P.value", "R2","R2.Adj")


ggplot(lmo4_melt_cellcounts, aes(x=value, y=cg20701183)) +
  geom_point() +
  geom_smooth(method='lm', se=F) +
  facet_wrap(vars(variable), scales="free") +
  ggtitle("LMO4 DNAm vs cell counts")



#######
#ICAM2#
#######

#add dnam to covariates
covariates_sub_geno_tech_cc$cg12076102 <- 
  adj_cb_6pcs_betas_geno_cc[rownames(adj_cb_6pcs_betas_geno_cc)=="cg12076102", ] 

#melt to long format for graphing
icam2_melt_PCs <- 
  reshape2::melt(covariates_sub_geno_tech_cc[ ,c("cg12076102", "PC1", "PC2",
                                                 "PC3", "PC4", "PC5", "PC6")],
                 id.vars="cg12076102")

#get linear regression equaions
icam2_lm_pcs <- plyr::ddply(icam2_melt_PCs, "variable" , lm_eqn)
#rename columns
colnames(icam2_lm_pcs) <- c("supp","slope","intercept","P.value", "R2","R2.Adj")

#plot
ggplot(icam2_melt_PCs, aes(x=value, y=cg12076102)) +
  geom_point() +
  geom_smooth(method='lm', se=F) +
  facet_wrap(vars(variable), scales="free") +
  ggtitle("ICAM2 DNAm vs cell type PCs")

#icam2 cell counts
icam2_melt_cellcounts <- 
  reshape2::melt(covariates_sub_geno_tech_cc[,c("cg12076102", 
                                                "counts.CD4T", 
                                                "counts.CD8T", 
                                                "counts.NK", 
                                                "counts.Bcell", 
                                                "counts.Mono", 
                                                "counts.nRBC")],
       id.vars="cg12076102")

#get linear regression equaions
icam2_lm_counts <- plyr::ddply(icam2_melt_cellcounts, "variable" , lm_eqn)
#rename columns
colnames(icam2_lm_counts) <- c("supp","slope","intercept","P.value", "R2","R2.Adj")


ggplot(icam2_melt_cellcounts, aes(x=value, y=cg12076102)) +
  geom_point() +
  geom_smooth(method='lm', se=F) +
  facet_wrap(vars(variable), scales="free") +
  ggtitle("ICAM2 DNAm vs cell counts")

#######
#TBX21#
#######

#add dnam to covariates
covariates_sub_geno_tech_cc$cg16928046 <- 
  adj_cb_6pcs_betas_geno_cc[rownames(adj_cb_6pcs_betas_geno_cc)=="cg16928046", ] 

#melt for plotting
tbx21_melt_PCs <- 
  reshape2::melt(covariates_sub_geno_tech_cc[,c("cg16928046", "PC1", "PC2", "PC3", "PC4", "PC5", "PC6")],
       id.vars="cg16928046")


#get linear regression equaions
tbx21_lm_PCs <- plyr::ddply(tbx21_melt_PCs, "variable" , lm_eqn)
#rename columns
colnames(tbx21_lm_PCs) <- c("supp","slope","intercept","P.value", "R2","R2.Adj")


#graphing
ggplot(tbx21_melt, aes(x=value, y=cg16928046)) +
  geom_point() +
  geom_smooth(method='lm', se=F) +
  facet_wrap(vars(variable), scales="free") +
  ggtitle("TBX21 DNAm vs cell PCs")


#tbx cell counts
tbx21_melt_cellcounts <- 
  reshape2::melt(covariates_sub_geno_tech_cc[,c("cg16928046", 
                                                "counts.CD4T", 
                                                "counts.CD8T", 
                                                "counts.NK", 
                                                "counts.Bcell", 
                                                "counts.Mono", 
                                                "counts.nRBC")],
       id.vars="cg16928046")


#get linear regression equaions
tbx21_lm_counts <- plyr::ddply(tbx21_melt_cellcounts, "variable" , lm_eqn)
#rename columns
colnames(tbx21_lm_counts) <- c("supp","slope","intercept","P.value", "R2","R2.Adj")




ggplot(tbx21_melt_cellcounts, aes(x=value, y=cg16928046)) +
  geom_point() +
  geom_smooth(method='lm', se=F) +
  facet_wrap(vars(variable), scales="free") +
  ggtitle("TBX21 DNAm vs cell counts")

```


Examining methylation at three nominally significant CpGs in LMO4, ICAM2, and TBX21.
```{r nom_sig_DNAm}

######
#LMO4#
######

covariates_sub_geno_tech_cc$cg20701183 <- 
  adj_cb_6pcs_betas_geno_cc[rownames(adj_cb_6pcs_betas_geno_cc)=="cg20701183", ] 

cg20701183_fit <- lm(cg20701183 ~ 1 + no2_preg_entire + 
                          sex + 
                          gestational_days +
                          prenatal_maternal_smoking + 
                          pss_18wk + 
                          maternal_education_length + 
                          genotyping.PC1 +
                          genotyping.PC2 +
                          genotyping.PC3 +
                          run + row, 
                        data=covariates_sub_geno_tech_cc)


#get partial residuals for no2 and dnam 
cg20701183_res <- partialResid(cg20701183 ~ no2_preg_entire, cg20701183_fit, 
              data=covariates_sub_geno_tech_cc)

#fit again to get r2
cg20701183_res_fit <- lm(cg20701183_res$yresid ~ cg20701183_res$xresid)

summary(cg20701183_res_fit)

ggplot(cg20701183_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, col = "#75A2D9", size=2) + 
  stat_smooth(method="lm", se=F, col="black") +
  theme_bw() +
  labs(x="Prenatal NO2 (ppb)", 
       y=str_wrap("LIM domain transription factor 4 (LMO4) cg20701183 DNA methyaltion (%)",width=40)) +
  ggtitle("Partial residual plot") +
  theme(panel.border=element_blank(),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=14)) 



#######
#ICAM2#
#######

covariates_sub_geno_tech_cc$cg12076102 <- 
  adj_cb_6pcs_betas_geno_cc[rownames(adj_cb_6pcs_betas_geno_cc)=="cg12076102", ] 

cg12076102_fit <- lm(cg12076102 ~ -1 + no2_preg_entire + 
                          sex + 
                          gestational_days +
                          prenatal_maternal_smoking + 
                          pss_18wk + 
                          maternal_education_length + 
                          genotyping.PC1 +
                          genotyping.PC2 +
                          genotyping.PC3 +
                          run + row, 
                        data=covariates_sub_geno_tech_cc)


#get partial residuals for no2 and dnam 
cg12076102_res <- partialResid(cg12076102 ~ no2_preg_entire, cg12076102_fit, 
              data=covariates_sub_geno_tech_cc)

#fit again to get r2
cg12076102_res_fit <- lm(cg12076102_res$yresid ~ cg12076102_res$xresid)

summary(cg12076102_res_fit)

ggplot(cg12076102_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, col = "#75A2D9", size=2) + 
  stat_smooth(method="lm", se=F, col="black") +
  theme_bw() +
  labs(x="Prenatal NO2 (ppb)", 
       y=str_wrap("Intercellular adhesion molecule 2 (ICAM2) cg12076102 DNA methyaltion (%)",width=47)) +
  ggtitle("Partial residual plot") +
  theme(panel.border=element_blank(),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=14)) 




#######
#TBX21#
#######

covariates_sub_geno_tech_cc$cg16928046 <- 
  adj_cb_6pcs_betas_geno_cc[rownames(adj_cb_6pcs_betas_geno_cc)=="cg16928046", ] 

cg16928046_fit <- lm(cg16928046 ~ no2_preg_entire + 
                          sex + 
                          gestational_days +
                          prenatal_maternal_smoking + 
                          pss_18wk + 
                          maternal_education_length + 
                          genotyping.PC1 +
                          genotyping.PC2 +
                          genotyping.PC3 +
                          run + row, 
                        data=covariates_sub_geno_tech_cc)


#get partial residuals for no2 and dnam 
cg16928046_res <- partialResid(cg16928046 ~ no2_preg_entire, cg16928046_fit, 
              data=covariates_sub_geno_tech_cc)

#fit again to get r2
cg16928046_res_fit <- lm(cg16928046_res$yresid ~ cg16928046_res$xresid)

summary(cg16928046_res_fit)

ggplot(cg16928046_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, col = "#75A2D9", size=2) + 
  stat_smooth(method="lm", se=F, col="black") +
  theme_bw() +
  labs(x="Prenatal NO2 (ppb)", 
       y=str_wrap("T-box transcription factor (TBX21) cg16928046 DNA methyaltion (%)",width=35)) +
  ggtitle("Partial residual plot") +
  theme(panel.border=element_blank(),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=14)) 
```



Try re-running limma with birth month/ interaction between birth month and air pollution.
```{r interaction_terms_to_assess_birthmonth}
#want to use anova to compare models usign anova
#anova only accepts models from lm
#use lm to compare how birth month affects cpgs above 10^-3 sig 

#You should be able to ask for confidence intervals for the slope coefficients for each month.  #even in a tabular display, you may see that “smoothly varying” pattern we were discussing.
#Then, you could use those to plot a sort of “here’s how the slope varies by month” picture.Sort #of like forest plots for meta analysis results, if that makes sense.

#Also, before we decide it doesn’t matter, the individual slopes might look fairly similar, but the appropriate overall test would be to check the anova(fit_with_interaction) and check the p-value on the interaction term.

test_cpgs <- subset(cb_pvals_tech, 
                    (pval_no2_preg_entire < 0.00001 & db > 0.05) | 
                      (pval_no2_preg_entire < 0.00001 & db < -0.05))

################################################
#LMO4

covariates_sub_geno_tech_cc$cg20701183 <- 
  adj_cb_6pcs_betas_geno_cc[rownames(adj_cb_6pcs_betas_geno_cc)=="cg20701183", ] 

cg20701183_fit_bm <- lm(cg20701183 ~ -1 + no2_preg_entire + 
                          sex + 
                          gestational_days +
                          prenatal_maternal_smoking + 
                          pss_18wk + 
                          maternal_education_length + 
                          genotyping.PC1 +
                          genotyping.PC2 +
                          genotyping.PC3 +
                          run + row + 
                          birth_month + 
                          no2_preg_entire*birth_month,
                        data=covariates_sub_geno_tech_cc)


tidy(cg20701183_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>%
  ggplot(aes(y = estimate, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low , ymax = conf.high)) +
  theme_bw() +
  ggtitle("LMO4 cg20701183") +
  scale_x_discrete(labels=c("NO2:February","NO2:March", "NO2:April", "NO2:May",
                            "NO2:June","NO2:July", "NO2:August", "NO2:September",
                            "NO2:October", "NO2:November","NO2:December")) +
  labs(x="Interaction term", y="Estimate") +
  theme(panel.border=element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=16)) 

anova(cg20701183_fit_bm)
  


################################################
#TBX21 cg16928046

covariates_sub_geno_tech_cc$cg16928046 <- 
  adj_cb_6pcs_betas_geno_cc[rownames(adj_cb_6pcs_betas_geno_cc)=="cg16928046", ] 


  
cg16928046_fit_bm <- lm(cg16928046 ~ -1 + no2_preg_entire + 
                          sex +
                          gestational_days +
                          maternal_education_length +
                          prenatal_maternal_smoking + 
                          pss_18wk + 
                          genotyping.PC1 +
                          genotyping.PC2 +
                          genotyping.PC3 +
                          run + row + 
                          birth_month +
                          no2_preg_entire*birth_month, 
                        data=covariates_sub_geno_tech_cc)



#most significant hypomethylated cpg cg12228123
tidy(cg16928046_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low , ymax = conf.high)) +
  scale_x_discrete(labels=c("NO2:February","NO2:March", "NO2:April", "NO2:May",
                            "NO2:June","NO2:July", "NO2:August", "NO2:September",
                            "NO2:October", "NO2:November","NO2:December")) +
  theme_bw() +
  theme(panel.border=element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=16)) +
  ggtitle("TBX21 cg16928046") +
    labs(x="Interaction term", y="Estimate") +
  ylim(-0.025, 0.025)

anova(cg16928046_fit_bm)

   
   
   
################################################
#ICAM2 cg12076102


covariates_sub_geno_tech_cc$cg12076102 <- 
  adj_cb_6pcs_betas_geno_cc[rownames(adj_cb_6pcs_betas_geno_cc)=="cg12076102", ] 


  
cg12076102_fit_bm <- lm(cg12076102 ~ -1 + no2_preg_entire + 
                          sex +
                          gestational_days +
                          maternal_education_length +
                          prenatal_maternal_smoking + 
                          pss_18wk + 
                          genotyping.PC1 +
                          genotyping.PC2 +
                          genotyping.PC3 +
                          run + row + 
                          birth_month +
                          no2_preg_entire*birth_month, 
                        data=covariates_sub_geno_tech_cc)



#most significant hypomethylated cpg cg12228123
tidy(cg12076102_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_preg_entire:birth_monthfebruary",
                                      "no2_preg_entire:birth_monthmarch",
                                      "no2_preg_entire:birth_monthapril",
                                      "no2_preg_entire:birth_monthmay",
                                      "no2_preg_entire:birth_monthjune",
                                      "no2_preg_entire:birth_monthjuly",
                                      "no2_preg_entire:birth_monthaugust",
                                      "no2_preg_entire:birth_monthseptember",
                                      "no2_preg_entire:birth_monthoctober",
                                      "no2_preg_entire:birth_monthnovember",
                                      "no2_preg_entire:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low , ymax = conf.high)) +
  scale_x_discrete(labels=c("NO2:February","NO2:March", "NO2:April", "NO2:May",
                            "NO2:June","NO2:July", "NO2:August", "NO2:September",
                            "NO2:October", "NO2:November","NO2:December")) +
  theme_bw() +
  theme(panel.border=element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=16)) +
  ggtitle("ICAM2 cg12076102") +
    labs(x="Interaction term", y="Estimate")

anova(cg12076102_fit_bm)

```




### Examine genome wide methylation changes vs no2 exposure


Examine mean methylation vs no2
```{r mean_meth}
#convert betas to df
adj_cb_6pcs_betas_geno_cc_df <- as.data.frame(adj_cb_6pcs_betas_geno_cc)

methmeans <- colMeans(adj_cb_6pcs_betas_geno_cc_df)

covariates_sub_geno_tech_cc <- cbind(covariates_sub_geno_tech_cc, methmeans)

#colmeans
meanmeth_lm <- lm(methmeans ~ no2_preg_entire +
                    sex + gestational_days +
                    prenatal_maternal_smoking + 
                    pss_18wk + 
                    maternal_education_length +
                    genotyping.PC1 +
                    genotyping.PC2 +
                    genotyping.PC3 +
                    run + row, 
                  data=covariates_sub_geno_tech_cc)

#get summary
summary(meanmeth_lm) #no2 not sig

#get partial residuals for no2 and dnam 
methmean_res <- partialResid( methmeans ~ no2_preg_entire, meanmeth_lm, 
              data=covariates_sub_geno_tech_cc)

#fit again to get r2
methmean_res_fit <- lm(methmean_res$yresid ~ methmean_res$xresid)

summary(methmean_res)

ggplot(methmean_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, col = "#75A2D9", size=2) + 
  stat_smooth(method="lm", se=F, col="black") +
  theme_bw() +
  labs(x="Prenatal NO2 (ppb)", y="Genome wide DNA methyaltion (%)") +
  ggtitle("Partial residual plot") +
  theme(panel.border=element_blank(),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=14)) 
```


### Examine LINE methylation vs NO2

Examine methylation of LINE1 elements
Based off of analysis by Andrea baccareli
https://link-springer-com.uml.idm.oclc.org/article/10.1186/s12989-014-0071-3
```{r line_annotation}

#load repeat element UCSC track
rmsk <- read.delim(file=here("raw_data", "repeat_element_anno", "rmsk.txt"), 
                   sep="\t",
                   header = F)

#populate colnames
colnames(rmsk) <- c("bin", "swscore", "milli_div", "milli_del",
                    "milli_ins", "genoname", "genostart", "genoend",
                    "genoleft", "strand", "repname", "repclass", "repfamily",
                    "repstart", "repend", "repleft", "id")

#get just LINE elements
rmsk_lines <- rmsk %>% filter(repclass=="LINE")

#add column to annotation indicating whether probe has at least 15 basepairs in LINE element

anno_lines <- pbapply(annotation, 1, function(x){
  
  probe_line <- rmsk_lines %>% 
  filter(genoname==x["chr"]) %>% 
  filter(genostart >= x["pos"] & genoend <= x["pos"])

  if(nrow(probe_line)!=0){
    return(TRUE)
  } else {return(FALSE)}
  
  
})

annotation$LINEs <- anno_lines

write.csv(annotation, file=here("output_data", "2020-11-29_annotation_LINEs.csv"))
```


```{r get_lineannotation}
#read in line annotation
annotationlines <- read.csv(here("output_data", "linear_regression", "2020-11-29_annotation_LINEs.csv"))

#subset 
annotationlines <- annotationlines[annotationlines$LINEs==T, ]

#subset betas
adj_cb_6pcs_betas_geno_cc_lines <- 
  adj_cb_6pcs_betas_geno_cc[rownames(adj_cb_6pcs_betas_geno_cc) %in%
                     annotationlines$X, ]

dim(adj_cb_6pcs_betas_geno_cc_lines)


#convert betas to df
adj_cb_6pcs_betas_geno_cc_lines_df <- as.data.frame(adj_cb_6pcs_betas_geno_cc_lines)

linemeth <- colMeans(adj_cb_6pcs_betas_geno_cc_lines)

covariates_sub_geno_tech_cc <- cbind(covariates_sub_geno_tech_cc, linemeth)

#colmeans
linemeth_lm <- lm(linemeth ~ no2_preg_entire + 
              sex + gestational_days +
              prenatal_maternal_smoking + 
              pss_18wk + 
              maternal_education_length + 
              genotyping.PC1 +
              genotyping.PC2 +
              genotyping.PC3 +
              run + row, 
              data=covariates_sub_geno_tech_cc)

summary(linemeth_lm) #no2 not sig




#get partial residuals for no2 and dnam 
linemeth_res <- partialResid( linemeth ~ no2_preg_entire, linemeth_lm, 
              data=covariates_sub_geno_tech_cc)

#fit again to get r2
linemeth_res_fit <- lm(linemeth_res$yresid ~ linemeth_res$xresid)

summary(linemeth_res_fit)

ggplot(linemeth_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, col = "#75A2D9", size=2) + 
  stat_smooth(method="lm", se=F, col="black") +
  theme_bw() +
  labs(x="Prenatal NO2 (ppb)", y="LINE DNA methyaltion (%)") +
  ggtitle("Partial residual plot") +
  theme(panel.border=element_blank(),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=14)) 

```


### Examine methylation changes from PMID 27448387
Dont run this since these changes are included in the list below
```{r validation, eval=FALSE}

covariates_ctpcs_geno_tech_cc$cg12283362 <- 
  cb_betas_geno_cc[rownames(cb_betas_geno_cc)=="cg12283362", ] 

lm_cg12283362 <- lm(cg12283362 ~ no2_preg_entire + 
              sex + gestational_days +
              maternal_education_length +
              prenatal_maternal_smoking + 
              pss_18wk + 
              csed_18wk + 
              cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
              cbmc.PC4 + cbmc.PC5 +
              genotyping.PC1 +
              genotyping.PC2 +
              genotyping.PC3 +
              run + row, data=covariates_ctpcs_geno_tech_cc)

summary(lm_cg12283362)

ggplot(covariates_ctpcs_geno_tech_cc, aes(x=no2_preg_entire, y=cg12283362*100)) +
  geom_point(alpha=0.8, col = "#75A2D9", size=2) +
  stat_smooth(method="lm", se=F, col="black") +
  #geom_abline(slope= 0.0005028644, intercept=0.5418461)+
  labs(y="Methylation %", x="Prenatal NO2 exposure (ppb)") +
  ggtitle("cg12283362") +
  theme_bw() +
  theme(panel.border=element_blank(),
        axis.text = element_text(size=14),
        axis.title = element_text(size=16),
        title = element_text(size=16)) 


##################
### cg24172570 ###
##################

covariates_ctpcs_geno_tech_cc$cg24172570 <- 
  cb_betas_geno_cc[rownames(cb_betas_geno_cc)=="cg24172570", ] 

lm_cg24172570 <- lm(cg24172570 ~ no2_preg_entire + 
              sex + gestational_days +
              maternal_education_length +
              prenatal_maternal_smoking + 
              pss_18wk + 
              csed_18wk + 
              cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
              cbmc.PC4 + cbmc.PC5 +
              genotyping.PC1 +
              genotyping.PC2 +
              genotyping.PC3 +
              run + row, data=covariates_ctpcs_geno_tech_cc)

summary(lm_cg24172570)

ggplot(covariates_ctpcs_geno_tech_cc, aes(x=no2_preg_entire, y=cg24172570*100)) +
  geom_point(alpha=0.8, col = "#75A2D9", size=2) +
  stat_smooth(method="lm", se=F, col="black") +
  labs(y="Methylation %", x="Prenatal NO2 exposure (ppb)") +
  ggtitle("cg24172570") +
  theme_bw() +
  theme(panel.border=element_blank(),
        axis.text = element_text(size=14),
        axis.title = element_text(size=16),
        title = element_text(size=16)) 

##################
### cg08973675 ###
##################

covariates_ctpcs_geno_tech_cc$cg08973675 <- 
  cb_betas_geno_cc[rownames(cb_betas_geno_cc)=="cg08973675", ] 

lm_cg08973675 <- lm(cg08973675 ~ no2_preg_entire + 
              sex + gestational_days +
              maternal_education_length +
              prenatal_maternal_smoking + 
              pss_18wk + 
              csed_18wk + 
              cbmc.PC1 + cbmc.PC2 + cbmc.PC3 + 
              cbmc.PC4 + cbmc.PC5 +
              genotyping.PC1 +
              genotyping.PC2 +
              genotyping.PC3 +
              run + row, data=covariates_ctpcs_geno_tech_cc)

summary(lm_cg08973675)


ggplot(covariates_ctpcs_geno_tech_cc, aes(x=no2_preg_entire, y=cg08973675*100)) +
  geom_point(alpha=0.8, col = "#75A2D9", size=2) +
  stat_smooth(method="lm", se=F, col="black") +
  labs(y="Methylation %", x="Prenatal NO2 exposure (ppb)") +
  ggtitle("cg08973675") +
  theme_bw() +
  theme(panel.border=element_blank(),
        axis.text = element_text(size=14),
        axis.title = element_text(size=16),
        title = element_text(size=16)) 

```



### Gene specific changes

These genes are related to the proposed pathway of air pollution action or have previously been shown to exhibit DNA methylation changes associated with prenatal air pollution exposure. 

```{r gene_specific}
#FHC is FTH1 (ferritin heavy chain 1)
#nothing maps to FOXP3
#TNFA is TNF
goi <- annotation[grep("RNF39|CYP2E1|LONP1|HIBADH|SLC25A28|PLVAP|GPR55|^CAT$|TPO|NFKB1|^NOX2|SOD1|TRX1|TRX2|GSTP1|^MT3|NQO1|HMOX1|GPX1|DDH1|^NOS2|CYP2C11|CYP7B|ALOX12$|ALOX5|^NRF2$|KEAP1|AHR$|^ARNT;|GCLC|GCLM|GSR1|SLC7A11|GPC2|GSTA1|GSTA2|GSTA3|GSTA5|GSTM1|GSTM2|GSTM3|TXN$|TXNRD1$|SRXN1|G6PD|CYP1A1|IL1A|^IL2$|^IL3$|^IL4.*IL4$|^IL5$|^IL6$|IL9$|IL10$|IL12A|IL13|IL15$|IL17A|IL18$|IL19|IL21$|IL22$|IL23A$|IL25$|IL27$|IL31$|IL33$|IFNG$|^TNF$|TGFB1$|FOXP3|FTH1", annotation$UCSC_RefGene_Name), ]

#transpose betas
adj_cb_6pcs_betas_geno_cc_t <- t(adj_cb_6pcs_betas_geno_cc)

#apply function for calculating significance of associations using mice and imputed covariate data
prenatal_goi_list <- 
  pbapply(adj_cb_6pcs_betas_geno_cc_t[,colnames(adj_cb_6pcs_betas_geno_cc_t) %in% rownames(goi)], 2, function(x) {
  
  combined <- cbind(covariates_sub_geno_tech_cc, x)
  
  #run the linear regression on this cpg using all 5 imputed covariate data sets
  fit <- lm(x ~ no2_preg_entire + 
              sex + 
              gestational_days + 
              maternal_education_length +
              prenatal_maternal_smoking + 
              pss_18wk + 
              genotyping.PC1 +
              genotyping.PC2 +
              genotyping.PC3 +
              run + row, data=combined)
  
  #get summary of lm fit
  #this includes coefficients and pvalues
  #coerce into df for manipulation
  #add in terms (rownames) as column for annotation
  fit_sum <- summary(fit)
  fit_sum_coeff <- as.data.frame(fit_sum$coefficients)
  fit_sum_coeff$term <- rownames(fit_sum_coeff)
  
  #this data frame has rows for each linear regression variable, and columns for statistics
  #want a dataframe of 1 row and many columns (one for each variable with each statistic type)
  #do this by using the spread function for each statistic
  #add cpg id onto dataframe for pvalues (to keep track of information)
  #then cbind dataframes together
  
  #coefficient
  estimate <- spread(fit_sum_coeff[,c("term", "Estimate")], term, Estimate)
  colnames(estimate) <- paste("estimate", colnames(estimate), sep=".")
  
  #std error
  std.error <- spread(fit_sum_coeff[,c("term", "Std. Error")], term, 'Std. Error')
  colnames(std.error) <- paste("std.error", colnames(std.error), sep=".")
  
  #t statistics
  statistic <- spread(fit_sum_coeff[,c("term", "t value")], term, 't value')
  colnames(statistic) <- paste("t.val", colnames(statistic), sep=".")

  #pvalue
  p.value <- spread(fit_sum_coeff[,c("term", "Pr(>|t|)")], term, 'Pr(>|t|)')
  colnames(p.value) <- paste("p.value", colnames(p.value), sep=".")
  
  #add in cpgid for annotation 
  p.value$cpg <- colnames(x)
  
  #cbind into wide dataframe
  widepool <- cbind(estimate, std.error, statistic, p.value)
  
  #return the wide dataframe
  #these will be do.call(rbind()) together after
  return(widepool)
  
})


prenatal_goi <- do.call(rbind, prenatal_goi_list)

prenatal_goi$p.value.no2_preg_entire_adjusted <- p.adjust(prenatal_goi$p.value.no2_preg_entire, method="bonferroni")

#reorder according ot unadjusted no2 pvalue
prenatal_goi_order <- prenatal_goi[order(prenatal_goi$p.value.no2_preg_entire),]

#reorder goi annotation to be in same order as goi stats
goi_order <- goi[rownames(prenatal_goi_order),]

#check identical
identical(rownames(goi_order), rownames(prenatal_goi_order)) #true

#bind
goi_df <- cbind(prenatal_goi_order, goi_order)

#write as .csv
write.csv(goi_df, file=here("output_data", "2020-12-21_prenatal_goi.csv"))




covariates_sub_geno_tech_cc$cg26227465_cb <- 
        adj_cb_6pcs_betas_geno_cc[rownames(adj_cb_6pcs_betas_geno_cc)=="cg26227465", ]

#colmeans
cg26227465_lm <- lm(cg26227465_cb ~ no2_preg_entire + 
              sex + gestational_days +
              prenatal_maternal_smoking + 
              pss_18wk + 
              maternal_education_length + 
              genotyping.PC1 +
              genotyping.PC2 +
              genotyping.PC3 +
              run + row, 
              data=covariates_sub_geno_tech_cc)

  


#get partial residuals for no2 and dnam 
cg26227465_res <- partialResid(cg26227465_cb ~ no2_preg_entire, cg26227465_lm, 
              data=covariates_sub_geno_tech_cc)

#fit again to get r2
cg26227465_res_fit <- lm(cg26227465_res$yresid ~ cg26227465_res$xresid)

summary(cg26227465_res_fit)

ggplot(cg26227465_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, col = "#75A2D9", size=2) + 
  stat_smooth(method="lm", se=F, col="black") +
  theme_bw() +
  labs(x="Prenatal NO2 (ppb)", 
       y=str_wrap("Interferon gamma (INFG) cg26227465 DNA methyaltion (%)",width=35)) +
  ggtitle("Partial residual plot") +
  theme(panel.border=element_blank(),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=14)) 





### cpg from gruzieva
covariates_sub_geno_tech_cc$cg12283362 <- 
  adj_cb_6pcs_betas_geno_cc[rownames(adj_cb_6pcs_betas_geno_cc)=="cg12283362", ] 


cg12283362_lm <- lm(cg12283362  ~ no2_preg_entire + 
              sex + gestational_days +
              prenatal_maternal_smoking + 
              pss_18wk + 
              maternal_education_length +
              genotyping.PC1 +
              genotyping.PC2 +
              genotyping.PC3 +
              run + row,  
              data=covariates_sub_geno_tech_cc)


#get partial residuals for no2 and dnam 
cg12283362_res <- partialResid(cg12283362 ~ no2_preg_entire, cg12283362_lm, 
              data=covariates_sub_geno_tech_cc)

#fit again to get r2
cg12283362_res_fit <- lm(cg12283362_res$yresid ~ cg12283362_res$xresid)

summary(cg12283362_res_fit)

ggplot(cg12283362_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, col = "#75A2D9", size=2) + 
  stat_smooth(method="lm", se=F, col="black") +
  theme_bw() +
  labs(x="Prenatal NO2 (ppb)", 
       y=str_wrap("Lon protease homolog (LONP1) cg12283362 DNA methyaltion (%)",width=35)) +
  ggtitle("Partial residual plot") +
  theme(panel.border=element_blank(),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=14)) 





#3rd cpg from gruzieve
covariates_sub_geno_tech_cc$cg08973675 <- 
  adj_cb_6pcs_betas_geno_cc[rownames(adj_cb_6pcs_betas_geno_cc)=="cg08973675", ] 



cg08973675_lm <- lm(cg08973675  ~ no2_preg_entire + 
              sex + gestational_days +
              prenatal_maternal_smoking + 
              pss_18wk + 
              csed_18wk + 
              maternal_education_length +
              run + row, 
              data=covariates_sub_geno_tech_cc)


#get partial residuals for no2 and dnam 
cg08973675_res <- partialResid(cg08973675 ~ no2_preg_entire, cg08973675_lm, 
              data=covariates_sub_geno_tech_cc)

#fit again to get r2
cg08973675_res_fit <- lm(cg08973675_res$yresid ~ cg08973675_res$xresid)

summary(cg08973675_res_fit)

ggplot(cg08973675_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, col = "#75A2D9", size=2) + 
  stat_smooth(method="lm", se=F, col="black") +
  theme_bw() +
  labs(x="Prenatal NO2 (ppb)", 
       y=str_wrap("Solute Carrier Family 25 Member 28 (SLC25A28) cg08973675 DNA methyaltion (%)",width=35)) +
  ggtitle("Partial residual plot") +
  theme(panel.border=element_blank(),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=14)) 


```




Looking at interferon gamma at age one.
```{r ifng_ageone}
#load cell counts
load(here("output_data", "deconvolution", "2020-12-10_PBMC_PCs.RData"))

#these children have data at both time points
y1_pdata <- subset(pdata_nodups, Tissue == "P")
dim(y1_pdata)

#subset cordblood DNAM samples
y1_betas <- betas[ ,colnames(betas) %in% rownames(y1_pdata)]

#rename colums with sample label (5 dig num)
identical(colnames(y1_betas), rownames(y1_pdata))#true
colnames(y1_betas) <- y1_pdata$Sample_Label

#subset y1 betas for samples in the complete case cord blood betas
y1_betas_cc <- y1_betas[,colnames(y1_betas) %in% colnames(cb_betas_geno_cc)]
dim(y1_betas_cc) #424644    101

#convert cell types to dataframe
#rename cols
colnames(y1_pcs) <- paste("pbmc", colnames(y1_pcs), sep=".")

#rename rwos of cell types
identical(rownames(y1_pcs), rownames(y1_pdata)) #true
rownames(y1_pcs) <- y1_pdata$Sample_Label

#subset for complete cases
y1_pcs_sub <- y1_pcs[rownames(y1_pcs) %in% covariates_ctpcs_geno_tech_cc$sampleid,]
dim(y1_pcs_sub) #101

#add to covariate df
covariates_ctpcs_geno_tech_cc <- cbind(covariates_ctpcs_geno_tech_cc, y1_pcs_sub)

covariates_ctpcs_geno_tech_cc$y1_cg26227465 <- 
        y1_betas_cc[rownames(y1_betas_cc)=="cg26227465", ]



sum(complete.cases(covariates_ctpcs_geno_tech_cc %>% dplyr::select(no2_preg_entire, 
                                                                sex,  
                                                                  maternal_smoking_y1,
                                                                  pss_6mos,
                                                                  csed_6mos, 
                                                                  pbmc.PC1, pbmc.PC2,
                                                                  pbmc.PC3, pbmc.PC4, 
                                                                  genotyping.PC1,
                                                                  genotyping.PC2,
                                                                  genotyping.PC3,
                                                                  run, row))) #97


#subset covariates
covariates_ctpcs_geno_tech_cc_y1 <- 
  covariates_ctpcs_geno_tech_cc[complete.cases(covariates_ctpcs_geno_tech_cc %>% 
                                              dplyr::select(no2_preg_entire, 
                                                                sex,  
                                                                  maternal_smoking_y1,
                                                                  pss_6mos,
                                                                  csed_6mos, 
                                                                  pbmc.PC1, pbmc.PC2,
                                                                  pbmc.PC3, pbmc.PC4, 
                                                                  genotyping.PC1,
                                                                  genotyping.PC2,
                                                                  genotyping.PC3,
                                                                  run, row)), ] 
#check size


#colmeans
yearone_cg26227465_lm <- lm(y1_cg26227465 ~ no2_preg_entire + 
              sex + 
              maternal_smoking_y1 + 
              pss_6mos + 
              csed_6mos + 
              pbmc.PC1 + pbmc.PC2 +
              pbmc.PC3 + pbmc.PC4 + 
              genotyping.PC1 +
              genotyping.PC2 +
              genotyping.PC3 +
              run + row, 
              data=covariates_ctpcs_geno_tech_cc_y1)


#get partial residuals for no2 and dnam 
yearone_cg26227465_res <- partialResid(y1_cg26227465 ~ no2_preg_entire, yearone_cg26227465_lm, 
              data=covariates_ctpcs_geno_tech_cc)

#fit again to get r2
yearone_cg26227465_res_fit <- lm(yearone_cg26227465_res$yresid ~ yearone_cg26227465_res$xresid)

summary(yearone_cg26227465_res_fit)

ggplot(yearone_cg26227465_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, col = "#75A2D9", size=2) + 
  stat_smooth(method="lm", se=F, col="black") +
  theme_bw() +
  labs(x="Prenatal NO2 (ppb)", 
       y=str_wrap("Interferon gamma (IFNG) cg26227465 DNA methyaltion at age one(%)",width=35)) +
  ggtitle("Partial residual plot") +
  theme(panel.border=element_blank(),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=14)) 





#colmeans
yearone_cg26227465_y1_lm <- lm(yearone_cg26227465 ~ no2_y1_entire + 
              sex + 
              maternal_smoking_y1 + 
              pss_6mos + 
              csed_6mos + 
              pbmc.PC1 + pbmc.PC2 +
              pbmc.PC3 + pbmc.PC4 + 
              genotyping.PC1 +
              genotyping.PC2 +
              genotyping.PC3 +
              run + row, 
              data=covariates_ctpcs_geno_tech_cc_y1)


#get partial residuals for no2 and dnam 
yearone_cg26227465_y1_res <- partialResid(yearone_cg26227465 ~ no2_y1_entire, yearone_cg26227465_y1_lm, 
              data=covariates_ctpcs_geno_tech_cc)

#fit again to get r2
yearone_cg26227465_y1_res_fit <- lm(yearone_cg26227465_y1_res$yresid ~ yearone_cg26227465_y1_res$xresid)

summary(yearone_cg26227465_y1_res_fit)

ggplot(yearone_cg26227465_y1_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, col = "#75A2D9", size=2) + 
  stat_smooth(method="lm", se=F, col="black") +
  theme_bw() +
  labs(x="Prenatal NO2 (ppb)", 
       y=str_wrap("Interferon gamma (IFNG) cg26227465 DNA methyaltion at age one(%)",width=35)) +
  ggtitle("Partial residual plot") +
  theme(panel.border=element_blank(),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=14)) 



#boxplot
ggplot(covariates_ctpcs_geno_tech_cc_y1[cg])

```





















Investigate interferon lambda (IL29) for meaghan
```{r ifn_lamda}
ifnl1_cpgs<- annotation[grep("IL29", annotation$UCSC_RefGene_Name), ]

#transpose betas
adj_cb_6pcs_betas_geno_cc_t <- t(adj_cb_6pcs_betas_geno_cc)


#apply function for calculating significance of associations using mice and imputed covariate data
ifnl1_list <- 
  pbapply(adj_cb_6pcs_betas_geno_cc_t [,colnames(adj_cb_6pcs_betas_geno_cc_t ) %in% rownames(ifnl1_cpgs)], 2, function(x) {
  
  combined <- cbind(covariates_sub_geno_tech_cc, x)
  
  #run the linear regression on this cpg using all 5 imputed covariate data sets
  fit <- lm(x ~no2_preg_entire + 
              sex + 
              maternal_education_length +
              prenatal_maternal_smoking + 
              pss_18wk + 
              maternal_education_length + 
              genotyping.PC1 +
              genotyping.PC2 +
              genotyping.PC3 +
              run + row, data=combined)
  
  #get summary of lm fit
  #this includes coefficients and pvalues
  #coerce into df for manipulation
  #add in terms (rownames) as column for annotation
  fit_sum <- summary(fit)
  fit_sum_coeff <- as.data.frame(fit_sum$coefficients)
  fit_sum_coeff$term <- rownames(fit_sum_coeff)
  
  #this data frame has rows for each linear regression variable, and columns for statistics
  #want a dataframe of 1 row and many columns (one for each variable with each statistic type)
  #do this by using the spread function for each statistic
  #add cpg id onto dataframe for pvalues (to keep track of information)
  #then cbind dataframes together
  
  #coefficient
  estimate <- spread(fit_sum_coeff[,c("term", "Estimate")], term, Estimate)
  colnames(estimate) <- paste("estimate", colnames(estimate), sep=".")
  
  #std error
  std.error <- spread(fit_sum_coeff[,c("term", "Std. Error")], term, 'Std. Error')
  colnames(std.error) <- paste("std.error", colnames(std.error), sep=".")
  
  #t statistics
  statistic <- spread(fit_sum_coeff[,c("term", "t value")], term, 't value')
  colnames(statistic) <- paste("t.val", colnames(statistic), sep=".")

  #pvalue
  p.value <- spread(fit_sum_coeff[,c("term", "Pr(>|t|)")], term, 'Pr(>|t|)')
  colnames(p.value) <- paste("p.value", colnames(p.value), sep=".")
  
  #add in cpgid for annotation 
  p.value$cpg <- colnames(x)
  
  #cbind into wide dataframe
  widepool <- cbind(estimate, std.error, statistic, p.value)
  
  #return the wide dataframe
  #these will be do.call(rbind()) together after
  return(widepool)
  
})


ifnl1_meth <- do.call(rbind, ifnl1_list)

ifnl1_meth$p.value.no2_preg_entire_adjusted <- p.adjust(ifnl1_meth$p.value.no2_preg_entire, method="bonferroni")

#reorder according ot unadjusted no2 pvalue
ifnl1_meth_order <- ifnl1_meth[order(ifnl1_meth$p.value.no2_preg_entire),]

#reorder goi annotation to be in same order as goi stats
ifnl1_cpgs_order <- ifnl1_cpgs[rownames(ifnl1_meth_order),]

#check identical
identical(rownames(ifnl1_cpgs_order), rownames(ifnl1_meth_order)) #true

#bind
ifnl1_df <- cbind(ifnl1_meth_order, ifnl1_cpgs_order)




covariates_sub_geno_tech_cc$cg07424400 <- 
  adj_cb_6pcs_betas_geno_cc[rownames(adj_cb_6pcs_betas_geno_cc)=="cg07424400", ] 


cg07424400_lm <- lm(cg07424400  ~ no2_preg_entire + 
              sex + gestational_days +
              prenatal_maternal_smoking + 
              pss_18wk + 
              maternal_education_length +
              genotyping.PC1 +
              genotyping.PC2 +
              genotyping.PC3 +
              run + row,  
              data=covariates_sub_geno_tech_cc)


#get partial residuals for no2 and dnam 
cg07424400_res <- partialResid(cg07424400 ~ no2_preg_entire, cg07424400_lm, 
              data=covariates_sub_geno_tech_cc)

#fit again to get r2
cg07424400_res_fit <- lm(cg07424400_res$yresid ~ cg07424400_res$xresid)

summary(cg07424400_res_fit)

ggplot(cg07424400_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, col = "#75A2D9", size=2) + 
  stat_smooth(method="lm", se=F, col="black") +
  theme_bw() +
  labs(x="Prenatal NO2 (ppb)", 
       y=str_wrap("Interferon lambda-1 (IL29) cg07424400 DNA methyaltion (%)",width=35)) +
  ggtitle("Partial residual plot") +
  theme(panel.border=element_blank(),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=14)) 
```