---
title: "Year one postnatal regression using limma with complete cases"
author: "SL"
date: "March 04, 2021"
output: html_document
---

## Analysis of postnatal specific DNAm changes associated with year one NO2 exposure

*Purpose*: The purpose of this script is to DNAm changes in year one PBMCs associated with year one air pollution exposure (proxied by no2). This script uses a subtraction method to "cancel" out effects of prenatal exposure, allowing the investigation of year one specific changes. 

Set global chunk options. 
```{r setup, include=FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Load required libraries.
```{r libraries}
library(tidyverse) # be tidy
library(limma) # for DNAm differences
library(here) # file calling
library(pbapply) # for running lm for each cpg
library(ggeffects) # for estimating marginal effects
library(broom) # tidy function for summarizing lm
library(sva) # for calculating SVs
library(ggcorrplot) # correlation matrix
library(ggsignif) # for significance comparison bars
library(table1)
library(GenomicRanges)
library(IRanges)
library(ENmix)
```

### Data import and cleaning

Load betas data
```{r covariates}
load(file=here("output_data", 
               "preprocessed_data",
               "2022-09-27_CHILD_preprocessed_betas_pdata_annotation.Rdata"))
```


Load covariate data
```{r covariates}
# load covariate file
load(file=here("output_data", 
               "covariates", 
               "2022-09-27_CHILD_covariates_na_with_means.Rdata"))

#######################################################
# subset out cord blood and year one pdata separately #
#######################################################

# cord blood pdata
cb_pdata <- subset(pdata_nodups, Tissue == "C")
# check size of cord blood pdata
dim(cb_pdata) # 144  17

# year one pdata
y1_pdata <- subset(pdata_nodups, Tissue == "P")
# check size of year one pdata
dim(y1_pdata) # 145  17

# year one data has one individual that is missing birth data
# remove this individual
y1_pdata_sub <- y1_pdata[y1_pdata$Sample_Label %in% cb_pdata$Sample_Label, ]
# recheck size of year one pdata
dim(y1_pdata_sub) # 144  17
```


Subset out cord blood and age one betas
```{r sub_betas}
####################
# cord blood betas #
####################

# subset cord blood betas based on pdata
# all children with cord blood betas also have year one DNAm data
cb_betas <- betas[,colnames(betas) %in% rownames(cb_pdata)]

#check size of cord blood data
dim(cb_betas) # 424644    144
min(cb_betas) # 0.002973941
max(cb_betas) # 0.9980669

# check if cord blood DNAm samples are in same order in cord blood pdata
identical(colnames(cb_betas), rownames(cb_pdata)) # TRUE

# rename columns of cb_betas to sample label (5 digit identifier)
colnames(cb_betas) <- cb_pdata$Sample_Label

##################
# year one betas #
##################

# subset out year one DNAm samples from beta matrix
y1_betas <- betas[,colnames(betas) %in% rownames(y1_pdata_sub)]

# check size of year one data
dim(y1_betas) # 424644    144
min(y1_betas) # 0.002650193
max(y1_betas) # 0.9983506

# check if year one DNAm samples are in same order in year one pdata
identical(colnames(y1_betas), rownames(y1_pdata_sub)) # TRUE

# rename columns of cb_betas to sample label (5 digit identifier)
colnames(y1_betas) <- y1_pdata_sub$Sample_Label

# check if cord blood samples and year one samples are in same order
identical(colnames(cb_betas), colnames(y1_betas)) # TRUE
```


Subtract cord blood DNAm data from year one DNAm data to isolate postnatal DNAm changes. 
```{r diff_betas}
# call postnatal DNAm data "diff betas"
diff_betas <- y1_betas - cb_betas

# check size of difference betas
dim(diff_betas) # 424644    144
min(diff_betas) # -0.9780622 
max(diff_betas) # 0.9790525
```


Load combined cell type PCs (CBMC and PBMC together).
```{r load_celltype_PCs}
# load cell type PCs for diff betas
load(here("output_data", 
          "deconvolution", 
          "2022-09-28_CHILD_CBMC_PBMC_PCs.Rdata"))

# pull out pcs
ccc_pcs <- comb_ilr$scores

# check size
dim(ccc_pcs) # 144 10

# no rownames, populate wiht pdata
rownames(ccc_pcs) <- cb_pdata$Sample_Label
```


Load genotyping PCs. There are PCs for all CHILD participants that were genotyped. Therefore we have to subset the REEGLE participants used in this study out.
```{r genotypingPCS}
# load genotyping data
genotyping <- read.csv(here("input_data", 
                            "genotyping", 
                            "CHILD_subjects_Qced_first10PCS_Duan.csv"))

# make rownames the sample label (5 digit identifier)
rownames(genotyping) <- genotyping$FID

# remove columns that contain naming info since this is now in rownames
genotyping <- subset(genotyping, select=-c(FID, IID))

# subset REEGLE genotyping PCs
genosub <- genotyping[rownames(genotyping) %in% cb_pdata$Sample_Label,]

# check size
dim(genosub) # 131 10

# rename columns of genotyping data for clarity
colnames(genosub) <- paste("genotyping.", colnames(genosub), sep="")

# put rows in numerical order according to sample id
genosub_order <- genosub[order(rownames(genosub)), ]
```


Subset covariates, betas, pdata, and cell type PCs for samples that also have genotyping.
```{r sub_on_geno}
############
# cb pdata #
############

# subset cb_pdata
cb_pdata_geno <- cb_pdata[as.character(cb_pdata$Sample_Label) %in%
                            rownames(genosub_order), ]
# check size
dim(cb_pdata_geno) # 131

# check order
identical(as.character(cb_pdata_geno$Sample_Label), 
          rownames(genosub_order)) # TRUE

############
# y1 pdata #
############

# subset y1_pdata
y1_pdata_geno <- y1_pdata_sub[as.character(y1_pdata_sub$Sample_Label) %in%
                                rownames(genosub_order), ]
# check size
dim(y1_pdata_geno) # 131 14

# check order
identical(as.character(y1_pdata_geno$Sample_Label), 
          rownames(genosub_order)) # TRUE

##############
# covariates #
##############

# subset covariates
covariates_geno <- 
  covariates_nameans[as.character(covariates_nameans$sample_id) %in%
                       rownames(genosub_order), ]

# check size
dim(covariates_geno) #131

# check order
identical(as.character(covariates_geno$sample_id), 
          rownames(genosub_order)) # TRUE

##############
# diff betas #
##############

# subset betas
diff_betas_geno <- diff_betas[,colnames(diff_betas) %in% 
                                rownames(genosub_order)]

# check size
dim(diff_betas_geno) # 131

# check order
identical(colnames(diff_betas_geno), rownames(genosub_order)) # TRUE

#################
# cell type PCs #
#################

# subset celltype pcs
ccc_pcs_geno <- ccc_pcs[rownames(ccc_pcs) %in% rownames(genosub_order), ]

# check size
dim(ccc_pcs_geno) # 131 10

# check order
identical(rownames(ccc_pcs_geno), rownames(genosub_order)) # TRUE
```


Combine covariates with genotyping data, cell type PCs, and batch variables from pdata. 
```{r combine_data}

################################
# clean pdata before combining #
################################

# refactor and label new columns for cord blood batch vars
cb_pdata_geno$cb_chip <- as.factor(cb_pdata_geno$Sentrix_ID)
cb_pdata_geno$cb_run <- as.factor(cb_pdata_geno$Run)
cb_pdata_geno$cb_row <-
  as.numeric(as.factor(substr(cb_pdata_geno$Sentrix_Position, 1, 3)))

# refactor and label  new columns for year one batch vars
y1_pdata_geno$y1_chip <- as.factor(y1_pdata_geno$Sentrix_ID)
y1_pdata_geno$y1_run <- as.factor(y1_pdata_geno$Run)
y1_pdata_geno$y1_row <-
  as.numeric(as.factor(substr(y1_pdata_geno$Sentrix_Position, 1, 3)))

#######################
# combine data frames #
#######################

covariates_geno_ctpcs_tech <- cbind(covariates_geno,
                                    ccc_pcs_geno,
                                    genosub_order,
                                    cb_pdata_geno[,c("cb_run", 
                                                     "cb_row",
                                                     "cb_chip")],
                                    y1_pdata_geno[,c("y1_run", 
                                                     "y1_row",
                                                     "y1_chip")])

# check classes
lapply(covariates_geno_ctpcs_tech, class)

# reclass sex
covariates_geno_ctpcs_tech$sex <- as.factor(covariates_geno_ctpcs_tech$sex)
``` 

### Complete cases

Subset covariates and betas for complete cases 
```{r complete_cases}
##############
# covariates #
##############

# covariate complete cases
covariates_geno_ctpcs_tech_cc <- 
  covariates_geno_ctpcs_tech[covariates_geno_ctpcs_tech %>% 
                               dplyr::select(no2_y1_2019v, sex,  
                                             maternal_smoking_y1,  
                                             maternal_education_length, 
                                             genotyping.PC1, genotyping.PC2,
                                             genotyping.PC3, Comp.1, Comp.2,
                                             Comp.3, Comp.4, Comp.5, Comp.6) %>% 
                               complete.cases,] 

# check size of complete case covariates
dim(covariates_geno_ctpcs_tech_cc) # 125 145
```


Remove rows where air pollution is 0.
```{r no_zeroes_in_no2}
sum(na.omit(covariates_geno_ctpcs_tech_cc$no2_y1_2019v==0))
```


```{r subset betas}

#########
# betas #
#########

#subset betas
diff_betas_geno_cc <- 
  diff_betas_geno[,colnames(diff_betas_geno) %in%
                    covariates_geno_ctpcs_tech_cc$sample_id]
# check size
dim(diff_betas_geno_cc) # 424644    125

# load lists of probes used in cell type prediction
load(here("output_data", 
          "deconvolution",
          "2022-09-28_cbmc_pbmc_deconvolution.Rdata"))

# remove betas that were used for cell type prediction
diff_betas_geno_cc_cpr <- 
  diff_betas_geno_cc[!rownames(diff_betas_geno_cc) %in%
                       unname(unlist(child_fscbc_ecc2_cb$probelist_all)) &
                       !rownames(diff_betas_geno_cc) %in%
                       unname(unlist(child_fscbc_ecc2_y1$probelist_all)), ]

# check size
dim(diff_betas_geno_cc_cpr) # 424079 125
```

### Model matrix

Create model matrix for investigating postnatal DNAm changes. 
```{r model_matrix}
# create model matrix
mod <- model.matrix(~ no2_y1_2019v + sex + maternal_smoking_y1 + 
                      maternal_education_length + genotyping.PC1 +
                      genotyping.PC2 + genotyping.PC3 + Comp.1 + Comp.2 + 
                      Comp.3 + Comp.4 + Comp.5 + Comp.6, 
                    data=covariates_geno_ctpcs_tech_cc)

# check size
dim(mod) # 125 14
```

### Surrogate variable analysis

Surrogate variable analysis using the sva packages. This will allow us to capture variance due to batch effects, as well as variance due to unkown biological variables. 
```{r sva}
#################
# calculate svs #
#################

# create null matrix
# this matrix does not include any of our covariates 
# this means that our covariates are "protected" and SVs will not capture them
mod0 <- model.matrix(~ sex + maternal_smoking_y1 + maternal_education_length + 
                       genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                       Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6, 
                     data=covariates_geno_ctpcs_tech_cc)

# to get SVs in a way that allows calculation of their variance
# we need to pull the code out of sva and run it line-by-line
# we are using the "Be" method for calculating SVs

######################################
# irwsva required internal functions #
######################################

# edge.lfdr function is internal to irwsva function below
# mono function is internal to edge.lfdr
# they have to be initialized before we can run the rest of the code

mono <- function(lfdr){
  .Call("monotone", as.numeric(lfdr), PACKAGE="sva")
}

edge.lfdr <- function(p, trunc=TRUE, monotone=TRUE, transf=c("probit", "logit"), adj=1.5, eps=10^-8, lambda=0.8, ...) {
  pi0 <- mean(p >= lambda)/(1 - lambda)
  pi0 <- min(pi0, 1)
  n <- length(p)
  transf <- match.arg(transf)
  if(transf=="probit") {
    p <- pmax(p, eps)
    p <- pmin(p, 1-eps)
    x <- qnorm(p)
    myd <- density(x, adjust=adj)
    mys <- smooth.spline(x=myd$x, y=myd$y)
    y <- predict(mys, x)$y
    lfdr <- pi0*dnorm(x)/y
  }
  if(transf=="logit") {
    x <- log((p+eps)/(1-p+eps))
    myd <- density(x, adjust=adj)
    mys <- smooth.spline(x=myd$x, y=myd$y)
    y <- predict(mys, x)$y
    dx <- exp(x) / (1+exp(x))^2
    lfdr <- pi0 * dx/y
  }
  if(trunc) {
    lfdr[lfdr > 1] <- 1
  }
  if(monotone) {	
    lfdr <- lfdr[order(p)]
    lfdr <- mono(lfdr)
    lfdr <- lfdr[rank(p)]
  }
  return(lfdr)
}



##################################
# numsv function internal to sva #
##################################

#determine the number of required SVs
#this is numsv function code pulled out

#this is what is fed into this function from the sva function
dat = diff_betas_geno_cc_cpr
mod = mod
#method="be" # this is the default and what is called by numsv
#vfilter=NULL # we dont apply a filter and its not used for be method
B=20
set.seed(1234)

# initialize some stuff
# get the residual matrix (the remaining variance)
n <- ncol(dat)
m <- nrow(dat)
H <- mod %*% solve(t(mod) %*% mod) %*% t(mod)
res <- dat - t(H %*% t(dat))

# run svd and get some eigenvalues
uu <- svd(res)
# get the num of "sig" eigenvalues
ndf <- min(m, n) - ceiling(sum(diag(H)))
# get percent variance explained by each eigenvalue
dstat <- uu$d[1:ndf]^2/sum(uu$d[1:ndf]^2)
# initialize null distribution matrix
dstat0 <- matrix(0, nrow = B, ncol = ndf)

# set seed here since we are calling sample
# create the null distribution
for (i in 1:B) {
  # this part of the code removes all relationships in the 
  res0 <- t(apply(res, 1, sample, replace = FALSE))
  res0 <- res0 - t(H %*% t(res0))
  # calculate svs on this
  uu0 <- svd(res0)
  # bind their explained variance in null matrix
  dstat0[i, ] <- uu0$d[1:ndf]^2/sum(uu0$d[1:ndf]^2)
  print(i) # print out the iteration we are on
}

# determine the number of svs required here based on pvalues
psv <- rep(1, n)
for (i in 1:ndf) {
  psv[i] <- mean(dstat0[, i] >= dstat[i])
}
for (i in 2:ndf) {
  psv[i] <- max(psv[(i - 1)], psv[i])
}

# this is the variable holding the number of SVs required 
# number is each to all the svs with a pval less than/equal to 0.1
nsv <- sum(psv <= 0.1)


####################################
# code from the main SVA function  #
####################################

n.sv = nsv # from code above (run internally when sva called)
B = 5 # number of iterations

# dat =  betas # already initialized
# mod = mod  # already initialized
# mod0 = mod0  # already initialized 
# controls = NULL # we dont include controls - only for supervised method
# method = "irw" # this isnt actually used but would be called by sva
# vfilter = NULL # we dont set this when running sva
# numSVmethod = "be" # this is the default called by sva (code above)


###################################
# irwsva function internal to sva #
###################################

# this code is really want does all the work in sva
# we use the irw method so this is the code for that method
# there are also supervised and two step methods (not used by us)

# irw function takes input from sva, so we have already defined below vars
# dat = dat 
# mod = mod
# mod0 = mod0
# n.sv = n.sv
# B = B

# intialize variables again
n <- ncol(dat)
m <- nrow(dat)
Id <- diag(n)
resid <- dat %*% (Id - mod %*% solve(t(mod) %*% mod) %*% t(mod))
uu <- eigen(t(resid) %*% resid)
vv <- uu$vectors
ndf <- n - dim(mod)[2]
pprob <- rep(1, m)
one <- rep(1, n)
Id <- diag(n)
df1 <- dim(mod)[2] + n.sv
df0 <- dim(mod0)[2] + n.sv
rm(resid)

cat(paste("Iteration (out of", B, "):"))
for (i in 1:B) {
  mod.b <- cbind(mod, uu$vectors[, 1:n.sv])
  mod0.b <- cbind(mod0, uu$vectors[, 1:n.sv])
  ptmp <- f.pvalue(dat, mod.b, mod0.b)
  pprob.b <- (1 - edge.lfdr(ptmp))
  mod.gam <- cbind(mod0, uu$vectors[, 1:n.sv])
  mod0.gam <- cbind(mod0)
  ptmp <- f.pvalue(dat, mod.gam, mod0.gam)
  pprob.gam <- (1 - edge.lfdr(ptmp))
  pprob <- pprob.gam * (1 - pprob.b)
  dats <- dat * pprob
  dats <- dats - rowMeans(dats)
  uu <- eigen(t(dats) %*% dats)
  cat(paste(i, " "))
}

# this is what is normally returned by sva
# problem is that it doesnt include the diagonal matrix we want
sv_y1 = svd(dats)$v[, 1:n.sv, drop = FALSE] 
svobj <- list(sv = sv_y1, pprob.gam = pprob.gam, pprob.b = pprob.b, 
              n.sv = n.sv)

# to get all info (including diagonal)
svobj_all <- svd(dats)

save(sv_y1, file = here("output_data", 
                        "linear_regression", 
                        "2022-10-07_postnatal_svs_nopss.RData"))

#####################
# heat scree of SVs #
#####################

# from elodie portales-casamar

# heat scree prepartion
vars <- svobj_all[['d']]^2
vars <- vars/sum(vars)
names(vars) <- paste0('SV', 1:length(vars))
xp <- 1:svobj$n.sv
varsdf <- as.data.frame(vars)
varsdf$svnum <- rep(1:length(vars))

# heat scree ggplot
pn_sv_scree <- ggplot(varsdf[1:20,], aes(x=svnum, y=vars*100)) +
  geom_col() +
  theme_bw() +
  labs(x="Surrogate variables", y = "Variance (%)")

ggsave(plot = pn_sv_scree, 
       filename = here("figures", 
                       "linear_regression", 
                       "postnatal", 
                       "2022-10-07_pn_sv_heat_scree.png"),
       height = 5, width = 15, units = "cm")


######################
# correlation matrix #
######################

# add svs to covariates
covariates_geno_ctpcs_tech_cc <- cbind(covariates_geno_ctpcs_tech_cc, sv_y1)

# rename columns with svs
colnames(covariates_geno_ctpcs_tech_cc)[146:160] <-
  paste("sv",colnames(covariates_geno_ctpcs_tech_cc)[146:160], sep="")

source(here("scripts", "2021-08-30_corr_mat_funs.R"))

corr <- corr.mat(covariates_geno_ctpcs_tech_cc %>% 
                   dplyr::select(no2_y1_2019v, sex, maternal_smoking_y1, 
                                 maternal_education_length,
                                 genotyping.PC1, genotyping.PC2, genotyping.PC3,
                                 Comp.1, Comp.2, Comp.3, 
                                 Comp.4, Comp.5, Comp.6,
                                 cb_chip, cb_run, cb_row,
                                 y1_chip, y1_run, y1_row, sv1:sv15) %>%
                   rename("Postnatal NO2" = no2_y1_2019v, 
                          Sex = sex,
                          "Postnatal maternal cigarette smoking" = maternal_smoking_y1,
                          "Maternal education (years)" = maternal_education_length,
                          "Genotyping PC1" = genotyping.PC1,
                          "Genotyping PC2" = genotyping.PC2,
                          "Genotyping PC3" = genotyping.PC3,
                          "Cord blood chip" = cb_chip,
                          "Cord blood run" = cb_run,
                          "Cord blood row" = cb_row,
                          "Peripheral blood chip" = y1_chip,
                          "Peripheral blood run" = y1_run,
                          "Peripheral blood row" = y1_row,
                          "Celltype PC1" = Comp.1, 
                          "Celltype PC2" = Comp.2,
                          "Celltype PC3" = Comp.3,
                          "Celltype PC4" = Comp.4,
                          "Celltype PC5" = Comp.5,
                          "Celltype PC6" = Comp.6,
                          SV1 = sv1, SV2 = sv2, SV3 = sv3, SV4 = sv4, SV5 = sv5,
                          SV6 = sv6, SV7 = sv7, SV8 = sv8, SV9 = sv9, 
                          SV10 = sv10, SV11 = sv11, SV12 = sv12, SV13 = sv13, 
                          SV14 = sv14, SV15 = sv15))

pval <- pval.mat(covariates_geno_ctpcs_tech_cc %>% 
                   dplyr::select(no2_y1_2019v, sex, maternal_smoking_y1, 
                                 maternal_education_length,
                                 genotyping.PC1, genotyping.PC2, genotyping.PC3,
                                 Comp.1, Comp.2, Comp.3, 
                                 Comp.4, Comp.5, Comp.6,
                                 cb_chip, cb_run, cb_row,
                                 y1_chip, y1_run, y1_row, sv1:sv15) %>%
                   rename("Postnatal NO2" = no2_y1_2019v, 
                          Sex = sex,
                          "Postnatal maternal cigarette smoking" = maternal_smoking_y1,
                          "Maternal education (years)" = maternal_education_length,
                          "Genotyping PC1" = genotyping.PC1,
                          "Genotyping PC2" = genotyping.PC2,
                          "Genotyping PC3" = genotyping.PC3,
                          "Cord blood chip" = cb_chip,
                          "Cord blood run" = cb_run,
                          "Cord blood row" = cb_row,
                          "Peripheral blood chip" = y1_chip,
                          "Peripheral blood run" = y1_run,
                          "Peripheral blood row" = y1_row,
                          "Celltype PC1" = Comp.1, 
                          "Celltype PC2" = Comp.2,
                          "Celltype PC3" = Comp.3,
                          "Celltype PC4" = Comp.4,
                          "Celltype PC5" = Comp.5,
                          "Celltype PC6" = Comp.6,
                          SV1 = sv1, SV2 = sv2, SV3 = sv3, SV4 = sv4, SV5 = sv5,
                          SV6 = sv6, SV7 = sv7, SV8 = sv8, SV9 = sv9, 
                          SV10 = sv10, SV11 = sv11, SV12 = sv12, SV13 = sv13, 
                          SV14 = sv14, SV15 = sv15))


svcorr <- corr.plot(pval = pval, 
                    corr = corr, 
                    label.sig = T,
                    shape = 8,shape.size = 1,
                    axis.text.size = 8, x.axis.text.angle = 90) +
  scale_y_discrete(position = "right") +
  theme(legend.position = "none",
        panel.grid = element_blank())
svcorr

ggsave(plot = svcorr,
       filename = here("figures", 
                       "linear_regression", 
                       "postnatal",
                       "2022-10-07_pn_sv_correlations.png"),
       height = 15,
       width = 15, 
       units = "cm")

######################################
# heat scree plot of svs against pcs #
######################################

# have to use betas for this as diff betas contain negative values
# cant convert negative beta to m-value

#pca 
uncor.dat <- t(scale(t(diff_betas_geno_cc_cpr)))
PCA_full<-princomp(na.omit(uncor.dat))
Loadings<-as.data.frame(unclass(PCA_full$loadings))
vars <- PCA_full$sdev^2
Importance <-vars/sum(vars)

source(here("scripts", "2021-09-14_heat_scree_plot.R"))

#Specify which covariates are categorical and/or continuous
meta_categorical <- 
  covariates_geno_ctpcs_tech_cc[,c("sex", "maternal_smoking_y1",
                                   "cb_run", "cb_chip", "y1_run", "y1_chip")] 
meta_continuous <- 
  covariates_geno_ctpcs_tech_cc[,c("no2_y1_entire",
                                   "maternal_education_length", 
                                   "pss_6mos", "csed_6mos",
                                   "cb_row", "y1_row", 
                                   "celltype.PC1", "celltype.PC2",
                                   "celltype.PC3", "celltype.PC4",
                                   "celltype.PC5", "celltype.PC6",
                                   "celltype.PC7", "celltype.PC8",
                                   "celltype.PC9", 
                                   "genotyping.PC1", "genotyping.PC2",
                                   "genotyping.PC3", "sv1", "sv2", "sv3",
                                   "sv4", "sv5", "sv6", "sv7", "sv8",
                                   "sv9", "sv10", "sv11", "sv12", "sv13",
                                   "sv14", "sv15")]

# Specify the number of PCs you want shown 
# pick enough to view most of the variance
Num<-10

# Designate what order you want the variables to appear (continuous variables rbinded to categorical variables in function)
Order<-c(1:40)

#Apply function on PCA results of uncorrected data, pulls in the meta data and beta values from above
heat_scree_plot(Loadings = Loadings, 
                Importance = Importance, 
                Num = Num, 
                Order = Order,
                Categorical = meta_categorical,
                Continuous = meta_continuous)

###########################
# add SVs to model matrix #
###########################

#Design matrix for limma with surrogate variables
modSv = cbind(mod,sv_y1)
```

### EWAS of postnatal specific DNAm changes associated with year one NO2

```{r limma}
set.seed(1234)

#fit methylation to no2
diff_fit <-  lmFit(diff_betas_geno_cc_cpr, modSv)
diff_ebayes <- eBayes(diff_fit)

#add annotation to cb_ebayes
#pull annotation for cpgs contained in diff_betas
annotation_cpgs <- annotation[rownames(annotation) %in% rownames(diff_ebayes),]
#check order
identical(rownames(annotation_cpgs), rownames(diff_ebayes)) #false
#match
annotation_cpgs_order <- annotation_cpgs[rownames(diff_ebayes), ]
identical(rownames(annotation_cpgs_order), rownames(diff_ebayes)) #true

diff_ebayes$annotation <- annotation_cpgs_order

#pull out pval, coefficients and annotation
diff_pvals_tech <- cbind(as.data.frame(diff_ebayes$p.value), 
                         as.data.frame(diff_ebayes$coefficients),
                         as.data.frame(diff_ebayes$annotation))

#rename columns to indicate pvalues vs coefficients 
colnames(diff_pvals_tech) <- c(paste("pval", colnames(diff_pvals_tech)[1:29],
                                     sep = "_"), 
                               paste("coeff", colnames(diff_pvals_tech)[30:58], 
                                     sep = "_"),
                               colnames(diff_pvals_tech)[59:91])

diff_pvals_tech$pval_no2_y1_2019v_adj <- 
  p.adjust(diff_pvals_tech$pval_no2_y1_2019v, method="BH")

# write.csv(diff_pvals_tech,
#           file=here("tables",
#                     "yearone",
#                     "2022-11-17_yearone_no2_epigenome_wide_analysis_postnatal_specific_dnam.csv"))
```


Examine qqplot and pvalue histogram.
```{r qqplot_pval_histo}
# use code from qqman::qq function 
# plot with ggplot to make it more aesthetically appealing 

pvector <- diff_pvals_tech$pval_no2_y1_2019v

pvector2 <- pvector[!is.na(pvector) & !is.nan(pvector) & !is.null(pvector) & 
                      is.finite(pvector) & pvector < 1 & pvector > 0]

pval_qq <- as.data.frame(cbind(o = -log10(sort(pvector2, decreasing = FALSE)),
                               e = -log10(ppoints(length(pvector2)))))

# calculate lambda
z = qnorm(pvector2/2)
lambda = round(median(z^2) / 0.454, 3)

qqplot <- ggplot(pval_qq, aes(x=e, y=o)) + 
  geom_point(col="#254A90", size=1) + 
  geom_abline(slope = 1, intercept = 0, 
              col = "red", size=0.25, linetype="dashed") +
  labs(x = "Expected -log10(p-value)",
       y = "Observed -log10(p-value)") +
  annotate("text", x=3, y=1.5, 
           label = paste("Inflation = ", signif(lambda, 2)), 
           size=3, hjust=0) +
  theme_bw() +
  theme(axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        axis.title = element_blank(),
        panel.grid = element_blank(),
        panel.border = element_blank(),
        axis.line = element_line(color ="black")) 
qqplot

ggsave(plot = qqplot, 
       filename= here("figures", 
                      "linear_regression", 
                      "postnatal", 
                      "2022-10-18_postnatal_qqplot.png"),
       width = 7.5, height = 6, units = "cm")
```


Effect size
```{r effect_size}
range = max(covariates_geno_ctpcs_tech_cc$no2_y1_2019v) -
  min(covariates_geno_ctpcs_tech_cc$no2_y1_2019v)

diff_pvals_tech$effect_size <- diff_pvals_tech$coeff_no2_y1_2019v * range
```


Volcano plot
```{r volcano_db_tech}
# points used in investigation of cord blood DNAm persistence 
persist_points <- c("cg03508112", "cg08568767", "cg12228123", "cg14491284", "cg02074714",
                    "cg14658149", "cg25472862", "cg11642106", "cg09936824", "cg08033031",
                    "cg22230559", "cg05026403", "cg20747577", "cg23745864", "cg00125048",
                    "cg07658809", "cg03529432", "cg05686950", "cg14795268", "cg06143362",
                    "cg06617093", "cg26881997", "cg18229717", "cg10021364", "cg17646392",
                    "cg17186073", "cg00155314", "cg04340874", "cg26330365", "cg26457160",
                    "cg24126361", "cg07576222", "cg22136098", "cg12764003", "cg26089165",
                    "cg12184221", "cg16317923", "cg06518781", "cg21737226", "cg16321846",
                    "cg08650639", "cg26859557", "cg17516539", "cg15696634", "cg20779966",
                    "cg17051207", "cg08303283", "cg04452203", "cg06178322", "cg09155362",
                    "cg01755551", "cg24336985", "cg03346521", "cg07053697", "cg13690280",
                    "cg02614016", "cg02659402", "cg04548926", "cg16118259", "cg08958294",
                    "cg15635480", "cg16660547", "cg08901632", "cg08460812", "cg24262654",
                    "cg12214032", "cg13864749", "cg06167719", "cg06295784", "cg19295068",
                    "cg19804488", "cg22336337", "cg25437304", "cg04058100", "cg26679367",
                    "cg21426218", "cg03303774", "cg03443762", "cg19913563", "cg03416601",
                    "cg17836141", "cg26745143", "cg18331909", "cg01008894", "cg26061593",
                    "cg07176692", "cg27151303", "cg26160086", "cg01468193", "cg11061177",
                    "cg21523908", "cg19124424", "cg18064256", "cg04508648", "cg06346505",
                    "cg26646558", "cg27627493", "cg18063278", "cg02446225", "cg17105886"
)

#make new column to label sig points with cpgid
diff_pvals_tech$cpgid <- rownames(diff_pvals_tech)

#volcano plot
volcano_plot <- ggplot(diff_pvals_tech %>%
                         subset(!rownames(.) %in% persist_points), 
                       aes(x=as.numeric(effect_size) , 
                           y=-log10(as.numeric(pval_no2_y1_2019v)))) +
  geom_point(alpha=0.75, col = "#254A90", size=1) +
  # colour significant cpgs; 2.5% effect size cut off and FDR < 0.25
  geom_point(data = diff_pvals_tech %>% 
               subset(rownames(.) %in% persist_points), 
             alpha=0.75, col = "#254A90", size=1, shape = 25, fill = "#254A90") +  #"#254A90", "#508FC2", "#92C5EB"
  scale_x_continuous(limits = c(-0.20, 0.20), 
                     breaks = seq(-0.20, 0.20, by = 0.1)) +
  
  labs(y="-log10(unadjusted p-value)", x = "Effect size (% DNAm)") +
  theme_bw() +
  theme(panel.grid = element_blank(),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        axis.title = element_blank()) 

ggsave(plot = volcano_plot,
       filename = here("figures", 
                       "linear_regression", 
                       "postnatal", 
                       "2022-10-18_pn_volcanoplot.png"),
       width = 7.5, height = 6, units = "cm")
```


Subset out important cpgs
```{r sig_cpgs_tables}
# add db to e bayes
diff_ebayes$effect_size <- diff_pvals_tech$effect_size

# subset based on delta beta
diff_ebayes_sig <- diff_ebayes[diff_ebayes$effect_size > 0.025 | diff_ebayes$effect_size < -0.025, ]

# get top ten significant genes with coefficients and write csv
write.csv(limma::topTable(diff_ebayes_sig, number=10, 
                          coef ="no2_y1_2019v", sort.by = "P", confint = T),
          here("tables", "yearone", "2022-10-07_toptable_yearone_10cpgs.csv"))

# get top 10 cpgs with manifest annotation
write.csv(subset(diff_pvals_tech, 
                 (pval_no2_y1_2019v < 0.00004 & effect_size > 0.025) | 
                   (pval_no2_y1_2019v < 0.00004 & effect_size < -0.025)), 
          here("tables", 
               "yearone", 
               "2022-10-07_top10_postnatal_cpgs.csv"))

# write out all prenatal findings
write.csv(diff_pvals_tech, 
          here("tables", 
               "yearone", 
               "2022-10-07_ALL_postnatal_cpgs.csv"))
```


### Gene specific changes

These genes are related to the proposed pathway of air pollution action or have previously been shown to exhibit DNA methylation changes associated with prenatal air pollution exposure. 

```{r gene_specific}
#FHC is FTH1 (ferritin heavy chain 1)
#nothing maps to FOXP3
#TNFA is TNF
goi <- annotation[grep("^RNF39|^CYP2E1|^LONP1|^SLC25A28|^PLVAP|^CAT$|^CAT;|^TPO|^NFKB1|^NOX2|^SOD1|^SOD2|^TRX1|^TRX2|^GSTP1|^MT3|^NQO1|^HMOX1|^TGPX1|^DDH1|^NOS2|^CYP2C11|^CYP7B|^ALOX12$|^ALOX5;|^ALOX5$|^NRF2$|^NRF2;|^KEAP1|^AHR$|^AHR;|^ARNT;|^ARNT$|^GCLC|^GCLM|^GSR1|^SLC7A11|^GSTA1|^GSTA2|^GSTA3|^GSTA5|^GSTM1|^GSTM2|^GSTM3|^TXN$|^TXN;|^TXNRD1$|^TXNRD1;|^SRXN1|^G6PD|^CYP1A1|^IL1A|^IL2$|^IL2;|^IL3$|^IL3;|^IL4;|^IL4$|^IL5$|^IL5;|^IL6$|^IL6;|^IL9$|^IL9;|^IL10$|^IL10;|^IL12A|^IL13|^IL15$|^IL15;|^IL17A|^IL18$|^IL18;|^IL19|^IL21$|^IL21;|^IL22$|^IL22;|^IL23A$|^IL23A;|^IL25$|^IL25;|^IL27$|^IL27;|^IL31$|^IL31;|^IL33$|^IL33;|^IFNG$|^IFNG;|^TNF$|^TNF;|^TGFB1$|^TGFB1;|^FOXP3|^FTH1", annotation$UCSC_RefGene_Name), ] %>% as.data.frame()

unique(goi$UCSC_RefGene_Name)
unique(gsub(";.*$", "", unique(goi$UCSC_RefGene_Name))) # 59

#transpose betas
diff_betas_geno_cc_cpr_t <- t(diff_betas_geno_cc_cpr)

#apply function for calculating significance of associations using mice and imputed covariate data
yearone_goi_list <- 
  pbapply(diff_betas_geno_cc_cpr_t [,colnames(diff_betas_geno_cc_cpr_t ) %in% rownames(goi)], 2, function(x) {
    
    combined <- cbind(covariates_geno_ctpcs_tech_cc, x)
    
    #run the linear regression on this cpg using all 5 imputed covariate data sets
    fit <- lm(x ~ no2_y1_2019v + sex + maternal_smoking_y1 + 
                maternal_education_length + Comp.1 + Comp.2 + Comp.3 + Comp.4 + 
                Comp.5 + Comp.6 + genotyping.PC1 + genotyping.PC2 + 
                genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 +
                sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + sv15, 
              data=combined)
    
    #get summary of lm fit
    #this includes coefficients and pvalues
    #coerce into df for manipulation
    #add in terms (rownames) as column for annotation
    fit_sum <- summary(fit)
    fit_sum_coeff <- as.data.frame(fit_sum$coefficients)
    fit_sum_coeff$term <- rownames(fit_sum_coeff)
    
    #this data frame has rows for each linear regression variable, and columns for statistics
    #want a dataframe of 1 row and many columns (one for each variable with each statistic type)
    #do this by using the spread function for each statistics
    #add cpg id onto dataframe for pvalues (to keep track of information)
    #then cbind dataframes together
    
    #coefficient
    estimate <- spread(fit_sum_coeff[,c("term", "Estimate")], term, Estimate)
    colnames(estimate) <- paste("estimate", colnames(estimate), sep=".")
    
    #std error
    std.error <- spread(fit_sum_coeff[,c("term", "Std. Error")], term, 'Std. Error')
    colnames(std.error) <- paste("std.error", colnames(std.error), sep=".")
    
    #t statistics
    statistic <- spread(fit_sum_coeff[,c("term", "t value")], term, 't value')
    colnames(statistic) <- paste("t.val", colnames(statistic), sep=".")
    
    #pvalue
    p.value <- spread(fit_sum_coeff[,c("term", "Pr(>|t|)")], term, 'Pr(>|t|)')
    colnames(p.value) <- paste("p.value", colnames(p.value), sep=".")
    
    #add in cpgid for annotation 
    p.value$cpg <- colnames(x)
    
    #cbind into wide dataframe
    widepool <- cbind(estimate, std.error, statistic, p.value)
    
    #return the wide dataframe
    #these will be do.call(rbind()) together after
    return(widepool)
    
  })


yearone_goi <- do.call(rbind, yearone_goi_list)

yearone_goi$p.value.no2_y1_2019v_adj <- p.adjust(yearone_goi$p.value.no2_y1_2019v, method="BH")

#reorder according ot unadjusted no2 pvalue
yearone_goi_order <- yearone_goi[order(yearone_goi$p.value.no2_y1_2019v),]

# caclculate effect size
range = max(covariates_geno_ctpcs_tech_cc$no2_y1_2019v) -
  min(covariates_geno_ctpcs_tech_cc$no2_y1_2019v)

yearone_goi_order$no2_y1_2019v_effect_size <- yearone_goi_order$estimate.no2_y1_2019v * range

# write out table for all GOIs
write.csv(yearone_goi_order, 
          file=here("tables", 
                    "yearone",
                    "2022-11-18_goi_postnatal_cpgs.csv"))


################
# gruzieva cps #
################

##################
#LONP1 cg12283362#
##################

covariates_geno_ctpcs_tech_cc$cg12283362 <- 
  diff_betas_geno_cc_cpr[rownames(diff_betas_geno_cc_cpr)=="cg12283362", ] 

cg12283362_lm <- lm(cg12283362 ~ no2_y1_2019v + sex + maternal_smoking_y1 + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + Comp.5 + Comp.6 + genotyping.PC1 + 
                      genotyping.PC2 + genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + 
                      sv5 + sv6 + sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + 
                      sv14 + sv15, 
                    data=covariates_geno_ctpcs_tech_cc)

# get lm info
summary(cg12283362_lm)

# estimate marginal effects of prenatal NO2 exposure on DNAm
cg12283362_no2_me <- ggpredict(cg12283362_lm, terms = "no2_y1_2019v") %>%
  as.data.frame() %>% 
  rename("cg12283362" = predicted,
         "no2_y1_2019v" = x)

# plot
cg12283362_plot <- ggplot(covariates_geno_ctpcs_tech_cc,
                          aes(x = no2_y1_2019v, 
                              y= cg12283362)) + 
  geom_line(data = cg12283362_no2_me, 
            aes(x=no2_y1_2019v, y = cg12283362),
            col="red", size=0.5, linetype="solid") +
  geom_ribbon(data = cg12283362_no2_me, 
              aes(x = no2_y1_2019v, 
                  ymin = conf.low, ymax = conf.high), 
              alpha = .2) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.title = element_blank()) 

ggsave(plot = cg12283362_plot, 
       filename = here("figures", 
                       "linear_regression", 
                       "postnatal", 
                       "2022-11-10_cg12283362_LONP1.png"),
       width = 4, height = 3.2, units = "cm")

# effect size
range_no2 = max(covariates_geno_ctpcs_tech_cc$no2_y1_2019v) -
  min(covariates_geno_ctpcs_tech_cc$no2_y1_2019v)

summary(cg12283362_lm)$coefficients["no2_y1_2019v", "Estimate"]*range_no2 # -0.01089209

# correlation 
cor(covariates_geno_ctpcs_tech_cc$cg12283362, 
    covariates_geno_ctpcs_tech_cc$no2_y1_2019v,
    method="pearson") # -0.003466747

# check slope of marginal effect
lm(cg12283362_no2_me$cg12283362 ~ cg12283362_no2_me$no2_y1_2019v) # -0.0004455 


#####################
#SLC25A28 cg08973675#
#####################

covariates_geno_ctpcs_tech_cc$cg08973675 <- 
  diff_betas_geno_cc_cpr[rownames(diff_betas_geno_cc_cpr)=="cg08973675", ] 

cg08973675_lm <- lm(cg08973675 ~ no2_y1_2019v + sex + maternal_smoking_y1 + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + Comp.5 + Comp.6 + genotyping.PC1 + 
                      genotyping.PC2 + genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + 
                      sv5 + sv6 + sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + 
                      sv14 + sv15, 
                    data=covariates_geno_ctpcs_tech_cc)

# get lm info
summary(cg08973675_lm)

# estimate marginal effects of prenatal NO2 exposure on DNAm
cg08973675_no2_me <- ggpredict(cg08973675_lm, terms = "no2_y1_2019v") %>%
  as.data.frame() %>% 
  rename("cg08973675" = predicted,
         "no2_y1_2019v" = x)

# plot
cg08973675_plot <- ggplot(covariates_geno_ctpcs_tech_cc,
                          aes(x = no2_y1_2019v, 
                              y= cg08973675)) + 
  geom_line(data = cg08973675_no2_me, 
            aes(x=no2_y1_2019v, y = cg08973675),
            col="red", size=0.5, linetype="solid") +
  geom_ribbon(data = cg08973675_no2_me, 
              aes(x = no2_y1_2019v, 
                  ymin = conf.low, ymax = conf.high), 
              alpha = .2) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.title = element_blank()) +
  scale_y_continuous(breaks = seq(-0.08, 0.08, by=0.04), limits=c(-0.08, 0.08))

ggsave(plot = cg08973675_plot, 
       filename = here("figures", 
                       "linear_regression", 
                       "postnatal", 
                       "2022-11-10_cg08973675_SLC25A28.png"),
       width = 4, height = 3.2, units = "cm")

# effect size
range_no2 = max(covariates_geno_ctpcs_tech_cc$no2_y1_2019v) -
  min(covariates_geno_ctpcs_tech_cc$no2_y1_2019v)

summary(cg08973675_lm)$coefficients["no2_y1_2019v", "Estimate"]*range_no2 # 0.01130771

# correlation 
cor(covariates_geno_ctpcs_tech_cc$cg08973675, 
    covariates_geno_ctpcs_tech_cc$no2_y1_2019v,
    method="pearson") # -0.03784814


```


### Differentially methylated regions

```{r dmr}
comb_dat <- diff_pvals_tech %>%
  select(pval_no2_y1_2019v, chr, pos, Name) %>%
  mutate(chr = str_sub(chr, 4,5),
         end = pos + 1) %>%
  dplyr::rename("p" =pval_no2_y1_2019v,
                "start" = pos,
                "probe" = Name)

# results stored in home directory 
combp(comb_dat, 
      seed = 0.01,
      verbose = T)

# read in csv of dmrs
resu_combp <- read_csv(here("output_data",
                            "linear_regression",
                            "2022-10-07_postnatalspecific_resu_combp.csv"))

# get annotation for DMRs in combp results
combp_anno <- apply(resu_combp, 1, function(x){
  annotation %>%
    as.data.frame() %>%
    subset(chr == paste("chr", x["chr"], sep = "")) %>%
    subset(pos >= x["start"] & pos <= x["end"]-1) %>%
    mutate(dmr = x["dmr"],
           analysis = x["analysis"])
})

# conver to data frame
combp_anno_df <- do.call(rbind, combp_anno)

# create new columns to specify DMR and analysis
resu_combp$dmr <- 1:nrow(resu_combp)
resu_combp$analysis <- "postnatalspecific_no2"

# specify which DMR the annotation belongs to
combp_anno_df$dmr <- sapply(1:nrow(resu_combp), function(i) 
  rep(i, resu_combp[i,"nprobe"]), simplify = "vector") %>%
  unlist()

combp_anno_df$analysis <- "postnatalspecific_no2"

# merge with rlm data to get effect size at each cpg
combp_anno_df_es <- merge(combp_anno_df, 
                          diff_pvals_tech %>%
                            subset(rownames(.) %in% rownames(combp_anno_df)) %>%
                            dplyr::select(effect_size, coeff_no2_y1_2019v),
                          by = "row.names")

# subset for dmrs with  more than 1 cpg
resu_combp_sub <- resu_combp %>%
  subset(nprobe >1 ) %>%
  mutate(dmr2 = 1:nrow(.))

# subset DMR annotation for those with more than 1 CpG
combp_anno_df_es_sub <- combp_anno_df_es %>%
  subset(dmr %in% resu_combp_sub$dmr)

# check mean effect sizes
combp_anno_df_es_sub %>%
  group_by(dmr) %>%
  summarise(mean(effect_size)) 


# write out annotation for dmrs
write.csv(combp_anno_df_es_sub,
          here("tables",
               "yearone",
               "2022-11-23_postnatalspecific_no2_combp_annotation_sigDMRs.csv"))
```

"CHARM-style" plots of DMRs
```{r charm_style_plots}

# subset out dmr 1 cpgs
covariates_geno_ctpcs_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpcs_tech_cc, 
        t(diff_betas_geno_cc_cpr[combp_anno_df_es_sub %>%
                                   subset(dmr == 1) %>%
                                   pull(Name),]))

# rename columns with numbers
colnames(covariates_geno_ctpcs_tech_cc_dmr_cpgs)[177:182] <-
  combp_anno_df_es_sub %>%
  subset(dmr == 1) %>%
  pull(pos)

# pivot to long
longdf_dmr1 <- 
  pivot_longer(covariates_geno_ctpcs_tech_cc_dmr_cpgs %>% 
                 dplyr::select("46681111", "46681154", "46681401", 
                               "46681135", "46681362", "46681316",
                               sample_id, 
                               no2_y1_2019v),
               cols = -c(sample_id, 
                         no2_y1_2019v), 
               names_to = "cpg", 
               values_to = "dnam") %>%
  #mutate(cpg = factor(cpg, levels = c("cg09594291", "cg02749105", "cg24052338"))) %>%
  mutate(no2_colour = ifelse(no2_y1_2019v < 4.687228, "first",
                             ifelse(no2_y1_2019v >=4.687228 & no2_y1_2019v < 10.172466 , "second", 
                                    ifelse(no2_y1_2019v >= 10.172466  & no2_y1_2019v < 14.943722, "third", "fourth")))) %>%
  mutate(no2_colour = factor(no2_colour, levels = c("first", "second",  "third", "fourth")))

# create charm-like visualization
dmr1_charm <- 
  ggplot(longdf_dmr1 %>% 
           subset(no2_colour %in% c("first", "fourth")) %>%
           group_by(cpg, no2_colour) %>%
           summarize(mean_dnam = mean(dnam)), 
         aes(x=as.numeric(cpg), y=mean_dnam, col = no2_colour)) +
  scale_color_manual(values = c("#254A90","red"), 
                     labels = c("First", "Third"),
                     name = "Prenatal NO2 quantile") +
  geom_line() + 
  geom_point(longdf_dmr1 %>% 
               subset(no2_colour %in% c("first", "fourth")), 
             mapping = aes(x=as.numeric(cpg), y=dnam, col = no2_colour),
             size = 0.25, alpha = 0.8) +
  theme_bw() +
  scale_x_continuous(name = "Genomic position (Chr17)",
                     breaks = seq(46681150, 46681350, by = 100)) +
  scale_y_continuous(name = "DNA methylation (%)") +
  theme(axis.text = element_text(size = 9, family = "sans"),
        axis.title = element_text(size = 9, family = "sans"),
        panel.grid = element_blank(),
        legend.position = "none",
        panel.background = element_rect(fill = '#D3D3D3'))

ggsave(plot = dmr1_charm,
       filename = here("figures",
                       "linear_regression",
                       "postnatal",
                       "2023-01-03_postnatal_dmr1_charm.png"),
       height = 6, width = 12, units = "cm")

# create genomic region visualization 
dmr1_from <- combp_anno_df_es_sub %>%
  subset(dmr == 1) %>%
  pull(pos) %>%
  min()
dmr1_to <- combp_anno_df_es_sub %>%
  subset(dmr == 1) %>%
  pull(pos) %>%
  max()

dmr1_refgenes <- UcscTrack(genome = "hg19", chromosome = "chr17", 
                           track = "NCBI RefSeq",
                           table = "refGene", 
                           from = dmr1_from, to = dmr1_to,
                           trackType = "GeneRegionTrack", 
                           rstarts = "exonStarts", rends = "exonEnds", 
                           gene = "name", symbol = "name", 
                           transcript = "name", strand = "strand", 
                           fill = "#254A90", name = "NCBI RefGene")

# add names
dmr1_ranges <- ranges(dmr1_refgenes)

# gene refseq gene symbiks abd add to annotation 
mcols(dmr1_ranges)$symbol <- mapIds(org.Hs.eg.db, 
                                    gsub("\\.[1-9]$", "", mcols(dmr1_ranges)$symbol),
                                    "SYMBOL","REFSEQ")
ranges(dmr1_refgenes) <- dmr1_ranges


# get regulatory info from ucsc
# gm12878 = b cell line
# hep g2 = epithelial like cell line 
# hu-vec = human umbilical cord vein 
# k562 = lymphoblasyt
# nhkf = lung fibroblasts
dmr1_chromHmmGm12878 <- UcscTrack(genome = "hg19", chromosome = "chr17", 
                                  track = "Broad ChromHMM",
                                  table = "wgEncodeBroadHmmGm12878HMM",
                                  trackType = "AnnotationTrack", 
                                  from = 46670000, 
                                  to  =  46685401,
                                  start = "chromStart", end = "chromEnd", 
                                  id = "name", feature = "score", shape = "box", 
                                  stacking = "dense", fill = "itemRgb",
                                  name = "GM12878")

# colour according to hmm table 
dmr1_chromHmmGm12878@dp@pars$fill <- recode(dmr1_chromHmmGm12878@dp@pars$fill, 
                                            "245,245,245" = "#f5f5f5",
                                            "255,105,105" = "#ff6969",
                                            "153,255,102" = "#99ff66",
                                              "255,0,0" = "#ff0000",
                                            "0,176,80" = "#00b050",
                                            "250,202,0" = "#faca00",
                                            "255,252,4" = "#fffc04",
                                            "127,127,127" = "#7f7f7f",
                                            "10,190,254" = "#0abefe",
                                            "207,11,198" = "#cf0bc6")

dmr1_chromHmmGmNhlf <- UcscTrack(genome = "hg19", chromosome = "chr17",
                                 track = "Broad ChromHMM",
                                 table = "wgEncodeBroadHmmNhlfHMM",
                                 trackType = "AnnotationTrack",
                                   from = 46670000, 
                                  to  =  46685401,
                                 start = "chromStart", end = "chromEnd",
                                 id = "name", feature = "score", shape = "box",
                                 stacking = "dense", fill = "itemRgb",
                                 name = "NHLF")

dmr1_chromHmmGmNhlf@dp@pars$fill <- recode(dmr1_chromHmmGmNhlf@dp@pars$fill, 
                                           "250,202,0" = "#faca00",
                                           "255,0,0" = "#ff0000",
                                            "0,176,80" = "#00b050",
                                           "255,105,105" = "#ff6969",
                                           "245,245,245" = "#f5f5f5",
                                           "255,252,4" = "#fffc04",
                                           "10,190,254"  = "#0abefe",
                                           "127,127,127" = "#7f7f7f",
                                           "153,255,102" = "#99ff66")

dmr1_chromHmmH1hesc <- UcscTrack(genome = "hg19", chromosome = "chr17", 
                                 track = "Broad ChromHMM",
                                 table = "wgEncodeBroadHmmH1hescHMM",
                                 trackType = "AnnotationTrack", 
                                   from = 46670000, 
                                  to  =  46685401,
                                 start = "chromStart", end = "chromEnd", 
                                 id = "name", feature = "score", shape = "box", 
                                 stacking = "dense", fill = "itemRgb",
                                 name = "HESC")

dmr1_chromHmmH1hesc@dp@pars$fill <- recode(dmr1_chromHmmH1hesc@dp@pars$fill,
                                           "127,127,127" = "#7f7f7f",
                                           "245,245,245" = "#f5f5f5",
                                            "255,105,105" = "#ff6969",
                                           "250,202,0" = "#faca00",
                                           "153,255,102" = "#99ff66",
                                            "0,176,80" = "#00b050",
                                             "255,0,0" = "#ff0000",
                                                  "255,252,4" = "#fffc04",
                                           "10,190,254"  = "#0abefe",
                                           "207,11,198" = "#cf0bc6")

axTrack <- GenomeAxisTrack()

# highlight track for DMR
dmr1_ht <- HighlightTrack(trackList = list(dmr1_refgenes, 
                                           dmr1_chromHmmGm12878, 
                                           dmr1_chromHmmH1hesc,
                                           dmr1_chromHmmGmNhlf,
                                           axTrack),
                          start = dmr1_from,
                          end = dmr1_to, 
                          chromosome = 17, fill = "lightgrey", col = "lightgrey")

unique(c(dmr1_chromHmmGm12878@dp@pars$fill, 
         dmr1_chromHmmGmNhlf@dp@pars$fill, 
         dmr1_chromHmmH1hesc@dp@pars$fill))

png(here("figures",
         "linear_regression",
         "postnatal",
         "2023-01-04_postnatal_dmr1_gviz.png"),
    height = 6, width = 10.5, units = "cm", res = 300)
plotTracks(list(dmr1_ht),  
           transcriptAnnotation = "symbol",
           geneAnnotation = "gene",
           showTitle = F)
dev.off()
```


Manhattan plot of CpGs in DMRS
```{r dmr_manhattan}
# Prepare data for manhattan plot.  
# nonsig pvals arent very informative and hae lots of overlap
# reduce number of nonsig pvals for plotting to improve/speed up performance
# randomly select subset of nonsig pvals for plotting
# if you dont subset, the plot will take a very long time to generate

sig.dmr.dat <- diff_pvals_tech %>% 
  subset(rownames(.) %in% combp_anno_df_es_sub$Name) %>%
  mutate(group = "dmr")

notsig.dat <- diff_pvals_tech %>% 
  subset(!rownames(.) %in% rownames(sig.dmr.dat)) %>%
  subset(pval_no2_y1_2019v_adj > 0.05) %>%
  mutate(group = "notsig")

# set seed before random sampling to ensure consistent results 
#set.seed(1234)
#notsig.dat.red <- notsig.dat[(sample(nrow(notsig.dat), nrow(notsig.dat) / 10)),]
mqtls.red <- rbind(sig.dmr.dat, notsig.dat)

mqtls.red$chr <- factor(mqtls.red$chr,
                        levels = c("chr1", "chr2", "chr3", "chr4", "chr5", 
                                   "chr6", "chr7", "chr8", "chr9", "chr10",
                                   "chr11", "chr12", "chr13", "chr14", "chr15",
                                   "chr16", "chr17", "chr18", "chr19", "chr20",
                                   "chr21", "chr22"))

# get max positions
chrsizes <- aggregate(pos ~ chr, data = mqtls.red, max) 

# get cumulative sizes of chrs
chrsizes$cumsum <- cumsum(as.numeric(chrsizes[, 2]))
# start cumsum at 0
chrsizes$cumsum <- chrsizes$cumsum - chrsizes$cumsum[1]

# Add this info to the initial dataset
mqtls.red2 <- left_join(mqtls.red, chrsizes[,c("chr", "cumsum")], by=c("chr"="chr"))

# Add a cumulative position of each mqtl
mqtls.red2$bpcum <- mqtls.red2$pos + mqtls.red2$cumsum

# Add a cumulative position of each SNP
# get chromosome center positions for x-axis
axisdf <- aggregate(mqtls.red2$bpcum, by = list(mqtls.red2$chr), function(x) {
  centre = (max(x) + min(x))/2
})
# add centres to cumsum
axisdf$cumsumcentre <- axisdf$x + chrsizes$cumsum

# custom colours
# shades of blue
colour_vals <- rep_len(c("#254A90", "#508FC2", "#92C5EB"), length.out=22)

# plot manhattan 
manhat <- ggplot(mqtls.red2 %>%
                   subset(group == "notsig"), 
                 aes(x=bpcum, y=effect_size)) +
  geom_point(aes(color=as.factor(chr)), alpha=0.6, size=1) + 
  scale_color_manual(values = colour_vals) +
  # dmrs
  geom_point(data = mqtls.red2 %>%
               subset(group == "dmr"),
             aes(x=bpcum, y=effect_size),
             color = "red2", size = 1, alpha = 0.8) +
  scale_x_continuous(label = c(1:22)[c(TRUE, FALSE)], 
                     breaks= axisdf$x[c(TRUE, FALSE)], 
                     name = "Chromosome") +
  # scale_y_continuous(expand = c(0, 0), limits= c(0, -log10(1e-8))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5),
        legend.position="none",
        panel.border = element_blank(),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        axis.text.x = element_text(size =  10),
        axis.text.y = element_text(size = 10),
        axis.title = element_text(size = 13)) +
  labs(x=element_blank(), y="Effect size (%DNAm)")

ggsave(plot = manhat,
       filename = here("figures",
                       "linear_regression",
                       "postnatal",
                       "2022-11-10_postnatalspecific_no2_manhattan_es.png"),
       height = 5, width = 15, units = "cm")
```

### mQTLdb cross check

Check whether CpGs that compose DMRs are annotated as potential mQTLs in mQTLdb.com. I copied the list of CpGs contained in DMRs into mQTLdb then saved the outputted data to compare whether the CpGs appear in hits here. 
```{r mqtl_check}
# load mQTL data
# birth_mqtls <- read.csv(here("input_data",
#                              "mQTLdb_lists", 
#                              "CHILD_postnatalspecific_dmr_cpgs_mqtldb_birth.csv"))
# 
# # subset mqtls for pval and effect size according to guant (2016; PMID: 27036880)
# birth_mqtls_sub <- birth_mqtls %>%  
#   subset(p.value< 1.0e-14 & Effect.Size > 0.02)
# 
# childhood_mqtls <- read.csv(here("input_data",
#                                  "mQTLdb_lists", 
#                                  "CHILD_postnatal_specific_dmr_cpgs_mqtldb_childhood.csv"))
# 
# # subset mqtls for pval and effect size according to guant (2016; PMID: 27036880)
childhood_mqtls_sub <- childhood_mqtls %>%  
  subset(p.value< 1.0e-14 & Effect.Size > 0.02)

# annotate individual cpgs within DMRs as a potential mqtl or not
# combp_anno_df_es_sub_mqtls <- combp_anno_df_es_sub %>%
#   mutate(birth_mqtl = ifelse(Name %in% (birth_mqtls %>% 
#                                     distinct(CpG) %>% 
#                                     pull(CpG)), TRUE, FALSE))
# 
# combp_anno_df_es_sub_mqtls <- combp_anno_df_es_sub_mqtls %>%
#   mutate(childhood_mqtl = ifelse(Name %in% (childhood_mqtls %>% 
#                                     distinct(CpG) %>% 
#                                     pull(CpG)), TRUE, FALSE))


# write out annotation for dmrs
# write.csv(combp_anno_df_es_sub_mqtls,
#           here("tables",
#                "yearone",
#                "2022-11-20_yearone_no2_combp_annotation_sigDMRs_mQTL_annotation.csv"))

```



Examine whether DNAm at significant cpgs/dmrs is associated with reported wheeze at age one. In previous CHILD paper [(PMID: 26424567)](https://www.science.org/doi/10.1126/scitranslmed.aab2271) wheeze was defined using wheeze questionnaires. If the child had wheezed with or without a cold during the first year of life (recorded via questionnaires answered by parents at 3, 6, and 12 months), the child was included in the wheezing group. Children were also included in the wheezing group if the CHILD clinician recorded a wheeze during the 1-year clinical assessment.

Need to multiple DNAm at each locus by 100 to put into percent to get an interpretable estimate and confidence interval.
```{r wheezing_associations}
covariates_geno_ctpcs_tech_cc$any_wheeze_y1 <- 
  ifelse(covariates_geno_ctpcs_tech_cc$reported_wheezing_y1 ==1 | 
           covariates_geno_ctpcs_tech_cc$clinician_wheeze_dx_y1 ==1, 1, 0)

# DMR 1
covariates_geno_ctpcs_tech_cc$dmrmean1 <- 
  diff_betas_geno_cc_cpr[combp_anno_df_es_sub %>% subset(dmr == 1) %>% pull(Name),] %>% 
  colMeans()

# dmr 1 effect size
combp_anno_df_es_sub %>% 
  subset(dmr==1) %>% 
  pull(effect_size) # increases

ggplot(covariates_geno_ctpcs_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean1)) +
  geom_boxplot()


# subset out dmr 1 cpgs
covariates_geno_ctpcs_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpcs_tech_cc, 
        t(diff_betas_geno_cc_cpr[combp_anno_df_es_sub %>%
                                   subset(dmr == 1) %>%
                                   pull(Name),]))

# wheeze
dmr1_mancova_wheeze <- manova(cbind(cg00072689, cg01986016, cg03803541, 
                                    cg15435170, cg17179862, cg20471691) ~ 
                                any_wheeze_y1 + sex + maternal_smoking_y1 + 
                                maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                                Comp.4 + Comp.5 + Comp.6 + genotyping.PC1 + 
                                genotyping.PC2 + genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + 
                                sv5 + sv6 + sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + 
                                sv14 + sv15, 
                              data = covariates_geno_ctpcs_tech_cc_dmr_cpgs)

summary(dmr1_mancova_wheeze, test="Pillai") # any wheeze p = 0.3864628 

# atopy
dmr1_mancova_atopy <- manova(cbind(cg00072689, cg01986016, cg03803541, 
                                   cg15435170, cg17179862, cg20471691) ~ 
                               clinician_atopy_dx_y1 +  sex + maternal_smoking_y1 + 
                               maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                               Comp.4 + Comp.5 + Comp.6 + genotyping.PC1 + 
                               genotyping.PC2 + genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + 
                               sv5 + sv6 + sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + 
                               sv14 + sv15, 
                             data = covariates_geno_ctpcs_tech_cc_dmr_cpgs)
summary(dmr1_mancova_atopy, test="Pillai") # atopy p = 0.0199429



longdf_dmr1 <- pivot_longer(covariates_geno_ctpcs_tech_cc_dmr_cpgs %>% 
                              select(cg00072689, cg01986016, cg03803541, 
                                     cg15435170, cg17179862, cg20471691, 
                                     sample_id, clinician_atopy_dx_y1, 
                                     no2_y1_2019v),
                            cols = -c(sample_id, clinician_atopy_dx_y1, 
                                      no2_y1_2019v), 
                            names_to = "cpg", 
                            values_to = "dnam") %>%
  mutate(cpg = factor(cpg, levels = c("cg00072689", "cg15435170", "cg01986016", 
                                      "cg20471691", "cg17179862", "cg03803541"))) %>%
  mutate(no2_colour = ifelse(no2_y1_2019v < 4.69, "first",
                             ifelse(no2_y1_2019v >=4.69 & no2_y1_2019v < 10.17 , "second", 
                                    ifelse(no2_y1_2019v >= 10.17  & no2_y1_2019v < 14.94, "third", "fourth")))) %>%
  mutate(no2_colour = factor(no2_colour, levels = c("first", "second",  "third", "fourth")))


dmr1_atopy_vis <- 
  ggplot(longdf_dmr1, aes(x=cpg, y=dnam, 
                          group = interaction(cpg, clinician_atopy_dx_y1), 
                          shape = as.factor(clinician_atopy_dx_y1),
                          color = as.factor(no2_colour),
                          fill = as.factor(clinician_atopy_dx_y1))) +
  scale_shape_manual(values = c(20, 17), labels = c("No atopy", "Atopy")) +
  scale_color_manual( values = c("#87a8e8", "#4484fc", "#254A90", "#10203d")) +
  scale_fill_manual(values = c("white", "grey"), labels = c("No atopy", "Atopy")) +
    geom_violin(draw_quantiles = c(0.5), trim = T, alpha =0.6) +
    ggbeeswarm::geom_beeswarm(dodge.width=0.9, cex = 1, size = 0.5, groupOnX = T) +
  theme_bw() +
  scale_x_discrete(labels = c("cg00072689",
                              "cg15435170",
                              "cg01986016",
                              "cg20471691",
                              "cg17179862",
                              "cg03803541"),
                   name = element_blank()) +
  scale_y_continuous(name = "DNA methylation (%)") +
  theme(legend.position = "none",
    legend.direction="horizontal",
    axis.text = element_text(size = 9),
    panel.grid.major.x = element_blank(),
    legend.key.size = unit(0.5, 'cm'))


ggsave(plot = dmr1_atopy_vis,
       filename  = here("figures",
                        "linear_regression",
                        "postnatal",
                        "2022-11-23_no2_pn_dmr1_atopy.png"), 
       width = 12, height = 5, units = "cm")

# check vcov and correlations
dmr1_mancova_wheeze_vcov <- vcov(dmr1_mancova_wheeze) 
colnames(dmr1_mancova_wheeze_vcov) <- rownames(coef(dmr1_mancova_wheeze))
rownames(dmr1_mancova_wheeze_vcov) <- rownames(coef(dmr1_mancova_wheeze))

dmr1_mancova_wheeze_cov2cor <- cov2cor(dmr1_mancova_wheeze_vcov)
colnames(dmr1_mancova_wheeze_cov2cor) <- rownames(coef(dmr1_mancova_wheeze))
rownames(dmr1_mancova_wheeze_cov2cor) <- rownames(coef(dmr1_mancova_wheeze))

```

### Genome wide changes in DNAm 

#### Mean array wide DNAm 

Examine changes in DNAm across all array probes
```{r mean_meth}
#convert betas to df
diff_betas_geno_cc_cpr_df <- as.data.frame(diff_betas_geno_cc_cpr)

methmeans <- colMeans(diff_betas_geno_cc_cpr_df)

covariates_geno_ctpcs_tech_cc <- cbind(covariates_geno_ctpcs_tech_cc, methmeans)

#colmeans
meanmeth_lm <- lm(methmeans ~ no2_y1_2019v + sex + maternal_smoking_y1 + 
                    maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                    Comp.4 + Comp.5 + Comp.6 + genotyping.PC1 + 
                    genotyping.PC2 + genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + 
                    sv5 + sv6 + sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + 
                    sv14 + sv15,
                  data=covariates_geno_ctpcs_tech_cc)

#get summary
summary(meanmeth_lm) # p = 0.337206

# estimate marginal effects of prenatal NO2 exposure on DNAm
methmean_no2_me <- ggpredict(meanmeth_lm, terms = "no2_y1_2019v") %>%
  as.data.frame() %>%
  dplyr::rename("methmeans" = predicted,
                "no2_y1_2019v" = x)

# plot
methmean_plot <- ggplot(covariates_geno_ctpcs_tech_cc,
                        aes(x = no2_y1_2019v, 
                            y= methmeans)) + 
  geom_line(data = methmean_no2_me, 
            aes(x=no2_y1_2019v, y = methmeans),
            col="red", size=0.5, linetype="solid") +
  geom_ribbon(data = methmean_no2_me, 
              aes(x = no2_y1_2019v, 
                  ymin = conf.low, ymax = conf.high), 
              alpha = .2) + 
  geom_point(alpha=0.8, col = "#254A90", size=1) + 
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.title = element_blank()) +
  scale_y_continuous(breaks = seq(-0.02, 0.06, by = 0.02), limits = c(-0.02,0.06))

ggsave(plot = methmean_plot, 
       filename = here("figures", 
                       "linear_regression", 
                       "postnatal", 
                       "2022-11-17_pn_global_dnam.png"),
       width = 4.0, height = 3.2, units = "cm")

# correlation 
cor(covariates_geno_ctpcs_tech_cc$methmeans, 
    covariates_geno_ctpcs_tech_cc$no2_y1_2019v,
    method="pearson") # -0.06392473

# effect size
summary(meanmeth_lm)$coefficients["no2_y1_2019v", "Estimate"]*range # -0.0002507753
```

#### LINE-1 specific DNAm changes

Use LINE1 DNAm as a surrogate for global DNAm.
```{r postnatal_line_dnam}

# read in line annotation from UCSC and repeat masker 
repeats_450k <- read.delim(file = gzfile(here("input_data", 
                                                "ucsc_rmsk_lines", 
                                                "ucsc_repeatmasker_enc_methyl450_gm12878.gz")),
                             header = T, skip = 1, sep = "\t")

# subset repeats for L1 LINE elements that are not in sex chromosomes
lines_450k  <- repeats_450k %>% 
  subset(rmsk.repClass == "LINE") %>%
  subset(rmsk.repFamily == "L1") %>%
  subset(!wgEncodeHaibMethyl450Gm12878SitesRep1.chrom %in% c("chrX", "chrY")) %>%
  subset(!is.na(wgEncodeHaibMethyl450Gm12878SitesRep1.chromStart))

# subset betas for those that overlap with line elements
diff_betas_geno_cc_cpr_lines <- diff_betas_geno_cc_cpr %>% 
  subset(rownames(.) %in% lines_450k$wgEncodeHaibMethyl450Gm12878SitesRep1.name)

# num of cpgs annotated to lines 
nrow(diff_betas_geno_cc_cpr_lines) # 7302

# get mean of line meth 
line_meth <- colMeans(diff_betas_geno_cc_cpr_lines)

# cbind line methylation to covariate data frame
covariates_geno_ctpcs_tech_cc <- cbind(covariates_geno_ctpcs_tech_cc, line_meth)

#colmeans
line_meth_lm <- lm(line_meth ~ no2_y1_2019v + sex + maternal_smoking_y1 + 
                     maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                     Comp.4 + Comp.5 + Comp.6 + genotyping.PC1 + 
                     genotyping.PC2 + genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + 
                     sv5 + sv6 + sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + 
                     sv14 + sv15, 
                   data=covariates_geno_ctpcs_tech_cc)

# summary of line regression
summary(line_meth_lm) # p = 0.252063    

# estimate marginal effects of prenatal NO2 exposure on DNAm
line_meth_no2_me <- ggpredict(line_meth_lm, terms = "no2_y1_2019v") %>%
  as.data.frame() %>% 
  dplyr::rename("line_meth" = predicted,
                "no2_y1_2019v" = x)

# plot 
line_dnam_plot <- ggplot(covariates_geno_ctpcs_tech_cc,
                         aes(x = no2_y1_2019v, y= line_meth)) + 
  geom_line(data = line_meth_no2_me, 
            aes(x=no2_y1_2019v, y = line_meth),
            col="red", size=0.5, linetype="solid") +
  geom_ribbon(data = line_meth_no2_me, 
              aes(x = no2_y1_2019v, 
                  ymin = conf.low, ymax = conf.high), 
              alpha = .2) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.title = element_blank()) +
  scale_y_continuous(breaks = seq(-0.02, 0.06, by = 0.02), limits = c(-0.02,0.06))

ggsave(plot = line_dnam_plot, 
       filename = here("figures", 
                       "linear_regression", 
                       "postnatal",
                       "2022-11-18_postnatal_line_dnam.png"),
       width = 4.0, height = 3.2, units = "cm")

# correlation 
cor(covariates_geno_ctpcs_tech_cc$line_meth, 
    covariates_geno_ctpcs_tech_cc$no2_y1_2019v,
    method="pearson") # -0.1189471

# effect size
summary(line_meth_lm)$coefficients["no2_y1_2019v", "Estimate"]*range # -0.0004671128
```


###  Variation in DNAm changes by birth month at top 10 CpGs

Try re-running limma with birth month/ interaction between birth month and air pollution.
```{r interaction_terms_to_assess_birthmonth}
#want to use anova to compare models usign anova
#anova only accepts models from lm
#use lm to compare how birth month affects cpgs above 10^-3 sig 

#You should be able to ask for confidence intervals for the slope coefficients for each month.  #even in a tabular display, you may see that smoothly varying pattern we were discussing.
#Then, you could use those to plot a sort of heres how the slope varies by month picture.Sort #of like forest plots for meta analysis results, if that makes sense.

#Also, before we decide it doesnt matter, the individual slopes might look fairly similar, but the appropriate overall test would be to check the anova(fit_with_interaction) and check the p-value on the interaction term.


##############
# cg19263548 #
##############

# add dnam to covariates for linear regression
covariates_geno_ctpcs_tech_cc$cg19263548 <- 
  diff_betas_geno_cc_cpr[rownames(diff_betas_geno_cc_cpr)=="cg19263548", ]

# fit wiht lm
cg19263548_lm_bm <- lm(cg19263548 ~ no2_y1_2019v*birth_month + sex + 
                         maternal_smoking_y1 + maternal_education_length + 
                         Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                         sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                         sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                       data=covariates_geno_ctpcs_tech_cc)

cg19263548_lm <- lm(cg19263548 ~ no2_y1_2019v + sex + 
                      maternal_smoking_y1 + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                      sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                    data=covariates_geno_ctpcs_tech_cc)


# anova
anova(cg19263548_lm, cg19263548_lm_bm, test = "LRT") # p = 0.9401
summary(cg19263548_lm_bm) 

# get plot of birth month effect with 95% CI
cg19263548_lm_bm_plot <- tidy(cg19263548_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_y1_2019v:birth_monthfebruary",
                                      "no2_y1_2019v:birth_monthmarch",
                                      "no2_y1_2019v:birth_monthapril",
                                      "no2_y1_2019v:birth_monthmay",
                                      "no2_y1_2019v:birth_monthjune",
                                      "no2_y1_2019v:birth_monthjuly",
                                      "no2_y1_2019v:birth_monthaugust",
                                      "no2_y1_2019v:birth_monthseptember",
                                      "no2_y1_2019v:birth_monthoctober",
                                      "no2_y1_2019v:birth_monthnovember",
                                      "no2_y1_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-5, 5) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg19263548_lm_bm_plot,
       filename =  here("figures", 
                        "linear_regression",
                        "postnatal",
                        "2022-10-07_pn_cg19263548_birthmonth.png"),
       width = 4.8, height = 4, units = "cm")


##############
# cg04823720 #
##############

# add dnam to covariates for linear regression
covariates_geno_ctpcs_tech_cc$cg04823720 <- 
  diff_betas_geno_cc_cpr[rownames(diff_betas_geno_cc_cpr)=="cg04823720", ]

# fit wiht lm
cg04823720_lm_bm <- lm(cg04823720 ~ no2_y1_2019v*birth_month + sex + 
                         maternal_smoking_y1 + maternal_education_length + 
                         Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                         sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                         sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                       data=covariates_geno_ctpcs_tech_cc)

cg04823720_lm <- lm(cg04823720 ~ no2_y1_2019v + sex + 
                      maternal_smoking_y1 + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                      sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                    data=covariates_geno_ctpcs_tech_cc)


# anova
anova(cg04823720_lm, cg04823720_lm_bm, test = "LRT") # p = 0.5323
summary(cg04823720_lm_bm) 

# get plot of birth month effect with 95% CI
cg04823720_lm_bm_plot <- tidy(cg04823720_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_y1_2019v:birth_monthfebruary",
                                      "no2_y1_2019v:birth_monthmarch",
                                      "no2_y1_2019v:birth_monthapril",
                                      "no2_y1_2019v:birth_monthmay",
                                      "no2_y1_2019v:birth_monthjune",
                                      "no2_y1_2019v:birth_monthjuly",
                                      "no2_y1_2019v:birth_monthaugust",
                                      "no2_y1_2019v:birth_monthseptember",
                                      "no2_y1_2019v:birth_monthoctober",
                                      "no2_y1_2019v:birth_monthnovember",
                                      "no2_y1_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-5, 5) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg04823720_lm_bm_plot,
       filename =  here("figures", 
                        "linear_regression",
                        "postnatal",
                        "2022-10-07_pn_cg04823720_birthmonth.png"),
       width = 4.8, height = 4, units = "cm")


##############
# cg08906030 #
##############

# add dnam to covariates for linear regression
covariates_geno_ctpcs_tech_cc$cg08906030 <- 
  diff_betas_geno_cc_cpr[rownames(diff_betas_geno_cc_cpr)=="cg08906030", ]

# fit wiht lm
cg08906030_lm_bm <- lm(cg08906030 ~ no2_y1_2019v*birth_month + sex + 
                         maternal_smoking_y1 + maternal_education_length + 
                         Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                         sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                         sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                       data=covariates_geno_ctpcs_tech_cc)

cg08906030_lm <- lm(cg08906030 ~ no2_y1_2019v + sex + 
                      maternal_smoking_y1 + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                      sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                    data=covariates_geno_ctpcs_tech_cc)


# anova
anova(cg08906030_lm, cg08906030_lm_bm, test = "LRT") # p = 0.7931
summary(cg08906030_lm_bm) 

# get plot of birth month effect with 95% CI
cg08906030_lm_bm_plot <- tidy(cg08906030_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_y1_2019v:birth_monthfebruary",
                                      "no2_y1_2019v:birth_monthmarch",
                                      "no2_y1_2019v:birth_monthapril",
                                      "no2_y1_2019v:birth_monthmay",
                                      "no2_y1_2019v:birth_monthjune",
                                      "no2_y1_2019v:birth_monthjuly",
                                      "no2_y1_2019v:birth_monthaugust",
                                      "no2_y1_2019v:birth_monthseptember",
                                      "no2_y1_2019v:birth_monthoctober",
                                      "no2_y1_2019v:birth_monthnovember",
                                      "no2_y1_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-5, 5) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg08906030_lm_bm_plot,
       filename =  here("figures", 
                        "linear_regression",
                        "postnatal",
                        "2022-10-07_pn_cg08906030_birthmonth.png"),
       width = 4.8, height = 4, units = "cm")


##############
# cg26989646 #
##############

# add dnam to covariates for linear regression
covariates_geno_ctpcs_tech_cc$cg26989646 <- 
  diff_betas_geno_cc_cpr[rownames(diff_betas_geno_cc_cpr)=="cg26989646", ]

# fit wiht lm
cg26989646_lm_bm <- lm(cg26989646 ~ no2_y1_2019v*birth_month + sex + 
                         maternal_smoking_y1 + maternal_education_length + 
                         Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                         sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                         sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                       data=covariates_geno_ctpcs_tech_cc)

cg26989646_lm <- lm(cg26989646 ~ no2_y1_2019v + sex + 
                      maternal_smoking_y1 + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                      sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                    data=covariates_geno_ctpcs_tech_cc)


# anova
anova(cg26989646_lm, cg26989646_lm_bm) # p = 0.9509
summary(cg26989646_lm_bm) 

# get plot of birth month effect with 95% CI
cg26989646_lm_bm_plot <- tidy(cg26989646_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_y1_2019v:birth_monthfebruary",
                                      "no2_y1_2019v:birth_monthmarch",
                                      "no2_y1_2019v:birth_monthapril",
                                      "no2_y1_2019v:birth_monthmay",
                                      "no2_y1_2019v:birth_monthjune",
                                      "no2_y1_2019v:birth_monthjuly",
                                      "no2_y1_2019v:birth_monthaugust",
                                      "no2_y1_2019v:birth_monthseptember",
                                      "no2_y1_2019v:birth_monthoctober",
                                      "no2_y1_2019v:birth_monthnovember",
                                      "no2_y1_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-5, 5) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg26989646_lm_bm_plot,
       filename =  here("figures", 
                        "linear_regression",
                        "postnatal",
                        "2022-10-07_pn_cg26989646_birthmonth.png"),
       width = 4.8, height = 4, units = "cm")


##############
# cg07472943 #
##############

# add dnam to covariates for linear regression
covariates_geno_ctpcs_tech_cc$cg07472943 <- 
  diff_betas_geno_cc_cpr[rownames(diff_betas_geno_cc_cpr)=="cg07472943", ]

# fit wiht lm
cg07472943_lm_bm <- lm(cg07472943 ~ no2_y1_2019v*birth_month + sex + 
                         maternal_smoking_y1 + maternal_education_length + 
                         Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                         sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                         sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                       data=covariates_geno_ctpcs_tech_cc)

cg07472943_lm <- lm(cg07472943 ~ no2_y1_2019v + sex + 
                      maternal_smoking_y1 + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                      sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                    data=covariates_geno_ctpcs_tech_cc)


# anova
anova(cg07472943_lm, cg07472943_lm_bm) # p = 0.6902
summary(cg07472943_lm_bm) 

# get plot of birth month effect with 95% CI
cg07472943_lm_bm_plot <- tidy(cg07472943_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_y1_2019v:birth_monthfebruary",
                                      "no2_y1_2019v:birth_monthmarch",
                                      "no2_y1_2019v:birth_monthapril",
                                      "no2_y1_2019v:birth_monthmay",
                                      "no2_y1_2019v:birth_monthjune",
                                      "no2_y1_2019v:birth_monthjuly",
                                      "no2_y1_2019v:birth_monthaugust",
                                      "no2_y1_2019v:birth_monthseptember",
                                      "no2_y1_2019v:birth_monthoctober",
                                      "no2_y1_2019v:birth_monthnovember",
                                      "no2_y1_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-5, 5) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg07472943_lm_bm_plot,
       filename =  here("figures", 
                        "linear_regression",
                        "postnatal",
                        "2022-10-07_pn_cg07472943_birthmonth.png"),
       width = 4.8, height = 4, units = "cm")



##############
# cg25530908 #
##############

# add dnam to covariates for linear regression
covariates_geno_ctpcs_tech_cc$cg25530908 <- 
  diff_betas_geno_cc_cpr[rownames(diff_betas_geno_cc_cpr)=="cg25530908", ]

# fit wiht lm
cg25530908_lm_bm <- lm(cg25530908 ~ no2_y1_2019v*birth_month + sex + 
                         maternal_smoking_y1 + maternal_education_length + 
                         Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                         sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                         sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                       data=covariates_geno_ctpcs_tech_cc)

cg25530908_lm <- lm(cg25530908 ~ no2_y1_2019v + sex + 
                      maternal_smoking_y1 + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                      sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                    data=covariates_geno_ctpcs_tech_cc)


# anova
anova(cg25530908_lm, cg25530908_lm_bm) # p = 0.2087
summary(cg25530908_lm_bm) 

# get plot of birth month effect with 95% CI
cg25530908_lm_bm_plot <- tidy(cg25530908_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_y1_2019v:birth_monthfebruary",
                                      "no2_y1_2019v:birth_monthmarch",
                                      "no2_y1_2019v:birth_monthapril",
                                      "no2_y1_2019v:birth_monthmay",
                                      "no2_y1_2019v:birth_monthjune",
                                      "no2_y1_2019v:birth_monthjuly",
                                      "no2_y1_2019v:birth_monthaugust",
                                      "no2_y1_2019v:birth_monthseptember",
                                      "no2_y1_2019v:birth_monthoctober",
                                      "no2_y1_2019v:birth_monthnovember",
                                      "no2_y1_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-5, 5) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg25530908_lm_bm_plot,
       filename =  here("figures", 
                        "linear_regression",
                        "postnatal",
                        "2022-10-07_pn_cg25530908_birthmonth.png"),
       width = 4.8, height = 4, units = "cm")


##############
# cg18792689 #
##############

# add dnam to covariates for linear regression
covariates_geno_ctpcs_tech_cc$cg18792689 <- 
  diff_betas_geno_cc_cpr[rownames(diff_betas_geno_cc_cpr)=="cg18792689", ]

# fit wiht lm
cg18792689_lm_bm <- lm(cg18792689 ~ no2_y1_2019v*birth_month + sex + 
                         maternal_smoking_y1 + maternal_education_length + 
                         Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                         sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                         sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                       data=covariates_geno_ctpcs_tech_cc)

cg18792689_lm <- lm(cg18792689 ~ no2_y1_2019v + sex + 
                      maternal_smoking_y1 + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                      sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                    data=covariates_geno_ctpcs_tech_cc)


# anova
anova(cg18792689_lm, cg18792689_lm_bm, test = "LRT") # p = 0.6678
summary(cg18792689_lm_bm) # mar (0.023729)

# get plot of birth month effect with 95% CI
cg18792689_lm_bm_plot <- tidy(cg18792689_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_y1_2019v:birth_monthfebruary",
                                      "no2_y1_2019v:birth_monthmarch",
                                      "no2_y1_2019v:birth_monthapril",
                                      "no2_y1_2019v:birth_monthmay",
                                      "no2_y1_2019v:birth_monthjune",
                                      "no2_y1_2019v:birth_monthjuly",
                                      "no2_y1_2019v:birth_monthaugust",
                                      "no2_y1_2019v:birth_monthseptember",
                                      "no2_y1_2019v:birth_monthoctober",
                                      "no2_y1_2019v:birth_monthnovember",
                                      "no2_y1_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-5, 5) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg18792689_lm_bm_plot,
       filename =  here("figures", 
                        "linear_regression",
                        "postnatal",
                        "2022-10-07_pn_cg18792689_birthmonth.png"),
       width = 4.8, height = 4, units = "cm")


##############
# cg15212210 #
##############

# add dnam to covariates for linear regression
covariates_geno_ctpcs_tech_cc$cg15212210 <- 
  diff_betas_geno_cc_cpr[rownames(diff_betas_geno_cc_cpr)=="cg15212210", ]

# fit wiht lm
cg15212210_lm_bm <- lm(cg15212210 ~ no2_y1_2019v*birth_month + sex + 
                         maternal_smoking_y1 + maternal_education_length + 
                         Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                         sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                         sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                       data=covariates_geno_ctpcs_tech_cc)

cg15212210_lm <- lm(cg15212210 ~ no2_y1_2019v + sex + 
                      maternal_smoking_y1 + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                      sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                    data=covariates_geno_ctpcs_tech_cc)


# anova
anova(cg15212210_lm, cg15212210_lm_bm, test = "LRT") # p = 0.4293
summary(cg15212210_lm_bm) 

# get plot of birth month effect with 95% CI
cg15212210_lm_bm_plot <- tidy(cg15212210_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_y1_2019v:birth_monthfebruary",
                                      "no2_y1_2019v:birth_monthmarch",
                                      "no2_y1_2019v:birth_monthapril",
                                      "no2_y1_2019v:birth_monthmay",
                                      "no2_y1_2019v:birth_monthjune",
                                      "no2_y1_2019v:birth_monthjuly",
                                      "no2_y1_2019v:birth_monthaugust",
                                      "no2_y1_2019v:birth_monthseptember",
                                      "no2_y1_2019v:birth_monthoctober",
                                      "no2_y1_2019v:birth_monthnovember",
                                      "no2_y1_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-10, 10) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg15212210_lm_bm_plot,
       filename =  here("figures", 
                        "linear_regression",
                        "postnatal",
                        "2022-10-07_pn_cg15212210_birthmonth.png"),
       width = 4.8, height = 4, units = "cm")


##############
# cg24353392 #
##############

# add dnam to covariates for linear regression
covariates_geno_ctpcs_tech_cc$cg24353392 <- 
  diff_betas_geno_cc_cpr[rownames(diff_betas_geno_cc_cpr)=="cg24353392", ]

# fit wiht lm
cg24353392_lm_bm <- lm(cg24353392 ~ no2_y1_2019v*birth_month + sex + 
                         maternal_smoking_y1 + maternal_education_length + 
                         Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                         sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                         sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                       data=covariates_geno_ctpcs_tech_cc)

cg24353392_lm <- lm(cg24353392 ~ no2_y1_2019v + sex + 
                      maternal_smoking_y1 + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                      sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                    data=covariates_geno_ctpcs_tech_cc)


# anova
anova(cg24353392_lm, cg24353392_lm_bm, test = "LRT") # p = 0.6804
summary(cg24353392_lm_bm) 

# get plot of birth month effect with 95% CI
cg24353392_lm_bm_plot <- tidy(cg24353392_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_y1_2019v:birth_monthfebruary",
                                      "no2_y1_2019v:birth_monthmarch",
                                      "no2_y1_2019v:birth_monthapril",
                                      "no2_y1_2019v:birth_monthmay",
                                      "no2_y1_2019v:birth_monthjune",
                                      "no2_y1_2019v:birth_monthjuly",
                                      "no2_y1_2019v:birth_monthaugust",
                                      "no2_y1_2019v:birth_monthseptember",
                                      "no2_y1_2019v:birth_monthoctober",
                                      "no2_y1_2019v:birth_monthnovember",
                                      "no2_y1_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-5, 5) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg24353392_lm_bm_plot,
       filename =  here("figures", 
                        "linear_regression",
                        "postnatal",
                        "2022-10-07_pn_cg24353392_birthmonth.png"),
       width = 4.8, height = 4, units = "cm")


##############
# cg06655349 #
##############

# add dnam to covariates for linear regression
covariates_geno_ctpcs_tech_cc$cg06655349 <- 
  diff_betas_geno_cc_cpr[rownames(diff_betas_geno_cc_cpr)=="cg06655349", ]

# fit wiht lm
cg06655349_lm_bm <- lm(cg06655349 ~ no2_y1_2019v*birth_month + sex + 
                         maternal_smoking_y1 + maternal_education_length + 
                         Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                         sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                         sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                       data=covariates_geno_ctpcs_tech_cc)

cg06655349_lm <- lm(cg06655349 ~ no2_y1_2019v + sex + 
                      maternal_smoking_y1 + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 + Comp.5 + Comp.6 + 
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + 
                      sv10 + sv11 + sv12 + sv13 + sv14 + sv15,
                    data=covariates_geno_ctpcs_tech_cc)


# anova
anova(cg06655349_lm, cg06655349_lm_bm) # p = 0.6721
summary(cg06655349_lm_bm) 

# get plot of birth month effect with 95% CI
cg06655349_lm_bm_plot <- tidy(cg06655349_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_y1_2019v:birth_monthfebruary",
                                      "no2_y1_2019v:birth_monthmarch",
                                      "no2_y1_2019v:birth_monthapril",
                                      "no2_y1_2019v:birth_monthmay",
                                      "no2_y1_2019v:birth_monthjune",
                                      "no2_y1_2019v:birth_monthjuly",
                                      "no2_y1_2019v:birth_monthaugust",
                                      "no2_y1_2019v:birth_monthseptember",
                                      "no2_y1_2019v:birth_monthoctober",
                                      "no2_y1_2019v:birth_monthnovember",
                                      "no2_y1_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-5, 5) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg06655349_lm_bm_plot,
       filename =  here("figures", 
                        "linear_regression",
                        "postnatal",
                        "2022-10-07_pn_cg06655349_birthmonth.png"),
       width = 4.8, height = 4, units = "cm")

```


### Table 1

"Table 1" of postnatal model
```{r postnatal_table1}
# table 1
table1(~no2_y1_2019v + sex + 
         maternal_smoking_y1 + maternal_education_length,
       data = covariates_geno_ctpcs_tech_cc)

# prenatal no2 quartiles
quantile(covariates_geno_ctpcs_tech_cc$no2_y1_2019v)
```




Boxplots of NO2, PM2.5
```{r airpollution_boxplots}

# prenatal no2
ggplot(covariates_geno_ctpcs_tech_cc, aes(y=no2_y1_entire, x=studycentre, fill=studycentre)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(height=0, width=0.25, alpha=0.8) +
  theme_bw() +
  scale_fill_manual(values=c("#254A90", "#508FC2", "#92C5EB", "white")) +
  labs(x="Study Centre", y=bquote("Year one" ~NO[2]~ "(ppb)")) +
  theme(panel.grid = element_blank(),
        legend.position = "none",
        axis.text = element_text(size=14),
        axis.title = element_text(size=16))

# year one no2
ggplot(covariates_geno_ctpc_tech_cc_y1_sub, aes(y=no2_y1_entire, x=studycentre,fill=studycentre)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(height=0, width=0.25, alpha=0.8) +
  scale_fill_manual(values=c("#254A90", "#508FC2", "#92C5EB", "white")) +
  labs(x="Study Centre", y=bquote("Year one" ~NO[2]~ "(ppb)")) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        legend.position = "none",
        axis.text = element_text(size=14),
        axis.title = element_text(size=16))

# Pm2.5 B
ggplot(covariates_geno_ctpc_tech, aes(y=PM25DALbYY_01, x=studycentre, fill=studycentre)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(height=0, width=0.25, alpha=0.8) +
  scale_fill_manual(values=c("#254A90", "#508FC2", "#92C5EB", "white")) +
  labs(x="Study Centre", y=bquote(~PM[2.5]~ "(ppb)")) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        legend.position = "none",
        axis.text = element_text(size=14),
        axis.title = element_text(size=16))
```

