---
title: "CHILD cord blood linear regression using limma with covariate missing values filled in with means"
author: "Samantha LEE"
date: "2021-09-24"
output: html_document
---

*Purpose*: The purpose of this script is to examine DNAm changes in cord blood associated with prenatal air pollution exposure (proxied by no2).


Set global chunk options. 
```{r setup, include=FALSE, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Libraries

Load required libraries.
```{r libraries}
library(tidyverse) # be tidy
library(limma) # for DNAm differences
library(here) # file calling
library(pbapply) # for running lm for each cpg
library(ggeffects) # for estimating marginal effects
library(broom) # tidy function for summarizing lm
library(sva) # for calculating SVs
library(ggcorrplot) # correlation matrix
library(bacon) # pvalue correction
library(ggsignif) # for significance comparison bars
```


## Analysis of prenatal air pollution exposure

### Load data

Load betas
```{r betas}
load(file=here("output_data", 
               "preprocessed_data",
               "2022-09-27_CHILD_preprocessed_betas_pdata_annotation.Rdata"))
```

Subset out cord blood betas
```{r nocombat_betas}
# check ranges to confirm DNAm data is in beta value format
dim(betas)
max(betas)
min(betas)

# subset pData into CBMCs (cord blood)
# these children have data at both time points
cb_pdata <- subset(pdata_nodups, Tissue == "C")
# check size of cord blood pdata
dim(cb_pdata)

# subset cordb lood DNAm samples for those included in cord blood pdata
cb_betas <- betas[ ,colnames(betas) %in% rownames(cb_pdata)]

# check size for correct sample number
dim(cb_betas) #144

# check that CBMC betas and pdata are in same order
identical(colnames(cb_betas), rownames(cb_pdata)) # TRUE

# relabel CBMC beta columns with sample label (5 digit identifier)
colnames(cb_betas) <- cb_pdata$Sample_Label
```

Load covariate data. Missing values in covariates of interest have been filled with the mean of that variable. 
```{r covariates}
load(here("output_data", 
          "covariates", 
          "2022-09-27_CHILD_covariates_na_with_means.Rdata"))

dim(covariates_nameans)
```

Subset covariates for individuals with CBMC DNAm.
```{r sub_covariates}
# subset covariates based on samples with DNAm data
covariates_sub <- covariates_nameans[as.character(covariates_nameans$sample_id) 
                                     %in% colnames(cb_betas), ]

# check size
dim(covariates_sub) # 144 120  

# check if subset of covariates and betas are in the same order 
identical(as.character(covariates_sub$sample_id), 
          colnames(cb_betas)) # TRUE
```

Add sample name to cord blood covariate data
```{r add_cb_sample_name}
identical(as.numeric(covariates_sub$sample_id),
          as.numeric(cb_pdata$Sample_Label)) # TRUE

covariates_sub$cb_sample_name <- rownames(cb_pdata)
```


Load CBMC PCs.
```{r load_PC_correctedbetas}
# load the cell type PCs
load(here("output_data", 
          "deconvolution", 
          "2022-09-28_CBMC_PCs.RData"))

# load deconvolution for later use - remove probes 
load(here("output_data",
          "deconvolution", 
          "2022-09-28_cbmc_pbmc_deconvolution.Rdata"))

# check size
dim(cb_ilr$scores) #144 6

# pull out pcs
cb_pcs <- cb_ilr$scores

# check if rownames of cell PCs are same as pdata
identical(rownames(cb_pcs), 
          rownames(cb_pdata)) # FALSE
# cb pcs do not have rownames

# rename rows using sample label (5 digit number)
rownames(cb_pcs) <- cb_pdata$Sample_Label
```

Load genotyping PCs and subset out REEGLE PCs. Genotyping data contains PCs for all CHILD participants that were genotyped.
```{r genotyping_PCS}
# read in genotyping data for CHILD cohort
genotyping <- read.csv(here("input_data", 
                            "genotyping", 
                            "CHILD_subjects_Qced_first10PCS_Duan.csv"))

# make rownames sample names (5 digit number)
rownames(genotyping) <- genotyping$FID
# removing sample naming columns - no longer needed now they are rownames 
genotyping <- subset(genotyping, select=-c(FID, IID))

# susbet genotyping data for REEGLE subset 
genosub <- genotyping[rownames(genotyping) %in% cb_pdata$Sample_Label,]

# check size
dim(genosub) # 131 of 144 children have genotyping

# rename columns of genotyping data for clarity
colnames(genosub) <- paste("genotyping.", colnames(genosub), sep="")

#put rows in numerical order according to sample id
genosub_order <- genosub[order(rownames(genosub)), ]
```

Examine genetic ancestry groups of participants based on genotyping PCs. Want to confirm that three PCs are required for genetic ancestry.
```{r genotyping_pc_groups, eval=FALSE}
# load genotyping info from Amirtha/Qingling 
# this info is for all CHILD participants that were genotyped
pca_info <- readxl::read_excel(here("input_data",
                                    "genotyping", 
                                    "PCA_ethnicity_child_comparison_duan_lab07102020.xlsx"))

# sub for individuals in this study (REEGLE subset)
pca_sub <- pca_info[as.character(pca_info$FID) %in% rownames(genosub),]

# plot and colour according to race specified on questionnaires
ggplot(pca_sub, aes(x=PC1, y=PC2, 
                    col=as.factor(race_mom), 
                    shape=as.factor(`PCA ethnicity`))) +
  geom_point(alpha=0.7, size=2) +
  scale_color_manual(labels = c("First Nations", "Mixed","South East Asian",
                                "East Asian","South Asian","Middle Eastern",
                                "Hispanic","Caucasian White"),
                     values=c("grey55","#4E2A7A" , "#B20D30","#C17817",
                              "#3F784C", "black", "#6DB1BF","#254A90"),
                     name ="Maternal race")  +
  scale_shape_manual(values=c(21,22,23,24),
                     labels=c("Admixed","Central European" , 
                              "East Asian","South Asian"),
                     name ="PCA Genetic Ancestry") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size=6),
        axis.text.y = element_text(size=6),
        axis.title.x = element_text(size=8),
        axis.title.y = element_text(size=8),
        legend.text=element_text(size=4),
        legend.title=element_text(size=5),
        legend.key.size = unit(0.5, 'cm'), #change legend key size
        legend.key.height = unit(0.3, 'cm'), #change legend key height
        legend.key.width = unit(0.5, 'cm')) 

#  ggsave( filename="2021-01-21_CHILD_genopcs_race_ancestry.tiff",
#          device = "tiff",
#          path =  here("figures", "genotyping"),
#          width = 3.5,
#          height = 2.5,
#          units = "in",
#          limitsize = TRUE)
```

Subset betas, pdata, covariates, and cell type PCS for participants with genotyping data (the limiting factor).
```{r subset}
# subset betas
cb_betas_geno <- cb_betas[, rownames(genosub_order)]
# check size of subbed betas
dim(cb_betas_geno) # 424644    131

# subset pdata
cb_pdata_geno <- cb_pdata[cb_pdata$Sample_Label %in% rownames(genosub_order), ]
# check size of subbed pdata
dim(cb_pdata_geno) # 131 17

# subset covariates
covariates_sub_geno <- covariates_sub[covariates_sub$sample_id %in% 
                                        rownames(genosub_order), ]
# check size of subbed covariates 
dim(covariates_sub_geno) # 131 120

# subset cell type PCs
cb_pcs_geno <- cb_pcs[rownames(genosub_order), ]
# check size of subbed cell type PCs
dim(cb_pcs_geno) # 131 5
```


Check that betas, pdata, covariates, cell type PCs are in same order as genotyping data for REEGLE participants.
```{r cb_batchcorr_order}
# pdata and genosub_order
identical(as.character(cb_pdata_geno$Sample_Label),
          rownames(genosub_order)) # TRUE

# betas and genosub_order
identical(colnames(cb_betas_geno), 
          rownames(genosub_order)) # TRUE

# covariates and genosub_order
identical(as.character(covariates_sub_geno$sample_id), 
          rownames(genosub_order)) # TRUE

# cell type PCs and genosub_order
identical(rownames(genosub_order), 
          rownames(cb_pcs_geno)) # TRUE
```


Combined covariates, genotyping, cell type PCs, and pData together.
```{r combine_cell_pcs_covariates}

###################################
# clean up pdata before combining #
###################################

# create new column for batch row variable in pdata
cb_pdata_geno$row <- as.numeric(as.factor(substr(cb_pdata_geno$Sentrix_Position,
                                                 1, 3)))
# examine
head(cb_pdata_geno$row)

# rename pdata sentrix Id column to chip
colnames(cb_pdata_geno)[8] <- "chip"

# rename pdata sentrix Id column to chip
colnames(cb_pdata_geno)[11] <- "run"

################################
# combine data frames together #
################################

# for pdata, only include batch variables
covariates_geno_ctpc_tech <- cbind(covariates_sub_geno, 
                                   cb_pcs_geno,
                                   genosub_order,
                                   cb_pdata_geno[,c("run", "row", "chip")])

dim(covariates_geno_ctpc_tech) # 131 138

# check classes to make sure they are what we expect before running linear model
lapply(covariates_geno_ctpc_tech, class)

# reclass run 
covariates_geno_ctpc_tech$run <- factor(covariates_geno_ctpc_tech$run, 
                                        ordered=F)
# reclass chip
covariates_geno_ctpc_tech$chip <- factor(covariates_geno_ctpc_tech$chip,
                                         ordered=F)
# reclass sex
covariates_geno_ctpc_tech$sex <- factor(covariates_geno_ctpc_tech$sex, 
                                        ordered=F)
```


Check 
```{r no_zeroes_in_no2}
# newest no2 variable
sum(na.omit(covariates_geno_ctpc_tech$no2_prenatal_avg_2019v)==0)
# old no2 variable
sum(na.omit(covariates_geno_ctpc_tech$no2_prenatal_avg_2015v)==0) # 6

# check correlation between two variables
ggplot(covariates_geno_ctpc_tech, aes(x= no2_prenatal_avg_2019v,
                                      y= no2_prenatal_avg_2015v)) +
  geom_point()
```


### Subset for complete cases

Subset complete cases in covariate data and betas in preparation for DNAm analysis.
```{r complete_cases}
# subset complete cases
covariates_geno_ctpc_tech_cc <- covariates_geno_ctpc_tech %>%
  subset(complete.cases(no2_prenatal_avg_2019v, sex, gestational_age,
                        prenatal_maternal_smoke, maternal_education_length,
                        Comp.1, Comp.2, Comp.3, Comp.4, 
                        genotyping.PC1, genotyping.PC2, genotyping.PC3))

# check number of complete cases
dim(covariates_geno_ctpc_tech_cc) # 128 139

# subset beta values for complete cases
cb_betas_geno_cc <- cb_betas_geno[,colnames(cb_betas_geno) %in% 
                                    covariates_geno_ctpc_tech_cc$sample_id]

# check size of complete case betas
dim(cb_betas_geno_cc) # 424644 128

# remove probes from betas that were used to predict cell type
cb_betas_geno_cc_clean <- 
  cb_betas_geno_cc[!rownames(cb_betas_geno_cc) %in%
                     (child_fscbc_ecc2_cb$probelist_all %>% unlist() %>% unname() %>% unique()), ]

# check size
dim(cb_betas_geno_cc_clean) # 424079 128; 565 probes removed
```

### Model matrix

Create model matrix for DNAm analysis. This model matrix does not contain batch variables as we are going to let surrogate variables capture their variance within the DNAm data. 
```{r model_matrix}
# create model matrix
mod <- model.matrix(~ no2_prenatal_avg_2019v + sex + gestational_age +
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3,
                    data=covariates_geno_ctpc_tech_cc)

# check size
dim(mod) # 128 15
```

### Surrogate variable analysis (SVA) to account for technical variation

Surrogate variable analysis using the sva packages. This will allow us to capture variance due to batch effects, as well as variance due to unkown biological variables. 
```{r sva}

###############
# null matrix #
###############

# create null matrix
# this matrix does not include any of our covariates 
# this means that our covariates are "protected" and SVs will not capture them
mod0 <- model.matrix(~ sex + gestational_age + prenatal_maternal_smoke + 
                       maternal_education_length + 
                       Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                       genotyping.PC1 + genotyping.PC2 + genotyping.PC3,  
                     data=covariates_geno_ctpc_tech_cc)

################
# sva analysis #
################

# to get SVs in a way that allows calculation of their variance
# we need to pull the code out of sva and run it line-by-line
# we are using the "Be" method for calculating SVs

######################################
# irwsva required internal functions #
######################################

# edge.lfdr function is internal to irwsva function below
# mono function is internal to edge.lfdr
# they have to be initialized before we can run the rest of the code

set.seed(1234)

mono <- function(lfdr){
  .Call("monotone", as.numeric(lfdr), PACKAGE="sva")
}

edge.lfdr <- function(p, trunc=TRUE, monotone=TRUE, transf=c("probit", "logit"), adj=1.5, eps=10^-8, lambda=0.8, ...) {
  pi0 <- mean(p >= lambda)/(1 - lambda)
  pi0 <- min(pi0, 1)
  n <- length(p)
  transf <- match.arg(transf)
  if(transf=="probit") {
    p <- pmax(p, eps)
    p <- pmin(p, 1-eps)
    x <- qnorm(p)
    myd <- density(x, adjust=adj)
    mys <- smooth.spline(x=myd$x, y=myd$y)
    y <- predict(mys, x)$y
    lfdr <- pi0*dnorm(x)/y
  }
  if(transf=="logit") {
    x <- log((p+eps)/(1-p+eps))
    myd <- density(x, adjust=adj)
    mys <- smooth.spline(x=myd$x, y=myd$y)
    y <- predict(mys, x)$y
    dx <- exp(x) / (1+exp(x))^2
    lfdr <- pi0 * dx/y
  }
  if(trunc) {
    lfdr[lfdr > 1] <- 1
  }
  if(monotone) {	
    lfdr <- lfdr[order(p)]
    lfdr <- mono(lfdr)
    lfdr <- lfdr[rank(p)]
  }
  return(lfdr)
}



##################################
# numsv function internal to sva #
##################################

#determine the number of required SVs
#this is numsv function code pulled out

#this is what is fed into this function from the sva function
dat = cb_betas_geno_cc_clean
mod = mod
#method="be" # this is the default and what is called by numsv
#vfilter=NULL # we dont apply a filter and its not used for be method
B=20
set.seed(1234)

# initialize some stuff
# get the residual matrix (the remaining variance)
n <- ncol(dat)
m <- nrow(dat)
H <- mod %*% solve(t(mod) %*% mod) %*% t(mod)
res <- dat - t(H %*% t(dat))

# run svd and get some eigenvalues
uu <- svd(res)
# get the num of "sig" eigenvalues
ndf <- min(m, n) - ceiling(sum(diag(H)))
# get percent variance explained by each eigenvalue
dstat <- uu$d[1:ndf]^2/sum(uu$d[1:ndf]^2)
# initialize null distribution matrix
dstat0 <- matrix(0, nrow = B, ncol = ndf)

# set seed here since we are calling sample
# create the null distribution
for (i in 1:B) {
  # this part of the code removes all relationships in the 
  res0 <- t(apply(res, 1, sample, replace = FALSE))
  res0 <- res0 - t(H %*% t(res0))
  # calculate svs on this
  uu0 <- svd(res0)
  # bind their explained variance in null matrix
  dstat0[i, ] <- uu0$d[1:ndf]^2/sum(uu0$d[1:ndf]^2)
  print(i) # print out the iteration we are on
}

# determine the number of svs required here based on pvalues
psv <- rep(1, n)
for (i in 1:ndf) {
  psv[i] <- mean(dstat0[, i] >= dstat[i])
}
for (i in 2:ndf) {
  psv[i] <- max(psv[(i - 1)], psv[i])
}

# this is the variable holding the number of SVs required 
# number is each to all the svs with a pval less than/equal to 0.1
nsv <- sum(psv <= 0.1)


####################################
# code from the main SVA function  #
####################################

n.sv = nsv # from code above (run internally when sva called)
B = 5 # number of iterations

# dat =  betas # already initialized
# mod = mod  # already initialized
# mod0 = mod0  # already initialized 
# controls = NULL # we dont include controls - only for supervised method
# method = "irw" # this isnt actually used but would be called by sva
# vfilter = NULL # we dont set this when running sva
# numSVmethod = "be" # this is the default called by sva (code above)


###################################
# irwsva function internal to sva #
###################################

# this code is really want does all the work in sva
# we use the irw method so this is the code for that method
# there are also supervised and two step methods (not used by us)

# irw function takes input from sva, so we have already defined below vars
# dat = dat 
# mod = mod
# mod0 = mod0
# n.sv = n.sv
# B = B

# intialize variables again
n <- ncol(dat)
m <- nrow(dat)
Id <- diag(n)
resid <- dat %*% (Id - mod %*% solve(t(mod) %*% mod) %*% t(mod))
uu <- eigen(t(resid) %*% resid)
vv <- uu$vectors
ndf <- n - dim(mod)[2]
pprob <- rep(1, m)
one <- rep(1, n)
Id <- diag(n)
df1 <- dim(mod)[2] + n.sv
df0 <- dim(mod0)[2] + n.sv
rm(resid)

cat(paste("Iteration (out of", B, "):"))
for (i in 1:B) {
  mod.b <- cbind(mod, uu$vectors[, 1:n.sv])
  mod0.b <- cbind(mod0, uu$vectors[, 1:n.sv])
  ptmp <- f.pvalue(dat, mod.b, mod0.b)
  pprob.b <- (1 - edge.lfdr(ptmp))
  mod.gam <- cbind(mod0, uu$vectors[, 1:n.sv])
  mod0.gam <- cbind(mod0)
  ptmp <- f.pvalue(dat, mod.gam, mod0.gam)
  pprob.gam <- (1 - edge.lfdr(ptmp))
  pprob <- pprob.gam * (1 - pprob.b)
  dats <- dat * pprob
  dats <- dats - rowMeans(dats)
  uu <- eigen(t(dats) %*% dats)
  cat(paste(i, " "))
}

# this is what is normally returned by sva
# problem is that it doesnt include the diagonal matrix we want
sv_cb = svd(dats)$v[, 1:n.sv, drop = FALSE] 
svobj <- list(sv = sv_cb, pprob.gam = pprob.gam, pprob.b = pprob.b, 
              n.sv = n.sv)

# to get all info (including diagonal)
svobj_all <- svd(dats)

svs_cb <- as.data.frame(svobj$sv)

# save 
save(sv_cb, file = here("output_data", 
                        "linear_regression", 
                        "2022-10-05_prenatal_svs_without_pss18wk.RData"))

###############
#scree of SVs #
###############

# from elodie portales-casamar

# heat scree prepartion
vars <- svobj_all[['d']]^2
vars <- vars/sum(vars)
names(vars) <- paste0('SV', 1:length(vars))
xp <- 1:svobj$n.sv
varsdf <- as.data.frame(vars)
varsdf$svnum <- rep(1:length(vars))

# heat scree ggplot!! 
cb_sv_scree <- ggplot(varsdf[1:20,], aes(x=svnum, y=vars*100)) +
  geom_col() +
  theme_bw() +
  labs(x="Surrogate variable", y = "Variance (%)")

cb_sv_scree

ggsave(plot = cb_sv_scree, 
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal", 
                       "2022-10-05_cb_sv_heat_scree.png"),
       height = 5, width = 15, units = "cm")

# it looks like the first two svs are most important, but SVA suggests 14
# examine a correlation matrix between all svs and covariates to see
# what variance they are capturing

######################
# correlation matrix #
######################

# add svs to covariates
covariates_geno_ctpc_tech_cc <- cbind(covariates_geno_ctpc_tech_cc, sv_cb)

# rename columns with svs
colnames(covariates_geno_ctpc_tech_cc)[141:157]  <-
  paste("sv",colnames(covariates_geno_ctpc_tech_cc)[141:157], sep="")

source(here("scripts", "2021-08-30_corr_mat_funs.R"))

# early exp and pn_18 are the same thing. 
# exclude early_exp from correlation matrices
corr <- corr.mat(covariates_geno_ctpc_tech_cc %>% 
                   dplyr::select(no2_prenatal_avg_2019v, sex, gestational_age,
                                 prenatal_maternal_smoke, maternal_education_length,
                                 Comp.1, Comp.2, Comp.3, Comp.4, 
                                 genotyping.PC1, genotyping.PC2, genotyping.PC3,
                                 chip, run, row, 
                                 sv1, sv2, sv3, sv4, sv5, sv6, sv7, sv8, sv9,
                                 sv10, sv11, sv12, sv13, sv14, sv15, sv16, sv17) %>%
                   rename("Prenatal NO2" = no2_prenatal_avg_2019v,
                          "Biological sex" = sex,
                          "Gestational age (days)" = gestational_age,
                          "Prenatal maternal cigarette smoking" = prenatal_maternal_smoke,
                          "Maternal education (years)" = maternal_education_length,
                          #"Prenatal maternal stress" = pss_pre18wk,
                          "Cell type PC1" = Comp.1, "Cell type PC2" = Comp.2,
                          "Cell type PC3" = Comp.3, "Cell type PC4" = Comp.4,
                          "Genetic ancestry PC1" = genotyping.PC1, 
                          "Genetic ancestry PC2" = genotyping.PC2, 
                          "Genetic ancestry PC3" = genotyping.PC3, 
                          "Chip" = chip, "Run" = run, "Row" = row,
                          "SV1" = sv1, "SV2" = sv2, "SV3" = sv3, "SV4" = sv4,
                          "SV5" = sv5, "SV6" = sv6, "SV7" = sv7, "SV8" = sv8,
                          "SV9" = sv9, "SV10" = sv10, "SV11" = sv11, 
                          "SV12" = sv12,"SV13" = sv13, "SV14" = sv14, 
                          "SV15" = sv15, "SV16" = sv16, "SV17" = sv17))

pval <- pval.mat(covariates_geno_ctpc_tech_cc %>% 
                   dplyr::select(no2_prenatal_avg_2019v, sex, gestational_age,
                                 prenatal_maternal_smoke, maternal_education_length,
                                 Comp.1, Comp.2, Comp.3, Comp.4, 
                                 genotyping.PC1, genotyping.PC2, genotyping.PC3,
                                 chip, run, row, 
                                 sv1, sv2, sv3, sv4, sv5, sv6, sv7, sv8, sv9,
                                 sv10, sv11, sv12, sv13, sv14, sv15, sv16,sv17) %>%
                   rename("Prenatal NO2" = no2_prenatal_avg_2019v,
                          "Biological sex" = sex,
                          "Gestational age (days)" = gestational_age,
                          "Prenatal maternal cigarette smoking" = prenatal_maternal_smoke,
                          "Maternal education (years)" = maternal_education_length,
                          #"Prenatal maternal stress" = pss_pre18wk,
                          "Cell type PC1" = Comp.1, "Cell type PC2" = Comp.2,
                          "Cell type PC3" = Comp.3, "Cell type PC4" = Comp.4,
                          "Genetic ancestry PC1" = genotyping.PC1, 
                          "Genetic ancestry PC2" = genotyping.PC2, 
                          "Genetic ancestry PC3" = genotyping.PC3, 
                          "Chip" = chip, "Run" = run, "Row" = row,
                          "SV1" = sv1, "SV2" = sv2, "SV3" = sv3, "SV4" = sv4,
                          "SV5" = sv5, "SV6" = sv6, "SV7" = sv7, "SV8" = sv8,
                          "SV9" = sv9, "SV10" = sv10, "SV11" = sv11, 
                          "SV12" = sv12,"SV13" = sv13, "SV14" = sv14, 
                          "SV15" = sv15, "SV16" = sv16,"SV17" = sv17))

svcorr <- corr.plot(pval = pval, 
                    corr = corr, 
                    label.sig = T,
                    shape = 8,shape.size = 1,
                    axis.text.size = 8, x.axis.text.angle = 90) +
  scale_y_discrete(position = "right") +
  theme(legend.position = "none",
        panel.grid = element_blank())

ggsave(plot = svcorr,
       filename = here("figures",
                       "linear_regression",
                       "prenatal", 
                       "2022-10-05_sv_correlations.png"),
       height = 15, width = 15, units = "cm")

###########################
# add SVs to model matrix #
###########################

colnames(sv_cb) <- c(1:17)

# use 6 SVs 
modSv = cbind(mod,sv_cb)
```

### EWAS examining the effects of prenatal NO2 exposure 

Run Limma for DNAm analysis.
```{r limma}
set.seed(1234)

# fit methylation to no2
cb_fit <-  lmFit(cb_betas_geno_cc_clean, modSv)
cb_ebayes <- eBayes(cb_fit)

# subset annotation file for cpgs used in limma
annotation_cpgs <- annotation[rownames(annotation) %in% rownames(cb_ebayes),]

# check that annotation file and limma cpgs are in the same order
identical(rownames(annotation_cpgs), rownames(cb_ebayes)) # false

# reorder annotation file to match limma output
annotation_cpgs_order <- annotation_cpgs[rownames(cb_ebayes), ]

# recheck order of annotation file against limma cpgs
identical(rownames(annotation_cpgs_order), rownames(cb_ebayes)) #true

# pull out pval, coefficients from limma and combine with annotation
cb_pvals_tech <- cbind(as.data.frame(cb_ebayes$p.value), 
                       as.data.frame(cb_ebayes$coefficients),
                       as.data.frame(annotation_cpgs_order))

# rename columns to indicate pvalues vs coeffcients 
colnames(cb_pvals_tech) <- c(paste("pval", 
                                   colnames(cb_pvals_tech)[1:30], 
                                   sep = "_"), 
                             paste("coeff", 
                                   colnames(cb_pvals_tech)[31:60], 
                                   sep = "_"),
                             colnames(cb_pvals_tech)[61:93])

cb_pvals_tech$pval_no2_prenatal_avg_2019v_adj  <-
  p.adjust(cb_pvals_tech$pval_no2_prenatal_avg_2019v , method="BH")
```

Examine the qqplot.
```{r qqplot_pval_histo}

pvector <- cb_pvals_tech$pval_no2_prenatal_avg_2019v

pvector2 <- pvector[!is.na(pvector) & !is.nan(pvector) & !is.null(pvector) & 
                      is.finite(pvector) & pvector < 1 & pvector > 0]

pval_qq <- as.data.frame(cbind(o = -log10(sort(pvector2, decreasing = FALSE)),
                               e = -log10(ppoints(length(pvector2)))))

# calculate lambda
z = qnorm(pvector2/2)
lambda = round(median(z^2) / 0.454, 3)


# qqplot of bacon corrected pvalues for main paper figure
qqplot <-  ggplot(pval_qq, aes(x=e, y=o)) + 
  geom_point(col="#254A90", size=1) + 
  geom_abline(slope = 1, intercept = 0, 
              col = "red", size=0.25, linetype="dashed") +
  labs(x = "Expected -log10(p-value)",
       y = "Observed -log10(p-value)") +
  annotate("text", x=3, y=1.5, 
           label = paste("Inflation = ", signif(lambda, 2)), 
           size=3, hjust=0) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        axis.title = element_blank())
qqplot

ggsave(plot = qqplot, 
       filename= here("figures", 
                      "linear_regression", 
                      "prenatal", 
                      "2022-10-18_prenatal_no2_qqplot.png"),
       width = 7.5, height = 6, units = "cm")
```

Effect size
```{r effect_size}
range = max(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v)

cb_pvals_tech$effect_size <- cb_pvals_tech$coeff_no2_prenatal_avg_2019v * range
```

Volcano plot
```{r volcano}

persist_points <- cb_pvals_tech %>% 
  subset(effect_size > 0.025 | effect_size < -0.025) %>%
  arrange(pval_no2_prenatal_avg_2019v) %>%
  slice_head(n=100) %>% 
  rownames()

cb_volcano <- ggplot(cb_pvals_tech %>%
                       subset(!rownames(.) %in% persist_points), 
                     aes(x=as.numeric(effect_size) , 
                         y=-log10(as.numeric(pval_no2_prenatal_avg_2019v)))) +
  geom_point(alpha=0.75, col = "#254A90", size=1) +
  # colour significant cpgs; 2.5% effect size cut off and FDR < 0.25
  geom_point(data = cb_pvals_tech %>% 
               subset(rownames(.) %in% persist_points), 
             alpha=0.75, col = "#254A90", size=1, shape = 25, fill = "#254A90") +  #"#254A90", "#508FC2", "#92C5EB"
  scale_x_continuous(breaks = seq(-0.50, 0.50, by = 0.10)) +
  
  labs(y="-log10(unadjusted p-value)", x = "Effect size (% DNAm)") +
  theme_bw() +
  theme(panel.grid = element_blank(),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        axis.title = element_blank()) 
cb_volcano

ggsave(plot = cb_volcano,
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal", 
                       "2022-10-18_cb_volcanoplot.png"),
       width = 7.5, height = 6, units = "cm")
```

### Gene specific changes

These genes are related to the proposed pathway of air pollution action or have previously been shown to exhibit DNA methylation changes associated with prenatal air pollution exposure. 

```{r gene_specific}
#FHC is FTH1 (ferritin heavy chain 1)
#nothing maps to FOXP3
#TNFA is TNF
goi <- annotation[grep("^RNF39|^CYP2E1|^LONP1|^SLC25A28|^PLVAP|^CAT$|^CAT;|^TPO|^NFKB1|^NOX2|^SOD1|^SOD2|^TRX1|^TRX2|^GSTP1|^MT3|^NQO1|^HMOX1|^TGPX1|^DDH1|^NOS2|^CYP2C11|^CYP7B|^ALOX12$|^ALOX5;|^ALOX5$|^NRF2$|^NRF2;|^KEAP1|^AHR$|^AHR;|^ARNT;|^ARNT$|^GCLC|^GCLM|^GSR1|^SLC7A11|^GSTA1|^GSTA2|^GSTA3|^GSTA5|^GSTM1|^GSTM2|^GSTM3|^TXN$|^TXN;|^TXNRD1$|^TXNRD1;|^SRXN1|^G6PD|^CYP1A1|^IL1A|^IL2$|^IL2;|^IL3$|^IL3;|^IL4;|^IL4$|^IL5$|^IL5;|^IL6$|^IL6;|^IL9$|^IL9;|^IL10$|^IL10;|^IL12A|^IL13|^IL15$|^IL15;|^IL17A|^IL18$|^IL18;|^IL19|^IL21$|^IL21;|^IL22$|^IL22;|^IL23A$|^IL23A;|^IL25$|^IL25;|^IL27$|^IL27;|^IL31$|^IL31;|^IL33$|^IL33;|^IFNG$|^IFNG;|^TNF$|^TNF;|^TGFB1$|^TGFB1;|^FOXP3|^FTH1", annotation$UCSC_RefGene_Name), ] %>% as.data.frame()

unique(goi$UCSC_RefGene_Name)
# only first annotate gene
unique(gsub(";.*$", "", unique(goi$UCSC_RefGene_Name))) # 59

# transpose for lm function
cb_betas_geno_cc_clean_t <- t(cb_betas_geno_cc_clean)

#apply function for calculating significance of associations using mice and imputed covariate data
prenatal_goi_list <- 
  pbapply(cb_betas_geno_cc_clean_t[,colnames(cb_betas_geno_cc_clean_t ) %in% rownames(goi)], 2, function(x) {
    
    combined <- cbind(covariates_geno_ctpc_tech_cc, x)
    
    #run the linear regression on this cpg using all 5 imputed covariate data sets
    fit <- lm(x ~ no2_prenatal_avg_2019v + sex + gestational_age  + 
                prenatal_maternal_smoke + maternal_education_length + 
                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
              data=combined)
    
    #get summary of lm fit
    #this includes coefficients and pvalues
    #coerce into df for manipulation
    #add in terms (rownames) as column for annotation
    fit_sum <- summary(fit)
    fit_sum_coeff <- as.data.frame(fit_sum$coefficients)
    fit_sum_coeff$term <- rownames(fit_sum_coeff)
    
    #this data frame has rows for each linear regression variable, and columns for statistics
    #want a dataframe of 1 row and many columns (one for each variable with each statistic type)
    #do this by using the spread function for each statistic
    #add cpg id onto dataframe for pvalues (to keep track of information)
    #then cbind dataframes together
    
    #coefficient
    estimate <- spread(fit_sum_coeff[,c("term", "Estimate")], term, Estimate)
    colnames(estimate) <- paste("estimate", colnames(estimate), sep="_")
    
    #std error
    std.error <- spread(fit_sum_coeff[,c("term", "Std. Error")], term, 'Std. Error')
    colnames(std.error) <- paste("std_error", colnames(std.error), sep="_")
    
    #t statistics
    statistic <- spread(fit_sum_coeff[,c("term", "t value")], term, 't value')
    colnames(statistic) <- paste("t_val", colnames(statistic), sep="_")
    
    #pvalue
    p.value <- spread(fit_sum_coeff[,c("term", "Pr(>|t|)")], term, 'Pr(>|t|)')
    colnames(p.value) <- paste("p_value", colnames(p.value), sep="_")
    
    #add in cpgid for annotation 
    p.value$cpg <- colnames(x)
    
    #cbind into wide dataframe
    widepool <- cbind(estimate, std.error, statistic, p.value)
    
    #return the wide dataframe
    #these will be do.call(rbind()) together after
    return(widepool)
    
  })


prenatal_goi <- do.call(rbind, prenatal_goi_list)

prenatal_goi$p_value_no2_prenatal_avg_2019v_adj <- 
  p.adjust(prenatal_goi$p_value_no2_prenatal_avg_2019v, method="BH")

#reorder according ot unadjusted no2 pvalue
prenatal_goi_order <- prenatal_goi[order(prenatal_goi$p_value_no2_prenatal_avg_2019v),]

#reorder goi annotation to be in same order as goi stats
goi_order <- goi[rownames(prenatal_goi_order),]

#check identical
identical(rownames(goi_order), rownames(prenatal_goi_order)) #true

#bind
goi_df <- cbind(prenatal_goi_order, goi_order)

#sort goi df by pvalue
goi_df_pval <- goi_df %>% arrange(p_value_no2_prenatal_avg_2019v)

# add effect size
goi_df_pval_effect_size <- merge(goi_df_pval, cb_pvals_tech, by="row.names", all.x=T) 

# write out table for all GOIs
write.csv(goi_df_pval_effect_size, 
          file=here("tables", 
                    "prenatal",
                    "2022-10-05_goi_prenatal_cpgs.csv"))

# write table of top ten 
write.csv(goi_df_pval_effect_size %>% 
            subset(effect_size > 0.025 | effect_size < -0.025) %>% 
            subset(-log10(p_value_no2_prenatal_avg_2019v) > 1.4) %>%
            arrange(p_value_no2_prenatal_avg_2019v),
          here("tables", 
               "prenatal", 
               "2022-10-05_top10goi_prenatal_cpgs.csv"))

# use top table to get 95% CI
goi_df_pval_db_sig <- goi_df_pval_effect_size %>% 
  subset(effect_size > 0.025 | effect_size < -0.025) %>% 
  subset(-log10(pval_no2_prenatal_avg_2019v) > 1.4)

write.csv(goi_df_pval_db_sig, here("tables", 
                                   "prenatal", 
                                   "2022-10-05_toptable_prenatal_goi.csv"))

#check cpgs from gruvieva 2017 paper

##################
#LONP1 cg12283362#
##################

covariates_geno_ctpc_tech_cc$cg12283362 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg12283362", ] 

cg12283362_lm <- lm(cg12283362 ~ no2_prenatal_avg_2019v + sex + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                    data=covariates_geno_ctpc_tech_cc)

# get lm info
summary(cg12283362_lm)

# estimate marginal effects of prenatal NO2 exposure on DNAm
cg12283362_no2_me <- ggpredict(cg12283362_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("cg12283362" = predicted,
         "no2_prenatal_avg_2019v" = x)

# plot 
cb_cg12283362_dnam_plot <- ggplot(covariates_geno_ctpc_tech_cc,
                                  aes(x = no2_prenatal_avg_2019v, 
                                      y= cg12283362)) + 
  geom_line(data = cg12283362_no2_me, 
            aes(x=no2_prenatal_avg_2019v, y = cg12283362),
            col="red", size=0.5, linetype="solid") +
  geom_ribbon(data = cg12283362_no2_me, 
              aes(x = no2_prenatal_avg_2019v, 
                  ymin = conf.low, ymax = conf.high), 
              alpha = .2) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  scale_y_continuous(breaks = seq(0.6,0.9,by=0.1), 
                     limits = c(0.6,0.9),
                     labels = scales::number_format(accuracy = 0.01)) +
  
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.title = element_blank()) 

ggsave(plot = cb_cg12283362_dnam_plot, 
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal",
                       "2022-11-10_cb_cg12283362_dnam.png"),
       width = 4.0, height = 3.2, units = "cm")

# correlation 
cor(covariates_geno_ctpc_tech_cc$cg12283362, 
    covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v,
    method="pearson") # 0.00301464

# effect size
range_cb = max(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v)

summary(cg12283362_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_cb # 0.0004828846


#####################
#SLC25A28 cg08973675#
#####################

covariates_geno_ctpc_tech_cc$cg08973675 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg08973675", ] 

cg08973675_lm <- lm(cg08973675~ no2_prenatal_avg_2019v + sex + gestational_age  + 
                      prenatal_maternal_smoke + maternal_education_length + 
                      Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                      genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                      sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                      sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                    data=covariates_geno_ctpc_tech_cc)

summary(cg08973675_lm)

# estimate marginal effects of prenatal NO2 exposure on DNAm
cg08973675_no2_me <- ggpredict(cg08973675_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("cg08973675" = predicted,
         "no2_prenatal_avg_2019v" = x)

# plot 
cb_cg08973675_dnam_plot <- ggplot(covariates_geno_ctpc_tech_cc,
                                  aes(x = no2_prenatal_avg_2019v, 
                                      y= cg08973675)) + 
  geom_line(data = cg08973675_no2_me, 
            aes(x=no2_prenatal_avg_2019v, y = cg08973675),
            col="red", size=0.5, linetype="solid") +
  geom_ribbon(data = cg08973675_no2_me, 
              aes(x = no2_prenatal_avg_2019v, 
                  ymin = conf.low, ymax = conf.high), 
              alpha = .2) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  scale_y_continuous(breaks = seq(0,0.16,by=0.04), limits = c(0,0.16)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.title = element_blank()) 

ggsave(plot = cb_cg08973675_dnam_plot, 
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal",
                       "2022-11-10_cb_cg08973675_dnam.png"),
       width = 4.0, height = 3.2, units = "cm")


# correlation 
cor(covariates_geno_ctpc_tech_cc$cg08973675, 
    covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v,
    method="pearson") # 0.1007488

# effect size
range_cb = max(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v)

summary(cg08973675_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_cb # 0.005189266
```


### Identification of differentially methylated regions (DMRs)


```{r dmr}
comb_dat <- cb_pvals_tech %>%
  select(pval_no2_prenatal_avg_2019v, chr, pos, Name) %>%
  mutate(chr = str_sub(chr, 4,5),
         end = pos + 1) %>%
  rename("p" =pval_no2_prenatal_avg_2019v,
         "start" = pos,
         "probe" = Name)

# need to run in "console" portion of RStudio
# results stored in home directory (~/CANDLE/)
# rename files, move to correct directory before loading 
ENmix::combp(comb_dat, 
             seed = 0.01,
             verbose = T)

# read in csv of dmrs
resu_combp <- read_csv(here("output_data",
                            "linear_regression",
                            "2022-10-05_prenatal_no2_resu_combp.csv"))

# get annotation for DMRs in combp results
combp_anno <- apply(resu_combp, 1, function(x){
  annotation_cpgs_order %>%
    as.data.frame() %>%
    subset(chr == paste("chr", x["chr"], sep = "")) %>%
    subset(pos >= x["start"] & pos <= x["end"]-1) %>%
    mutate(dmr = x["dmr"],
           analysis = x["analysis"])
})

# conver to data frame
combp_anno_df <- do.call(rbind, combp_anno)

# create new columns to specify DMR and analysis
resu_combp$dmr <- 1:nrow(resu_combp)
resu_combp$analysis <- "prenatal_no2"

# specify which DMR the annotation belongs to
combp_anno_df$dmr <- sapply(1:nrow(resu_combp), function(i) 
  rep(i, resu_combp[i,"nprobe"]), simplify = "vector") %>%
  unlist()

combp_anno_df$analysis <- "prenatal_no2"

# merge with rlm data to get effect size at each cpg
combp_anno_df_es <- merge(combp_anno_df, 
                          cb_pvals_tech %>%
                            subset(rownames(.) %in% rownames(combp_anno_df)) %>%
                            dplyr::select(effect_size, coeff_no2_prenatal_avg_2019v),
                          by = "row.names")

# subset for dmrs with  more than 1 cpg
resu_combp_sub <- resu_combp %>%
  subset(nprobe >1 ) %>%
  mutate(dmr2 = 1:nrow(.))

# subset DMR annotation for those with more than 1 CpG
combp_anno_df_es_sub <- combp_anno_df_es %>%
  subset(dmr %in% resu_combp_sub$dmr)

# check mean effect sizes
combp_anno_df_es_sub %>%
  group_by(dmr) %>%
  summarise(mean(effect_size)) 

# omit dmr 22 as effect size < 0.025
combp_anno_df_es_sub_escutoff <- combp_anno_df_es_sub %>%
  subset(!dmr %in% c(22))

resu_combp_sub_escutoff <- resu_combp_sub %>%
  subset(!dmr %in% c(22))

# write out annotation for dmrs
write.csv(combp_anno_df_es_sub_escutoff,
          here("output_data",
               "linear_regression",
               "2022-10-05_prenatal_no2_combp_annotation_sigDMRs.csv"))

# write out annotation for dmrs
write.csv(resu_combp_sub_escutoff,
          here("output_data",
               "linear_regression",
               "2022-10-05_prenatal_no2_resu_combp_sub_escutoff_sigDMRs.csv"))
```


Manhattan plot of CpGs in DMRS
```{r dmr_manhattan}
# Prepare data for manhattan plot.  
# nonsig pvals arent very informative and hae lots of overlap
# reduce number of nonsig pvals for plotting to improve/speed up performance
# randomly select subset of nonsig pvals for plotting
# if you dont subset, the plot will take a very long time to generate

sig.dmr.dat <- cb_pvals_tech %>% 
  subset(rownames(.) %in% combp_anno_df_es_sub_escutoff$Name) %>%
  mutate(group = "dmr")

notsig.dat <- cb_pvals_tech %>% 
  subset(!rownames(.) %in% rownames(sig.dmr.dat)) %>%
  subset(pval_no2_prenatal_avg_2019v_adj > 0.05) %>%
  mutate(group = "notsig")

# set seed before random sampling to ensure consistent results 
#set.seed(1234)
#notsig.dat.red <- notsig.dat[(sample(nrow(notsig.dat), nrow(notsig.dat) / 10)),]
mqtls.red <- rbind(sig.dmr.dat, notsig.dat)

mqtls.red$chr <- factor(mqtls.red$chr,
                        levels = c("chr1", "chr2", "chr3", "chr4", "chr5", 
                                   "chr6", "chr7", "chr8", "chr9", "chr10",
                                   "chr11", "chr12", "chr13", "chr14", "chr15",
                                   "chr16", "chr17", "chr18", "chr19", "chr20",
                                   "chr21", "chr22"))

# get max positions
chrsizes <- aggregate(pos ~ chr, data = mqtls.red, max) 

# get cumulative sizes of chrs
chrsizes$cumsum <- cumsum(as.numeric(chrsizes[, 2]))
# start cumsum at 0
chrsizes$cumsum <- chrsizes$cumsum - chrsizes$cumsum[1]

# Add this info to the initial dataset
mqtls.red2 <- left_join(mqtls.red, chrsizes[,c("chr", "cumsum")], by=c("chr"="chr"))

# Add a cumulative position of each mqtl
mqtls.red2$bpcum <- mqtls.red2$pos + mqtls.red2$cumsum

# Add a cumulative position of each SNP
# get chromosome center positions for x-axis
axisdf <- aggregate(mqtls.red2$bpcum, by = list(mqtls.red2$chr), function(x) {
  centre = (max(x) + min(x))/2
})
# add centres to cumsum
axisdf$cumsumcentre <- axisdf$x + chrsizes$cumsum

# custom colours
# shades of blue
colour_vals <- rep_len(c("#254A90", "#508FC2", "#92C5EB"), length.out=22)

# plot manhattan 
manhat <- ggplot(mqtls.red2 %>%
                   subset(group == "notsig"), 
                 aes(x=bpcum, y=effect_size)) +
  geom_point(aes(color=as.factor(chr)), alpha=0.6, size=1) + 
  scale_color_manual(values = colour_vals) +
  # dmrs
  geom_point(data = mqtls.red2 %>%
               subset(group == "dmr"),
             aes(x=bpcum, y=effect_size),
             color = "red2", size = 1, alpha = 0.8) +
  scale_x_continuous(label = c(1:22)[c(TRUE, FALSE)], 
                     breaks= axisdf$x[c(TRUE, FALSE)], 
                     name = "Chromosome") +
  # scale_y_continuous(expand = c(0, 0), limits= c(0, -log10(1e-8))) +
  theme_bw() +
  theme(plot.title = element_text(hjust = 0.5),
        legend.position="none",
        panel.border = element_blank(),
        panel.grid.major.x = element_blank(),
        panel.grid.minor.x = element_blank(),
        axis.text.x = element_text(size = 9),
        axis.text.y = element_text(size = 9)) +
  labs(x=element_blank(), y="Effect size (%DNAm)")

ggsave(plot = manhat,
       filename = here("figures",
                       "linear_regression",
                       "prenatal",
                       "2022-11-10_no2_manhattan_es.png"),
       height = 5, width = 15, units = "cm")
```



```{r dmr_gene_set_enrichment}
library(GenomicRanges)
library(IRanges)

# create GRanges object from combp output
combp_gr <- GRanges(
  seqnames = Rle(resu_combp_sub_escutoff$chr, c(rep(1, int = nrow(resu_combp_sub_escutoff)))),
  ranges = IRanges(start = resu_combp_sub_escutoff$start, end = resu_combp_sub_escutoff$end),
  strand = Rle(combp_anno_df_es_sub_escutoff %>%
                 distinct(dmr, .keep_all = T) %>%
                 pull(strand), rep(1, int = nrow(resu_combp_sub_escutoff))))

# perform enrichment analysis for GO terms
gst.region <-  goregion(regions = combp_gr, 
                        all.cpg = cb_pvals_tech %>%
                          pull(Name),
                        collection ="GO",
                        array.type = "450k",
                        sig.genes = T)
# NS
```

Examine whether DNAm at significant cpgs is associated with reported wheeze at age one. In previous CHILD paper [(PMID: 26424567)](https://www.science.org/doi/10.1126/scitranslmed.aab2271) wheeze was defined using wheeze questionnaires. If the child had wheezed with or without a cold during the first year of life (recorded via questionnaires answered by parents at 3, 6, and 12 months), the child was included in the wheezing group. Children were also included in the wheezing group if the CHILD clinician recorded a wheeze during the 1-year clinical assessment.
```{r ageone_wheeze_atopy_associations}

covariates_geno_ctpc_tech_cc$any_wheeze_y1 <- 
  ifelse(covariates_geno_ctpc_tech_cc$reported_wheezing_y1 ==1 | 
           covariates_geno_ctpc_tech_cc$clinician_wheeze_dx_y1 ==1, 1, 0)

# DMR 1
covariates_geno_ctpc_tech_cc$dmrmean1 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es %>% subset(dmr == 1) %>% pull(Name),] %>% 
  colMeans()

# dmr 1 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==1) %>% 
  pull(effect_size) # increases

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean1)) +
  geom_boxplot()

# https://stats.stackexchange.com/questions/124694/how-to-do-mancova-in-r
# mancova
# Y <- cbind(VAR1, VAR2, VAR3)
# fit <- manova(Y ~ GRP+age+gender)
# summary(fit, test="Pillai")
# 
# Other test options are "Wilks", "Hotelling-Lawley", and "Roy".
# 
# summary.aov(fit) # for univariate statistics.

# subset out dmr 1 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 1) %>%
                                   pull(Name),]))

# wheeze
dmr1_mancova_wheeze <- manova(cbind(cg00969405,cg01323381,cg01370449,cg01748892,cg02005600,
                                    cg02106682,cg02248486,cg02646423,cg02916332,cg03207666,
                                    cg03368099,cg03744763,cg04863892,cg05579037,cg05774699,
                                    cg07049592,cg08070327,cg09549073,cg09880291,cg12015737,
                                    cg12128839,cg13512268,cg13694927,cg14013695,cg14014955,
                                    cg14058329,cg14658493,cg14882265,cg16997642,cg17432857,
                                    cg17569124,cg19196335,cg19643053,cg19759481,cg20517050,
                                    cg20817131,cg23204968,cg23454797,cg23936031,cg24389585,
                                    cg25307665,cg25390165,cg25506432,cg25866143,cg26023912,
                                    cg27151303) ~ any_wheeze_y1 + sex + gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr1_mancova_wheeze, test="Pillai") # any wheeze p = 0.0601170

# atopy
dmr1_mancova_atopy <- manova(cbind(cg00969405,cg01323381,cg01370449,cg01748892,cg02005600,
                                   cg02106682,cg02248486,cg02646423,cg02916332,cg03207666,
                                   cg03368099,cg03744763,cg04863892,cg05579037,cg05774699,
                                   cg07049592,cg08070327,cg09549073,cg09880291,cg12015737,
                                   cg12128839,cg13512268,cg13694927,cg14013695,cg14014955,
                                   cg14058329,cg14658493,cg14882265,cg16997642,cg17432857,
                                   cg17569124,cg19196335,cg19643053,cg19759481,cg20517050,
                                   cg20817131,cg23204968,cg23454797,cg23936031,cg24389585,
                                   cg25307665,cg25390165,cg25506432,cg25866143,cg26023912,
                                   cg27151303) ~ clinician_atopy_dx_y1 + sex + gestational_age + 
                               prenatal_maternal_smoke + maternal_education_length + 
                               Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                               genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                               sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                               sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                             data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr1_mancova_atopy, test="Pillai") # atopy p = 0.1988312


# check vcov and correlations
dmr1_mancova_wheeze_vcov <- vcov(dmr1_mancova_wheeze) 
colnames(dmr1_mancova_wheeze_vcov) <- rownames(coef(dmr1_mancova_wheeze))
rownames(dmr1_mancova_wheeze_vcov) <- rownames(coef(dmr1_mancova_wheeze))

dmr1_mancova_wheeze_cov2cor <- cov2cor(dmr1_mancova_wheeze_vcov)
colnames(dmr1_mancova_wheeze_cov2cor) <- rownames(coef(dmr1_mancova_wheeze))
rownames(dmr1_mancova_wheeze_cov2cor) <- rownames(coef(dmr1_mancova_wheeze))

# DMR 2
covariates_geno_ctpc_tech_cc$dmrmean2 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>% subset(dmr == 2) %>% pull(Name),] %>% 
  colMeans()

# dmr 2 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==2) %>% 
  pull(effect_size) # increases

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean2)) +
  geom_boxplot()

# subset out dmr2 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 2) %>%
                                   pull(Name),]))

# wheeze
dmr2_mancova_wheeze <- manova(cbind(cg03529432, cg04265576, cg05928186, cg06237983, 
                                    cg09936824, cg12110087, cg12810523, cg14044640, 
                                    cg17994139, cg22469274) ~ any_wheeze_y1 + sex + 
                                gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr2_mancova_wheeze, test="Pillai") # any wheeze p = 0.1841408 

# atopy
dmr2_mancova_atopy <- manova(cbind(cg03529432, cg04265576, cg05928186, cg06237983, 
                                   cg09936824, cg12110087, cg12810523, cg14044640, 
                                   cg17994139, cg22469274) ~ clinician_atopy_dx_y1 + sex + 
                               gestational_age + prenatal_maternal_smoke + maternal_education_length + 
                               Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                               genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                               sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                               sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                             data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr2_mancova_atopy, test="Pillai") # any wheeze p = 0.7162367 


# DMR 3
covariates_geno_ctpc_tech_cc$dmrmean3 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>% subset(dmr == 3) %>% pull(Name),] %>% 
  colMeans()

# dmr 3 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==3) %>% 
  pull(effect_size) # lower DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean3)) +
  geom_boxplot()

# subset out dmr 3 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 3) %>%
                                   pull(Name),]))

# wheeze
dmr3_mancova_wheeze <- manova(cbind(cg02463440, cg02788857, cg07298985, cg09517075, 
                                    cg10137597, cg10298898, cg14487702, cg18135555, 
                                    cg21471580, cg26039829) ~ any_wheeze_y1 + sex + 
                                gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr3_mancova_wheeze, test="Pillai") # any wheeze p = 0.1084379 

# atopy
dmr3_mancova_atopy <- manova(cbind(cg02463440, cg02788857, cg07298985, cg09517075, 
                                   cg10137597, cg10298898, cg14487702, cg18135555, 
                                   cg21471580, cg26039829) ~ clinician_atopy_dx_y1 + sex + 
                               gestational_age + 
                               prenatal_maternal_smoke + maternal_education_length + 
                               Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                               genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                               sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                               sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                             data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr3_mancova_atopy, test="Pillai") # any atopy p = 0.1024228  

# DMR 4
covariates_geno_ctpc_tech_cc$dmrmean4 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es %>% subset(dmr == 4) %>% pull(Name),] %>% 
  colMeans()

# dmr 4 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==4) %>% 
  pull(effect_size) # higher DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean4)) +
  geom_boxplot()

# subset dmr 4 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 4) %>%
                                   pull(Name),]))

# wheeze
dmr4_mancova_wheeze <- manova(cbind(cg01564268, cg04645039, cg24534926) ~ any_wheeze_y1 + sex + 
                                gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr4_mancova_wheeze, test="Pillai") # any wheeze p = 0.4767293

# atopy
dmr4_mancova_atopy <- manova(cbind(cg01564268, cg04645039, cg24534926) ~ 
                               clinician_atopy_dx_y1 + sex + gestational_age + 
                               prenatal_maternal_smoke + maternal_education_length + 
                               Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                               genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                               sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                               sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                             data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr4_mancova_atopy, test="Pillai") # any wheeze p = 0.2695668    

# DMR 5
covariates_geno_ctpc_tech_cc$dmrmean5 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es %>% subset(dmr == 5) %>% pull(Name),] %>% 
  colMeans()

# dmr 5 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==5) %>% 
  pull(effect_size) # lower DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean5)) +
  geom_boxplot()

# subset out dmr 5 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 5) %>%
                                   pull(Name),]))

# wheeze
dmr5_mancova_wheeze <- manova(cbind(cg05016408, cg06953773, cg07499553, cg13790719, 
                                    cg15996534, cg19585597, cg26212163, cg27079680) ~ 
                                any_wheeze_y1 + sex + 
                                gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr5_mancova_wheeze, test="Pillai") # any wheeze p = 0.106488 

# atopy
dmr5_mancova_atopy <- manova(cbind(cg05016408, cg06953773, cg07499553, cg13790719, 
                                   cg15996534, cg19585597, cg26212163, cg27079680) ~ 
                               clinician_atopy_dx_y1 + sex + gestational_age + 
                               prenatal_maternal_smoke + maternal_education_length + 
                               Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                               genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                               sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                               sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                             data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr5_mancova_atopy, test="Pillai") # any atopy p = 0.554216     

# DMR 6
covariates_geno_ctpc_tech_cc$dmrmean6 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es %>% subset(dmr == 6) %>% pull(Name),] %>% 
  colMeans()

# dmr 6 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==6) %>% 
  pull(effect_size) # lower DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean6)) +
  geom_boxplot()

covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 6) %>%
                                   pull(Name),]))

# wheeze
dmr6_mancova_wheeze <- manova(cbind(cg06888209, cg17646392, cg26089165) ~ any_wheeze_y1 + sex + 
                                gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr6_mancova_wheeze, test="Pillai") # any wheeze p = 0.8307230   

# atopy
dmr6_mancova_atopy <- manova(cbind(cg06888209, cg17646392, cg26089165) ~ 
                               clinician_atopy_dx_y1 + sex + gestational_age + 
                               prenatal_maternal_smoke + maternal_education_length + 
                               Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                               genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                               sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                               sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                             data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr6_mancova_atopy, test="Pillai") # any atopy p = 0.0636551   


# DMR 7
covariates_geno_ctpc_tech_cc$dmrmean7 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es %>% subset(dmr == 7) %>% pull(Name),] %>% 
  colMeans()

# dmr 7 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==7) %>% 
  pull(effect_size) # higher DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean7)) +
  geom_boxplot()

# subset dmr 7 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 7) %>%
                                   pull(Name),]))

# wheeze
dmr7_mancova_wheeze <- manova(cbind(cg09426383, cg19216792, cg20747577) ~
                                any_wheeze_y1 + sex + gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr7_mancova_wheeze, test="Pillai") # any wheeze p = 0.544303     

# atopy
dmr7_mancova_atopy <- manova(cbind(cg09426383, cg19216792, cg20747577) ~ 
                               clinician_atopy_dx_y1 + sex + gestational_age + 
                               prenatal_maternal_smoke + maternal_education_length + 
                               Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                               genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                               sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                               sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                             data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr7_mancova_atopy, test="Pillai") # any atopy p = 0.1109498    


# DMR 8
covariates_geno_ctpc_tech_cc$dmrmean8 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es %>% subset(dmr == 8) %>% pull(Name),] %>% 
  colMeans()

# dmr 8 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==8) %>% 
  pull(effect_size) # lower DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean8)) +
  geom_boxplot()

# subset dmr 8 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 8) %>%
                                   pull(Name),]))

# wheeze
dmr8_mancova_wheeze <- manova(cbind(cg00785941, cg03748376, cg04028570, cg08260406, 
                                    cg08944170, cg20434529, cg20507276) ~ any_wheeze_y1 + sex + 
                                gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr8_mancova_wheeze, test="Pillai") # any wheeze p = 0.582027    

# atopy
dmr8_mancova_atopy <- manova(cbind(cg00785941, cg03748376, cg04028570, cg08260406, 
                                   cg08944170, cg20434529, cg20507276) ~ 
                               clinician_atopy_dx_y1 + sex + gestational_age + 
                               prenatal_maternal_smoke + maternal_education_length + 
                               Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                               genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                               sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                               sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                             data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr8_mancova_atopy, test="Pillai") # any wheeze p = 0.152508    


# DMR 9
covariates_geno_ctpc_tech_cc$dmrmean9 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es %>% subset(dmr == 9) %>% pull(Name),] %>% 
  colMeans()

# dmr 9 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==9) %>% 
  pull(effect_size) # higher DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean9)) +
  geom_boxplot()

# subset dmr 9 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 9) %>%
                                   pull(Name),]))

# wheeze
dmr9_mancova_wheeze <- manova(cbind(cg02574073, cg05888755, cg10906729, cg16848873, 
                                    cg18684142, cg19828220) ~ any_wheeze_y1 + sex + 
                                gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr9_mancova_wheeze, test="Pillai") # any wheeze p = 0.344249    

# atopy
dmr9_mancova_atopy <- manova(cbind(cg02574073, cg05888755, cg10906729, cg16848873, 
                                   cg18684142, cg19828220) ~ 
                               clinician_atopy_dx_y1 + sex + gestational_age + 
                               prenatal_maternal_smoke + maternal_education_length + 
                               Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                               genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                               sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                               sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                             data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr9_mancova_atopy, test="Pillai") # atopy p = 0.0057741 **     


# DMR 10
covariates_geno_ctpc_tech_cc$dmrmean10 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es %>% subset(dmr == 10) %>% pull(Name),] %>% 
  colMeans()

# dmr 10 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==10) %>% 
  pull(effect_size) # lower DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean10)) +
  geom_boxplot()

# subset dmr 10 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 10) %>%
                                   pull(Name),]))

# wheeze
dmr10_mancova_wheeze <- manova(cbind(cg03724423, cg04321618, cg06942814, cg08657492, 
                                     cg11015251, cg14359292, cg16651126, cg17457637, 
                                     cg17591595, cg24169822, cg25952581) ~ any_wheeze_y1 + sex + 
                                 gestational_age + 
                                 prenatal_maternal_smoke + maternal_education_length + 
                                 Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                 genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                 sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                 sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                               data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr10_mancova_wheeze, test="Pillai") # any wheeze p = 0.1706771

# atopy
dmr10_mancova_atopy <- manova(cbind(cg03724423, cg04321618, cg06942814, cg08657492, 
                                    cg11015251, cg14359292, cg16651126, cg17457637, 
                                    cg17591595, cg24169822, cg25952581) ~ 
                                clinician_atopy_dx_y1 + sex + gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr10_mancova_atopy, test="Pillai") # any wheeze p = 0.6550467    


# DMR 11
covariates_geno_ctpc_tech_cc$dmrmean11 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es %>% subset(dmr == 11) %>% pull(Name),] %>% 
  colMeans()

# dmr 11 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==11) %>% 
  pull(effect_size) # lower DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean11)) +
  geom_boxplot()

# subset out dmr 11 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 11) %>%
                                   pull(Name),]))

# wheeze
dmr11_mancova_wheeze <- manova(cbind(cg02649248, cg07658809) ~ any_wheeze_y1 + sex + 
                                 gestational_age + 
                                 prenatal_maternal_smoke + maternal_education_length + 
                                 Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                 genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                 sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                 sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                               data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr11_mancova_wheeze, test="Pillai") # any wheeze p = 0.762636

# atopy
dmr11_mancova_atopy <- manova(cbind(cg02649248, cg07658809) ~ 
                                clinician_atopy_dx_y1 + sex + gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr11_mancova_atopy, test="Pillai") # any atopy p = 0.0251976

# DMR 12
covariates_geno_ctpc_tech_cc$dmrmean12 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es %>% subset(dmr == 12) %>% pull(Name),] %>% 
  colMeans()

# dmr 12 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==12) %>% 
  pull(effect_size) # higher DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean12)) +
  geom_boxplot()

# subset dmr 12 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 12) %>%
                                   pull(Name),]))

# wheeze
dmr12_mancova_wheeze <- manova(cbind(cg00701946, cg01876338, cg05414908, cg08611810, 
                                     cg12451530, cg13356253, cg14832904, cg19878627, 
                                     cg19961545) ~ any_wheeze_y1 + sex + 
                                 gestational_age + 
                                 prenatal_maternal_smoke + maternal_education_length + 
                                 Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                 genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                 sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                 sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                               data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr12_mancova_wheeze, test="Pillai") # any wheeze p = 0.3110278 

# atopy
dmr12_mancova_atopy <- manova(cbind(cg00701946, cg01876338, cg05414908, cg08611810, 
                                    cg12451530, cg13356253, cg14832904, cg19878627, 
                                    cg19961545) ~ clinician_atopy_dx_y1 + sex + 
                                gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr12_mancova_atopy, test="Pillai") # atopy p = 0.0045016  


# DMR 13
covariates_geno_ctpc_tech_cc$dmrmean13 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es %>% subset(dmr == 13) %>% pull(Name),] %>% 
  colMeans()

# dmr 13 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==13) %>% 
  pull(effect_size) # higher DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean13)) +
  geom_boxplot()

# subset dmr 13 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 13) %>%
                                   pull(Name),]))

# wheeze
dmr13_mancova_wheeze <- manova(cbind(cg02749105, cg09594291, cg24052338) ~ 
                                 any_wheeze_y1 + sex + gestational_age + 
                                 prenatal_maternal_smoke + maternal_education_length + 
                                 Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                 genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                 sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                 sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                               data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr13_mancova_wheeze, test="Pillai") # any wheeze p = 0.254663 

# atopy
dmr13_mancova_atopy <- manova(cbind(cg02749105, cg09594291, cg24052338) ~ 
                                clinician_atopy_dx_y1 + sex + gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr13_mancova_atopy, test="Pillai") # atopy p = 5.546e-05 


longdf_dmr13 <- pivot_longer(covariates_geno_ctpc_tech_cc_dmr_cpgs %>% 
                               select(cg02749105, cg09594291, cg24052338, 
                                      sample_id, clinician_atopy_dx_y1, 
                                      no2_prenatal_avg_2019v),
                             cols = -c(sample_id, clinician_atopy_dx_y1, 
                                       no2_prenatal_avg_2019v), 
                             names_to = "cpg", 
                             values_to = "dnam") %>%
  mutate(cpg = factor(cpg, levels = c("cg09594291", "cg02749105", "cg24052338"))) 

dmr13_atopy_vis <- 
  ggplot(longdf_dmr13, aes(x=cpg, y=dnam, 
                           group = interaction(cpg, clinician_atopy_dx_y1), 
                           shape = as.factor(clinician_atopy_dx_y1),
                           color = no2_prenatal_avg_2019v,
                           fill = as.factor(clinician_atopy_dx_y1))) +
  scale_shape_manual(values = c(20, 17)) +
  scale_fill_manual(values = c("white", "grey")) +
  geom_violin(draw_quantiles = c(0.5), trim = T, alpha =0.6) +
  ggbeeswarm::geom_beeswarm(dodge.width=0.9, cex = 1.5, size = 1) +
  theme_bw() +
  scale_x_discrete(labels = c("cg09594291",
                              "cg02749105",
                              "cg24052338"),
                   name = element_blank()) +
  scale_y_continuous(name = "DNA methylation (%)") +
  theme(legend.position = "none",
        axis.text = element_text(size = 9),
        panel.grid.major.x = element_blank())

ggsave(plot = dmr13_atopy_vis,
       filename  = here("figures",
                        "linear_regression",
                        "prenatal",
                        "2022-11-11_no2_cb_dmr13_atopy.png"), 
       width = 7.5, height = 5, units = "cm")


# DMR 14
covariates_geno_ctpc_tech_cc$dmrmean14 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es %>% subset(dmr == 14) %>% pull(Name),] %>% 
  colMeans()

# dmr 14 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==14) %>% 
  pull(effect_size) # lower DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean14)) +
  geom_boxplot()

# MANCOVA
# subset dmr 14 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 14) %>%
                                   pull(Name),]))

# wheeze
dmr14_mancova_wheeze <- manova(cbind(cg04231094, cg05353869, cg06329735) ~ any_wheeze_y1 + sex + 
                                 gestational_age + 
                                 prenatal_maternal_smoke + maternal_education_length + 
                                 Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                 genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                 sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                 sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                               data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr14_mancova_wheeze, test="Pillai") # any wheeze p = 0.1621576  

# atopy
dmr14_mancova_atopy <- manova(cbind(cg04231094, cg05353869, cg06329735) ~ 
                                clinician_atopy_dx_y1 + sex + gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr14_mancova_atopy, test="Pillai") # any wheeze p = 0.2409701    


# DMR 15
covariates_geno_ctpc_tech_cc$dmrmean15 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es %>% subset(dmr == 15) %>% pull(Name),] %>% 
  colMeans()

# dmr 15 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==15) %>% 
  pull(effect_size) # higher DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean15)) +
  geom_boxplot()

# MANCOVA
# subset dmr 15 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 15) %>%
                                   pull(Name),]))

# wheeze
dmr15_mancova_wheeze <- manova(cbind(cg00256046, cg04217177, cg17434154) ~ any_wheeze_y1 + sex + 
                                 gestational_age + 
                                 prenatal_maternal_smoke + maternal_education_length + 
                                 Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                 genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                 sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                 sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                               data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr15_mancova_wheeze, test="Pillai") # any wheeze p = 0.4586307 

# atopy
dmr15_mancova_atopy <- manova(cbind(cg00256046, cg04217177, cg17434154) ~ 
                                clinician_atopy_dx_y1 + sex + gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr15_mancova_atopy, test="Pillai") # any wheeze p = 0.0180669 


# DMR 16
covariates_geno_ctpc_tech_cc$dmrmean16 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es %>% subset(dmr == 16) %>% pull(Name),] %>% 
  colMeans()

# dmr 16 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==16) %>% 
  pull(effect_size) # higher DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean16)) +
  geom_boxplot()

# MANCOVA
# subset dmr 16 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 16) %>%
                                   pull(Name),]))

# wheeze
dmr16_mancova_wheeze <- manova(cbind(cg14583973, cg18352616, cg22655196) ~ any_wheeze_y1 + sex + 
                                 gestational_age + 
                                 prenatal_maternal_smoke + maternal_education_length + 
                                 Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                 genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                 sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                 sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                               data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr16_mancova_wheeze, test="Pillai") # any wheeze p = 0.0429344****

# atopy
dmr16_mancova_atopy <- manova(cbind(cg14583973, cg18352616, cg22655196) ~ 
                                clinician_atopy_dx_y1 + sex + gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr16_mancova_atopy, test="Pillai") # any wheeze p = 0.0063444

# DMR 18
covariates_geno_ctpc_tech_cc$dmrmean18 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>% subset(dmr == 18) %>% pull(Name),] %>% 
  colMeans()

# dmr 18 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==18) %>% 
  pull(effect_size) # lower DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean18)) +
  geom_boxplot()

# MANCOVA
# subset dmr 18 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 18) %>%
                                   pull(Name),]))

# wheeze
dmr18_mancova_wheeze <- manova(cbind(cg06853455, cg12764003) ~ any_wheeze_y1 + sex + 
                                 gestational_age + 
                                 prenatal_maternal_smoke + maternal_education_length + 
                                 Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                 genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                 sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                 sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                               data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr18_mancova_wheeze, test="Pillai") # any wheeze p = 0.35775 

# atopy
dmr18_mancova_atopy <- manova(cbind(cg06853455, cg12764003) ~ 
                                clinician_atopy_dx_y1 + sex + gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr18_mancova_atopy, test="Pillai") # atopy p = 0.66060 


# DMR 19
covariates_geno_ctpc_tech_cc$dmrmean19 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>% subset(dmr == 19) %>% pull(Name),] %>% 
  colMeans()

# dmr 19 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==19) %>% 
  pull(effect_size) # higher DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean19)) +
  geom_boxplot()

# MANCOVA
# subset dmr 19 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 19) %>%
                                   pull(Name),]))

# wheeze
dmr19_mancova_wheeze <- manova(cbind(cg09512891, cg25437304) ~ any_wheeze_y1 + sex + 
                                 gestational_age + 
                                 prenatal_maternal_smoke + maternal_education_length + 
                                 Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                 genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                 sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                 sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                               data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr19_mancova_wheeze, test="Pillai") # any wheeze p = 0.779689  

# atopy
dmr19_mancova_atopy <- manova(cbind(cg09512891, cg25437304) ~ 
                                clinician_atopy_dx_y1 + sex + gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr19_mancova_atopy, test="Pillai") # atopy p = 0.254903 


# DMR 20
covariates_geno_ctpc_tech_cc$dmrmean20 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>% subset(dmr == 20) %>% pull(Name),] %>% 
  colMeans()

# dmr 20 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==20) %>% 
  pull(effect_size) # lower DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean20)) +
  geom_boxplot()

# MANCOVA
# subset dmr 20 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 20) %>%
                                   pull(Name),]))

# wheeze
dmr20_mancova_wheeze <- manova(cbind(cg02105458, cg02614024, cg03829137, cg11179621) ~ 
                                 any_wheeze_y1 + sex + gestational_age + 
                                 prenatal_maternal_smoke + maternal_education_length + 
                                 Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                 genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                 sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                 sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                               data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr20_mancova_wheeze, test="Pillai") # any wheeze p = 0.1855319   

# atopy
dmr20_mancova_atopy <- manova(cbind(cg02105458, cg02614024, cg03829137, cg11179621) ~ 
                                clinician_atopy_dx_y1 + sex + gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr20_mancova_atopy, test="Pillai") # atopy p = 0.5985048     

# DMR 21
covariates_geno_ctpc_tech_cc$dmrmean21 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>% subset(dmr == 21) %>% pull(Name),] %>% 
  colMeans()

# dmr 21 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==21) %>% 
  pull(effect_size) # higher DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean21)) +
  geom_boxplot()

# MANCOVA
# subset dmr 21 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 21) %>%
                                   pull(Name),]))

# wheeze
dmr21_mancova_wheeze <- manova(cbind(cg04534697, cg05170671, cg12793096, cg26656672) ~ 
                                 any_wheeze_y1 + sex + gestational_age + 
                                 prenatal_maternal_smoke + maternal_education_length + 
                                 Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                 genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                 sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                 sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                               data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr21_mancova_wheeze, test="Pillai") # any wheeze p =  0.0240144

# atopy
dmr21_mancova_atopy <- manova(cbind(cg04534697, cg05170671, cg12793096, cg26656672) ~ 
                                clinician_atopy_dx_y1 + sex + gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr21_mancova_atopy, test="Pillai") # atopy p = 0.1119310      

# DMR 23
covariates_geno_ctpc_tech_cc$dmrmean23 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>% subset(dmr == 23) %>% pull(Name),] %>% 
  colMeans()

# dmr 23 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==23) %>% 
  pull(effect_size) # higher DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean23)) +
  geom_boxplot()

# subset dmr 23 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 23) %>%
                                   pull(Name),]))

# wheeze
dmr23_mancova_wheeze <- manova(cbind(cg05086567, cg11874331, cg17887478) ~ 
                                 any_wheeze_y1 + sex + gestational_age + 
                                 prenatal_maternal_smoke + maternal_education_length + 
                                 Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                 genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                 sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                 sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                               data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr23_mancova_wheeze, test="Pillai") # any wheeze p = 0.001144

# plot wheeze
longdf_dmr23 <- pivot_longer(covariates_geno_ctpc_tech_cc_dmr_cpgs %>% 
                               select(cg05086567 , cg11874331 , cg17887478 , 
                                      sample_id, any_wheeze_y1, no2_prenatal_avg_2019v),
                             cols = -c(sample_id, any_wheeze_y1, no2_prenatal_avg_2019v), 
                             names_to = "cpg", 
                             values_to = "dnam") %>%
  mutate(cpg = factor(cpg, levels = c("cg17887478", "cg11874331", "cg05086567"))) %>%
  na.omit()

# violin plots
dmr23_wheeze_vis <-  ggplot(longdf_dmr23, 
                            aes(x=cpg, y=dnam, 
                                group = interaction(cpg, any_wheeze_y1), 
                                shape = as.factor(any_wheeze_y1),
                                color = no2_prenatal_avg_2019v,
                                fill = as.factor(any_wheeze_y1))) +
  scale_shape_manual(values = c(20, 17)) +
  scale_fill_manual(values = c("white", "grey"), 
                    name = "Any wheeze age one",
                    labels = c("No", "Yes")) +
  geom_violin(draw_quantiles = c(0.5), trim = T, alpha =0.6) +
  ggbeeswarm::geom_beeswarm(dodge.width=0.9, cex = 1.5, size = 1) +
  theme_bw() +
  scale_x_discrete(labels = c("cg17887478",
                              "cg11874331",
                              "cg05086567"),
                   name = element_blank()) +
  scale_y_continuous(name = "DNA methylation (%)") +
  theme(legend.position = "none",
        axis.text = element_text(size =9),
        panel.grid.major.x = element_blank())

ggsave(plot = dmr23_wheeze_vis,
       filename  = here("figures",
                        "linear_regression",
                        "prenatal",
                        "2022-11-11_no2_cb_dmr23_wheeze.png"), 
       width = 7.5, height = 5, units = "cm")


# atopy
dmr23_mancova_atopy <- manova(cbind(cg05086567, cg11874331, cg17887478) ~ 
                                clinician_atopy_dx_y1 + sex + gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr23_mancova_atopy, test="Pillai") # atopy p = 0.0164543    

# DMR 24
covariates_geno_ctpc_tech_cc$dmrmean24 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>% subset(dmr == 24) %>% pull(Name),] %>% 
  colMeans()

# dmr 24 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==24) %>% 
  pull(effect_size) # less DNAm

# MANCOVA
# subset dmr 24 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 24) %>%
                                   pull(Name),]))

# wheeze
dmr24_mancova_wheeze <- manova(cbind(cg16608407, cg18978493) ~ 
                                 any_wheeze_y1 + sex + gestational_age + 
                                 prenatal_maternal_smoke + maternal_education_length + 
                                 Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                 genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                 sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                 sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                               data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr24_mancova_wheeze, test="Pillai") # any wheeze p = 0.27635  

# atopy
dmr24_mancova_atopy <- manova(cbind(cg16608407, cg18978493) ~ 
                                clinician_atopy_dx_y1 + sex + gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr24_mancova_atopy, test="Pillai") # atopy p = 0.296082  


# DMR 25
covariates_geno_ctpc_tech_cc$dmrmean25 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>% subset(dmr == 25) %>% pull(Name),] %>% 
  colMeans()

# dmr 25 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==25) %>% 
  pull(effect_size) # higher DNAm

ggplot(covariates_geno_ctpc_tech_cc %>% subset(!is.na(any_wheeze_y1)),
       aes(x=as.factor(any_wheeze_y1), y=dmrmean25)) +
  geom_boxplot()

# MANCOVA
# subset dmr 25 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 25) %>%
                                   pull(Name),]))

# wheeze
dmr25_mancova_wheeze <- manova(cbind(cg08383929, cg15946312) ~ 
                                 any_wheeze_y1 + sex + gestational_age + 
                                 prenatal_maternal_smoke + maternal_education_length + 
                                 Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                 genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                 sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                 sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                               data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr25_mancova_wheeze, test="Pillai") # any wheeze p = 0.1170872  

# atopy
dmr25_mancova_atopy <- manova(cbind(cg08383929, cg15946312) ~ 
                                clinician_atopy_dx_y1 + sex + gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr25_mancova_atopy, test="Pillai") # atopy p = 0.0045309  


# DMR 27
covariates_geno_ctpc_tech_cc$dmrmean27 <- 
  cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>% subset(dmr == 27) %>% pull(Name),] %>% 
  colMeans()

# dmr 27 effect size
combp_anno_df_es_sub_escutoff %>% 
  subset(dmr==27) %>% 
  pull(effect_size) # lower DNAm

# MANCOVA
# subset dmr 27 cpgs
covariates_geno_ctpc_tech_cc_dmr_cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc, 
        t(cb_betas_geno_cc_clean[combp_anno_df_es_sub_escutoff %>%
                                   subset(dmr == 27) %>%
                                   pull(Name),]))

# wheeze
dmr27_mancova_wheeze <- manova(cbind(cg10200291, cg18710053) ~ 
                                 any_wheeze_y1 + sex + gestational_age + 
                                 prenatal_maternal_smoke + maternal_education_length + 
                                 Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                 genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                 sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                 sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                               data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr27_mancova_wheeze, test="Pillai") # any wheeze p = 9.098e-05  


longdf_dmr27 <- pivot_longer(covariates_geno_ctpc_tech_cc_dmr_cpgs %>% 
                               select(cg10200291 , cg18710053, 
                                      sample_id, any_wheeze_y1, no2_prenatal_avg_2019v),
                             cols = -c(sample_id, any_wheeze_y1, no2_prenatal_avg_2019v), 
                             names_to = "cpg", 
                             values_to = "dnam") %>%
  mutate(cpg = factor(cpg, levels = c("cg10200291", "cg18710053"))) %>%
  na.omit()

# violin plots
dmr27_wheeze_vis <-  ggplot(longdf_dmr27, 
                            aes(x=cpg, y=dnam, 
                                group = interaction(cpg, any_wheeze_y1), 
                                shape = as.factor(any_wheeze_y1),
                                color = no2_prenatal_avg_2019v,
                                fill = as.factor(any_wheeze_y1))) +
  scale_shape_manual(values = c(20, 17)) +
  scale_fill_manual(values = c("white", "grey"), 
                    name = "Any wheeze age one",
                    labels = c("No", "Yes")) +
  geom_violin(draw_quantiles = c(0.5), trim = T, alpha =0.6) +
  ggbeeswarm::geom_beeswarm(dodge.width=0.9, cex = 1.5, size = 1) +
  theme_bw() +
  scale_x_discrete(labels = c("cg10200291", 
                              "cg18710053"),
                   name = element_blank()) +
  scale_y_continuous(name = "DNA methylation (%)") +
  theme(legend.position = "none",
        axis.text = element_text(size =9),
        panel.grid.major.x = element_blank())

ggsave(plot = dmr27_wheeze_vis,
       filename  = here("figures",
                        "linear_regression",
                        "prenatal",
                        "2022-11-11_no2_cb_dmr27_wheeze.png"), 
       width = 7.5, height = 5, units = "cm")

# atopy
dmr27_mancova_atopy <- manova(cbind(cg10200291, cg18710053) ~ 
                                clinician_atopy_dx_y1 + sex + gestational_age + 
                                prenatal_maternal_smoke + maternal_education_length + 
                                Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                                genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                                sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                                sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                              data = covariates_geno_ctpc_tech_cc_dmr_cpgs)

summary(dmr27_mancova_atopy, test="Pillai") # atopy p = 0.4054433  

dmr27_wheeze_plot <- ggplot(covariates_geno_ctpc_tech_cc %>% 
                              subset(!is.na(any_wheeze_y1)),
                            aes(x = as.factor(any_wheeze_y1), 
                                y = dmrmean27,
                                color = no2_prenatal_avg_2019v)) +
  geom_violin(draw_quantiles = c(0.5)) +
  ggbeeswarm::geom_beeswarm(cex = 3, alpha = 0.9, size = 1) +
  scale_y_continuous(name = "Average DNAm (%)", limits = c(0,0.4)) +
  scale_x_discrete(name = "Age one wheeze", labels = c("No", "Yes")) +
  theme_bw() +
  theme(panel.grid.major.x = element_blank(),
        axis.text.x = element_text(size=9),
        axis.text.y = element_text(size=9),
        legend.key.height = unit(0.3, "cm"), 
        legend.key.width = unit(0.3, "cm"), 
        axis.title = element_blank()) +
  scale_color_continuous(name = "") +
  ggsignif::geom_signif(annotations = "adj p = 0.12", 
                        xmin = 1, 
                        xmax = 2, 
                        y_position = 0.325, 
                        vjust = -0.25, textsize = 3.5)

ggsave(plot = dmr27_wheeze_plot,
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal", 
                       "2022-10-12_dmrmean27_wheeze_violin.png"),
       width = 6, height = 5, units = "cm")
```


### Genome-wide DNA methylation changes 

Mean array-wide DNAm chanages
```{r mean_array_wide_dnam}
# convert betas to df
cb_betas_geno_cc_clean_df <- as.data.frame(cb_betas_geno_cc_clean)

methmeans <- colMeans(cb_betas_geno_cc_clean_df)

covariates_geno_ctpc_tech_cc <- cbind(covariates_geno_ctpc_tech_cc, methmeans)

#colmeans
meanmeth_lm <- 
  lm(methmeans ~ no2_prenatal_avg_2019v + sex + gestational_age  + 
       prenatal_maternal_smoke + maternal_education_length + 
       Comp.1 + Comp.2 + Comp.3 + Comp.4 +
       genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
       sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
       sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
     data=covariates_geno_ctpc_tech_cc)

#get summary
summary(meanmeth_lm) #no2 not sig

# estimate marginal effects of prenatal NO2 exposure on DNAm
globaldna_no2_me <- ggpredict(meanmeth_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  dplyr::rename("methmeans" = predicted,
                "no2_prenatal_avg_2019v" = x)

# plot 
global_dnam_plot <- ggplot(covariates_geno_ctpc_tech_cc,
                           aes(x = no2_prenatal_avg_2019v, 
                               y= methmeans)) + 
  geom_line(data = globaldna_no2_me, 
            aes(x=no2_prenatal_avg_2019v, y = methmeans),
            col="red", size=0.5, linetype="solid") +
  geom_ribbon(data = globaldna_no2_me, 
              aes(x = no2_prenatal_avg_2019v, 
                  ymin = conf.low, ymax = conf.high), 
              alpha = .2) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.title = element_blank()) +
  lims(y=c(0.48,0.54))

ggsave(plot = global_dnam_plot, 
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal",
                       "2022-10-05_cb_global_dnam.png"),
       width = 4.0, height = 3.2, units = "cm")

# correlation 
cor(covariates_geno_ctpc_tech_cc$methmeans, 
    covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v,
    method="pearson") # 0.0489746

# effect size
range_cb = max(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v)

summary(meanmeth_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_cb # -1.476332e-05
```

LINE-1 specific DNAm
```{r mean_line1_dnam}
# read in line annotation from UCSC and repeat masker 
repeats_450k <- read.delim(file = gzfile(here("input_data", 
                                              "ucsc_rmsk_lines", 
                                              "ucsc_repeatmasker_enc_methyl450_gm12878.gz")),
                           header = T, skip = 1, sep = "\t")

# subset repeats for L1 LINE elements that are not in sex chromosomes
lines_450k  <- repeats_450k %>% 
  subset(rmsk.repClass == "LINE") %>%
  subset(rmsk.repFamily == "L1") %>%
  subset(!wgEncodeHaibMethyl450Gm12878SitesRep1.chrom %in% c("chrX", "chrY")) %>%
  subset(!is.na(wgEncodeHaibMethyl450Gm12878SitesRep1.chromStart))

# subset betas for those that overlap with line elements
cb_betas_geno_cc_clean_df_lines <- cb_betas_geno_cc_clean_df %>% 
  subset(rownames(.) %in% lines_450k$wgEncodeHaibMethyl450Gm12878SitesRep1.name)

# num of cpgs annotated to lines 
nrow(cb_betas_geno_cc_clean_df_lines) # 7302 

# get mean of line meth 
line_meth <- colMeans(cb_betas_geno_cc_clean_df_lines)

# cbind line methylation to covariate data frame
covariates_geno_ctpc_tech_cc <- cbind(covariates_geno_ctpc_tech_cc, line_meth)

#colmeans
line_meth_lm <- lm(line_meth ~ no2_prenatal_avg_2019v + sex + gestational_age  + 
                     prenatal_maternal_smoke + maternal_education_length + 
                     Comp.1 + Comp.2 + Comp.3 + Comp.4 +
                     genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                     sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                     sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17, 
                   data=covariates_geno_ctpc_tech_cc)

# summary of line regression
summary(line_meth_lm) #no2 not sig

# estimate marginal effects of prenatal NO2 exposure on DNAm
line_meth_no2_me <- ggpredict(line_meth_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  dplyr::rename("line_meth" = predicted,
                "no2_prenatal_avg_2019v" = x)

# plot 
line_dnam_plot <- ggplot(covariates_geno_ctpc_tech_cc,
                         aes(x = no2_prenatal_avg_2019v, 
                             y= line_meth)) + 
  geom_line(data = line_meth_no2_me, 
            aes(x=no2_prenatal_avg_2019v, y = line_meth),
            col="red", size=0.5, linetype="solid") +
  geom_ribbon(data = line_meth_no2_me, 
              aes(x = no2_prenatal_avg_2019v, 
                  ymin = conf.low, ymax = conf.high), 
              alpha = .2) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.title = element_blank()) +
  lims(y=c(0.80, 0.86))

ggsave(plot = line_dnam_plot, 
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal",
                       "2022-10-05_cb_line_dnam.png"),
       width = 4.0, height = 3.2, units = "cm")

# correlation 
cor(covariates_geno_ctpc_tech_cc$line_meth, 
    covariates_geno_ctpc_tech_cc$no2_prenatal_avg_2019v,
    method="pearson") # 0.03949858

# effect size
summary(line_meth_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_cb # -4.694871e-05
```

### Interaction effect with birth month in prenatal model 

Try re-running limma with birth month/ interaction between birth month and air pollution.
```{r interaction_terms_to_assess_birthmonth}
#want to use anova to compare models usign anova
#anova only accepts models from lm
#use lm to compare how birth month affects cpgs above 10^-3 sig 

#You should be able to ask for confidence intervals for the slope coefficients for each month.  #even in a tabular display, you may see that “smoothly varying” pattern we were discussing.
#Then, you could use those to plot a sort of “here’s how the slope varies by month” picture.Sort #of like forest plots for meta analysis results, if that makes sense.

#Also, before we decide it doesn’t matter, the individual slopes might look fairly similar, but the appropriate overall test would be to check the anova(fit_with_interaction) and check the p-value on the interaction term.

#start by investigating cpgs identified as nominally significant in volcano plot

#  get names of top 10 cpgs based on effect size and pvalue from ewas
cb_pvals_tech %>%
  subset(effect_size > 0.025 | effect_size < -0.025) %>%
  arrange(pval_no2_prenatal_avg_2019v) %>%
  slice_head(n=10) %>%
  rownames()

#############
#cg03508112 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg03508112 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg03508112", ] 

# fit wiht lm
cg03508112_lm_bm <- lm(cb_cg03508112 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg03508112_lm <- lm(cb_cg03508112 ~  no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg03508112_lm, cg03508112_lm_bm, test = "LRT") # p =0.3891
summary(cg03508112_lm_bm) # feb (0.0317)

# get plot of birth month effect with 95% CI
cg03508112_lm_bm_plot <- tidy(cg03508112_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-30, 30) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(filename="2022-10-07_CHILD_prenatal_no2_cg03508112_bm.png",
       path =  here("figures", "linear_regression", "prenatal"),
       width = 4.8, height = 4, units = "cm")


#############
#cg08568767 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg08568767 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg08568767", ] 

# fit wiht lm
cg08568767_lm_bm <- lm(cb_cg08568767 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg08568767_lm <- lm(cb_cg08568767 ~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg08568767_lm, cg08568767_lm_bm, test = "LRT") # p = 0.9724
summary(cg08568767_lm_bm) 

# get plot of birth month effect with 95% CI
cg08568767_lm_bm_plot <- tidy(cg08568767_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-35, 35) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg08568767_lm_bm_plot,
       filename="2022-10-07_CHILD_prenatal_no2_cg08568767_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm")


#############
#cg12228123 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg12228123 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg12228123", ] 

# fit wiht lm
cg12228123_lm_bm <- lm(cb_cg12228123 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg12228123_lm <- lm(cb_cg12228123 ~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg12228123_lm, cg12228123_lm_bm, test = "LRT") # p = 0.444
summary(cg12228123_lm_bm) # dec (0.032987)

# get plot of birth month effect with 95% CI
cg12228123_lm_bm_plot <- tidy(cg12228123_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-80, 80) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg12228123_lm_bm_plot,
       filename="2022-10-07_CHILD_prenatal_no2_cg12228123_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm")


#############
#cg14491284 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg14491284 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg14491284", ] 

# fit wiht lm
cg14491284_lm_bm <- lm(cb_cg14491284 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg14491284_lm <- lm(cb_cg14491284 ~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg14491284_lm, cg14491284_lm_bm, test = "LRT") # p = 0.7233
summary(cg14491284_lm_bm) 

# get plot of birth month effect with 95% CI
cg14491284_lm_bm_plot <- tidy(cg14491284_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-30, 30) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg14491284_lm_bm_plot,
       filename="2022-10-07_CHILD_prenatal_no2_cg14491284_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm")


#############
#cg02074714 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg02074714 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg02074714", ] 

# fit wiht lm
cg02074714_lm_bm <- lm(cb_cg02074714 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg02074714_lm <- lm(cb_cg02074714 ~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg02074714_lm, cg02074714_lm_bm, test = "LRT") # p = 0.958
summary(cg02074714_lm_bm) 

# get plot of birth month effect with 95% CI
cg02074714_lm_bm_plot <- tidy(cg02074714_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-50, 50) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg02074714_lm_bm_plot,
       filename="2022-10-07_CHILD_prenatal_no2_cg02074714_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm")


#############
#cg14658149 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg14658149 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg14658149", ] 

# fit wiht lm
cg14658149_lm_bm <- lm(cb_cg14658149 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg14658149_lm <- lm(cb_cg14658149 ~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg14658149_lm, cg14658149_lm_bm) # p = 0.159
summary(cg14658149_lm_bm) # nov (0.044061)

# get plot of birth month effect with 95% CI
cg14658149_lm_bm_plot <- tidy(cg14658149_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-20, 20) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg14658149_lm_bm_plot,
       filename="2022-10-07_CHILD_prenatal_no2_cg14658149_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm")


#############
#cg25472862 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg25472862 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg25472862", ] 

# fit wiht lm
cg25472862_lm_bm <- lm(cb_cg25472862 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg25472862_lm <- lm(cb_cg25472862 ~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg25472862_lm, cg25472862_lm_bm, test = "LRT") # p = 0.5835
summary(cg25472862_lm_bm) 

# get plot of birth month effect with 95% CI
cg25472862_lm_bm_plot <- tidy(cg25472862_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-30, 30) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg25472862_lm_bm_plot,
       filename="2022-10-07_CHILD_prenatal_no2_cg25472862_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm")


#############
#cg11642106 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg11642106 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg11642106", ] 

# fit wiht lm
cg11642106_lm_bm <- lm(cb_cg11642106 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg11642106_lm <- lm(cb_cg11642106 ~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg11642106_lm, cg11642106_lm_bm, test = "LRT") # p = 0.6545
summary(cg11642106_lm_bm) 

# get plot of birth month effect with 95% CI
cg11642106_lm_bm_plot <- tidy(cg11642106_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-20, 20) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg11642106_lm_bm_plot,
       filename="2022-10-07_CHILD_prenatal_no2_cg11642106_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm")


#############
#cg09936824 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg09936824 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg09936824", ] 

# fit wiht lm
cg09936824_lm_bm <- lm(cb_cg09936824 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg09936824_lm <- lm(cb_cg09936824 ~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg09936824_lm, cg09936824_lm_bm, test = "LRT") # p = 0.6164
summary(cg09936824_lm_bm) 

# get plot of birth month effect with 95% CI
cg09936824_lm_bm_plot <- tidy(cg09936824_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-30, 30) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg09936824_lm_bm_plot,
       filename="2022-10-07_CHILD_prenatal_no2_cg09936824_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm")


#############
#cg08033031 #             
#############

# add to covariate data frame
covariates_geno_ctpc_tech_cc$cb_cg08033031 <- 
  cb_betas_geno_cc_clean[rownames(cb_betas_geno_cc_clean)=="cg08033031", ] 

# fit wiht lm
cg08033031_lm_bm <- lm(cb_cg08033031 ~ no2_prenatal_avg_2019v*birth_month + 
                         sex + gestational_age  + prenatal_maternal_smoke + 
                         maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                         Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                         genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                         sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                         sv15 + sv16 + sv17,
                       data=covariates_geno_ctpc_tech_cc)

cg08033031_lm <- lm(cb_cg08033031 ~ no2_prenatal_avg_2019v + 
                      sex + gestational_age  + prenatal_maternal_smoke + 
                      maternal_education_length + Comp.1 + Comp.2 + Comp.3 + 
                      Comp.4 + genotyping.PC1 + genotyping.PC2 + 
                      genotyping.PC3 + sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + 
                      sv7 + sv8 + sv9 + sv10 + sv11 + sv12 + sv13 + sv14 + 
                      sv15 + sv16 + sv17,
                    data=covariates_geno_ctpc_tech_cc)


# anova
anova(cg08033031_lm, cg08033031_lm_bm, test = "LRT") # p = 0.7881
summary(cg08033031_lm_bm) 

# get plot of birth month effect with 95% CI
cg08033031_lm_bm_plot <- tidy(cg08033031_lm_bm, conf.int = 0.95) %>% 
  filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_prenatal_avg_2019v:birth_monthfebruary",
                                      "no2_prenatal_avg_2019v:birth_monthmarch",
                                      "no2_prenatal_avg_2019v:birth_monthapril",
                                      "no2_prenatal_avg_2019v:birth_monthmay",
                                      "no2_prenatal_avg_2019v:birth_monthjune",
                                      "no2_prenatal_avg_2019v:birth_monthjuly",
                                      "no2_prenatal_avg_2019v:birth_monthaugust",
                                      "no2_prenatal_avg_2019v:birth_monthseptember",
                                      "no2_prenatal_avg_2019v:birth_monthoctober",
                                      "no2_prenatal_avg_2019v:birth_monthnovember",
                                      "no2_prenatal_avg_2019v:birth_monthdecember"))) %>% 
  ggplot(aes(y = estimate*100, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low*100 , ymax = conf.high*100)) +
  scale_x_discrete(labels=c("February","March", "April", "May",
                            "June","July", "August", "September",
                            "October", "November","December")) +
  ylim(-20, 20) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=9, hjust=0.9),
        axis.title = element_blank()) 

ggsave(plot = cg08033031_lm_bm_plot,
       filename="2022-10-07_CHILD_prenatal_no2_cg08033031_bm.png",
       path =  here("figures", 
                    "linear_regression", 
                    "prenatal"),
       width = 4.8, height = 4, units = "cm")
```

### mQTLdb cross check

Check whether CpGs that compose DMRs are annotated as potential mQTLs in mQTLdb.com. I copied the list of CpGs contained in DMRs into mQTLdb then saved the outputted data to compare whether the CpGs appear in hits here. 
```{r mqtl_check}

for(x in unique(combp_anno_df_es_sub_escutoff$dmr)){
  
  cgids <- combp_anno_df_es_sub_escutoff %>%
    subset(dmr == x) %>%
    pull(Name)
  print(cgids)
  
  print(ltm::cronbach.alpha(cb_betas_geno_cc_clean_df[cgids, ]))
}

# load mQTL data
mqtls <- read.csv(here("input_data",
                       "mQTLdb_lists", 
                       "CHILD_cordblood_dmr_cpgs_mqtldb.csv"))

# annotate individual cpgs within DMRs as a potential mqtl or not
combp_anno_df_es_sub_escutoff_mqtls <- combp_anno_df_es_sub_escutoff %>%
  mutate(mqtl = ifelse(Name %in% (mqtls %>% 
                                    distinct(CpG) %>% 
                                    pull(CpG)), TRUE, FALSE))

# write out annotation for dmrs
write.csv(combp_anno_df_es_sub_escutoff_mqtls,
          here("output_data",
               "linear_regression",
               "2022-10-13_prenatal_no2_combp_annotation_sigDMRs_mQTL_annotation.csv"))
```


### Table 1 of prenatal model

"Table 1" of prenatal model
```{r prenatal_table1}
library(table1)

# get cell type estimates (not pcs)
cb_cell_estimates <- child_fscbc_ecc2_cb$prop
dim(cb_cell_estimates)

# subset out duplicates
cb_cell_estimates_sub <- cb_cell_estimates %>% 
  subset(rownames(.) %in% paste(cb_pdata$Sentrix_ID, cb_pdata$Sentrix_Position, sep = "_"))
dim(cb_cell_estimates_sub)

# rename rows
identical(rownames(cb_cell_estimates_sub),
          paste(cb_pdata$Sentrix_ID, cb_pdata$Sentrix_Position, sep = "_")) # TRUE
rownames(cb_cell_estimates_sub) <- cb_pdata$Sample_Label

# subset for participants with complete data
cb_cell_estimates_sub_cc <- cb_cell_estimates_sub %>%
  subset(rownames(.) %in% covariates_geno_ctpc_tech_cc$sample_id)
dim(cb_cell_estimates_sub_cc) # 128 6

# check order

identical(rownames(cb_cell_estimates_sub_cc), 
          as.character(covariates_geno_ctpc_tech_cc$sample_id)) # TRUE


# bind cell type estimates with covariate info
covariates_geno_ctpc_tech_cc_cbmc <- cbind(covariates_geno_ctpc_tech_cc, 
                                           cb_cell_estimates_sub_cc)

# table 1
table1(~sex + gestational_age + no2_prenatal_avg_2019v + 
         maternal_education_length +  maternal_race + 
         prenatal_maternal_smoke + CD4T + CD8T + nRBC + Bcell + Mono + NK,
       data = covariates_geno_ctpc_tech_cc_cbmc)

# prenatal no2 quartiles
quantile(covariates_geno_ctpc_tech_cc_cbmc$no2_prenatal_avg_2019v)
```






## Persistence of prenatal DNAm changes at age one

First we need to prepare the age one data for the linear model. Then we need to calculate the number of SVs needed for this model. 

### Load and clean age one data 

Get age one pdata
```{r y1_pdata}
#subset out age one pdata
y1_pdata <- subset(pdata_nodups, Tissue == "P")

#check size
dim(y1_pdata) # 145 17
```


Get age one betas
```{r y1_betas}
#subset out PBMC betas 
y1_betas <- betas[ ,colnames(betas) %in% rownames(y1_pdata)]

#check if age one betas and pdata are in same order
identical(colnames(y1_betas), rownames(y1_pdata)) # TRUE

#rename colums with sample label (5 dig num)
colnames(y1_betas) <- y1_pdata$Sample_Label

dim(y1_betas) # 145
```


Load age one cell type PCs
```{r y1_celltypePCs}
#load age one PBMC estimates
load(here("output_data", 
          "deconvolution", 
          "2022-09-28_PBMC_PCs.RData"))

# load deconvolution for later use - remove probes 
load(here("output_data",
          "deconvolution", 
          "2022-09-28_cbmc_pbmc_deconvolution.Rdata"))

y1_pcs <- y1_ilr$scores

#rename columns of PBMCs for clarity
colnames(y1_pcs) <- paste("pbmc", colnames(y1_pcs), sep=".")

#rename row of cell type PCs to sample label (5 digit num)
identical(rownames(y1_pcs), rownames(y1_pdata)) #true
rownames(y1_pcs) <- y1_pdata$Sample_Label

#subset cell type PCs for complete cases
y1_pcs_sub <- y1_pcs[rownames(y1_pcs) %in% covariates_geno_ctpc_tech_cc$sample_id,]
dim(y1_pcs_sub) # 128

# add age one PBMCs to covariate data frame
covariates_geno_ctpc_tech_cc_y1 <- cbind(covariates_geno_ctpc_tech_cc, y1_pcs_sub)
```


Add age one batch variables to covariate data frame
```{r y1_data_prep}
#add age one batch variables to covariate matrix
y1_pdata$y1_run <- as.factor(y1_pdata$Run) 
y1_pdata$y1_row  <- as.factor(substr(y1_pdata$Sentrix_Position,1,3)) 
y1_pdata$y1_row <- as.numeric(y1_pdata$y1_row)
y1_pdata$y1_chip <- as.factor(y1_pdata$Sentrix_ID)

covariates_geno_ctpc_tech_cc_y1 <- cbind(covariates_geno_ctpc_tech_cc_y1, 
                                         y1_pdata[y1_pdata$Sample_Label %in% covariates_geno_ctpc_tech_cc_y1$sample_id, c("y1_run", "y1_row", "y1_chip")])
```


Subset for complete cases in age one data
```{r y1_complete_cases}
#subset covariates with age one cell type PCs for linear regression
covariates_geno_ctpc_tech_cc_y1_sub <- covariates_geno_ctpc_tech_cc_y1 %>%
  subset(complete.cases(no2_prenatal_avg_2019v, sex, maternal_smoking_y1,
                        maternal_education_length,
                        pbmc.Comp.1 , pbmc.Comp.2, pbmc.Comp.3,
                        genotyping.PC1, genotyping.PC2, genotyping.PC3))

#check size
dim(covariates_geno_ctpc_tech_cc_y1_sub) # 124

#subset age one betas for samples in the complete case cord blood betas 
y1_betas_cc <- y1_betas[,colnames(y1_betas) %in%
                          as.character(covariates_geno_ctpc_tech_cc_y1_sub$sample_id)]

#check size
dim(y1_betas_cc) # 424644    124

#remove probes that were used for PBMC cell type predict from age one betas
y1_betas_cc_sub <- 
  y1_betas_cc[!rownames(y1_betas_cc) %in%
                rownames(child_fscbc_ecc2_y1$probelist_deconvo_coeffEsts), ]

#check size
dim(y1_betas_cc_sub) # 424079    124
# 565 probes removed

#check that covariates and betas in same order
identical(colnames(y1_betas_cc_sub), 
          as.character(covariates_geno_ctpc_tech_cc_y1_sub$sample_id)) # TRUE
```

Add sample name (sentrix id and row) to y1 data
```{r add_sentrix_info_y1}
# check if any y1 one samples duplicated
y1_pdata$Sample_Label %>% duplicated() %>% sum()

# subset for covariate data
y1_pdata_sub <- y1_pdata %>%
  subset(Sample_Label %in% covariates_geno_ctpc_tech_cc_y1_sub$sample_id)

# check order
identical(as.numeric(y1_pdata_sub$Sample_Label), 
          as.numeric(covariates_geno_ctpc_tech_cc_y1_sub$sample_id)) # TRUE
# check order
covariates_geno_ctpc_tech_cc_y1_sub$y1_sample_name <- rownames(y1_pdata_sub)

```

### Model matrix

Create model matrix for age one analysis
```{r model_matrix}
# create model matrix
y1_mod <- model.matrix(~ no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                         maternal_education_length +
                         pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3,
                       data=covariates_geno_ctpc_tech_cc_y1_sub)

# check size
dim(y1_mod) # 124 11
```

### Surrogate variable analysis to account for technical variables

Calculate the number of SVs needed for age one model.
```{r y1_sv_analysis}

# create null matrix
# this matrix does not include any of our covariates 
# this means that our covariates are "protected" and SVs will not capture them
y1_mod0 <- model.matrix(~ sex + maternal_smoking_y1 + maternal_education_length +
                          pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                        data=covariates_geno_ctpc_tech_cc_y1_sub)


################
# sva analysis #
################

##################################
# numsv function internal to sva #
##################################

#determine the number of required SVs
#this is numsv function code pulled out

set.seed(1234)

#this is what is fed into this function from the sva function
dat = y1_betas_cc_sub
mod = y1_mod
mod0 = y1_mod0
#method="be" # this is the default and what is called by numsv
#vfilter=NULL # we dont apply a filter and its not used for be method
B=20


# initialize some stuff
# get the residual matrix (the remaining variance)
n <- ncol(dat)
m <- nrow(dat)
H <- mod %*% solve(t(mod) %*% mod) %*% t(mod)
res <- dat - t(H %*% t(dat))

# run svd and get some eigenvalues
uu <- svd(res)
# get the num of "sig" eigenvalues
ndf <- min(m, n) - ceiling(sum(diag(H)))
# get percent variance explained by each eigenvalue
dstat <- uu$d[1:ndf]^2/sum(uu$d[1:ndf]^2)
# initialize null distribution matrix
dstat0 <- matrix(0, nrow = B, ncol = ndf)

# set seed here since we are calling sample
# create the null distribution
for (i in 1:B) {
  # this part of the code removes all relationships in the 
  res0 <- t(apply(res, 1, sample, replace = FALSE))
  res0 <- res0 - t(H %*% t(res0))
  # calculate svs on this
  uu0 <- svd(res0)
  # bind their explained variance in null matrix
  dstat0[i, ] <- uu0$d[1:ndf]^2/sum(uu0$d[1:ndf]^2)
  print(i) # print out the iteration we are on
}

# determine the number of svs required here based on pvalues
psv <- rep(1, n)
for (i in 1:ndf) {
  psv[i] <- mean(dstat0[, i] >= dstat[i])
}
for (i in 2:ndf) {
  psv[i] <- max(psv[(i - 1)], psv[i])
}

# this is the variable holding the number of SVs required 
# number is each to all the svs with a pval less than/equal to 0.1
nsv <- sum(psv <= 0.1)


####################################
# code from the main SVA function  #
####################################

n.sv = nsv # from code above (run internally when sva called)
B = 5 # number of iterations

# dat =  betas # already initialized
# mod = mod  # already initialized
# mod0 = mod0  # already initialized 
# controls = NULL # we dont include controls - only for supervised method
# method = "irw" # this isnt actually used but would be called by sva
# vfilter = NULL # we dont set this when running sva
# numSVmethod = "be" # this is the default called by sva (code above)


###################################
# irwsva function internal to sva #
###################################

# this code is really want does all the work in sva
# we use the irw method so this is the code for that method
# there are also supervised and two step methods (not used by us)

# irw function takes input from sva, so we have already defined below vars
# dat = dat 
# mod = mod
# mod0 = mod0
# n.sv = n.sv
# B = B

# intialize variables again
n <- ncol(dat)
m <- nrow(dat)
Id <- diag(n)
resid <- dat %*% (Id - mod %*% solve(t(mod) %*% mod) %*% t(mod))
uu <- eigen(t(resid) %*% resid)
vv <- uu$vectors
ndf <- n - dim(mod)[2]
pprob <- rep(1, m)
one <- rep(1, n)
Id <- diag(n)
df1 <- dim(mod)[2] + n.sv
df0 <- dim(mod0)[2] + n.sv
rm(resid)

cat(paste("Iteration (out of", B, "):"))
for (i in 1:B) {
  mod.b <- cbind(mod, uu$vectors[, 1:n.sv])
  mod0.b <- cbind(mod0, uu$vectors[, 1:n.sv])
  ptmp <- f.pvalue(dat, mod.b, mod0.b)
  pprob.b <- (1 - edge.lfdr(ptmp))
  mod.gam <- cbind(mod0, uu$vectors[, 1:n.sv])
  mod0.gam <- cbind(mod0)
  ptmp <- f.pvalue(dat, mod.gam, mod0.gam)
  pprob.gam <- (1 - edge.lfdr(ptmp))
  pprob <- pprob.gam * (1 - pprob.b)
  dats <- dat * pprob
  dats <- dats - rowMeans(dats)
  uu <- eigen(t(dats) %*% dats)
  cat(paste(i, " "))
}

# this is what is normally returned by sva
# problem is that it doesnt include the diagonal matrix we want
sv_y1 = svd(dats)$v[, 1:n.sv, drop = FALSE] 
svobj_y1 <- list(sv = sv_y1, pprob.gam = pprob.gam, pprob.b = pprob.b, 
                 n.sv = n.sv)

# to get all info (including diagonal)
svobj_all_y1 <- svd(dats)

# populate rownames with sample ids
colnames(svobj_all_y1$v) <- covariates_geno_ctpc_tech_cc_y1_sub$sampleid


# save 
save(sv_y1, file = here("output_data", 
                        "linear_regression", 
                        "2022-10-05_ageone_svs_no_pss6m.RData"))

###############
#scree of SVs #
###############

# from elodie portales-casamar

# heat scree prepartion
vars <- svobj_all_y1[['d']]^2
vars <- vars/sum(vars)
names(vars) <- paste0('SV', 1:length(vars))
xp <- 1:svobj_y1$n.sv
varsdf <- as.data.frame(vars)
varsdf$svnum <- rep(1:length(vars))

# heat scree ggplot!! 
y1_sv_scree <- ggplot(varsdf[1:20,], aes(x=svnum, y=vars*100)) +
  geom_col() +
  theme_bw() +
  labs(x="Surrogate Variables", y = "Percentage of Variance")

y1_sv_scree

ggsave(plot = y1_sv_scree, 
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal", 
                       "2022-10-07_y1_sv_heat_scree.png"),
       height = 5, width = 15, units = "cm")


######################
# correlation matrix #
######################

# add svs to covariates
covariates_geno_ctpc_tech_cc_y1_sub <- cbind(covariates_geno_ctpc_tech_cc_y1_sub, sv_y1)

# rename columns with svs
colnames(covariates_geno_ctpc_tech_cc_y1_sub)[204:222] <-
  paste("y1sv",colnames(covariates_geno_ctpc_tech_cc_y1_sub)[204:222], sep="")


# early exp and pn_18 are the same thing. 
# exclude early_exp from correlation matrices
corr_y1 <- corr.mat(covariates_geno_ctpc_tech_cc_y1_sub %>% 
                      dplyr::select(no2_prenatal_avg_2019v, 
                                    sex, maternal_smoking_y1,
                                    maternal_education_length,
                                    pbmc.Comp.1, pbmc.Comp.2, pbmc.Comp.3,
                                    genotyping.PC1, genotyping.PC2,
                                    genotyping.PC3, y1_chip, y1_run, y1_row, 
                                    y1sv1:y1sv19) %>%
                      rename("Average prenatal NO2" = no2_prenatal_avg_2019v,
                             Sex = sex,
                             "Year one maternal smoking" = maternal_smoking_y1,
                             "Maternal education (years)" = maternal_education_length,
                             "PBMC PC1" = pbmc.Comp.1, "PBMC PC2" = pbmc.Comp.2,
                             "PBMC PC3" = pbmc.Comp.3, 
                             Chip = y1_chip, Run = y1_run, Row = y1_row,
                             SV1 = y1sv1, SV2 = y1sv2, SV3 = y1sv3,
                             SV4 = y1sv4, SV5 = y1sv5, SV6 = y1sv6,
                             SV7 = y1sv7, SV8 = y1sv8, SV9 = y1sv9,
                             SV10 = y1sv10, SV11 = y1sv11, SV12 = y1sv12,
                             SV13 = y1sv13, SV14 = y1sv14, SV15 = y1sv15,
                             SV16 = y1sv16, SV17 = y1sv17, SV18 = y1sv18,
                             SV19 = y1sv19))

pval_y1 <- pval.mat(covariates_geno_ctpc_tech_cc_y1_sub %>% 
                      dplyr::select(no2_prenatal_avg_2019v, 
                                    sex, maternal_smoking_y1,
                                    maternal_education_length,
                                    pbmc.Comp.1, pbmc.Comp.2, pbmc.Comp.3,
                                    genotyping.PC1, genotyping.PC2,
                                    genotyping.PC3, y1_chip, y1_run, y1_row, 
                                    y1sv1:y1sv19) %>%
                      rename("Average prenatal NO2" = no2_prenatal_avg_2019v,
                             Sex = sex,
                             "Year one maternal smoking" = maternal_smoking_y1,
                             "Maternal education (years)" = maternal_education_length,
                             "PBMC PC1" = pbmc.Comp.1, "PBMC PC2" = pbmc.Comp.2,
                             "PBMC PC3" = pbmc.Comp.3, 
                             Chip = y1_chip, Run = y1_run, Row = y1_row,
                             SV1 = y1sv1, SV2 = y1sv2, SV3 = y1sv3,
                             SV4 = y1sv4, SV5 = y1sv5, SV6 = y1sv6,
                             SV7 = y1sv7, SV8 = y1sv8, SV9 = y1sv9,
                             SV10 = y1sv10, SV11 = y1sv11, SV12 = y1sv12,
                             SV13 = y1sv13, SV14 = y1sv14, SV15 = y1sv15,
                             SV16 = y1sv16, SV17 = y1sv17, SV18 = y1sv18,
                             SV19 = y1sv19))


svcorr_y1 <- corr.plot(pval = pval_y1, 
                       corr = corr_y1, 
                       label.sig = T,
                       shape = 8,shape.size = 1,
                       axis.text.size = 8, x.axis.text.angle = 90) +
  scale_y_discrete(position = "right") +
  theme(legend.position = "none",
        panel.grid = element_blank())
svcorr_y1

ggsave(plot = svcorr_y1,
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal",
                       "2022-10-07_y1_sv_correlations.png"),
       height = 15,
       width = 15, 
       units = "cm")


######################################
# heat scree plot of svs against pcs #
######################################

#convert raw betas to mvals
mvals <- lumi::beta2m(y1_betas_cc_sub)
max(mvals)
min(mvals)

#pca 
uncor.dat <- t(scale(t(mvals)))
PCA_full<-princomp(na.omit(uncor.dat))
Loadings<-as.data.frame(unclass(PCA_full$loadings))
vars <- PCA_full$sdev^2
Importance <-vars/sum(vars)

#Specify which covariates are categorical and/or continuous
meta_categorical <- 
  covariates_geno_ctpc_tech_cc_y1_sub[,c("sex", "maternal_smoking_y1",
                                         "y1_run", "y1_chip")] 
meta_continuous <- 
  covariates_geno_ctpc_tech_cc_y1_sub[,c("no2_preg_entire",
                                         "maternal_education_length", 
                                         "pss_6mos", "csed_6mos","y1_row",
                                         "pbmc.PC1", "pbmc.PC2", "pbmc.PC3",
                                         "pbmc.PC4",
                                         "genotyping.PC1", "genotyping.PC2",
                                         "genotyping.PC3", "sv1", "sv2", "sv3",
                                         "sv4", "sv5", "sv6", "sv7", "sv8",
                                         "sv9", "sv10", "sv11", "sv12", "sv13",
                                         "sv14", "y1_row")]

# Specify the number of PCs you want shown 
# pick enough to view most of the variance
Num<-10

# Designate what order you want the variables to appear (continuous variables rbinded to categorical variables in function)
Order<-c(1:40)

#Apply function on PCA results of uncorrected data, pulls in the meta data and beta values from above
heat_scree_plot(Loadings = Loadings, 
                Importance = Importance, 
                Num = Num, 
                Order = Order,
                Categorical = meta_categorical,
                Continuous = meta_continuous)
```




### Combined cell type PCs across two timepoint

```{r combined_cellpcs}
# load cord blood and age one blood cell type estimates
load(here("output_data",
          "deconvolution",
          "2022-09-28_cbmc_pbmc_deconvolution.Rdata"))

# pull out cell type proportion estimates'
y1_counts <- as.data.frame(child_fscbc_ecc2_y1$prop)
cb_counts <- as.data.frame(child_fscbc_ecc2_cb$prop)

# create new row in age one blood for nRBCs - set to 0
y1_counts$nRBC <- 0
# relocate nRBC column to match structure of cb data
y1_counts <- y1_counts %>% 
  relocate(nRBC, .after = NK)

# create columns for study ID in cord blood data
# check order of pdata and cell counts
identical(rownames(cb_counts), rownames(cb_pdata)) # FALSE
# remove samples in cell counts not present in pdata
cb_counts_sub <- cb_counts %>%
  subset(rownames(.) %in% rownames(cb_pdata))
# check order of pdata and cell counts
identical(rownames(cb_counts_sub), rownames(cb_pdata)) # TRUE
cb_counts_sub$Sample_Label <- cb_pdata$Sample_Label

# create columns for study ID in age one blood data
# check order of pdata and cell counts
identical(rownames(y1_counts), rownames(y1_pdata))
y1_counts$Sample_Label <- y1_pdata$Sample_Label


# subset age one cell counts for samples with cord blood cell counts
y1_counts_sub <- y1_counts %>%
  subset(Sample_Label %in% cb_counts_sub$Sample_Label)

dim(cb_counts_sub)
dim(y1_counts_sub)
identical(cb_counts_sub$Sample_Label, y1_counts_sub$Sample_Label) # TRUE

# combine cord blood and year one blood cell types into one data frame
comb_counts <- rbind(cb_counts_sub, y1_counts_sub)

# run isometric log transformation and robust PCA
comb_counts_ilr <- suppressWarnings(princomp(comb_counts[,c(1:6)], 
                                             covmat = MASS::cov.rob(comb_counts[,c(1:6)]), 
                                             cor = FALSE))

# scree plot of variance
var_explained = comb_counts_ilr$sdev^2 /sum(comb_counts_ilr$sdev^2)
ggplot(var_explained %>% as.data.frame(), aes(y=., x=(c(1:6)))) + geom_col()
cumsum(var_explained) 
# use first 3 PCs in linear mixed models
#    Comp.1    Comp.2    Comp.3    Comp.4    Comp.5    Comp.6 
# 0.6419671 0.7946265 0.9135715 0.9588014 0.9993992 1.0000000

comb_counts_pcs <- as.data.frame(comb_counts_ilr$scores)
dim(comb_counts_pcs) # 288
dim(covariates_geno_ctpc_tech_cc_y1_sub) #124

# subset combined cell counts for participants that have both birth and age one data
identical(rownames(comb_counts), rownames(comb_counts_pcs)) # TRUE
comb_counts_pcs$Sample_Label <- comb_counts$Sample_Label

comb_counts_pcs_cc <- comb_counts_pcs %>%
  subset(Sample_Label %in% covariates_geno_ctpc_tech_cc_y1_sub$sample_id)
dim(comb_counts_pcs_cc) # 248
```





### Surrogate variable analysis to account for technical variables in combined cord blood and age one blood DNAm data


```{r combined_dnam_data}
# subset for samples with data at both time points
cb_betas_geno_cc_sub <- cb_betas_geno_cc[, c(colnames(cb_betas_geno_cc) %in%
                                               colnames(y1_betas_cc_sub))]

# check columns are in the same order 
identical(colnames(cb_betas_geno_cc_sub), 
          colnames(y1_betas_cc_sub)) # TRUE

dim(cb_betas_geno_cc_sub)
dim(y1_betas_cc_sub)

# subset for cpgs shared between data
cb_betas_geno_cc_sub_cpgs <- cb_betas_geno_cc_sub[rownames(y1_betas_cc_sub), ]

dim(cb_betas_geno_cc_sub_cpgs)
dim(y1_betas_cc_sub)

# rename columns
colnames(cb_betas_geno_cc_sub_cpgs) <-
  paste("cb_", colnames(cb_betas_geno_cc_sub_cpgs), sep ="")

colnames(y1_betas_cc_sub) <-
  paste("y1_", colnames(y1_betas_cc_sub), sep ="")

comb_betas <- cbind(cb_betas_geno_cc_sub_cpgs,y1_betas_cc_sub)
```


Add combined cell type pcs to covariate data
```{r add_comb_pcs_to_y1_covariate}


 cg_df <- rbind(cbind(covariates_geno_ctpc_tech_cc_y1_sub %>%
                         select(no2_prenatal_avg_2019v,
                                gestational_age,
                                prenatal_maternal_smoke, sex, 
                                maternal_education_length, 
                                genotyping.PC1, genotyping.PC2, genotyping.PC3, 
                                sample_id, cb_sample_name,
                                chip, run, row) %>%
                         dplyr::rename("no2" = no2_prenatal_avg_2019v,
                                       "smoking" = prenatal_maternal_smoke,
                                       "sample_name" = cb_sample_name),
                       timepoint = "birth"),
                 cbind(covariates_geno_ctpc_tech_cc_y1_sub %>%
                         dplyr::select(no2_prenatal_avg_2019v, 
                                       gestational_age,
                                       maternal_smoking_y1, sex, 
                                       maternal_education_length,
                                       genotyping.PC1, genotyping.PC2, genotyping.PC3, 
                                       sample_id, y1_sample_name, 
                                       y1_chip, y1_row, y1_run)  %>%
                         dplyr::rename("no2" = no2_prenatal_avg_2019v,
                                       "smoking" = maternal_smoking_y1,
                                       "sample_name" = y1_sample_name,
                                       "chip" = y1_chip, "run" = y1_run,
                                       "row" = y1_row), 
                       timepoint = "ageone")) %>%
    mutate(timepoint = factor(timepoint, levels =c("birth", "ageone"))) %>%
    merge(x = ., y = comb_counts_pcs_cc, by.x = "sample_name", by.y = "row.names")

# reorder data frame to match order of betas  
cg_df_order <- cg_df %>%
  arrange(timepoint, sample_id)

# check that the order is the same
# 124 samples repeated over 2 time points
identical(as.numeric(cg_df_order$sample_id[1:124]), 
          str_sub(colnames(comb_betas)[1:124], start = 4, end = 8) %>% as.numeric())

# 
# 
# # subset birth time point for pcs
# comb_counts_pcs_cc_birth <- comb_counts_pcs_cc %>%
#   subset(rownames(.) %in% covariates_geno_ctpc_tech_cc_y1_sub$cb_sample_name)
# dim(comb_counts_pcs_cc_birth)
# 
# # rename columns
# colnames(comb_counts_pcs_cc_birth)[1:6] <-
#   paste("Birth", colnames(comb_counts_pcs_cc_birth)[1:6], sep="_")
# 
# # merge with covariates
# covariates_geno_ctpc_tech_cc_y1_sub <- merge(x = covariates_geno_ctpc_tech_cc_y1_sub,
#                                              y = comb_counts_pcs_cc_birth,
#                                              by.x = "sample_id",
#                                              by.y = "Sample_Label")
# 
# # subset age one time point for pcs
# comb_counts_pcs_cc_age_one <- comb_counts_pcs_cc %>%
#   subset(rownames(.) %in% covariates_geno_ctpc_tech_cc_y1_sub$y1_sample_name)
# dim(comb_counts_pcs_cc_age_one) # 124
# 
# colnames(comb_counts_pcs_cc_age_one)[1:6] <-
#   paste("AgeOne", colnames(comb_counts_pcs_cc_age_one)[1:6], sep="_")
# 
# # mergre with covariates
# covariates_geno_ctpc_tech_cc_y1_sub <- merge(x = covariates_geno_ctpc_tech_cc_y1_sub,
#                                              y = comb_counts_pcs_cc_age_one,
#                                              by.x = "sample_id",
#                                              by.y = "Sample_Label")
```

Create model matrix for combined betas
```{r model_matrix}

# create model matrix
comb_mod <- model.matrix(~ timepoint*(no2 + gestational_age) + 
                           sex + maternal_education_length + smoking +
                           genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
                           Comp.1 + Comp.2 + Comp.3, 
                         data=cg_df_order)

# check size
dim(comb_mod) # 248 15
```

Calculate the number of SVs needed for age one model.
```{r comb_sv_analysis}

# create null matrix
# this matrix does not include any of our covariates 
# this means that our covariates are "protected" and SVs will not capture them
comb_mod0 <- model.matrix(~ timepoint*(gestational_age) + 
                           sex + maternal_education_length + smoking +
                           genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
                           Comp.1 + Comp.2 + Comp.3, 
                         data=cg_df_order)


################
# sva analysis #
################

##################################
# numsv function internal to sva #
##################################

#determine the number of required SVs
#this is numsv function code pulled out

set.seed(1234)

#this is what is fed into this function from the sva function
dat = as.matrix(comb_betas)
mod = comb_mod
mod0 = comb_mod0
#method="be" # this is the default and what is called by numsv
#vfilter=NULL # we dont apply a filter and its not used for be method
B=20


# initialize some stuff
# get the residual matrix (the remaining variance)
n <- ncol(dat)
m <- nrow(dat)
H <- mod %*% solve(t(mod) %*% mod) %*% t(mod)
res <- dat - t(H %*% t(dat))

# run svd and get some eigenvalues
uu <- svd(res)
# get the num of "sig" eigenvalues
ndf <- min(m, n) - ceiling(sum(diag(H)))
# get percent variance explained by each eigenvalue
dstat <- uu$d[1:ndf]^2/sum(uu$d[1:ndf]^2)
# initialize null distribution matrix
dstat0 <- matrix(0, nrow = B, ncol = ndf)

# set seed here since we are calling sample
# create the null distribution
for (i in 1:B) {
  # this part of the code removes all relationships in the 
  res0 <- t(apply(res, 1, sample, replace = FALSE))
  res0 <- res0 - t(H %*% t(res0))
  # calculate svs on this
  uu0 <- svd(res0)
  # bind their explained variance in null matrix
  dstat0[i, ] <- uu0$d[1:ndf]^2/sum(uu0$d[1:ndf]^2)
  print(i) # print out the iteration we are on
}

# determine the number of svs required here based on pvalues
psv <- rep(1, n)
for (i in 1:ndf) {
  psv[i] <- mean(dstat0[, i] >= dstat[i])
}
for (i in 2:ndf) {
  psv[i] <- max(psv[(i - 1)], psv[i])
}

# this is the variable holding the number of SVs required 
# number is each to all the svs with a pval less than/equal to 0.1
nsv <- sum(psv <= 0.1)


####################################
# code from the main SVA function  #
####################################

n.sv = nsv # from code above (run internally when sva called)
B = 5 # number of iterations

# dat =  betas # already initialized
# mod = mod  # already initialized
# mod0 = mod0  # already initialized 
# controls = NULL # we dont include controls - only for supervised method
# method = "irw" # this isnt actually used but would be called by sva
# vfilter = NULL # we dont set this when running sva
# numSVmethod = "be" # this is the default called by sva (code above)


###################################
# irwsva function internal to sva #
###################################

# this code is really want does all the work in sva
# we use the irw method so this is the code for that method
# there are also supervised and two step methods (not used by us)

# irw function takes input from sva, so we have already defined below vars
# dat = dat 
# mod = mod
# mod0 = mod0
# n.sv = n.sv
# B = B

# intialize variables again
n <- ncol(dat)
m <- nrow(dat)
Id <- diag(n)
resid <- dat %*% (Id - mod %*% solve(t(mod) %*% mod) %*% t(mod))
uu <- eigen(t(resid) %*% resid)
vv <- uu$vectors
ndf <- n - dim(mod)[2]
pprob <- rep(1, m)
one <- rep(1, n)
Id <- diag(n)
df1 <- dim(mod)[2] + n.sv
df0 <- dim(mod0)[2] + n.sv
rm(resid)

cat(paste("Iteration (out of", B, "):"))
for (i in 1:B) {
  mod.b <- cbind(mod, uu$vectors[, 1:n.sv])
  mod0.b <- cbind(mod0, uu$vectors[, 1:n.sv])
  ptmp <- f.pvalue(dat, mod.b, mod0.b)
  pprob.b <- (1 - edge.lfdr(ptmp))
  mod.gam <- cbind(mod0, uu$vectors[, 1:n.sv])
  mod0.gam <- cbind(mod0)
  ptmp <- f.pvalue(dat, mod.gam, mod0.gam)
  pprob.gam <- (1 - edge.lfdr(ptmp))
  pprob <- pprob.gam * (1 - pprob.b)
  dats <- dat * pprob
  dats <- dats - rowMeans(dats)
  uu <- eigen(t(dats) %*% dats)
  cat(paste(i, " "))
}

# this is what is normally returned by sva
# problem is that it doesnt include the diagonal matrix we want
sv_comb = svd(dats)$v[, 1:n.sv, drop = FALSE] 
svobj_comb <- list(sv = sv_comb, pprob.gam = pprob.gam, pprob.b = pprob.b, 
                   n.sv = n.sv)

# to get all info (including diagonal)
svobj_all_comb <- svd(dats)

# populate rownames with sample ids
colnames(svobj_all_comb$v) <- covariates_geno_ctpc_tech_cc_y1_sub$sampleid


# save 
save(sv_comb, file = here("output_data", 
                          "linear_regression", 
                          "2022-11-16_combined_cb_y1_svs_with_interaction.RData"))

###############
#scree of SVs #
###############

# from elodie portales-casamar

# heat scree prepartion
vars <- svobj_all_comb[['d']]^2
vars <- vars/sum(vars)
names(vars) <- paste0('SV', 1:length(vars))
xp <- 1:svobj_comb$n.sv
varsdf <- as.data.frame(vars)
varsdf$svnum <- rep(1:length(vars))

# heat scree ggplot!! 
comb_sv_scree <- ggplot(varsdf[1:20,], aes(x=svnum, y=vars*100)) +
  geom_col() +
  theme_bw() +
  labs(x="Surrogate Variables", y = "Percentage of Variance")

comb_sv_scree

ggsave(plot = comb_sv_scree, 
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal", 
                       "2022-10-07_y1_sv_heat_scree.png"),
       height = 5, width = 15, units = "cm")


######################
# correlation matrix #
######################

# add svs to covariates
cg_df_order_svs <- cbind(cg_df_order, sv_comb)

# rename columns with svs
colnames(cg_df_order_svs)[22:117] <-
  paste("comb_sv",colnames(cg_df_order_svs)[22:117], sep="")

source(here("scripts", "2021-08-30_corr_mat_funs.R"))

# early exp and pn_18 are the same thing. 
# exclude early_exp from correlation matrices
corr_comb <- corr.mat(cg_df_order_svs %>% 
                        dplyr::select(no2, gestational_age, sex, 
                                      maternal_education_length, smoking,
                                      genotyping.PC1, genotyping.PC2,
                                      genotyping.PC3, Comp.1, Comp.2, Comp.3,
                                      chip, run, row, 
                                      comb_sv1:comb_sv15)) 

pval_comb <- pval.mat(cg_df_order_svs %>% 
                        dplyr::select(no2, gestational_age, sex, 
                                      maternal_education_length, smoking,
                                      genotyping.PC1, genotyping.PC2,
                                      genotyping.PC3, Comp.1, Comp.2, Comp.3,
                                      chip, run, row, 
                                      comb_sv1:comb_sv15)) 


svcorr_comb <- corr.plot(pval = pval_comb, 
                       corr = corr_comb, 
                       label.sig = T, 
                       shape = 8,shape.size = 1,
                       axis.text.size = 8, x.axis.text.angle = 90) +
  scale_y_discrete(position = "right") +
  theme(legend.position = "none",
        panel.grid = element_blank())
svcorr_comb

ggsave(plot = svcorr_comb,
       filename = here("figures", 
                       "linear_regression", 
                       "prenatal",
                       "2022-11-16_comb_cb_y1_sv_correlations.png"),
       height = 15,
       width = 15, 
       units = "cm")
```


### Linear mixed models regressions examining persistence at significant CpGs/DMRs

Examine persistence of prenatal changes at age one.
```{r persistence}

library(lme4)
library(lmerTest)

# check order of data and dnam
identical(as.numeric(str_sub(colnames(comb_betas), 4, 8)), 
          as.numeric(cg_df_order_svs_cpgs$sample_id)) # true

cg_df_order_svs_cpgs <- 
  cbind(cg_df_order_svs, 
        t(comb_betas[combp_anno_df_es_sub_escutoff %>% pull(Name), ]))

lmer_results <- NULL
lmer_pvals <- NULL
lmer_coeffs <- NULL

for(i in 118:268){
  
  cg_df_order_svs_cpgs$dnam <- cg_df_order_svs_cpgs[,i] 
  cg_lmer <- lmer(dnam ~  timepoint*(no2 + gestational_age)  + sex +  
                    maternal_education_length + smoking +
                    genotyping.PC1 + genotyping.PC2 + genotyping.PC3 + 
                    Comp.1 + Comp.2 + Comp.3 + comb_sv1 + comb_sv2 +
                    comb_sv3 + comb_sv4 + (1 |sample_id), 
                  data = cg_df_order_svs_cpgs)
  
  lmer_results <- c(lmer_results, cg_lmer)
  lmer_pvals <- cbind(lmer_pvals, summary(cg_lmer)$coefficients[c(1,2,3,18),5])
  lmer_coeffs <- cbind(lmer_coeffs, summary(cg_lmer)$coefficients[c(1,2,3,18),1])
  
}

names(lmer_results) <- combp_anno_df_es_sub_escutoff %>% pull(Name)

lmer_pvals <- lmer_pvals %>% 
  t() %>% 
  as.data.frame() %>% 
  rename_with(., ~ paste("pval", .x, sep="_"))

lmer_coeffs <- lmer_coeffs %>% 
  t() %>% 
  as.data.frame() %>% 
  rename_with(., ~ paste("coef", .x, sep="_"))

test <- lmer_pvals %>% 
  cbind(., lmer_coeffs) %>%
  as.data.frame() %>%
  #rename("pval" = 1) %>%
  mutate(dmr = combp_anno_df_es_sub_escutoff$dmr,
         cpg = combp_anno_df_es_sub_escutoff$Name,
         singular = lapply(lmer_results, isSingular) %>% do.call(rbind,.),
         pval_no2_adj = p.adjust(pval_no2, method = "bonferroni"),
         pval_timepointageone_adj = p.adjust(pval_timepointageone, method = "bonferroni"),
         `pval_timepointageone:no2_adj` = p.adjust(`pval_timepointageone:no2`, method = "bonferroni")) 
# 
# # test with cg10906729
# covariates_geno_ctpc_tech_cc_y1_sub$y1_cg12764003 <- y1_betas_cc_sub["cg12764003",] 
# 
# # linear regression
# y1_cg12764003_lm <- lm(y1_cg12764003 ~  
#                          no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
#                          maternal_education_length +
#                          pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
#                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
#                          y1sv1 + y1sv2 + y1sv3 + y1sv4 + y1sv5 + y1sv6 + y1sv7 + y1sv8 + y1sv9 +
#                          y1sv10 + y1sv11 + y1sv12 + y1sv13 + y1sv14 + y1sv15 + y1sv16 + y1sv17 + 
#                          y1sv18 + y1sv19, 
#                        data=covariates_geno_ctpc_tech_cc_y1_sub)
# 
# # summary of lm
# summary(y1_cg12764003_lm) # pvalue = 0.063860
# # (Intercept)                0.7879325  0.0279074  28.234  < 2e-16 ***
# # no2_prenatal_avg_2019v    -0.0030214  0.0007943  -3.804 0.000253 ***
# 
# # pval_(Intercept) pval_no2_prenatal_avg_2019v
# # 9.134913e-05                0.0001500842
# # 
# #  coeff_(Intercept)      coeff_no2_prenatal_avg_2019v 
# #  0.8596572               -0.003790429
# 
# # test with cg02574073
# 
# # ggplot(data=NULL) +
# # +     geom_point(aes(y=covariates_geno_ctpc_tech_cc_y1_sub_cpgs$birth_cg02649248,
# # +                    x=covariates_geno_ctpc_tech_cc_y1_sub_cpgs$no2_prenatal_avg_2019v)) +
# # +     geom_point(aes(y=covariates_geno_ctpc_tech_cc_y1_sub_cpgs$age_one_cg02649248,
# # +                    x=covariates_geno_ctpc_tech_cc_y1_sub_cpgs$no2_prenatal_avg_2019v, col = "red"))
# 
# covariates_geno_ctpc_tech_cc_y1_sub$y1_cg02574073 <- y1_betas_cc_sub["cg02574073",] 
# 
# # linear regression
# y1_cg02574073_lm <- lm(y1_cg02574073 ~  
#                          no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
#                          maternal_education_length +
#                          pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
#                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
#                          y1sv1 + y1sv2 + y1sv3 + y1sv4 + y1sv5 + y1sv6 + y1sv7 + y1sv8 + y1sv9 +
#                          y1sv10 + y1sv11 + y1sv12 + y1sv13 + y1sv14 + y1sv15 + y1sv16 + y1sv17 + 
#                          y1sv18 + y1sv19, 
#                        data=covariates_geno_ctpc_tech_cc_y1_sub)
# 
# # summary of lm
# summary(y1_cg02574073_lm) # pvalue = 0.063860
# 
# 
# # test with cg05888755
# 
# covariates_geno_ctpc_tech_cc_y1_sub$y1_cg05888755 <- y1_betas_cc_sub["cg05888755",] 
# 
# # linear regression
# y1_cg05888755_lm <- lm(y1_cg05888755 ~  
#                          no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
#                          maternal_education_length +
#                          pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
#                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
#                          y1sv1 + y1sv2 + y1sv3 + y1sv4 + y1sv5 + y1sv6 + y1sv7 + y1sv8 + y1sv9 +
#                          y1sv10 + y1sv11 + y1sv12 + y1sv13 + y1sv14 + y1sv15 + y1sv16 + y1sv17 + 
#                          y1sv18 + y1sv19, 
#                        data=covariates_geno_ctpc_tech_cc_y1_sub)
# 
# # summary of lm
# summary(y1_cg05888755_lm) # pvalue = 0.063860
# 
# # test with cg01564268
# 
# covariates_geno_ctpc_tech_cc_y1_sub$y1_cg01564268 <- y1_betas_cc_sub["cg01564268",] 
# 
# # linear regression
# y1_cg01564268_lm <- lm(y1_cg01564268 ~  
#                          no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
#                          maternal_education_length +
#                          pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
#                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
#                          y1sv1 + y1sv2 + y1sv3 + y1sv4 + y1sv5 + y1sv6 + y1sv7 + y1sv8 + y1sv9 +
#                          y1sv10 + y1sv11 + y1sv12 + y1sv13 + y1sv14 + y1sv15 + y1sv16 + y1sv17 + 
#                          y1sv18 + y1sv19, 
#                        data=covariates_geno_ctpc_tech_cc_y1_sub)
# 
# # summary of lm
# summary(y1_cg01564268_lm) # pvalue = 0.063860
# 
# # test with cg04645039
# 
# covariates_geno_ctpc_tech_cc_y1_sub$y1_cg04645039 <- y1_betas_cc_sub["cg04645039",] 
# 
# # linear regression
# y1_cg04645039_lm <- lm(y1_cg04645039 ~  
#                          no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
#                          maternal_education_length +
#                          pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
#                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
#                          y1sv1 + y1sv2 + y1sv3 + y1sv4 + y1sv5 + y1sv6 + y1sv7 + y1sv8 + y1sv9 +
#                          y1sv10 + y1sv11 + y1sv12 + y1sv13 + y1sv14 + y1sv15 + y1sv16 + y1sv17 + 
#                          y1sv18 + y1sv19, 
#                        data=covariates_geno_ctpc_tech_cc_y1_sub)
# 
# # summary of lm
# summary(y1_cg04645039_lm) # pvalue = 0.063860
# 
# 
# # test with cg24534926
# 
# covariates_geno_ctpc_tech_cc_y1_sub$y1_cg24534926 <- y1_betas_cc_sub["cg24534926",] 
# 
# # linear regression
# y1_cg24534926_lm <- lm(y1_cg24534926 ~  
#                          no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
#                          maternal_education_length +
#                          pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
#                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
#                          y1sv1 + y1sv2 + y1sv3 + y1sv4 + y1sv5 + y1sv6 + y1sv7 + y1sv8 + y1sv9 +
#                          y1sv10 + y1sv11 + y1sv12 + y1sv13 + y1sv14 + y1sv15 + y1sv16 + y1sv17 + 
#                          y1sv18 + y1sv19, 
#                        data=covariates_geno_ctpc_tech_cc_y1_sub)
# 
# # summary of lm
# summary(y1_cg24534926_lm) # pvalue = 0.063860
# 
# # test with cg24534926
# 
# covariates_geno_ctpc_tech_cc_y1_sub$y1_cg05579037 <- y1_betas_cc_sub["cg05579037",] 
# 
# # linear regression
# y1_cg05579037_lm <- lm(y1_cg05579037 ~  
#                          no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
#                          maternal_education_length +
#                          pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
#                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
#                          y1sv1 + y1sv2 + y1sv3 + y1sv4 + y1sv5 + y1sv6 + y1sv7 + y1sv8 + y1sv9 +
#                          y1sv10 + y1sv11 + y1sv12 + y1sv13 + y1sv14 + y1sv15 + y1sv16 + y1sv17 + 
#                          y1sv18 + y1sv19, 
#                        data=covariates_geno_ctpc_tech_cc_y1_sub)
# 
# # summary of lm
# summary(y1_cg05579037_lm) # pvalue = 0.063860
# 
# # test with cg05774699
# 
# covariates_geno_ctpc_tech_cc_y1_sub$y1_cg05774699 <- y1_betas_cc_sub["cg05774699",] 
# 
# # linear regression
# y1_cg05774699_lm <- lm(y1_cg05774699 ~  
#                          no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
#                          maternal_education_length +
#                          pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
#                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
#                          y1sv1 + y1sv2 + y1sv3 + y1sv4 + y1sv5 + y1sv6 + y1sv7 + y1sv8 + y1sv9 +
#                          y1sv10 + y1sv11 + y1sv12 + y1sv13 + y1sv14 + y1sv15 + y1sv16 + y1sv17 + 
#                          y1sv18 + y1sv19, 
#                        data=covariates_geno_ctpc_tech_cc_y1_sub)
# 
# # summary of lm
# summary(y1_cg05774699_lm) # pvalue = 0.063860
# 
# 
# # test with cg07049592
# 
# covariates_geno_ctpc_tech_cc_y1_sub$y1_cg07049592 <- y1_betas_cc_sub["cg07049592",] 
# 
# # linear regression
# y1_cg07049592_lm <- lm(y1_cg07049592 ~  
#                          no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
#                          maternal_education_length +
#                          pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
#                          genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
#                          y1sv1 + y1sv2 + y1sv3 + y1sv4 + y1sv5 + y1sv6 + y1sv7 + y1sv8 + y1sv9 +
#                          y1sv10 + y1sv11 + y1sv12 + y1sv13 + y1sv14 + y1sv15 + y1sv16 + y1sv17 + 
#                          y1sv18 + y1sv19, 
#                        data=covariates_geno_ctpc_tech_cc_y1_sub)
# 
# # summary of lm
# summary(y1_cg07049592_lm) # pvalue = 0.063860
```



```{r old_persistence}
# dmr 1
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean1 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 1) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean1_lm <- lm(y1_dmrmean1 ~  
                       no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                       maternal_education_length +
                       pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                       genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                     data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean1_lm) # pvalue = 0.2735

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean1_no2_me <- ggpredict(y1_dmrmean1_lm, 
                                terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean1" = predicted,
         "no2_prenatal_avg_2019v" = x)

# plot 
y1_dmrmean1_plot <- ggplot(covariates_geno_ctpc_tech_cc_y1_sub,
                           aes(x = no2_prenatal_avg_2019v, 
                               y= y1_dmrmean1)) + 
  geom_line(data = y1_dmrmean1_no2_me, 
            aes(x=no2_prenatal_avg_2019v, y = y1_dmrmean1),
            col="red", size=0.5, linetype="solid") +
  geom_ribbon(data = y1_dmrmean1_no2_me, 
              aes(x = no2_prenatal_avg_2019v, 
                  ymin = conf.low, ymax = conf.high), 
              alpha = .2) + 
  geom_point(alpha=0.6, col = "#254A90", size=1) + 
  theme_bw() +
  theme(panel.grid = element_blank(),
        axis.text = element_text(size=9),
        panel.border = element_blank(),
        axis.line = element_line(color ="black"),
        axis.title = element_blank()) 

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean1, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # 0.02900599

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean1_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # 0.02995307


# dmr 2
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean2 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 2) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean2_lm <- lm(y1_dmrmean2 ~  
                       no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                       maternal_education_length +
                       pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                       genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                     data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean2_lm) # pvalue = 0.1275

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean2_no2_me <- ggpredict(y1_dmrmean2_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean2" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean2, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # 0.06678491

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean2_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # 0.02047185


# dmr 3
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean3 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 3) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean3_lm <- lm(y1_dmrmean3 ~  
                       no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                       maternal_education_length +
                       pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                       genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                     data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean3_lm) # pvalue = 0.006483

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean3_no2_me <- ggpredict(y1_dmrmean3_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean3" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean3, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # -0.1627883

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean3_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # -0.04806303


# dmr 4
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean4 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 4) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean4_lm <- lm(y1_dmrmean4 ~  
                       no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                       maternal_education_length +
                       pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                       genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                     data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean4_lm) # pvalue = 0.0207

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean4_no2_me <- ggpredict(y1_dmrmean4_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean4" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean4, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # 0.2201343

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean4_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # 0.03650168


# dmr 5
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean5 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 5) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean5_lm <- lm(y1_dmrmean5 ~  
                       no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                       maternal_education_length +
                       pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                       genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                     data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean5_lm) # pvalue = 0.00335

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean5_no2_me <- ggpredict(y1_dmrmean5_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean5" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean5, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = -0.217798

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean5_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # -0.04602454


# dmr 6
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean6 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 6) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean6_lm <- lm(y1_dmrmean6 ~  
                       no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                       maternal_education_length +
                       pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                       genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                     data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean6_lm) # pvalue = 0.000324 

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean6_no2_me <- ggpredict(y1_dmrmean6_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean6" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean6, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = -0.3621699

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean6_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # -0.0731885


# plot dmr 6 individual cpgs


# dmr 7
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean7 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 7) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean7_lm <- lm(y1_dmrmean7 ~  
                       no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                       maternal_education_length +
                       pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                       genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                     data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean7_lm) # pvalue = 0.0071  

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean7_no2_me <- ggpredict(y1_dmrmean7_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean7" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean7, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = 0.1904801

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean7_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # 0.03597229


# dmr 8
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean8 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 8) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean8_lm <- lm(y1_dmrmean8 ~  
                       no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                       maternal_education_length +
                       pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                       genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                     data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean8_lm) # pvalue = 0.0482   

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean8_no2_me <- ggpredict(y1_dmrmean8_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean8" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean8, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = -0.1914197

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean8_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # -0.1431807


# dmr 9
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean9 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 9) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean9_lm <- lm(y1_dmrmean9 ~  
                       no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                       maternal_education_length +
                       pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                       genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                     data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean9_lm) # pvalue = 0.0299    

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean9_no2_me <- ggpredict(y1_dmrmean9_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean9" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean9, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = 0.2372674

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean9_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # 0.1011445


# dmr 10
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean10 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 10) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean10_lm <- lm(y1_dmrmean10 ~  
                        no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                        maternal_education_length +
                        pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                        genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                      data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean10_lm) # pvalue = 0.276    

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean10_no2_me <- ggpredict(y1_dmrmean10_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean10" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean10, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = -0.1385322

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean10_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # -0.03584968


# dmr 11
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean11 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 11) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean11_lm <- lm(y1_dmrmean11 ~  
                        no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                        maternal_education_length +
                        pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                        genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                      data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean11_lm) # pvalue = 0.04978     

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean11_no2_me <- ggpredict(y1_dmrmean11_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean11" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean11, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = -0.1092971

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean11_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # -0.0113446


# dmr 12
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean12 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 12) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean12_lm <- lm(y1_dmrmean12 ~  
                        no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                        maternal_education_length +
                        pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                        genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                      data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean12_lm) # pvalue = 0.128619      

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean12_no2_me <- ggpredict(y1_dmrmean12_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean12" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean12, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = -0.002083642

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean12_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # -0.0123446


# dmr 13
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean13 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 13) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean13_lm <- lm(y1_dmrmean13 ~  
                        no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                        maternal_education_length +
                        pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                        genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                      data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean13_lm) # pvalue = 0.000829       

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean13_no2_me <- ggpredict(y1_dmrmean13_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean13" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean13, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = 0.2716437

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean13_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # 0.07593806


# dmr 14
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean14 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 14) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean14_lm <- lm(y1_dmrmean14 ~  
                        no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                        maternal_education_length +
                        pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                        genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                      data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean14_lm) # pvalue = 0.07349        

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean14_no2_me <- ggpredict(y1_dmrmean14_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean14" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean14, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = -0.2660887

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean14_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # -0.09085841


# dmr 15
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean15 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 15) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean15_lm <- lm(y1_dmrmean15 ~  
                        no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                        maternal_education_length +
                        pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                        genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                      data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean15_lm) # pvalue = 0.1158        

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean15_no2_me <- ggpredict(y1_dmrmean15_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean15" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean15, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = 0.1401879

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean15_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # 0.05257575


# dmr 16
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean16 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 16) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean16_lm <- lm(y1_dmrmean16 ~  
                        no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                        maternal_education_length +
                        pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                        genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                      data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean16_lm) # pvalue = 0.0635     

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean16_no2_me <- ggpredict(y1_dmrmean16_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean16" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean16, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = 0.176961

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean16_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # 0.08468159


# dmr 18
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean18 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 18) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean18_lm <- lm(y1_dmrmean18 ~  
                        no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                        maternal_education_length +
                        pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                        genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                      data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean18_lm) # pvalue =  0.000144     

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean18_no2_me <- ggpredict(y1_dmrmean18_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean18" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean18, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = -0.2861818

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean18_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # -0.07930344


# dmr 19
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean19 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 19) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean19_lm <- lm(y1_dmrmean19 ~  
                        no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                        maternal_education_length +
                        pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                        genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                      data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean19_lm) # pvalue = 0.0603      

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean19_no2_me <- ggpredict(y1_dmrmean19_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean19" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean19, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = 0.1772126

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean19_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # 0.08921469


# dmr 20
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean20 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 20) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean20_lm <- lm(y1_dmrmean20 ~  
                        no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                        maternal_education_length +
                        pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                        genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                      data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean20_lm) # pvalue = 0.0428       

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean20_no2_me <- ggpredict(y1_dmrmean20_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean20" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean20, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = -0.07730894

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean20_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # -0.03880709


# dmr 21
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean21 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 21) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean21_lm <- lm(y1_dmrmean21 ~  
                        no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                        maternal_education_length +
                        pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                        genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                      data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean21_lm) # pvalue = 0.00617       

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean21_no2_me <- ggpredict(y1_dmrmean21_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean21" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean21, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = 0.2467306

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean21_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # 0.04360987


# dmr 23
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean23 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 23) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean23_lm <- lm(y1_dmrmean23 ~  
                        no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                        maternal_education_length +
                        pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                        genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                      data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean23_lm) # pvalue = 0.105134         

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean23_no2_me <- ggpredict(y1_dmrmean23_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean23" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean23, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = 0.1895335

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean23_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # 0.01340701


# dmr 24
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean24 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 24) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean24_lm <- lm(y1_dmrmean24 ~  
                        no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                        maternal_education_length +
                        pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                        genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                      data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean24_lm) # pvalue = 0.0410          

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean24_no2_me <- ggpredict(y1_dmrmean24_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean24" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean24, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = -0.1195407

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean24_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # -0.0326896


# dmr 25
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean25 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 25) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean25_lm <- lm(y1_dmrmean25 ~  
                        no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                        maternal_education_length +
                        pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                        genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                      data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean25_lm) # pvalue = 0.40356         

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean25_no2_me <- ggpredict(y1_dmrmean25_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean25" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean25, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = 0.1511256

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean25_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 # 0.01290266




# dmr 27
covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean27 <- 
  y1_betas_cc_sub[combp_anno_df_es_sub_escutoff %>% subset(dmr == 27) %>% pull(Name),] %>% 
  colMeans()

# linear regression
y1_dmrmean27_lm <- lm(y1_dmrmean27 ~  
                        no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                        maternal_education_length +
                        pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                        genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
                      data=covariates_geno_ctpc_tech_cc_y1_sub)

# summary of lm
summary(y1_dmrmean27_lm) # pvalue =  0.1785        

# estimate marginal effects of prenatal NO2 exposure on DNAm
y1_dmrmean27_no2_me <- ggpredict(y1_dmrmean27_lm, terms = "no2_prenatal_avg_2019v") %>%
  as.data.frame() %>% 
  rename("y1_dmrmean27" = predicted,
         "no2_prenatal_avg_2019v" = x)

# correlation 
cor(covariates_geno_ctpc_tech_cc_y1_sub$y1_dmrmean27, 
    covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v,
    method="pearson") # pvalue = -0.1361932

# effect size
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

summary(y1_dmrmean27_lm)$coefficients["no2_prenatal_avg_2019v", "Estimate"]*range_y1 #  -0.02625926

``` 

### Examine changes in effect size at top 100 CpGs from EWAS over the first year of life 


```{r top100_effect_size_persistence_test}

# get top 100 cord blood cpgs 
cb100 <- cb_pvals_tech %>% 
  subset(effect_size > 0.025 | effect_size < -0.025) %>%
  arrange(pval_no2_prenatal_avg_2019v) %>%
  slice_head(n=100)


# get birth and age one dnam for top 100 cord blood cpgs
cb_top100_betas <- 
  cb_betas_geno_cc_clean_df[rownames(cb100), 
                            as.character(covariates_geno_ctpc_tech_cc_y1_sub$sample_id)] %>%
  t() %>%
  as.data.frame() %>%
  rename_with(., ~ paste("cb", .x, sep="_"))

y1_top100_betas <- 
  y1_betas_cc_sub[rownames(cb100), 
                  as.character(covariates_geno_ctpc_tech_cc_y1_sub$sample_id)] %>%
  t() %>%
  as.data.frame() %>%
  rename_with(., ~ paste("y1", .x, sep="_"))

covariates_geno_ctpc_tech_cc_y1_sub_top100cpgs <- 
  cbind(covariates_geno_ctpc_tech_cc_y1_sub, cb_top100_betas, y1_top100_betas)



lmer_results_cpgs <- NULL
lmer_pvals_cpgs <- NULL
lmer_coeffs_cpgs <- NULL

for(i in 224:323){
  
  
  cg_df <- rbind(cbind(covariates_geno_ctpc_tech_cc_y1_sub_top100cpgs %>%
                         select(no2_prenatal_avg_2019v, all_of(i), 
                                prenatal_maternal_smoke, sex, 
                                maternal_education_length, 
                                genotyping.PC1, genotyping.PC2, genotyping.PC3, 
                                Comp.1, Comp.2, Comp.3, Comp.4, 
                                sv1, sv2, sv3, sv4, sv5, sv6, sv7, sv8, sv9,
                                sv10, sv11, sv12, sv13, sv14, sv15, sv16, sv17) %>%
                         mutate(sv18 = 0, sv19 = 0, 
                                sample_id = covariates_geno_ctpc_tech_cc_y1_sub_top100cpgs$sample_id) %>% 
                         rename("no2" = no2_prenatal_avg_2019v,
                                "cg" = 2,
                                "smoking" = prenatal_maternal_smoke,
                                "PC1" = Comp.1, "PC2" = Comp.2 ,
                                "PC3" = Comp.3, "PC4" = Comp.4),
                       timepoint = "birth",
                       age = scale(covariates_geno_ctpc_tech_cc_y1_sub_top100cpgs$gestational_age, 
                                   scale = T)),
                 cbind(covariates_geno_ctpc_tech_cc_y1_sub_top100cpgs %>%
                         select(no2_prenatal_avg_2019v, all_of(i+100), 
                                maternal_smoking_y1, sex, 
                                maternal_education_length,
                                genotyping.PC1, genotyping.PC2, genotyping.PC3, 
                                pbmc.Comp.1, pbmc.Comp.2, pbmc.Comp.3, pbmc.Comp.4, 
                                y1sv1, y1sv2, y1sv3, y1sv4, y1sv5, y1sv6, y1sv7, y1sv8, y1sv9,
                                y1sv10, y1sv11, y1sv12, y1sv13, y1sv14, y1sv15, y1sv16, y1sv17, 
                                y1sv18, y1sv19, sample_id)  %>%
                         mutate(pbmc.Comp.4 = 0) %>% 
                         rename("no2" = no2_prenatal_avg_2019v,
                                "cg" = 2,
                                "smoking" = maternal_smoking_y1,
                                "PC1" = pbmc.Comp.1, "PC2" = pbmc.Comp.2,
                                "PC3" = pbmc.Comp.3, "PC4" = pbmc.Comp.4,
                                "sv1" = y1sv1, "sv2" = y1sv2, "sv3" = y1sv3, 
                                "sv4" = y1sv4, "sv5" = y1sv5, "sv6" = y1sv6, 
                                "sv7" = y1sv7, "sv8" = y1sv8, "sv9" = y1sv9, 
                                "sv10" = y1sv10, "sv11" = y1sv11, 
                                "sv12" = y1sv12, "sv13" = y1sv13, 
                                "sv14" = y1sv14, "sv15" = y1sv15, 
                                "sv16" = y1sv16, "sv17" = y1sv17, 
                                "sv18" = y1sv18, "sv19" = y1sv19), 
                       timepoint = "ageone",
                       age = 0)) %>%
    mutate(timepoint = factor(timepoint, levels =c("birth", "ageone")))
  
  cg_lmer_cpgs <- lmer(cg ~ no2*timepoint + sex + age + maternal_education_length + smoking +
                         genotyping.PC1 + genotyping.PC2 + genotyping.PC3 +
                         PC1 + PC2 + PC3 + PC4 + 
                         sv1 + sv2 + sv3 + sv4 + sv5 + sv6 + sv7 + sv8 + sv9 + sv10 +
                         sv11 + sv12 + sv13 + sv14 + sv15 + sv16 + sv17 + sv18 + 
                         sv19 + (1 |sample_id), data = cg_df)
  
  lmer_results_cpgs <- c(lmer_results_cpgs, cg_lmer_cpgs)
  lmer_pvals_cpgs <- cbind(lmer_pvals_cpgs, summary(cg_lmer_cpgs)$coefficients[c(1,2,3,34),5])
  lmer_coeffs_cpgs <- cbind(lmer_coeffs_cpgs, summary(cg_lmer_cpgs)$coefficients[c(1,2,3,34),1])
}



lmer_pvals_cpgs <- lmer_pvals_cpgs %>% 
  t() %>% 
  as.data.frame() %>% 
  rename_with(., ~ paste("pval", .x, sep="_"))

lmer_coeffs_cpgs <- lmer_coeffs_cpgs %>% 
  t() %>% 
  as.data.frame() %>% 
  rename_with(., ~ paste("coef", .x, sep="_"))

lmer_pvals_cpgs %>% 
  cbind(., lmer_coeffs_cpgs) %>%
  as.data.frame() %>%
  mutate(cpg = cb_pvals_tech %>% 
           subset(effect_size > 0.025 | effect_size < -0.025) %>%
           arrange(pval_no2_prenatal_avg_2019v) %>%
           slice_head(n=100) %>% 
           pull(Name),
         cb_effectsize = cb_pvals_tech %>% 
           subset(effect_size > 0.025 | effect_size < -0.025) %>%
           arrange(pval_no2_prenatal_avg_2019v) %>%
           slice_head(n=100) %>% 
           pull(effect_size),
         y1_effectsize = pb100_pvals_df$effect_size, 
         pval_no2_adj = pval_no2*100,
         pval_timepointageone_adj = pval_timepointageone*100,
         `pval_no2:timepointageone_adj` = `pval_no2:timepointageone`*100,
         singular = (lapply(lmer_results_cpgs, isSingular) %>% do.call(rbind,.)) ) %>%
  View() 


```

Examine how effect size changes over first year of life at top 100 cord blood cpgs.
```{r top100_effect_size_persistence}

# get top 100 cord blood cpgs 
cb100 <- cb_pvals_tech %>% 
  subset(effect_size > 0.025 | effect_size < -0.025) %>%
  arrange(pval_no2_prenatal_avg_2019v) %>%
  slice_head(n=100)


# get age one dnam for top 100 cord blood cpgs
pb100_betas <- y1_betas_cc_sub[rownames(cb100),]

# check size
dim(pb100_betas)

# check order
identical(rownames(cb100), rownames(pb100_betas)) # TRUE

# transpose for db calculations
pb100_betas_t <- t(pb100_betas)

# calculate delta beta for each of the 100 age one cpgs
pb100_pvals <- 
  pbapply(pb100_betas_t, 2, function(x) {
    
    combined <- cbind(covariates_geno_ctpc_tech_cc_y1_sub, x)
    
    #run the linear regression on this cpg using all 5 imputed covariate data sets
    fit <- lm(x ~ no2_prenatal_avg_2019v + sex + maternal_smoking_y1 +
                maternal_education_length +
                pbmc.Comp.1 + pbmc.Comp.2 + pbmc.Comp.3 +
                genotyping.PC1 + genotyping.PC2 + genotyping.PC3, 
              data=combined)
    
    #get summary of lm fit
    #this includes coefficients and pvalues
    #coerce into df for manipulation
    #add in terms (rownames) as column for annotation
    fit_sum <- summary(fit)
    fit_sum_coeff <- as.data.frame(fit_sum$coefficients)
    fit_sum_coeff$term <- rownames(fit_sum_coeff)
    
    #this data frame has rows for each linear regression variable, and columns for statistics
    #want a dataframe of 1 row and many columns (one for each variable with each statistic type)
    #do this by using the spread function for each statistic
    #add cpg id onto dataframe for pvalues (to keep track of information)
    #then cbind dataframes together
    
    #coefficient
    estimate <- spread(fit_sum_coeff[,c("term", "Estimate")], term, Estimate)
    colnames(estimate) <- paste("estimate", colnames(estimate), sep=".")
    
    #std error
    std.error <- spread(fit_sum_coeff[,c("term", "Std. Error")], term, 'Std. Error')
    colnames(std.error) <- paste("std.error", colnames(std.error), sep=".")
    
    #t statistics
    statistic <- spread(fit_sum_coeff[,c("term", "t value")], term, 't value')
    colnames(statistic) <- paste("t.val", colnames(statistic), sep=".")
    
    #pvalue
    p.value <- spread(fit_sum_coeff[,c("term", "Pr(>|t|)")], term, 'Pr(>|t|)')
    colnames(p.value) <- paste("p.value", colnames(p.value), sep=".")
    
    #add in cpgid for annotation 
    p.value$cpg <- colnames(x)
    
    #cbind into wide dataframe
    widepool <- cbind(estimate, std.error, statistic, p.value)
    
    #return the wide dataframe
    #these will be do.call(rbind()) together after
    return(widepool)
    
  })

# rbind list to data frame 
pb100_pvals_df <- do.call(rbind, pb100_pvals)

# calculate effect size at age one
range_y1 = max(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v) -
  min(covariates_geno_ctpc_tech_cc_y1_sub$no2_prenatal_avg_2019v)

pb100_pvals_df$effect_size <- pb100_pvals_df$estimate.no2_prenatal_avg_2019v * range_y1

pb100_pvals_df$p.value.no2_prenatal_avg_2019v_adj <- 
  p.adjust(pb100_pvals_df$p.value.no2_prenatal_avg_2019v, method = "bonferroni")

# check that rownames match between cord blood and peripheral blood
identical(rownames(cb100), rownames(pb100_pvals_df)) # TRUE

# bind age one effect size to cord blood data frame
cb100$effect_size_y1 <- pb100_pvals_df$effect_size
cb100$p.value.no2_prenatal_avg_2019v_y1 <- pb100_pvals_df$p.value.no2_prenatal_avg_2019v

# calculate the change in db between birth and age one
cb100$delta_effect_size <- cb100$effect_size_y1 - cb100$effect_size

# change to long format for plotting 
cb100_long <- cb100 %>% 
  mutate(dnam_dir = ifelse(effect_size >= 0, 
                           "Higher DNAm at birth", 
                           "Lower DNAm at birth"),
         effect_size_cb = effect_size) %>%
  pivot_longer(cols = c(effect_size_cb, effect_size_y1)) 

# plot the cpgs exhibited lower 
top100_es_lower <- ggplot(cb100_long %>% 
                            subset(dnam_dir == "Lower DNAm at birth"), 
                          aes(x=name, y = value, group = Name)) +
  geom_point(alpha = 0.5, color = "#254A90", fill = "#254A90", size = 1, shape = 25) +
  geom_line(alpha = 0.5, color = "#254A90") +
  facet_wrap(~"Lower cord blood \nDNAm associated \nwith prenatal NO2") +
  scale_x_discrete(labels = c("Birth", "Age one"), name = "Time point") +
  scale_y_continuous(name = "Effect size (% DNA methylation)",
                     limits = c(-0.25, 0.10),
                     breaks = seq(-0.25, 0.10, 0.05)) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        panel.border = element_blank(),
        axis.line = element_line(color = "black"),
        strip.background = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank(),
        legend.position = "none") +
  geom_signif(comparisons = list(c("effect_size_cb", "effect_size_y1")),
              colour = "black",
              y_position = 0.05,
              annotations = "p = 2.2e-16",
              vjust=-0.25,
              textsize = 3.5) 

top100_es_higher <- ggplot(cb100_long %>% 
                             subset(dnam_dir == "Higher DNAm at birth"), 
                           aes(x=name, y = value, group = Name)) +
  geom_point(alpha = 0.5, color = "#254A90", fill = "#254A90", size = 1, shape = 25) +
  geom_line(alpha = 0.5, color = "#254A90") +
  facet_wrap(~"Higher cord blood \nDNAm associated \nwith prenatal NO2") +
  scale_x_discrete(labels = c("Birth", "Age one"), name = "Time point") +
  theme_bw() +
  theme(panel.grid = element_blank(),
        panel.border = element_blank(),
        axis.line = element_line(color = "black"),
        strip.background = element_blank(),
        axis.text = element_text(size=9),
        axis.title = element_blank(),
        legend.position = "none") +
  geom_signif(comparisons = list(c("effect_size_cb", "effect_size_y1")),
              y_position = 0.25, 
              colour = "black",
              annotations = "p = 1.2e-14",
              vjust=-0.25,
              textsize = 3.5) +
  scale_y_continuous(limits = c(-0.05, 0.30),
                     breaks = seq(-0.05, 0.30, 0.05),
                     name= "")

top100_es <- cowplot::plot_grid(top100_es_lower, top100_es_higher, 
                                ncol=2, nrow=1)


ggsave(plot = top100_es,
       filename = here("figures",
                       "linear_regression", 
                       "prenatal",
                       "2022-10-18_top100_es_change.png"),
       height = 7, width = 8, units = "cm")


# one sample paired t-test for decreased dnam
# alternative = "greater" is the alternative that x has a larger mean than y.
t.test(x=cb100 %>% subset(effect_size<=0) %>% pull(effect_size_y1),
       y=cb100 %>% subset(effect_size<=0) %>% pull(effect_size),
       paired = TRUE, alternative = "greater")
# p-value = 2.2e-16

# one sample pairwd t-test for increased dnam
t.test(x=cb100 %>% subset(effect_size>0) %>% pull(effect_size_y1),
       y=cb100 %>% subset(effect_size>0) %>% pull(effect_size),
       paired = TRUE, alternative = "less")
# p-value = 1.145e-14
```

### Table 1 for persistence model


```{r age_one_persistence_table1}
library(table1)

# get cell type estimates (not pcs)
y1_cell_estimates <- child_fscbc_ecc2_y1$prop
dim(y1_cell_estimates) # 145

# rename rows
identical(rownames(y1_cell_estimates_sub),
          paste(y1_pdata$Sentrix_ID, y1_pdata$Sentrix_Position, sep = "_")) # TRUE
rownames(y1_cell_estimates_sub) <- y1_pdata$Sample_Label

# subset for participants with complete data
y1_cell_estimates_sub_cc <- y1_cell_estimates_sub %>%
  subset(rownames(.) %in% covariates_geno_ctpc_tech_cc_y1_sub$sample_id)
dim(y1_cell_estimates_sub_cc) # 124 6

# check order
identical(rownames(y1_cell_estimates_sub_cc), 
          as.character(covariates_geno_ctpc_tech_cc_y1_sub$sample_id)) # TRUE

# bind cell type estimates with covariate info
covariates_geno_ctpc_tech_cc_y1_sub_pbmc <- cbind(covariates_geno_ctpc_tech_cc_y1_sub, 
                                                  y1_cell_estimates_sub_cc)

# table 1
table1(~sex + no2_prenatal_avg_2019v + 
         maternal_education_length +  maternal_race + 
         maternal_smoking_y1 + CD4T + CD8T  + Bcell + Mono + NK,
       data = covariates_geno_ctpc_tech_cc_y1_sub_pbmc)

# prenatal no2 quartiles
quantile(covariates_geno_ctpc_tech_cc_y1_sub_pbmc$no2_prenatal_avg_2019v)

```


## Prenatal and postnatal air pollution exposure boxplots

```{r airpollution_boxplots}

# prenatal no2
ggplot(covariates_geno_ctpc_tech_cc, aes(y=no2_preg_entire, x=studycentre, fill=studycentre)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(height=0, width=0.25, alpha=0.8) +
  theme_bw() +
  scale_fill_manual(values=c("#254A90", "#508FC2", "#92C5EB", "white")) +
  labs(x="Study Centre", y=bquote("Prenatal" ~NO[2]~ "(ppb)")) +
  theme(panel.grid = element_blank(),
        legend.position = "none",
        axis.text = element_text(size=14),
        axis.title = element_text(size=16))

# year one no2
ggplot(covariates_geno_ctpc_tech_cc_y1_sub, aes(y=no2_y1_entire, x=studycentre,fill=studycentre)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(height=0, width=0.25, alpha=0.8) +
  scale_fill_manual(values=c("#254A90", "#508FC2", "#92C5EB", "white")) +
  labs(x="Study Centre", y=bquote("Year one" ~NO[2]~ "(ppb)")) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        legend.position = "none",
        axis.text = element_text(size=14),
        axis.title = element_text(size=16))

# Pm2.5 B
ggplot(covariates_geno_ctpc_tech, aes(y=PM25DALbYY_01, x=studycentre, fill=studycentre)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(height=0, width=0.25, alpha=0.8) +
  scale_fill_manual(values=c("#254A90", "#508FC2", "#92C5EB", "white")) +
  labs(x="Study Centre", y=bquote(~PM[2.5]~ "(ppb)")) +
  theme_bw() +
  theme(panel.grid = element_blank(),
        legend.position = "none",
        axis.text = element_text(size=14),
        axis.title = element_text(size=16))
```
