---
title: "Age one linear regression using limma with complete cases"
author: "SL"
date: "12/09/2020"
output: html_document
---

*Purpose*: The purpose of this script is to DNAm changes in year one PBMCs associated with year one air pollution exposure (proxied by no2). This script uses a subtraction method to "cancel" out effects of prenatal exposure, allowing the investigation of year one specific changes. 


Load required libraries.
```{r libraries}
library(tidyverse)
library(limma)
library(here)
library(pbapply)
library(piecewiseSEM)
```


Load betas
```{r covariates}
load(file=here("output_data", "preprocessed_data",
               "2021-01-14_CHILD_preprocessed_betas_pdata_annotation.Rdata"))

#subset out cord blood and year one pData
#cord blood pdata
cb_pdata <- subset(pdata_nodups, Tissue == "C")
dim(cb_pdata) # 144  17

#year one pdatga
y1_pdata <- subset(pdata_nodups, Tissue == "P")
dim(y1_pdata) # 145  17

#year one data has one individual that is missing birth data
#remove this individual
y1_pdata_sub <- y1_pdata[y1_pdata$Sample_Label %in% cb_pdata$Sample_Label, ]
dim(y1_pdata_sub) # 144  17
```


Subset out cord blood and age one betas
```{r sub_betas}
#subset pData into CBMCs (cord blood)
#these children have data at both time points
cb_betas <- betas[,colnames(betas) %in% rownames(cb_pdata)]
#check size of cord blood data
dim(cb_betas) # 424644    144
min(cb_betas) # 0.002973941
max(cb_betas) # 0.9980669

#check if cord blood DNAm samples are in same order in cord blood pdata
identical(colnames(cb_betas), rownames(cb_pdata)) #true
#rename columns of cb_betas
colnames(cb_betas) <- cb_pdata$Sample_Label


#subset out cordblood samples from beta matrix
y1_betas <- betas[,colnames(betas) %in% rownames(y1_pdata_sub)]
#check size of year one data
dim(y1_betas) # 424644    144
min(y1_betas) #0.002650193
max(y1_betas) #0.9983506

#check if cord blood DNAm samples are in same order in cord blood pdata
identical(colnames(y1_betas), rownames(y1_pdata_sub)) #true
#rename columns of cb_betas
colnames(y1_betas) <- y1_pdata_sub$Sample_Label

#check if cord blood samples and year one samples are in same order
identical(colnames(cb_betas), colnames(y1_betas))
```


Subtract cord blood DNAm data from year one DNAm data. 
```{r diff_betas}
#check if cord blood samples and year one samples are in same order
identical(colnames(cb_betas), colnames(y1_betas))

#subtract cord blood DNAm from year one DNAm ("difference betas")
diff_betas <- y1_betas - cb_betas
#check size of difference betas
dim(diff_betas) # 424644    144
min(diff_betas) # -0.9793121 
max(diff_betas) # 0.9797916
```


Load covariate data
```{r covariates}
load(file=here("output_data", "covariates", "2020-11-12_CHILD_covariates_na_with_means.Rdata"))
```


Subset covariates and pdata for samples in diff_betas
```{r sub_covariates_pdata}
#subset covariates
covariates_sub <- 
  covariates_nameans[as.character(covariates_nameans$sampleid) %in% colnames(diff_betas), ]

#check order
identical(as.character(covariates_sub$sampleid), colnames(diff_betas)) #true
dim(covariates_sub) #144

#check order
identical(as.character(cb_pdata$Sample_Label), colnames(diff_betas)) #true
dim(cb_pdata) #144

#check order
identical(as.character(y1_pdata_sub$Sample_Label), colnames(diff_betas)) #true
dim(y1_pdata_sub) #144
```


Load combined cell type PCs (CBMC and PBMC together).
```{r load_celltype_PCs}
load(here("output_data", "deconvolution",
                 "2021-02-04_CHILD_CBMC_PBMC_PCs.Rdata"))

#check size
dim(ccc_pcs) #144 11

#rename columns for clarity
colnames(ccc_pcs) <- paste("celltype.", colnames(ccc_pcs), sep="")

#check that cell type pcs and covariates are in the same order
identical(as.character(covariates_sub$sampleid), rownames(ccc_pcs)) #true

#combine cell type pcs and covariates
covariates_ctpcs <- cbind(covariates_sub, ccc_pcs)
dim(covariates_ctpcs)
```


Load genotyping PCs and subset out REEGLE PCs (genotyping data contains PCs for a few thousand participants).
```{r genotypingPCS}
genotyping <- read.csv(here("input_data", "genotyping", "CHILD_subjects_Qced_first10PCS_Duan.csv"))

#make rownames sample names then rmove this row
rownames(genotyping) <- genotyping$FID
genotyping <- subset(genotyping, select=-c(FID, IID))

genosub <- genotyping[rownames(genotyping) %in% cb_pdata$Sample_Label,]
#missing genotyping data for 14 REEGLE children?

#rename columns of genotyping data for clarity
colnames(genosub) <- paste("genotyping.", colnames(genosub), sep="")

#put rows in numerical order according to sample id
genosub_order <- genosub[order(rownames(genosub)), ]
```


Subset covariates, betas, cb_pdata, and y1_pdata for samples that also have genotyping.
```{r sub_on_geno}
#subset cb_pdata
cb_pdata_geno <- cb_pdata[as.character(cb_pdata$Sample_Label) %in% rownames(genosub_order), ]
#check size
dim(cb_pdata_geno) #131
#check order
identical(as.character(cb_pdata_geno$Sample_Label), rownames(genosub_order)) #true

#subset y1_pdata
y1_pdata_geno <- y1_pdata_sub[as.character(y1_pdata_sub$Sample_Label) %in% rownames(genosub_order), ]
#check size
dim(y1_pdata_geno) #131
#check order
identical(as.character(y1_pdata_geno$Sample_Label), rownames(genosub_order)) #true

#subset covariates
covariates_ctpcs_geno <- covariates_ctpcs[as.character(covariates_ctpcs$sampleid) %in% rownames(genosub_order), ]
#check size
dim(covariates_ctpcs_geno) #131
#check order
identical(as.character(covariates_ctpcs_geno$sampleid), rownames(genosub_order)) #true

#combine covariates with genotyping data
covariates_ctpcs_geno <- cbind(covariates_ctpcs_geno, genosub_order)

#subset betas
diff_betas_geno <- diff_betas[,colnames(diff_betas) %in% rownames(genosub_order)]
#check size
dim(diff_betas_geno) #131
#check order
identical(colnames(diff_betas_geno), rownames(genosub_order)) #true

#### add row run and chip to covariates

#check that covariates and dpata are in same order
identical(as.character(cb_pdata_geno$Sample_Label), 
          as.character(covariates_ctpcs_geno$sampleid)) #true

#create new column for technical variables in y1 pdata
y1_pdata_geno$y1_chip <- as.factor(y1_pdata_geno$Sentrix_ID)
y1_pdata_geno$y1_run <- as.factor(y1_pdata_geno$Run)
y1_pdata_geno$y1_row <- as.numeric(as.factor(substr(y1_pdata_geno$Sentrix_Position, 1, 3)))

y1_pdata_geno$cb_chip <- as.factor(cb_pdata_geno$Sentrix_ID)
y1_pdata_geno$cb_run <- as.factor(cb_pdata_geno$Run)
y1_pdata_geno$cb_row <- as.numeric(as.factor(substr(cb_pdata_geno$Sentrix_Position, 1, 3)))

#add technical columns of y1 pdata to covariates data
covariates_ctpcs_geno_tech <- cbind(covariates_ctpcs_geno, 
                                    y1_pdata_geno[,c("cb_run", "cb_row", "cb_chip",
                                                     "y1_run", "y1_row", "y1_chip")])
``` 




Linear regression using limma and complete cases
```{r limma}

##################
#Data preparation#
##################

#how many complete cases now?
sum(complete.cases(covariates_ctpcs_geno_tech %>% 
                     dplyr::select(no2_y1_entire, sex,  
                                   maternal_smoking_y1,  
                                   maternal_education_length, 
                                   pss_6mos,
                                   csed_6mos,
                                   genotyping.PC1, 
                                   genotyping.PC2,
                                   genotyping.PC3,
                                   celltype.PC1, celltype.PC2,
                                   celltype.PC3, celltype.PC4,
                                   celltype.PC5, celltype.PC6,
                                   celltype.PC7, celltype.PC8,
                                   celltype.PC9,
                                   cb_run, cb_row, 
                                   y1_row, y1_run))) #125 complete cases now


#create model matrix
diff_design <- 
  model.matrix(~no2_y1_entire + sex +
                 maternal_smoking_y1 + 
                 maternal_education_length +
                 pss_6mos + csed_6mos + 
                 genotyping.PC1 + genotyping.PC2 + 
                 genotyping.PC3 + 
                 celltype.PC1 + celltype.PC2+
                 celltype.PC3+ celltype.PC4+
                 celltype.PC5+ celltype.PC6+
                 celltype.PC7+ celltype.PC8+
                 celltype.PC9+
                 cb_run+ cb_row+ 
                 y1_row+ y1_run, 
               data=covariates_ctpcs_geno_tech)

#check size
dim(diff_design) # 125 26

#further subset covariates and beta values for complete cases
#subset covariates
covariates_ctpcs_geno_tech_cc <- 
  covariates_ctpcs_geno_tech[covariates_ctpcs_geno_tech %>% 
                             dplyr::select(no2_y1_entire, sex,  
                                           maternal_smoking_y1,  
                                           maternal_education_length, 
                                           pss_6mos,csed_6mos,
                                           genotyping.PC1, 
                                           genotyping.PC2,
                                           genotyping.PC3,
                                           celltype.PC1, celltype.PC2,
                                           celltype.PC3, celltype.PC4,
                                           celltype.PC5, celltype.PC6,
                                           celltype.PC7, celltype.PC8,
                                           celltype.PC9,
                                           cb_run, cb_row, 
                                           y1_row, y1_run) %>% complete.cases,] 

#check size
dim(covariates_ctpcs_geno_tech_cc) # 125 140 

#subset betas
diff_betas_geno_cc <- 
  diff_betas_geno[,colnames(diff_betas_geno) %in% covariates_ctpcs_geno_tech_cc$sampleid]

#check size
dim(diff_betas_geno_cc) # 424644    125

#remove betas that were used for cell type prediction
diff_betas_geno_cc_sub <- 
  diff_betas_geno_cc[!rownames(diff_betas_geno_cc) %in% rownames(cb_deconvolutionprobes$coefEsts) & !rownames(diff_betas_geno_cc) %in% rownames(y1_deconvolutionprobes$coefEsts), ]

#check size
dim(diff_betas_geno_cc_sub) #424079    125
#565 probes removed

#check that covariates and betas in same order
identical(colnames(diff_betas_geno_cc_sub), 
          as.character(covariates_ctpcs_geno_tech_cc$sampleid)) # TRUE

###############
#### LIMMA ####
###############

#fit methylation to no2
diff_fit <-  lmFit(diff_betas_geno_cc, diff_design)
diff_ebayes <- eBayes(diff_fit)

#add annotation to cb_ebayes
#pull annotation for cpgs contained in diff_betas
annotation_cpgs <- annotation[rownames(annotation) %in% rownames(diff_ebayes),]

#check order
identical(rownames(annotation_cpgs), rownames(diff_ebayes)) #false

#match
annotation_cpgs_order <- annotation_cpgs[rownames(diff_ebayes), ]

#recheck order
identical(rownames(annotation_cpgs_order), rownames(diff_ebayes)) #true

#pull out pval, coefficients and annotation
diff_pvals_tech <- cbind(as.data.frame(diff_ebayes$p.value), 
                  as.data.frame(diff_ebayes$coefficients),
                  as.data.frame(annotation_cpgs_order))

#rename columns to indicate pvalues vs coefficients 
colnames(diff_pvals_tech) <- c(paste("pval", colnames(diff_pvals_tech)[1:27], sep = "_"), 
                         paste("coeff", colnames(diff_pvals_tech)[28:54], sep = "_"),
                         colnames(diff_pvals_tech)[55:87])


diff_pvals_tech$adj_pval_no2_y1_entire <- p.adjust(diff_pvals_tech$pval_no2_y1_entire, method="BH")
```


Examine qqplot and pvalue histogram.
```{r qqplot_pval_histo}

########
#qqplot#
########

# use code from qqman::qq function 
# plot with ggplot to make it more aesthetically appealling 
pvector <- diff_pvals_tech$pval_no2_y1_entire

pvector2 <- pvector[!is.na(pvector) & !is.nan(pvector) & !is.null(pvector) & 
                      is.finite(pvector) & pvector < 1 & pvector > 0]

pval_qq <- as.data.frame(cbind(o = -log10(sort(pvector2, decreasing = FALSE)),
                               e = -log10(ppoints(length(pvector2)))))


#calculate lambda
chisq <- qchisq(1-pvector,1)
lambda <-  median(chisq)/qchisq(0.5,1)
 
ggplot(pval_qq, aes(x=e, y=o)) + 
  geom_point(col="#254A90", size=1) + 
  geom_abline(slope = 1, intercept = 0, col = "red", size=0.25) +
  labs(x = "Expected -log10(p-value)",
       y = "Observed -log10(p-value)") +
  annotate("text", x=4, y=1, label = paste("λ = ", round(lambda, 2)), size=3) +
  scale_x_continuous(breaks=c(0,2,4,6),
                     labels=c("-0.00", "-2.00", "-4.00", "-6.00")) +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size=5,angle=90, vjust = 0.5, hjust=1),
        axis.text.y = element_text(size=5),
        axis.title.x = element_text(size=7),
        axis.title.y = element_text(size=7)) +
    ggsave( filename="2021-01-21_CHILD_y1_qqplot.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression", "yearone"),
          width = 2,
          height = 2,
          units = "in",
          limitsize = TRUE)
 

###########        
#histogram#
###########

ggplot( data = diff_pvals_tech, aes(x=pval_no2_y1_entire)) +
  geom_histogram(bins=100, fill = "#254A90") +
  labs(y="Count", x="Unadjusted p-value") +
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size=6),
        axis.text.y = element_text(size=6),
        axis.title.x = element_text(size=8),
        axis.title.y = element_text(size=8)) +
    ggsave( filename="2021-01-21_CHILD_y1_pval_histogram.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression", "yearone"),
          width = 3,
          height = 2,
          units = "in",
          limitsize = TRUE)
 

```


Delta beta
```{r db}
range = max(covariates_ctpcs_geno_tech_cc$no2_y1_entire) -
    min(covariates_ctpcs_geno_tech_cc$no2_y1_entire)

diff_pvals_tech$db <- diff_pvals_tech$coeff_no2_y1_entire * range
```



Volcano plot
```{r volcano_db_tech}
#make new column to label sig points with cpgid
diff_pvals_tech$cpgid <- rownames(diff_pvals_tech)


ggplot(diff_pvals_tech, aes(x=as.numeric(db)*100 , y=-log10(as.numeric(pval_no2_y1_entire)))) +
  geom_point(alpha=0.4, col = "#254A90", size=1) +
  
  #nominal p value cut off
  geom_hline(yintercept = -log10(0.00001), linetype="dashed", size=0.25) + 
  
  #10% effect size cut-off
  geom_point(data=subset(diff_pvals_tech, 
                         (pval_no2_y1_entire < 0.00001 & db > 0.1) | 
                          (pval_no2_y1_entire < 0.00001 & db < -0.1)), 
    alpha=0.4, col = "red", size=1) + 
  geom_vline(xintercept = c(10, -10), linetype = "dashed", size=0.25) +

  # 5% effect size cut off
  geom_point(data=subset(diff_pvals_tech, 
                         (pval_no2_y1_entire < 0.00001 & db > 0.05) | 
                          (pval_no2_y1_entire < 0.00001 & db < -0.05)), 
    alpha=0.4, col = "red", size=1) + 
  geom_vline(xintercept = c(5, -5), linetype = "dashed", size=0.25) +
  # 2.5% effect size cut off
    geom_point(data=subset(diff_pvals_tech, 
                         (pval_no2_y1_entire < 0.00001 & db > 0.025) | 
                          (pval_no2_y1_entire < 0.00001 & db < -0.025)), 
    alpha=0.4, col = "red", size=1) + 
  geom_vline(xintercept = c(2.5, -2.5), linetype = "dashed", size=0.25) +

  scale_x_continuous(breaks=c(-20, -10, -5, -2.5, 0, 2.5, 5, 10, 20)) +
  labs(y="-log10(unadjusted p-value)", x = "Delta beta") +
  
  theme_bw() +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text.x = element_text(size=5, angle=90, vjust = 0.5, hjust=1),
        axis.text.y = element_text(size=5),
        axis.title.x = element_text(size=7),
        axis.title.y = element_text(size=7)) +
  
  ggsave( filename="2021-01-25_CHILD_y1_volcanoplot.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression", "yearone"),
          width = 3,
          height = 2,
          units = "in",
          limitsize = TRUE)

```



Plot nominally significant cpgs
```{r sig_cpg}

###################
#MMP23B cg24535439#
###################

covariates_ctpcs_geno_tech_cc$cg24535439 <- diff_betas_geno_cc[rownames(diff_betas_geno_cc)=="cg24535439", ]


cg24535439_lm <- lm(cg24535439  ~ no2_y1_entire + sex +
                 maternal_smoking_y1 + 
                 maternal_education_length +
                 pss_6mos + csed_6mos + 
                 genotyping.PC1 + genotyping.PC2 + 
                 genotyping.PC3 + 
                 celltype.PC1 + celltype.PC2+
                 celltype.PC3+ celltype.PC4+
                 celltype.PC5+ celltype.PC6+
                 celltype.PC7+ celltype.PC8+
                 celltype.PC9+
                 cb_run+ cb_row+ 
                 y1_row+ y1_run, data=covariates_ctpcs_geno_tech_cc)


#get partial residuals for no2 and dnam 
cg24535439_res <- partialResid(cg24535439  ~ no2_y1_entire, cg24535439_lm, 
              data=covariates_ctpcs_geno_tech_cc)

#fit again to get r2
cg24535439_res_fit <- lm(cg24535439_res$yresid ~ cg24535439_res$xresid)

summary(cg24535439_res_fit)

ggplot(cg24535439_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, fill = "#254A90", shape=21, size=1.5) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Year one" ~NO[2]~ "(ppb) partial residual"), 
       y="Methylation (%) partial residual") +
  annotate("text",x=8,y=0.06, label="slope = 0.1558", size=2.25) +
  ggtitle("MMP23B cg24535439") +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7.5),
        title = element_text(size=7.5)) +
  ylim(-0.13, 0.13) +
  
  ggsave( filename="2021-01-25_CHILD_y1_cg24535439_MMP23B.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression", "yearone"),
          width = 2.5,
          height = 2,
          units = "in",
          limitsize = TRUE)


############
#cg24727343#
############

covariates_ctpcs_geno_tech_cc$cg24727343 <- diff_betas_geno_cc[rownames(diff_betas_geno_cc)=="cg24727343", ]


cg24727343_lm <- lm(cg24727343  ~ no2_y1_entire + sex +
                 maternal_smoking_y1 + 
                 maternal_education_length +
                 pss_6mos + csed_6mos + 
                 genotyping.PC1 + genotyping.PC2 + 
                 genotyping.PC3 + 
                 celltype.PC1 + celltype.PC2+
                 celltype.PC3+ celltype.PC4+
                 celltype.PC5+ celltype.PC6+
                 celltype.PC7+ celltype.PC8+
                 celltype.PC9+
                 cb_run+ cb_row+ 
                 y1_row+ y1_run, data=covariates_ctpcs_geno_tech_cc)


#get partial residuals for no2 and dnam 
cg24727343_res <- partialResid(cg24727343 ~ no2_y1_entire, cg24727343_lm, 
              data=covariates_ctpcs_geno_tech_cc)

#fit again to get r2
cg24727343_res_fit <- lm(cg24727343_res$yresid ~ cg24727343_res$xresid)

summary(cg24727343_res_fit)

ggplot(cg24727343_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, fill = "#254A90", shape=21, size=1.5) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Year one" ~NO[2]~ "(ppb) partial residual"), 
       y="Methylation (%) partial residual") +
  annotate("text",x=8,y=0.06, label="slope = -0.2028", size=2.25) +
  ggtitle("cg24727343") +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7.5),
        title = element_text(size=7.5)) +
  ylim(-0.13, 0.13) +
  
  ggsave( filename="2021-01-25_CHILD_y1_cg24727343.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression", "yearone"),
          width = 2.5,
          height = 2,
          units = "in",
          limitsize = TRUE)


################
#cg19867107 DMR#
################

covariates_ctpcs_geno_tech_cc$cg19867107 <- diff_betas_geno_cc[rownames(diff_betas_geno_cc)=="cg19867107", ]


cg19867107_lm <- lm(cg19867107  ~ no2_y1_entire + sex +
                 maternal_smoking_y1 + 
                 maternal_education_length +
                 pss_6mos + csed_6mos + 
                 genotyping.PC1 + genotyping.PC2 + 
                 genotyping.PC3 + 
                 celltype.PC1 + celltype.PC2+
                 celltype.PC3+ celltype.PC4+
                 celltype.PC5+ celltype.PC6+
                 celltype.PC7+ celltype.PC8+
                 celltype.PC9+
                 cb_run+ cb_row+ 
                 y1_row+ y1_run, data=covariates_ctpcs_geno_tech_cc)


#get partial residuals for no2 and dnam 
cg19867107_res <- partialResid(cg19867107 ~ no2_y1_entire, cg19867107_lm, 
              data=covariates_ctpcs_geno_tech_cc)

#fit again to get r2
cg19867107_res_fit <- lm(cg19867107_res$yresid ~ cg19867107_res$xresid)

summary(cg19867107_res_fit)

ggplot(cg19867107_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, fill = "#254A90", shape=21, size=1.5) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Year one" ~NO[2]~ "(ppb) partial residual"), 
       y="Methylation (%) partial residual") +
  annotate("text",x=8,y=0.06, label="slope = 0.1327", size=2.25) +
  ggtitle("cg19867107") +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7.5),
        title = element_text(size=7.5)) +
  ylim(-0.13, 0.13) +
  
  ggsave( filename="2021-01-25_CHILD_y1_cg19867107.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression", "yearone"),
          width = 2.5,
          height = 2,
          units = "in",
          limitsize = TRUE)


###################
#NANOS1 cg14916079#
###################


covariates_ctpcs_geno_tech_cc$cg14916079 <- diff_betas_geno_cc[rownames(diff_betas_geno_cc)=="cg14916079", ]


cg14916079_lm <- lm(cg14916079  ~ no2_y1_entire + sex +
                 maternal_smoking_y1 + 
                 maternal_education_length +
                 pss_6mos + csed_6mos + 
                 genotyping.PC1 + genotyping.PC2 + 
                 genotyping.PC3 + 
                 celltype.PC1 + celltype.PC2+
                 celltype.PC3+ celltype.PC4+
                 celltype.PC5+ celltype.PC6+
                 celltype.PC7+ celltype.PC8+
                 celltype.PC9+
                 cb_run+ cb_row+ 
                 y1_row+ y1_run, data=covariates_ctpcs_geno_tech_cc)

#get partial residuals for no2 and dnam 
cg14916079_res <- partialResid(cg14916079 ~ no2_y1_entire, cg14916079_lm, 
              data=covariates_ctpcs_geno_tech_cc)

#fit again to get r2
cg14916079_res_fit <- lm(cg14916079_res$yresid ~ cg14916079_res$xresid)

summary(cg14916079_res_fit)

ggplot(cg14916079_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, fill = "#254A90", shape=21, size=1.5) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Year one" ~NO[2]~ "(ppb) partial residual"), 
       y="Methylation (%) partial residual") +
  annotate("text",x=8,y=0.06, label="slope = 0.3559", size=2.25) +
  ggtitle("NANOS1 cg14916079") +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7.5),
        title = element_text(size=7.5)) +
  ylim(-0.13, 0.13) +
  
  ggsave( filename="2021-01-25_CHILD_y1_cg14916079_NANOs1.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression", "yearone"),
          width = 2.5,
          height = 2,
          units = "in",
          limitsize = TRUE)

###################
#AHNAK cg22365167 #
###################


covariates_ctpcs_geno_tech_cc$cg22365167 <- diff_betas_geno_cc[rownames(diff_betas_geno_cc)=="cg22365167", ]


cg22365167_lm <- lm(cg22365167  ~ no2_y1_entire + sex +
                 maternal_smoking_y1 + 
                 maternal_education_length +
                 pss_6mos + csed_6mos + 
                 genotyping.PC1 + genotyping.PC2 + 
                 genotyping.PC3 + 
                 celltype.PC1 + celltype.PC2+
                 celltype.PC3+ celltype.PC4+
                 celltype.PC5+ celltype.PC6+
                 celltype.PC7+ celltype.PC8+
                 celltype.PC9+
                 cb_run+ cb_row+ 
                 y1_row+ y1_run, data=covariates_ctpcs_geno_tech_cc)


#get partial residuals for no2 and dnam 
cg22365167_res <- partialResid(cg22365167 ~ no2_y1_entire, cg22365167_lm, 
              data=covariates_ctpcs_geno_tech_cc)

#fit again to get r2
cg22365167_res_fit <- lm(cg22365167_res$yresid ~ cg22365167_res$xresid)

summary(cg22365167_res_fit)

ggplot(cg22365167_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, fill = "#254A90", shape=21, size=1.5) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Year one" ~NO[2]~ "(ppb) partial residual"), 
       y="Methylation (%) partial residual") +
  annotate("text",x=8,y=0.06, label="slope = 0.2388", size=2.25) +
  ggtitle("AHNAK cg22365167") +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7.5),
        title = element_text(size=7.5)) +
  ylim(-0.13, 0.13) +
  
  ggsave( filename="2021-01-25_CHILD_y1_cg22365167_AHNAK.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression", "yearone"),
          width = 2.5,
          height = 2,
          units = "in",
          limitsize = TRUE)

##################
#P2RX4 cg16955074#
##################

covariates_ctpcs_geno_tech_cc$cg16955074 <- diff_betas_geno_cc[rownames(diff_betas_geno_cc)=="cg16955074", ]


cg16955074_lm <- lm(cg16955074  ~ no2_y1_entire + sex +
                 maternal_smoking_y1 + 
                 maternal_education_length +
                 pss_6mos + csed_6mos + 
                 genotyping.PC1 + genotyping.PC2 + 
                 genotyping.PC3 + 
                 celltype.PC1 + celltype.PC2+
                 celltype.PC3+ celltype.PC4+
                 celltype.PC5+ celltype.PC6+
                 celltype.PC7+ celltype.PC8+
                 celltype.PC9+
                 cb_run+ cb_row+ 
                 y1_row+ y1_run, data=covariates_ctpcs_geno_tech_cc)


#get partial residuals for no2 and dnam 
cg16955074_res <- partialResid(cg16955074 ~ no2_y1_entire, cg16955074_lm, 
              data=covariates_ctpcs_geno_tech_cc)

#fit again to get r2
cg16955074_res_fit <- lm(cg16955074_res$yresid ~ cg16955074_res$xresid)

summary(cg16955074_res_fit)

ggplot(cg16955074_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, fill = "#254A90", shape=21, size=1.5) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Year one" ~NO[2]~ "(ppb) partial residual"), 
       y="Methylation (%) partial residual") +
  annotate("text",x=8,y=0.06, label="slope = -0.1235", size=2.25) +
  ggtitle("P2RX4 cg16955074") +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7.5),
        title = element_text(size=7.5)) +
  ylim(-0.13, 0.13) +
  
  ggsave( filename="2021-01-25_CHILD_y1_cg16955074_P2RX4.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression", "yearone"),
          width = 2.5,
          height = 2,
          units = "in",
          limitsize = TRUE)


##################
#LONP1 cg18668679#
##################

covariates_ctpcs_geno_tech_cc$cg18668679 <- diff_betas_geno_cc[rownames(diff_betas_geno_cc)=="cg18668679", ]


cg18668679_lm <- lm(cg18668679  ~ no2_y1_entire + sex +
                 maternal_smoking_y1 + 
                 maternal_education_length +
                 pss_6mos + csed_6mos + 
                 genotyping.PC1 + genotyping.PC2 + 
                 genotyping.PC3 + 
                 celltype.PC1 + celltype.PC2+
                 celltype.PC3+ celltype.PC4+
                 celltype.PC5+ celltype.PC6+
                 celltype.PC7+ celltype.PC8+
                 celltype.PC9+
                 cb_run+ cb_row+ 
                 y1_row+ y1_run, data=covariates_ctpcs_geno_tech_cc)


#get partial residuals for no2 and dnam 
cg18668679_res <- partialResid(cg18668679 ~ no2_y1_entire, cg18668679_lm, 
              data=covariates_ctpcs_geno_tech_cc)

#fit again to get r2
cg18668679_res_fit <- lm(cg18668679_res$yresid ~ cg18668679_res$xresid)

summary(cg18668679_res_fit)

ggplot(cg18668679_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.8, fill = "#254A90", shape=21, size=1.5) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Year one" ~NO[2]~ "(ppb) partial residual"), 
       y="Methylation (%) partial residual") +
  annotate("text",x=8,y=0.06, label="slope = -0.1791", size=2.25) +
  ggtitle("LONP1 cg18668679") +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7.5),
        title = element_text(size=7.5)) +
  ylim(-0.13, 0.13) +
  
  ggsave( filename="2021-01-25_CHILD_y1_cg18668679_LONP1.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression", "yearone"),
          width = 2.5,
          height = 2,
          units = "in",
          limitsize = TRUE)
  


###############
#can we facet?#
###############

# using raw data (as opposed to residuals) shows subtle grouping of cpgs 
# possibly suggests influence of mQTLs?


#subset out methylation
y1_dnam <- covariates_ctpcs_geno_tech_cc[,c("cg24535439", "cg24727343", "cg19867107",
                                            "cg14916079","cg22365167", "cg16955074",
                                            "cg18668679", "no2_y1_entire")]

#melt for faceting
y1_dnam_melt <- gather(y1_dnam, cpg, methylation, cg24535439:cg18668679)

#set cpg to factor
y1_dnam_melt$cpg <- as.factor(y1_dnam_melt$cpg)

#set levels of factors so they arent reordered
y1_dnam_melt$cpg <- factor(y1_dnam_melt$cpg, levels = c("cg24535439", "cg24727343", "cg19867107",
                                             "cg14916079","cg22365167", "cg16955074",
                                             "cg18668679"))

#create new facet labels that include genes in a labelled vector called supp.labs
supp.labs <- c("MMP23B cg24535439", "cg24727343", "cg19867107",
               "NANOS1 cg14916079","AHNAK cg22365167", "P2RX4 cg16955074",
               "LONP1 cg18668679")
names(supp.labs) <- c("cg24535439", "cg24727343", "cg19867107",
                      "cg14916079","cg22365167", "cg16955074",
                      "cg18668679")


#create labels for slopes
dat_text <- data.frame(label = signif(c(cg24535439_res_fit$coefficients[[2]],
                                        cg24727343_res_fit$coefficients[[2]],
                                        cg19867107_res_fit$coefficients[[2]],
                                        cg14916079_res_fit$coefficients[[2]],
                                        cg22365167_res_fit$coefficients[[2]],
                                        cg16955074_res_fit$coefficients[[2]],
                                        cg18668679_res_fit$coefficients[[2]])*100, 4),
                       cpg  = c("cg24535439", "cg24727343", "cg19867107",
                                "cg14916079","cg22365167", "cg16955074",
                                "cg18668679"))

# facet plottttt what up
ggplot(y1_dnam_melt, aes(x=no2_y1_entire, y=methylation*100)) + 
  geom_point(alpha=0.8, fill = "#254A90", shape=21, size=1.5) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Year one" ~NO[2]~ "(ppb)"), 
       y="Methylation %") +
  # facet wrap according to cpg
  facet_wrap(.~as.factor(cpg),
             labeller = labeller(.cols = supp.labs)) + 
  #apply slope labels to each facet
  geom_text(data = dat_text,
            mapping = aes(x = 15, y = -13, 
                          label = paste("slope = ", label, sep="")), 
            size=3) + 
  #make it pretty
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=8),
        axis.title = element_text(size=10),
        strip.text.x = element_text(size = 7.5)) +
  ylim(-15, 15) + 

  ggsave( filename="2021-01-25_CHILD_y1_sig_cpgs_facet.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression", "yearone"),
          width = 6,
          height = 5,
          units = "in",
          limitsize = TRUE)

#############################
#create facet with residuals#
#############################

test <- cbind(cg24535439_res, cg24727343_res, cg19867107_res, cg14916079_res,
              cg22365167_res, cg16955074_res, cg18668679_res)

colnames(test) <- c("cg24535439.yresid", "cg24535439.xresid",
                    "cg24727343.yresid", "cg24727343.xresid",
                    "cg19867107.yresid", "cg19867107.xresid",
                    "cg14916079.yresid", "cg14916079.xresid",
                    "cg22365167.yresid", "cg22365167.xresid",
                    "cg16955074.yresid", "cg16955074.xresid",
                    "cg18668679.yresid", "cg18668679.xresid")
  
test2y <- gather(test, cpg, yresid, 
                cg24535439.yresid, cg24727343.yresid,
                cg19867107.yresid, cg14916079.yresid,
                cg22365167.yresid,cg16955074.yresid,
                cg18668679.yresid)


test2x <- gather(test, cpg, xresid, 
                cg24535439.xresid, cg24727343.xresid,
                cg19867107.xresid, cg14916079.xresid,
                cg22365167.xresid,cg16955074.xresid,
                cg18668679.xresid)

test3 <- data.frame(yresid = test2y$yresid,
                    xresid = test2x$xresid,
                    cpg = test2$cpg)

test3$cpg <- substr(test3$cpg, 1, 10)


test3$cpg <- factor(test3$cpg, levels = c("cg24535439", "cg24727343", "cg19867107",
                                             "cg14916079","cg22365167", "cg16955074",
                                             "cg18668679"))

#create labels for slopes
dat_text <- data.frame(label = signif(c(cg24535439_res_fit$coefficients[[2]],
                                        cg24727343_res_fit$coefficients[[2]],
                                        cg19867107_res_fit$coefficients[[2]],
                                        cg14916079_res_fit$coefficients[[2]],
                                        cg22365167_res_fit$coefficients[[2]],
                                        cg16955074_res_fit$coefficients[[2]],
                                        cg18668679_res_fit$coefficients[[2]])*100, 4),
                       cpg  = c("cg24535439", "cg24727343", "cg19867107",
                                "cg14916079","cg22365167", "cg16955074",
                                "cg18668679"))


ggplot(test3, aes(x=xresid, y=yresid*100)) + 
  geom_point(alpha=0.8, fill = "#254A90", shape=21, size=1.5) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Year one" ~NO[2]~ "(ppb)"), 
       y="Methylation %") +
  # facet wrap according to cpg
  facet_wrap(.~as.factor(cpg)) + 
  #apply slope labels to each facet
  geom_text(data = dat_text,
            mapping = aes(x = 15, y = -13, 
                          label = paste("slope = ", label, sep="")), 
            size=3) + 
  #make it pretty
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=8),
        axis.title = element_text(size=10),
        strip.text.x = element_text(size = 7.5))



#overplot residual and raw?
df <- data.frame(no2 = covariates_ctpcs_geno_tech_cc$no2_y1_entire,
                 dnam = covariates_ctpcs_geno_tech_cc$cg18668679,
                 xresid = cg18668679_res$xresid,
                 yresid = cg18668679_res$yresid)

ggplot(df, aes(x=no2, y=dnam)) +
  geom_point(col="black", alpha=0.5) +
  ylim(-0.1, 0.1) +
  xlim(-10, 30) +
  geom_point(df, mapping=aes(x=xresid, y=yresid), col="red", alpha=0.5) +
  theme_bw() + 
    theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=8),
        axis.title = element_text(size=10),
        strip.text.x = element_text(size = 7.5))
  
```





### Gene specific changes

These genes are related to the proposed pathway of air pollution action or have previously been shown to exhibit DNA methylation changes associated with prenatal air pollution exposure. 

```{r gene_specific}
#FHC is FTH1 (ferritin heavy chain 1)
#nothing maps to FOXP3
#TNFA is TNF
goi <- annotation[grep("RNF39|CYP2E1|LONP1|HIBADH|SLC25A28|PLVAP|GPR55|^CAT$|TPO|NFKB1|^NOX2|SOD1|TRX1|TRX2|GSTP1|^MT3|NQO1|HMOX1|GPX1|DDH1|^NOS2|CYP2C11|CYP7B|ALOX12$|ALOX5|^NRF2$|KEAP1|AHR$|^ARNT;|GCLC|GCLM|GSR1|SLC7A11|GPC2|GSTA1|GSTA2|GSTA3|GSTA5|GSTM1|GSTM2|GSTM3|TXN$|TXNRD1$|SRXN1|G6PD|CYP1A1|IL1A|^IL2$|^IL3$|^IL4.*IL4$|^IL5$|^IL6$|IL9$|IL10$|IL12A|IL13|IL15$|IL17A|IL18$|IL19|IL21$|IL22$|IL23A$|IL25$|IL27$|IL31$|IL33$|IFNG$|^TNF$|TGFB1$|FOXP3|FTH1", annotation$UCSC_RefGene_Name), ]

#transpose betas
diff_betas_geno_cc_t <- t(diff_betas_geno_cc)

#apply function for calculating significance of associations using mice and imputed covariate data
yearone_goi_list <- 
  pbapply(diff_betas_geno_cc_t [,colnames(diff_betas_geno_cc_t ) %in% rownames(goi)], 2, function(x) {
  
  combined <- cbind(covariates_ctpcs_geno_tech_cc, x)
  
  #run the linear regression on this cpg using all 5 imputed covariate data sets
  fit <- lm(x ~ no2_y1_entire + sex +
                 maternal_smoking_y1 + 
                 maternal_education_length +
                 pss_6mos + csed_6mos + 
                 genotyping.PC1 + genotyping.PC2 + 
                 genotyping.PC3 + 
                 celltype.PC1 + celltype.PC2+
                 celltype.PC3+ celltype.PC4+
                 celltype.PC5+ celltype.PC6+
                 celltype.PC7+ celltype.PC8+
                 celltype.PC9+
                 cb_run+ cb_row+ 
                 y1_row+ y1_run, data=combined)
  
  #get summary of lm fit
  #this includes coefficients and pvalues
  #coerce into df for manipulation
  #add in terms (rownames) as column for annotation
  fit_sum <- summary(fit)
  fit_sum_coeff <- as.data.frame(fit_sum$coefficients)
  fit_sum_coeff$term <- rownames(fit_sum_coeff)
  
  #this data frame has rows for each linear regression variable, and columns for statistics
  #want a dataframe of 1 row and many columns (one for each variable with each statistic type)
  #do this by using the spread function for each statistics
  #add cpg id onto dataframe for pvalues (to keep track of information)
  #then cbind dataframes together
  
  #coefficient
  estimate <- spread(fit_sum_coeff[,c("term", "Estimate")], term, Estimate)
  colnames(estimate) <- paste("estimate", colnames(estimate), sep=".")
  
  #std error
  std.error <- spread(fit_sum_coeff[,c("term", "Std. Error")], term, 'Std. Error')
  colnames(std.error) <- paste("std.error", colnames(std.error), sep=".")
  
  #t statistics
  statistic <- spread(fit_sum_coeff[,c("term", "t value")], term, 't value')
  colnames(statistic) <- paste("t.val", colnames(statistic), sep=".")

  #pvalue
  p.value <- spread(fit_sum_coeff[,c("term", "Pr(>|t|)")], term, 'Pr(>|t|)')
  colnames(p.value) <- paste("p.value", colnames(p.value), sep=".")
  
  #add in cpgid for annotation 
  p.value$cpg <- colnames(x)
  
  #cbind into wide dataframe
  widepool <- cbind(estimate, std.error, statistic, p.value)
  
  #return the wide dataframe
  #these will be do.call(rbind()) together after
  return(widepool)
  
})


yearone_goi <- do.call(rbind, yearone_goi_list)

yearone_goi$p.value.no2_y1_entire_adjusted <- p.adjust(yearone_goi$p.value.no2_y1_entire, method="bonferroni")

#reorder according ot unadjusted no2 pvalue
yearone_goi_order <- yearone_goi[order(yearone_goi$p.value.no2_y1_entire),]

#reorder goi annotation to be in same order as goi stats
goi_order <- goi[rownames(yearone_goi_order),]

#check identical
identical(rownames(goi_order), rownames(yearone_goi_order)) #true

#bind
yearone_goi_df <- cbind(yearone_goi_order, goi_order)

#write as .csv
#write.csv(yearone_goi_df, file=here("output_data", "2020-11-29_yearone_goi.csv"))


#####
#AHR#
#####

#AHR cpg that is close to sig
covariates_ctpcs_geno_tech_cc$cg07793849 <- diff_betas_geno_cc[rownames(diff_betas_geno_cc)=="cg07793849", ]


cg07793849_lm <- lm(cg07793849 ~ no2_y1_entire + sex +
                 maternal_smoking_y1 + 
                 maternal_education_length +
                 pss_6mos + csed_6mos + 
                 genotyping.PC1 + genotyping.PC2 + 
                 genotyping.PC3 + 
                 celltype.PC1 + celltype.PC2+
                 celltype.PC3+ celltype.PC4+
                 celltype.PC5+ celltype.PC6+
                 celltype.PC7+ celltype.PC8+
                 celltype.PC9+
                 cb_run+ cb_row+ 
                 y1_row+ y1_run, data=covariates_ctpcs_geno_tech_cc)


#get partial residuals for no2 and dnam 
cg07793849_res <- partialResid(cg07793849 ~ no2_y1_entire, cg07793849_lm, 
              data=covariates_ctpcs_geno_tech_cc)

#fit again to get r2
cg07793849_res_fit <- lm(cg07793849_res$yresid ~ cg07793849_res$xresid)

summary(cg07793849_res_fit)

ggplot(cg07793849_res, aes(x=xresid, y=yresid)) + 
    geom_point(alpha=0.8, fill = "#254A90", shape=21, size=1.5) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Year one" ~NO[2]~ "(ppb) partial residual"), 
       y="Methylation (%) partial residual") +
  annotate("text",x=8,y=0.01, label="slope = 0.03273", size=2.25) +
  ggtitle("AHR cg07793849") +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7.5),
        title = element_text(size=7.5)) +
 # ylim(-0.13, 0.13) +
  
  ggsave( filename="2021-01-25_CHILD_y1_cg07793849_AHR.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression", "yearone"),
          width = 2.5,
          height = 2,
          units = "in",
          limitsize = TRUE)


ggplot(covariates_ctpcs_geno_tech_cc, aes(x=no2_y1_entire, y=cg07793849)) + 
  geom_point(alpha=0.8, fill = "#254A90", shape=21, size=1.5) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Year one" ~NO[2]~ "(ppb)"), 
       y="Methylation (%)") +
  annotate("text",x=20,y=-0.01, label="slope = 0.03273", size=2.25) +
  ggtitle("AHR cg07793849") +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7.5),
        title = element_text(size=7.5)) +
  # ylim(-0.13, 0.13) +
  
  ggsave( filename="2021-01-25_CHILD_y1_cg07793849_AHR_raw.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression", "yearone"),
          width = 2.5,
          height = 2,
          units = "in",
          limitsize = TRUE)

```




Examine mean methylation vs no2
```{r mean_meth}
#convert betas to df
diff_betas_geno_cc_df <- as.data.frame(diff_betas_geno_cc)

methmeans <- colMeans(diff_betas_geno_cc_df)

covariates_ctpcs_geno_tech_cc <- cbind(covariates_ctpcs_geno_tech_cc, methmeans)

#colmeans
meanmeth_lm <- lm(methmeans ~ no2_y1_entire + sex +
                 maternal_smoking_y1 + 
                 maternal_education_length +
                 pss_6mos + csed_6mos + 
                 genotyping.PC1 + genotyping.PC2 + 
                 genotyping.PC3 + 
                 celltype.PC1 + celltype.PC2+
                 celltype.PC3+ celltype.PC4+
                 celltype.PC5+ celltype.PC6+
                 celltype.PC7+ celltype.PC8+
                 celltype.PC9+
                 cb_run+ cb_row+ 
                 y1_row+ y1_run, data=covariates_ctpcs_geno_tech_cc)

#get summary
summary(meanmeth_lm) #no2 not sig

#get partial residuals for no2 and dnam 
methmean_res <- partialResid( methmeans ~ no2_y1_entire, meanmeth_lm, 
              data=covariates_ctpcs_geno_tech_cc)

#fit again to get r2
methmean_res_fit <- lm(methmean_res$yresid ~ methmean_res$xresid)

summary(methmean_res_fit)

ggplot(methmean_res, aes(x=xresid, y=yresid)) + 
  
    geom_point(alpha=0.8, fill = "#254A90", shape=21, size=1.5) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x=bquote("Year one" ~NO[2]~ "(ppb) partial residual"), 
       y="Methylation (%)") +
  annotate("text",x=5,y=0.01, label="slope =0.001419", size=2.25) +
  ggtitle("Global DNA methylation") +
  theme(panel.grid.major = element_blank(), 
        panel.grid.minor = element_blank(),
        axis.text = element_text(size=6),
        axis.title = element_text(size=7.5),
        title = element_text(size=7.5)) +
  ylim(-0.012, 0.012) +
  
  ggsave( filename="2021-01-25_CHILD_y1_global_dnam.tiff",
          device = "tiff",
          path =  here("figures", "linear_regression", "yearone"),
          width = 2.5,
          height = 2,
          units = "in",
          limitsize = TRUE)
```





```{r get_lineannotation}
#read in line annotation
annotationlines <- read.csv(here("output_data", "linear_regression", "2020-11-29_annotation_LINEs.csv"))

#subset 
annotationlines <- annotationlines[annotationlines$LINEs==T, ]

#subset betas
diff_betas_geno_cc_lines <- 
  diff_betas_geno_cc[rownames(diff_betas_geno_cc) %in%
                     annotationlines$X, ]

dim(diff_betas_geno_cc_lines)


#convert betas to df
diff_betas_geno_cc_lines_df <- as.data.frame(diff_betas_geno_cc_lines)

linemeth <- colMeans(diff_betas_geno_cc_lines_df)

covariates_ctpcs_geno_tech_cc$linemeth <- linemeth

#colmeans
linemeth_lm <- lm(linemeth ~ no2_y1_entire + sex +
                 maternal_smoking_y1 + 
                 maternal_education_length +
                 pss_6mos + csed_6mos + 
                 genotyping.PC1 + genotyping.PC2 + 
                 genotyping.PC3 + 
                 celltype.PC1 + celltype.PC2+
                 celltype.PC3+ celltype.PC4+
                 celltype.PC5+ celltype.PC6+
                 celltype.PC7+ celltype.PC8+
                 celltype.PC9+
                 cb_run+ cb_row+ 
                 y1_row+ y1_run, data=covariates_ctpcs_geno_tech_cc)

summary(linemeth_lm) #no2 not sig

#get partial residuals for no2 and dnam 
linemeth_res <- partialResid( linemeth ~ no2_y1_entire, linemeth_lm, 
              data=covariates_ctpcs_geno_tech_cc)

#fit again to get r2
linemeth_res_fit <- lm(linemeth_res$yresid ~ linemeth_res$xresid)

summary(linemeth_res_fit)

ggplot(linemeth_res, aes(x=xresid, y=yresid)) + 
  geom_point(alpha=0.6, fill = "#254A90", shape=21, size=3) + 
  stat_smooth(method="lm", se=F, col="red") + 
  theme_bw() +
  labs(x="Year one NO2 (ppb)", y="LINE DNA methyaltion (%)") +
  ggtitle("Partial residual plot") +
  theme(panel.border=element_blank(),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=14)) 

```








Try re-running limma with birth month/ interaction between birth month and air pollution.
```{r interaction_terms_to_assess_birthmonth}
#want to use anova to compare models usign anova
#anova only accepts models from lm
#use lm to compare how birth month affects cpgs above 10^-3 sig 

#You should be able to ask for confidence intervals for the slope coefficients for each month.  #even in a tabular display, you may see that “smoothly varying” pattern we were discussing.
#Then, you could use those to plot a sort of “here’s how the slope varies by month” picture.Sort #of like forest plots for meta analysis results, if that makes sense.

#Also, before we decide it doesn’t matter, the individual slopes might look fairly similar, but the appropriate overall test would be to check the anova(fit_with_interaction) and check the p-value on the interaction term.


################################################
#sig cpg
library(broom)

covariates_ctpcs_geno_tech_cc$cg13668823 <- adj_diff_betas_geno_cc[rownames(adj_diff_betas_geno_cc)==cg24535439, ]



cg13668823_fit_bm <- lm(cg13668823 ~ no2_y1_entire + sex +
              maternal_smoking_y1 + 
              maternal_education_length +
              pss_6mos + 
              csed_6mos + 
              celltype.PC1 + celltype.PC2 + 
              celltype.PC3 + celltype.PC4 + 
              celltype.PC5 + celltype.PC6 + 
              celltype.PC7 + celltype.PC8 +
              celltype.PC9 + 
              genotyping.PC1 + genotyping.PC2 + 
              genotyping.PC3 +
                birth_month +
                no2_y1_entire*birth_month +
              run + row, data=covariates_ctpcs_geno_tech_cc)



tidy(cg13668823_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_y1_entire:birth_monthfebruary",
                                      "no2_y1_entire:birth_monthmarch",
                                      "no2_y1_entire:birth_monthapril",
                                      "no2_y1_entire:birth_monthmay",
                                      "no2_y1_entire:birth_monthjune",
                                      "no2_y1_entire:birth_monthjuly",
                                      "no2_y1_entire:birth_monthaugust",
                                      "no2_y1_entire:birth_monthseptember",
                                      "no2_y1_entire:birth_monthoctober",
                                      "no2_y1_entire:birth_monthnovember",
                                      "no2_y1_entire:birth_monthdecember"))) %>%
  ggplot(aes(y = estimate, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low , ymax = conf.high)) +
  theme_bw() +
  ggtitle(cg24535439) +
  scale_x_discrete(labels=c("NO2:February","NO2:March", "NO2:April", "NO2:May",
                            "NO2:June","NO2:July", "NO2:August", "NO2:September",
                            "NO2:October", "NO2:November","NO2:December")) +
  labs(x="Interaction term", y="Estimate") +
  theme(panel.border=element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=16)) 

anova(cg13668823_fit_bm)






cg18668679_fit_bm <- lm(cg18668679 ~ no2_y1_entire + sex +
              maternal_smoking_y1 + 
              maternal_education_length +
              pss_6mos + 
              csed_6mos + 
              celltype.PC1 + celltype.PC2 + 
              celltype.PC3 + celltype.PC4 + 
              celltype.PC5 + celltype.PC6 + 
              celltype.PC7 + celltype.PC8 +
              celltype.PC9 + 
              genotyping.PC1 + genotyping.PC2 + 
              genotyping.PC3 +
                birth_month +
                no2_y1_entire*birth_month +
              run + row, data=covariates_ctpcs_geno_tech_cc)



tidy(cg18668679_fit_bm, conf.int = 0.95) %>% filter(term %>% str_detect(":")) %>% 
  mutate(term = factor(term, levels=c("no2_y1_entire:birth_monthfebruary",
                                      "no2_y1_entire:birth_monthmarch",
                                      "no2_y1_entire:birth_monthapril",
                                      "no2_y1_entire:birth_monthmay",
                                      "no2_y1_entire:birth_monthjune",
                                      "no2_y1_entire:birth_monthjuly",
                                      "no2_y1_entire:birth_monthaugust",
                                      "no2_y1_entire:birth_monthseptember",
                                      "no2_y1_entire:birth_monthoctober",
                                      "no2_y1_entire:birth_monthnovember",
                                      "no2_y1_entire:birth_monthdecember"))) %>%
  ggplot(aes(y = estimate, x = as.factor(term))) + 
  geom_point() +
  geom_errorbar(aes(ymin = conf.low , ymax = conf.high)) +
  theme_bw() +
  ggtitle("cg18668679") +
  scale_x_discrete(labels=c("NO2:February","NO2:March", "NO2:April", "NO2:May",
                            "NO2:June","NO2:July", "NO2:August", "NO2:September",
                            "NO2:October", "NO2:November","NO2:December")) +
  labs(x="Interaction term", y="Estimate") +
  theme(panel.border=element_blank(),
        axis.text.x = element_text(angle=90),
        axis.text = element_text(size=12),
        axis.title = element_text(size=14),
        title = element_text(size=16)) 

anova(cg18668679_fit_bm)
  
```



